// @spartacus/cart/esm2015/saved-cart/components/details/saved-cart-details.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AuthGuard, ConfigModule, I18nModule, UrlModule, } from '@spartacus/core';
import { AddToCartModule, CardModule, CartSharedModule, IconModule, MediaModule, SpinnerModule, } from '@spartacus/storefront';
import { SavedCartDetailsActionComponent } from './saved-cart-details-action/saved-cart-details-action.component';
import { SavedCartDetailsItemsComponent } from './saved-cart-details-items/saved-cart-details-items.component';
import { SavedCartDetailsOverviewComponent } from './saved-cart-details-overview/saved-cart-details-overview.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class SavedCartDetailsModule {
}
SavedCartDetailsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartDetailsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SavedCartDetailsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartDetailsModule, declarations: [SavedCartDetailsOverviewComponent,
        SavedCartDetailsActionComponent,
        SavedCartDetailsItemsComponent], imports: [CartSharedModule,
        CommonModule,
        I18nModule,
        UrlModule,
        RouterModule,
        CardModule,
        MediaModule,
        AddToCartModule,
        IconModule,
        SpinnerModule, i1.ConfigModule], exports: [SavedCartDetailsOverviewComponent,
        SavedCartDetailsActionComponent,
        SavedCartDetailsItemsComponent] });
SavedCartDetailsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartDetailsModule, imports: [[
            CartSharedModule,
            CommonModule,
            I18nModule,
            UrlModule,
            RouterModule,
            CardModule,
            MediaModule,
            AddToCartModule,
            IconModule,
            SpinnerModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    SavedCartDetailsOverviewComponent: {
                        component: SavedCartDetailsOverviewComponent,
                        guards: [AuthGuard],
                    },
                    SavedCartDetailsItemsComponent: {
                        component: SavedCartDetailsItemsComponent,
                        guards: [AuthGuard],
                    },
                    SavedCartDetailsActionComponent: {
                        component: SavedCartDetailsActionComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartDetailsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CartSharedModule,
                        CommonModule,
                        I18nModule,
                        UrlModule,
                        RouterModule,
                        CardModule,
                        MediaModule,
                        AddToCartModule,
                        IconModule,
                        SpinnerModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                SavedCartDetailsOverviewComponent: {
                                    component: SavedCartDetailsOverviewComponent,
                                    guards: [AuthGuard],
                                },
                                SavedCartDetailsItemsComponent: {
                                    component: SavedCartDetailsItemsComponent,
                                    guards: [AuthGuard],
                                },
                                SavedCartDetailsActionComponent: {
                                    component: SavedCartDetailsActionComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [
                        SavedCartDetailsOverviewComponent,
                        SavedCartDetailsActionComponent,
                        SavedCartDetailsItemsComponent,
                    ],
                    exports: [
                        SavedCartDetailsOverviewComponent,
                        SavedCartDetailsActionComponent,
                        SavedCartDetailsItemsComponent,
                    ],
                }]
        }] });
