// @spartacus/product/esm2015/variants/components/variant-size-selector/product-variant-size-selector.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { UrlModule, I18nModule } from '@spartacus/core';
import { ProductVariantSizeSelectorComponent } from './product-variant-size-selector.component';
import { RouterModule } from '@angular/router';
import * as i0 from "@angular/core";
export class ProductVariantSizeSelectorModule {
}
ProductVariantSizeSelectorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantSizeSelectorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductVariantSizeSelectorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantSizeSelectorModule, declarations: [ProductVariantSizeSelectorComponent], imports: [CommonModule, RouterModule, UrlModule, I18nModule], exports: [ProductVariantSizeSelectorComponent] });
ProductVariantSizeSelectorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantSizeSelectorModule, imports: [[CommonModule, RouterModule, UrlModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantSizeSelectorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RouterModule, UrlModule, I18nModule],
                    declarations: [ProductVariantSizeSelectorComponent],
                    exports: [ProductVariantSizeSelectorComponent],
                }]
        }] });
