// @spartacus/core/esm2015/src/cart/connectors/save-cart/save-cart.connecter.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./save-cart.adapter";
export class SaveCartConnector {
    constructor(adapter) {
        this.adapter = adapter;
    }
    saveCart(userId, cartId, saveCartName, saveCartDescription) {
        return this.adapter.saveCart(userId, cartId, saveCartName, saveCartDescription);
    }
}
SaveCartConnector.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SaveCartConnector, deps: [{ token: i1.SaveCartAdapter }], target: i0.ɵɵFactoryTarget.Injectable });
SaveCartConnector.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SaveCartConnector, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SaveCartConnector, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.SaveCartAdapter }]; } });
