// @spartacus/product/esm2015/bulk-pricing/root/bulk-pricing-root.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfigFactory } from '@spartacus/core';
import { PRODUCT_BULK_PRICING_FEATURE } from './feature-name';
import * as i0 from "@angular/core";
// TODO: Inline this factory when we start releasing Ivy compiled libraries
export function defaultProductBulkPricingComponentsConfig() {
    const config = {
        featureModules: {
            [PRODUCT_BULK_PRICING_FEATURE]: {
                cmsComponents: ['BulkPricingTableComponent'],
            },
        },
    };
    return config;
}
export class BulkPricingRootModule {
}
BulkPricingRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BulkPricingRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule });
BulkPricingRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule, providers: [
        provideDefaultConfigFactory(defaultProductBulkPricingComponentsConfig),
    ], imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    providers: [
                        provideDefaultConfigFactory(defaultProductBulkPricingComponentsConfig),
                    ],
                }]
        }] });
