// @spartacus/storefront/esm2015/layout/a11y/skip-link/directive/skip-link.directive.js
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../service/skip-link.service";
export class SkipLinkDirective {
    constructor(elementRef, skipLinkService) {
        this.elementRef = elementRef;
        this.skipLinkService = skipLinkService;
    }
    ngOnInit() {
        this.skipLinkService.add(this.cxSkipLink, this.elementRef.nativeElement);
    }
    ngOnDestroy() {
        this.skipLinkService.remove(this.cxSkipLink);
    }
}
SkipLinkDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SkipLinkDirective, deps: [{ token: i0.ElementRef }, { token: i1.SkipLinkService }], target: i0.ɵɵFactoryTarget.Directive });
SkipLinkDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: SkipLinkDirective, selector: "[cxSkipLink]", inputs: { cxSkipLink: "cxSkipLink" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SkipLinkDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxSkipLink]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i1.SkipLinkService }]; }, propDecorators: { cxSkipLink: [{
                type: Input
            }] } });
