// @spartacus/cart/esm2015/quick-order/components/quick-order/quick-order-list.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { FormErrorsModule, IconModule, ItemCounterModule, MediaModule, MessageComponentModule, ProgressButtonModule, } from '@spartacus/storefront';
import { QuickOrderComponent } from './quick-order.component';
import { QuickOrderFormComponent } from './form/quick-order-form.component';
import { QuickOrderItemComponent } from './table/item/quick-order-item.component';
import { QuickOrderTableComponent } from './table/quick-order-table.component';
import * as i0 from "@angular/core";
export class QuickOrderListModule {
}
QuickOrderListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
QuickOrderListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderListModule, declarations: [QuickOrderComponent,
        QuickOrderFormComponent,
        QuickOrderItemComponent,
        QuickOrderTableComponent], imports: [CommonModule,
        ReactiveFormsModule,
        RouterModule,
        FormErrorsModule,
        I18nModule,
        IconModule,
        ItemCounterModule,
        MediaModule,
        UrlModule,
        ProgressButtonModule,
        MessageComponentModule], exports: [QuickOrderComponent,
        QuickOrderFormComponent,
        QuickOrderItemComponent,
        QuickOrderTableComponent] });
QuickOrderListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderListModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                QuickOrderComponent: {
                    component: QuickOrderComponent,
                    data: {
                        quickOrderListLimit: 10,
                    },
                },
            },
        }),
    ], imports: [[
            CommonModule,
            ReactiveFormsModule,
            RouterModule,
            FormErrorsModule,
            I18nModule,
            IconModule,
            ItemCounterModule,
            MediaModule,
            UrlModule,
            ProgressButtonModule,
            MessageComponentModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        RouterModule,
                        FormErrorsModule,
                        I18nModule,
                        IconModule,
                        ItemCounterModule,
                        MediaModule,
                        UrlModule,
                        ProgressButtonModule,
                        MessageComponentModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                QuickOrderComponent: {
                                    component: QuickOrderComponent,
                                    data: {
                                        quickOrderListLimit: 10,
                                    },
                                },
                            },
                        }),
                    ],
                    declarations: [
                        QuickOrderComponent,
                        QuickOrderFormComponent,
                        QuickOrderItemComponent,
                        QuickOrderTableComponent,
                    ],
                    exports: [
                        QuickOrderComponent,
                        QuickOrderFormComponent,
                        QuickOrderItemComponent,
                        QuickOrderTableComponent,
                    ],
                }]
        }] });
