// @spartacus/core/esm2015/src/cart/connectors/validation/cart-validation.connector.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./cart-validation.adapter";
export class CartValidationConnector {
    constructor(adapter) {
        this.adapter = adapter;
    }
    validate(cartId, userId) {
        return this.adapter.validate(cartId, userId);
    }
}
CartValidationConnector.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationConnector, deps: [{ token: i1.CartValidationAdapter }], target: i0.ɵɵFactoryTarget.Injectable });
CartValidationConnector.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationConnector, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationConnector, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CartValidationAdapter }]; } });
