// @spartacus/cart/esm2015/saved-cart/occ/saved-cart-occ.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { SavedCartAdapter } from '@spartacus/cart/saved-cart/core';
import { provideDefaultConfig } from '@spartacus/core';
import { OccSavedCartAdapter } from './adapters/occ-saved-cart.adapter';
import { defaultOccSavedCartConfig } from './config/default-occ-saved-cart-config';
import * as i0 from "@angular/core";
export class SavedCartOccModule {
}
SavedCartOccModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SavedCartOccModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartOccModule, imports: [CommonModule] });
SavedCartOccModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartOccModule, providers: [
        provideDefaultConfig(defaultOccSavedCartConfig),
        {
            provide: SavedCartAdapter,
            useClass: OccSavedCartAdapter,
        },
    ], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccSavedCartConfig),
                        {
                            provide: SavedCartAdapter,
                            useClass: OccSavedCartAdapter,
                        },
                    ],
                }]
        }] });
