// @spartacus/storefront/esm2015/cms-structure/seo/structured-data/builders/json-ld-builder.module.js
import { NgModule } from '@angular/core';
import { BreadcrumbSchemaBuilder } from './breadcrumb/index';
import { JsonLdBaseProductBuilder, JsonLdProductOfferBuilder, JsonLdProductReviewBuilder, ProductSchemaBuilder, } from './product/index';
import { JSONLD_PRODUCT_BUILDER, SCHEMA_BUILDER } from './tokens';
import * as i0 from "@angular/core";
/**
 * Provides several standard json-ld builders that contribute
 * to collecting and building json-ld data.
 */
export class JsonLdBuilderModule {
}
JsonLdBuilderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: JsonLdBuilderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
JsonLdBuilderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: JsonLdBuilderModule });
JsonLdBuilderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: JsonLdBuilderModule, providers: [
        {
            provide: SCHEMA_BUILDER,
            useExisting: ProductSchemaBuilder,
            multi: true,
        },
        {
            provide: SCHEMA_BUILDER,
            useExisting: BreadcrumbSchemaBuilder,
            multi: true,
        },
        // lower level json-ld builder classes offering fine-grained control
        // for product related schemas
        {
            provide: JSONLD_PRODUCT_BUILDER,
            useExisting: JsonLdBaseProductBuilder,
            multi: true,
        },
        {
            provide: JSONLD_PRODUCT_BUILDER,
            useExisting: JsonLdProductOfferBuilder,
            multi: true,
        },
        {
            provide: JSONLD_PRODUCT_BUILDER,
            useExisting: JsonLdProductReviewBuilder,
            multi: true,
        },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: JsonLdBuilderModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        {
                            provide: SCHEMA_BUILDER,
                            useExisting: ProductSchemaBuilder,
                            multi: true,
                        },
                        {
                            provide: SCHEMA_BUILDER,
                            useExisting: BreadcrumbSchemaBuilder,
                            multi: true,
                        },
                        // lower level json-ld builder classes offering fine-grained control
                        // for product related schemas
                        {
                            provide: JSONLD_PRODUCT_BUILDER,
                            useExisting: JsonLdBaseProductBuilder,
                            multi: true,
                        },
                        {
                            provide: JSONLD_PRODUCT_BUILDER,
                            useExisting: JsonLdProductOfferBuilder,
                            multi: true,
                        },
                        {
                            provide: JSONLD_PRODUCT_BUILDER,
                            useExisting: JsonLdProductReviewBuilder,
                            multi: true,
                        },
                    ],
                }]
        }] });
