// @spartacus/cart/esm2015/saved-cart/occ/adapters/occ-saved-cart.adapter.js
import { Injectable } from '@angular/core';
import { CART_NORMALIZER, } from '@spartacus/core';
import { map, pluck } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "@spartacus/core";
export class OccSavedCartAdapter {
    constructor(http, occEndpoints, converter) {
        this.http = http;
        this.occEndpoints = occEndpoints;
        this.converter = converter;
    }
    load(userId, cartId) {
        return this.http
            .get(this.getSavedCartEndpoint(userId, cartId))
            .pipe(pluck('savedCartData'), this.converter.pipeable(CART_NORMALIZER));
    }
    loadList(userId) {
        return this.http
            .get(this.getSavedCartListEndpoint(userId))
            .pipe(pluck('carts'), map((carts) => carts !== null && carts !== void 0 ? carts : []), this.converter.pipeableMany(CART_NORMALIZER));
    }
    restoreSavedCart(userId, cartId) {
        return this.http
            .patch(this.getRestoreSavedCartEndpoint(userId, cartId), cartId)
            .pipe(pluck('savedCartData'), this.converter.pipeable(CART_NORMALIZER));
    }
    saveCart(userId, cartId, saveCartName, saveCartDescription) {
        return this.http
            .patch(this.getSaveCartEndpoint(userId, cartId, saveCartName, saveCartDescription), cartId)
            .pipe(pluck('savedCartData'), this.converter.pipeable(CART_NORMALIZER));
    }
    cloneSavedCart(userId, cartId, saveCartName) {
        return this.http
            .post(this.getCloneSavedCartEndpoint(userId, cartId, saveCartName), cartId)
            .pipe(pluck('savedCartData'), this.converter.pipeable(CART_NORMALIZER));
    }
    getSavedCartEndpoint(userId, cartId) {
        return this.occEndpoints.buildUrl('savedCart', {
            urlParams: { userId, cartId },
        });
    }
    getSavedCartListEndpoint(userId) {
        return this.occEndpoints.buildUrl('savedCarts', { urlParams: { userId } });
    }
    getRestoreSavedCartEndpoint(userId, cartId) {
        return this.occEndpoints.buildUrl('restoreSavedCart', {
            urlParams: { userId, cartId },
        });
    }
    getSaveCartEndpoint(userId, cartId, saveCartName, saveCartDescription) {
        return this.occEndpoints.buildUrl('saveCart', {
            urlParams: {
                userId,
                cartId,
                saveCartName,
                saveCartDescription,
            },
        });
    }
    getCloneSavedCartEndpoint(userId, cartId, saveCartName) {
        return this.occEndpoints.buildUrl('cloneSavedCart', {
            urlParams: { userId, cartId, saveCartName },
        });
    }
}
OccSavedCartAdapter.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccSavedCartAdapter, deps: [{ token: i1.HttpClient }, { token: i2.OccEndpointsService }, { token: i2.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable });
OccSavedCartAdapter.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccSavedCartAdapter });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccSavedCartAdapter, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: i2.OccEndpointsService }, { type: i2.ConverterService }]; } });
