// @spartacus/storefront/esm2015/cms-components/navigation/page-header/page-title.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { PageMetaModule, provideDefaultConfig, } from '@spartacus/core';
import { PageTitleComponent } from './page-title.component';
import * as i0 from "@angular/core";
export class PageTitleModule {
}
PageTitleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageTitleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PageTitleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageTitleModule, declarations: [PageTitleComponent], imports: [CommonModule, RouterModule, PageMetaModule], exports: [PageTitleComponent] });
PageTitleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageTitleModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                PageTitleComponent: {
                    component: PageTitleComponent,
                },
            },
        }),
    ], imports: [[CommonModule, RouterModule, PageMetaModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageTitleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RouterModule, PageMetaModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                PageTitleComponent: {
                                    component: PageTitleComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [PageTitleComponent],
                    exports: [PageTitleComponent],
                    entryComponents: [PageTitleComponent],
                }]
        }] });
