// @spartacus/storefront/esm2015/cms-components/navigation/page-header/page-title.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { isNotNullable, } from '@spartacus/core';
import { filter, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../../cms-structure/page/model/cms-component-data";
import * as i2 from "@spartacus/core";
import * as i3 from "@angular/common";
export class PageTitleComponent {
    constructor(component, pageMetaService) {
        this.component = component;
        this.pageMetaService = pageMetaService;
    }
    ngOnInit() {
        this.setTitle();
    }
    setTitle() {
        this.title$ = this.pageMetaService.getMeta().pipe(filter(isNotNullable), map((meta) => { var _a; return (_a = (meta.heading || meta.title)) !== null && _a !== void 0 ? _a : ''; }));
    }
}
PageTitleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageTitleComponent, deps: [{ token: i1.CmsComponentData }, { token: i2.PageMetaService }], target: i0.ɵɵFactoryTarget.Component });
PageTitleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: PageTitleComponent, selector: "cx-page-title", ngImport: i0, template: "<h1 class=\"cx-visually-hidden\">{{ title$ | async }}</h1>\n", pipes: { "async": i3.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-page-title',
                    templateUrl: './page-title.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.CmsComponentData }, { type: i2.PageMetaService }]; } });
