// @spartacus/storefront/esm2015/cms-components/misc/site-context-selector/language-currency.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./site-context-selector.component";
export class LanguageCurrencyComponent {
}
LanguageCurrencyComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LanguageCurrencyComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LanguageCurrencyComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: LanguageCurrencyComponent, selector: "cx-language-currency-selector", ngImport: i0, template: `
    <cx-site-context-selector context="language"></cx-site-context-selector>
    <cx-site-context-selector context="currency"></cx-site-context-selector>
  `, isInline: true, components: [{ type: i1.SiteContextSelectorComponent, selector: "cx-site-context-selector", inputs: ["context"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LanguageCurrencyComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-language-currency-selector',
                    template: `
    <cx-site-context-selector context="language"></cx-site-context-selector>
    <cx-site-context-selector context="currency"></cx-site-context-selector>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }] });
