// @spartacus/storefront/esm2015/shared/services/file/files-form-validators.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class FilesFormValidators {
    /**
     * Checks max size of file
     *
     * @param {number} maxSize Max size [MB]
     * @returns Uses 'tooLarge' validator error with maxSize property
     * @memberOf FilesFormValidators
     */
    maxSize(maxSize) {
        return (control) => {
            const errors = {};
            if (maxSize) {
                const files = Array.from(control.value);
                files.forEach((file) => {
                    var _a, _b;
                    if (file.size > maxSize * 1000000) {
                        const invalidFiles = (_b = (_a = errors.tooLarge) === null || _a === void 0 ? void 0 : _a.invalidFiles) !== null && _b !== void 0 ? _b : [];
                        errors.tooLarge = {
                            maxSize,
                            invalidFiles: [...invalidFiles, file.name],
                        };
                    }
                });
            }
            return Object.keys(errors).length === 0 ? null : errors;
        };
    }
}
FilesFormValidators.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FilesFormValidators, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
FilesFormValidators.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FilesFormValidators, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FilesFormValidators, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
