// @spartacus/storefront/esm2015/shared/components/media/media.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MediaComponent } from './media.component';
import * as i0 from "@angular/core";
export class MediaModule {
    static forRoot() {
        return {
            ngModule: MediaModule,
        };
    }
}
MediaModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MediaModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MediaModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MediaModule, declarations: [MediaComponent], imports: [CommonModule], exports: [MediaComponent] });
MediaModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MediaModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MediaModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [MediaComponent],
                    exports: [MediaComponent],
                }]
        }] });
