// @spartacus/user/esm2015/profile/components/update-email/update-email.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { AuthGuard, AuthRedirectService, AuthService, GlobalMessageService, I18nModule, provideDefaultConfig, RoutingService, UrlModule, } from '@spartacus/core';
import { FormErrorsModule, SpinnerModule } from '@spartacus/storefront';
import { UserEmailFacade } from '@spartacus/user/profile/root';
import { UpdateEmailComponentService } from './update-email-component.service';
import { UpdateEmailComponent } from './update-email.component';
import * as i0 from "@angular/core";
export class UpdateEmailModule {
}
UpdateEmailModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UpdateEmailModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UpdateEmailModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UpdateEmailModule, declarations: [UpdateEmailComponent], imports: [CommonModule,
        FormsModule,
        ReactiveFormsModule,
        SpinnerModule,
        UrlModule,
        RouterModule,
        I18nModule,
        FormErrorsModule] });
UpdateEmailModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UpdateEmailModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                UpdateEmailComponent: {
                    component: UpdateEmailComponent,
                    guards: [AuthGuard],
                    providers: [
                        {
                            provide: UpdateEmailComponentService,
                            useClass: UpdateEmailComponentService,
                            deps: [
                                UserEmailFacade,
                                RoutingService,
                                GlobalMessageService,
                                AuthService,
                                AuthRedirectService,
                            ],
                        },
                    ],
                },
            },
        }),
    ], imports: [[
            CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            UrlModule,
            RouterModule,
            I18nModule,
            FormErrorsModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UpdateEmailModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        SpinnerModule,
                        UrlModule,
                        RouterModule,
                        I18nModule,
                        FormErrorsModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                UpdateEmailComponent: {
                                    component: UpdateEmailComponent,
                                    guards: [AuthGuard],
                                    providers: [
                                        {
                                            provide: UpdateEmailComponentService,
                                            useClass: UpdateEmailComponentService,
                                            deps: [
                                                UserEmailFacade,
                                                RoutingService,
                                                GlobalMessageService,
                                                AuthService,
                                                AuthRedirectService,
                                            ],
                                        },
                                    ],
                                },
                            },
                        }),
                    ],
                    declarations: [UpdateEmailComponent],
                }]
        }] });
