// @spartacus/storefront/esm2015/cms-components/myaccount/order/order-details/order-details.service.js
import { Injectable } from '@angular/core';
import { getLastValueSync, } from '@spartacus/core';
import { distinctUntilChanged, map, shareReplay, switchMap, tap, } from 'rxjs/operators';
import { OrderDetailsServiceTransitionalToken } from '../order-transitional-tokens';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class OrderDetailsService {
    constructor(userOrderService, routingService, unifiedInjector) {
        this.userOrderService = userOrderService;
        this.routingService = routingService;
        this.unifiedInjector = unifiedInjector;
        this.orderCode$ = this.routingService.getRouterState().pipe(map((routingData) => routingData.state.params.orderCode), distinctUntilChanged());
        this.orderLoad$ = this.orderCode$.pipe(tap((orderCode) => {
            if (orderCode) {
                this.userOrderService.loadOrderDetails(orderCode);
            }
            else {
                this.userOrderService.clearOrderDetails();
            }
        }), shareReplay({ bufferSize: 1, refCount: true }));
    }
    getOrderDetails() {
        if (this.unifiedInjector) {
            const serivce = getLastValueSync(this.unifiedInjector.get(OrderDetailsServiceTransitionalToken));
            if (serivce) {
                return serivce.getOrderDetails();
            }
        }
        return this.orderLoad$.pipe(switchMap(() => this.userOrderService.getOrderDetails()));
    }
}
OrderDetailsService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailsService, deps: [{ token: i1.UserOrderService }, { token: i1.RoutingService }, { token: i1.UnifiedInjector }], target: i0.ɵɵFactoryTarget.Injectable });
OrderDetailsService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailsService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.UserOrderService }, { type: i1.RoutingService }, { type: i1.UnifiedInjector }]; } });
