// @spartacus/storefront/esm2015/layout/theme/theme.module.js
import { APP_BOOTSTRAP_LISTENER, NgModule } from '@angular/core';
import { ThemeService } from './theme.service';
import * as i0 from "@angular/core";
export function initTheme(themeService) {
    const result = (component) => themeService.init(component);
    return result;
}
export class ThemeModule {
}
ThemeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ThemeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThemeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ThemeModule });
ThemeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ThemeModule, providers: [
        {
            provide: APP_BOOTSTRAP_LISTENER,
            multi: true,
            useFactory: initTheme,
            deps: [ThemeService],
        },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ThemeModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        {
                            provide: APP_BOOTSTRAP_LISTENER,
                            multi: true,
                            useFactory: initTheme,
                            deps: [ThemeService],
                        },
                    ],
                }]
        }] });
