// @spartacus/storefront/esm2015/cms-structure/seo/seo.module.js
import { APP_INITIALIZER, Injector, NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';
import { defaultSeoConfig } from './config';
import { htmlLangProvider } from './html-lang-provider';
import { SeoMetaService } from './seo-meta.service';
import { StructuredDataModule } from './structured-data/structured-data.module';
import * as i0 from "@angular/core";
export function initSeoService(injector) {
    const result = () => {
        const service = injector.get(SeoMetaService);
        service.init();
    };
    return result;
}
export class SeoModule {
}
SeoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SeoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SeoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SeoModule, imports: [StructuredDataModule] });
SeoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SeoModule, providers: [
        provideDefaultConfig(defaultSeoConfig),
        {
            provide: APP_INITIALIZER,
            useFactory: initSeoService,
            deps: [Injector],
            multi: true,
        },
        htmlLangProvider,
    ], imports: [[StructuredDataModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SeoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [StructuredDataModule],
                    providers: [
                        provideDefaultConfig(defaultSeoConfig),
                        {
                            provide: APP_INITIALIZER,
                            useFactory: initSeoService,
                            deps: [Injector],
                            multi: true,
                        },
                        htmlLangProvider,
                    ],
                }]
        }] });
