// @spartacus/storefinder/schematics/add-storefinder/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addStorefinderFeatures = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const schematics_2 = require("@spartacus/schematics");
const package_json_1 = require("../../package.json");
const constants_1 = require("../constants");
function addStorefinderFeatures(options) {
    return (tree, _context) => {
        const packageJson = schematics_2.readPackageJson(tree);
        schematics_2.validateSpartacusInstallation(packageJson);
        return schematics_1.chain([
            schematics_2.addPackageJsonDependenciesForLibrary(package_json_1.peerDependencies, options),
            schematics_2.shouldAddFeature(schematics_2.CLI_STOREFINDER_FEATURE, options.features)
                ? addStorefinderFeature(options)
                : schematics_1.noop(),
        ]);
    };
}
exports.addStorefinderFeatures = addStorefinderFeatures;
function addStorefinderFeature(options) {
    return schematics_2.addLibraryFeature(options, {
        folderName: constants_1.STOREFINDER_FOLDER_NAME,
        moduleName: constants_1.STOREFINDER_MODULE_NAME,
        featureModule: {
            name: schematics_2.STOREFINDER_MODULE,
            importPath: schematics_2.SPARTACUS_STOREFINDER,
        },
        rootModule: {
            name: constants_1.STOREFINDER_ROOT_MODULE,
            importPath: constants_1.SPARTACUS_STOREFINDER_ROOT,
        },
        lazyLoadingChunk: {
            moduleSpecifier: constants_1.SPARTACUS_STOREFINDER_ROOT,
            namedImports: [constants_1.STOREFINDER_FEATURE_NAME_CONSTANT],
        },
        i18n: {
            resources: constants_1.STOREFINDER_TRANSLATIONS,
            chunks: constants_1.STOREFINDER_TRANSLATION_CHUNKS_CONFIG,
            importPath: constants_1.SPARTACUS_STOREFINDER_ASSETS,
        },
        styles: {
            scssFileName: schematics_2.STORE_FINDER_SCSS_FILE_NAME,
            importStyle: schematics_2.SPARTACUS_STOREFINDER,
        },
    });
}
