// @spartacus/storefront/esm2015/layout/a11y/keyboard-focus/focus.directive.js
import { Directive, Input } from '@angular/core';
import { LockFocusDirective } from './lock/lock-focus.directive';
import * as i0 from "@angular/core";
import * as i1 from "./services/keyboard-focus.service";
export class FocusDirective extends LockFocusDirective {
    constructor(elementRef, service, renderer) {
        super(elementRef, service, renderer);
        this.elementRef = elementRef;
        this.service = service;
        this.renderer = renderer;
        this.defaultConfig = {};
        this.config = {};
    }
}
FocusDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FocusDirective, deps: [{ token: i0.ElementRef }, { token: i1.KeyboardFocusService }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
FocusDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: FocusDirective, selector: "[cxFocus]", inputs: { config: ["cxFocus", "config"] }, usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FocusDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxFocus]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i1.KeyboardFocusService }, { type: i0.Renderer2 }]; }, propDecorators: { config: [{
                type: Input,
                args: ['cxFocus']
            }] } });
