// @spartacus/storefront/esm2015/layout/a11y/keyboard-focus/focus-testing.module.js
import { Directive, Input, NgModule } from '@angular/core';
import * as i0 from "@angular/core";
export class MockKeyboardFocusDirective {
    constructor() {
        this.config = {};
    }
}
MockKeyboardFocusDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MockKeyboardFocusDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MockKeyboardFocusDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: MockKeyboardFocusDirective, selector: "[cxFocus]", inputs: { config: ["cxFocus", "config"] }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MockKeyboardFocusDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxFocus]',
                }]
        }], propDecorators: { config: [{
                type: Input,
                args: ['cxFocus']
            }] } });
export class KeyboardFocusTestingModule {
}
KeyboardFocusTestingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusTestingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
KeyboardFocusTestingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusTestingModule, declarations: [MockKeyboardFocusDirective], exports: [MockKeyboardFocusDirective] });
KeyboardFocusTestingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusTestingModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusTestingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MockKeyboardFocusDirective],
                    exports: [MockKeyboardFocusDirective],
                }]
        }] });
