// @spartacus/storefront/esm2015/layout/a11y/keyboard-focus/keyboard-focus.module.js
import { CommonModule } from '@angular/common';
import { APP_INITIALIZER, NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ConfigModule, provideDefaultConfig } from '@spartacus/core';
import { defaultKeyboardFocusConfig } from './config/default-keyboard-focus.config';
import { FocusDirective } from './focus.directive';
import { OnNavigateFocusService } from './on-navigate/on-navigate-focus.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
const directives = [
    // PersistFocusDirective,
    // VisibleFocusDirective,
    // BlockFocusDirective,
    // AutoFocusDirective,
    // EscapeFocusDirective,
    // LockFocusDirective,
    // TrapFocusDirective,
    // TabFocusDirective,
    FocusDirective,
];
/**
 * @deprecated since 4.2, refer to spartacus issues (#13762)
 * Remove the router module as well as it was part of the old initiative for scroll positiioning
 */
export class KeyboardFocusModule {
}
KeyboardFocusModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
KeyboardFocusModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusModule, declarations: [
        // PersistFocusDirective,
        // VisibleFocusDirective,
        // BlockFocusDirective,
        // AutoFocusDirective,
        // EscapeFocusDirective,
        // LockFocusDirective,
        // TrapFocusDirective,
        // TabFocusDirective,
        FocusDirective], imports: [CommonModule, i1.ConfigModule, RouterModule], exports: [
        // PersistFocusDirective,
        // VisibleFocusDirective,
        // BlockFocusDirective,
        // AutoFocusDirective,
        // EscapeFocusDirective,
        // LockFocusDirective,
        // TrapFocusDirective,
        // TabFocusDirective,
        FocusDirective] });
KeyboardFocusModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusModule, providers: [
        provideDefaultConfig(defaultKeyboardFocusConfig),
        {
            provide: APP_INITIALIZER,
            useFactory: keyboardFocusFactory,
            deps: [OnNavigateFocusService],
            multi: true,
        },
    ], imports: [[
            CommonModule,
            ConfigModule.withConfig(defaultKeyboardFocusConfig),
            RouterModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: KeyboardFocusModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ConfigModule.withConfig(defaultKeyboardFocusConfig),
                        RouterModule,
                    ],
                    declarations: [...directives],
                    providers: [
                        provideDefaultConfig(defaultKeyboardFocusConfig),
                        {
                            provide: APP_INITIALIZER,
                            useFactory: keyboardFocusFactory,
                            deps: [OnNavigateFocusService],
                            multi: true,
                        },
                    ],
                    exports: [...directives],
                }]
        }] });
/**
 *  @deprecated since 4.2, refer to spartacus issues (#13762)
 *  Start keyboard focus services on app initialization.
 */
export function keyboardFocusFactory(onNavigateFocusService) {
    const isReady = () => onNavigateFocusService.initializeWithConfig();
    return isReady;
}
