// @spartacus/storefront/esm2015/shared/components/table/table-data-cell/table-data-cell.component.js
import { ChangeDetectionStrategy, Component, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../../../cms-structure/outlet/outlet.model";
export class TableDataCellComponent {
    constructor(outlet) {
        this.outlet = outlet;
    }
    get value() {
        return this.model[this.field];
    }
    get model() {
        var _a;
        return (_a = this.outlet) === null || _a === void 0 ? void 0 : _a.context;
    }
    get field() {
        var _a, _b;
        return (_b = (_a = this.outlet) === null || _a === void 0 ? void 0 : _a.context) === null || _b === void 0 ? void 0 : _b._field;
    }
}
TableDataCellComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: TableDataCellComponent, deps: [{ token: i1.OutletContextData }], target: i0.ɵɵFactoryTarget.Component });
TableDataCellComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: TableDataCellComponent, selector: "cx-table-data-cell", host: { properties: { "attr.title": "this.value" } }, ngImport: i0, template: `{{ value }}`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: TableDataCellComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-table-data-cell',
                    template: `{{ value }}`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.OutletContextData }]; }, propDecorators: { value: [{
                type: HostBinding,
                args: ['attr.title']
            }] } });
