// @spartacus/asm/schematics/add-asm/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addAsmFeatures = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const schematics_2 = require("@spartacus/schematics");
const package_json_1 = require("../../package.json");
const constants_1 = require("../constants");
function addAsmFeatures(options) {
    return (tree, _context) => {
        const packageJson = schematics_2.readPackageJson(tree);
        schematics_2.validateSpartacusInstallation(packageJson);
        return schematics_1.chain([
            schematics_2.addPackageJsonDependenciesForLibrary(package_json_1.peerDependencies, options),
            schematics_2.shouldAddFeature(schematics_2.CLI_ASM_FEATURE, options.features)
                ? addAsmFeature(options)
                : schematics_1.noop(),
        ]);
    };
}
exports.addAsmFeatures = addAsmFeatures;
function addAsmFeature(options) {
    return schematics_2.addLibraryFeature(options, {
        folderName: constants_1.ASM_FOLDER_NAME,
        moduleName: constants_1.ASM_MODULE_NAME,
        featureModule: {
            name: constants_1.ASM_MODULE,
            importPath: schematics_2.SPARTACUS_ASM,
        },
        rootModule: {
            name: constants_1.ASM_ROOT_MODULE,
            importPath: constants_1.SPARTACUS_ASM_ROOT,
        },
        lazyLoadingChunk: {
            moduleSpecifier: constants_1.SPARTACUS_ASM_ROOT,
            namedImports: [constants_1.ASM_FEATURE_NAME_CONSTANT],
        },
        styles: {
            scssFileName: constants_1.SCSS_FILE_NAME,
            importStyle: schematics_2.SPARTACUS_ASM,
        },
        i18n: {
            resources: constants_1.ASM_TRANSLATIONS,
            chunks: constants_1.ASM_TRANSLATION_CHUNKS_CONFIG,
            importPath: constants_1.SPARTACUS_ASM_ASSETS,
        },
    });
}
