// @spartacus/storefinder/bundles/spartacus-storefinder-occ.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('rxjs/operators'), require('@spartacus/storefinder/core'), require('@spartacus/core'), require('@angular/common/http')) :
    typeof define === 'function' && define.amd ? define('@spartacus/storefinder/occ', ['exports', '@angular/core', 'rxjs/operators', '@spartacus/storefinder/core', '@spartacus/core', '@angular/common/http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.storefinder = global.spartacus.storefinder || {}, global.spartacus.storefinder.occ = {}), global.ng.core, global.rxjs.operators, global.spartacus.storefinder.core, global.core, global.ng.common.http));
}(this, (function (exports, i0, operators, core, i2, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var OccStoreFinderAdapter = /** @class */ (function () {
        function OccStoreFinderAdapter(http, occEndpointsService, converterService) {
            this.http = http;
            this.occEndpointsService = occEndpointsService;
            this.converterService = converterService;
        }
        OccStoreFinderAdapter.prototype.search = function (query, searchConfig, longitudeLatitude, radius) {
            return this.callOccFindStores(query, searchConfig, longitudeLatitude, radius).pipe(this.converterService.pipeable(core.STORE_FINDER_SEARCH_PAGE_NORMALIZER));
        };
        OccStoreFinderAdapter.prototype.loadCounts = function () {
            return this.http
                .get(this.occEndpointsService.buildUrl('storescounts'))
                .pipe(operators.map(function (_a) {
                var countriesAndRegionsStoreCount = _a.countriesAndRegionsStoreCount;
                return countriesAndRegionsStoreCount;
            }), this.converterService.pipeableMany(core.STORE_COUNT_NORMALIZER));
        };
        OccStoreFinderAdapter.prototype.load = function (storeId) {
            return this.http
                .get(this.occEndpointsService.buildUrl('store', { urlParams: { storeId: storeId } }))
                .pipe(this.converterService.pipeable(i2.POINT_OF_SERVICE_NORMALIZER));
        };
        OccStoreFinderAdapter.prototype.callOccFindStores = function (query, searchConfig, longitudeLatitude, radius) {
            var params = {};
            if (longitudeLatitude) {
                params['longitude'] = String(longitudeLatitude.longitude);
                params['latitude'] = String(longitudeLatitude.latitude);
                params['radius'] = String(radius);
            }
            else {
                params['query'] = query;
            }
            if (searchConfig.pageSize) {
                params['pageSize'] = String(searchConfig.pageSize);
            }
            if (searchConfig.currentPage) {
                params['currentPage'] = String(searchConfig.currentPage);
            }
            if (searchConfig.sort) {
                params['sort'] = searchConfig.sort;
            }
            return this.http.get(this.occEndpointsService.buildUrl('stores', { queryParams: params }));
        };
        return OccStoreFinderAdapter;
    }());
    OccStoreFinderAdapter.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccStoreFinderAdapter, deps: [{ token: i1__namespace.HttpClient }, { token: i2__namespace.OccEndpointsService }, { token: i2__namespace.ConverterService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    OccStoreFinderAdapter.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccStoreFinderAdapter });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccStoreFinderAdapter, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.HttpClient }, { type: i2__namespace.OccEndpointsService }, { type: i2__namespace.ConverterService }]; } });

    // Imported for side effects (module augmentation)

    var defaultOccStoreFinderConfig = {
        backend: {
            occ: {
                endpoints: {
                    store: 'stores/${storeId}?fields=FULL',
                    stores: 'stores?fields=stores(name,displayName,formattedDistance,openingHours(weekDayOpeningList(FULL),specialDayOpeningList(FULL)),geoPoint(latitude,longitude),address(line1,line2,town,region(FULL),postalCode,phone,country,email), features),pagination(DEFAULT),sorts(DEFAULT)',
                    storescounts: 'stores/storescounts',
                },
            },
        },
    };

    var StoreFinderOccModule = /** @class */ (function () {
        function StoreFinderOccModule() {
        }
        return StoreFinderOccModule;
    }());
    StoreFinderOccModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderOccModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    StoreFinderOccModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderOccModule });
    StoreFinderOccModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderOccModule, providers: [
            i2.provideDefaultConfig(defaultOccStoreFinderConfig),
            { provide: core.StoreFinderAdapter, useClass: OccStoreFinderAdapter },
        ] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderOccModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        providers: [
                            i2.provideDefaultConfig(defaultOccStoreFinderConfig),
                            { provide: core.StoreFinderAdapter, useClass: OccStoreFinderAdapter },
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.OccStoreFinderAdapter = OccStoreFinderAdapter;
    exports.StoreFinderOccModule = StoreFinderOccModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
