// @spartacus/storefinder/bundles/spartacus-storefinder-assets.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define('@spartacus/storefinder/assets', ['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.storefinder = global.spartacus.storefinder || {}, global.spartacus.storefinder.assets = {})));
}(this, (function (exports) { 'use strict';

    var storeFinder = {
        storeFinder: {
            openUntil: 'Open until',
            closed: 'Closed',
            back: 'Back',
            call: 'Call',
            getDirections: 'Get Directions',
            ariaLabelGetDirections: 'Get Directions, this link opens in a new window',
            listView: 'List View',
            mapView: 'Map View',
            noStoresFound: 'No Stores Found.',
            storeHours: 'Store hours',
            storeFeatures: 'Store features',
            fromStoresFound: 'from {{ count }} store found',
            fromStoresFound_plural: 'from {{ count }} stores found',
            findStore: 'Find store',
            useMyLocation: 'Use my location',
            viewAllStores: 'View all stores',
            contactUs: 'Contact us',
            searchBox: 'Enter postal code, town or address',
            backToList: 'Back to list',
            noStoresMessage: 'No stores found in database...',
            geolocationNotEnabled: 'Cannot locate nearby stores. Geolocation consent has not been enabled',
        },
    };

    var en = {
        storeFinder: storeFinder,
    };

    var storeFinderTranslations = {
        en: en,
    };
    var storeFinderTranslationChunksConfig = {
        storeFinder: ['storeFinder'],
    };

    /**
     * Generated bundle index. Do not edit.
     */

    exports.storeFinderTranslationChunksConfig = storeFinderTranslationChunksConfig;
    exports.storeFinderTranslations = storeFinderTranslations;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
