// @spartacus/storefinder/bundles/spartacus-storefinder-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/storefinder/root', ['exports', '@angular/core', '@spartacus/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.storefinder = global.spartacus.storefinder || {}, global.spartacus.storefinder.root = {}), global.ng.core, global.core));
}(this, (function (exports, i0, core) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var defaultStoreFinderLayoutConfig = {
        layoutSlots: {
            StoreFinderPageTemplate: {
                slots: ['MiddleContent', 'SideContent'],
            },
        },
    };

    var STORE_FINDER_FEATURE = 'storeFinder';

    // TODO: Inline this factory when we start releasing Ivy compiled libraries
    function defaultStoreFinderComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[STORE_FINDER_FEATURE] = {
                    cmsComponents: ['StoreFinderComponent'],
                },
                _a),
        };
        return config;
    }
    var StoreFinderRootModule = /** @class */ (function () {
        function StoreFinderRootModule() {
        }
        return StoreFinderRootModule;
    }());
    StoreFinderRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    StoreFinderRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderRootModule });
    StoreFinderRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderRootModule, providers: [
            core.provideDefaultConfig(defaultStoreFinderLayoutConfig),
            core.provideDefaultConfigFactory(defaultStoreFinderComponentsConfig),
        ] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: StoreFinderRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [],
                        providers: [
                            core.provideDefaultConfig(defaultStoreFinderLayoutConfig),
                            core.provideDefaultConfigFactory(defaultStoreFinderComponentsConfig),
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.STORE_FINDER_FEATURE = STORE_FINDER_FEATURE;
    exports.StoreFinderRootModule = StoreFinderRootModule;
    exports.defaultStoreFinderComponentsConfig = defaultStoreFinderComponentsConfig;
    exports.defaultStoreFinderLayoutConfig = defaultStoreFinderLayoutConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
