// @spartacus/storefront/esm2015/cms-components/product/carousel/product-references/product-references.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { provideDefaultConfig, UrlModule } from '@spartacus/core';
import { CarouselModule } from '../../../../shared/components/carousel/carousel.module';
import { MediaModule } from '../../../../shared/components/media/media.module';
import { ProductReferencesComponent } from './product-references.component';
import * as i0 from "@angular/core";
export class ProductReferencesModule {
}
ProductReferencesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductReferencesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductReferencesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductReferencesModule, declarations: [ProductReferencesComponent], imports: [CommonModule, CarouselModule, MediaModule, RouterModule, UrlModule], exports: [ProductReferencesComponent] });
ProductReferencesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductReferencesModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                ProductReferencesComponent: {
                    component: ProductReferencesComponent,
                },
            },
        }),
    ], imports: [[CommonModule, CarouselModule, MediaModule, RouterModule, UrlModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductReferencesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CarouselModule, MediaModule, RouterModule, UrlModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ProductReferencesComponent: {
                                    component: ProductReferencesComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [ProductReferencesComponent],
                    exports: [ProductReferencesComponent],
                }]
        }] });
