// @spartacus/storefront/esm2015/layout/header/hamburger-menu/hamburger-menu.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { HamburgerMenuComponent } from './hamburger-menu.component';
import * as i0 from "@angular/core";
export class HamburgerMenuModule {
}
HamburgerMenuModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: HamburgerMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
HamburgerMenuModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: HamburgerMenuModule, declarations: [HamburgerMenuComponent], imports: [CommonModule, I18nModule], exports: [HamburgerMenuComponent] });
HamburgerMenuModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: HamburgerMenuModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                HamburgerMenuComponent: {
                    component: HamburgerMenuComponent,
                },
            },
        }),
    ], imports: [[CommonModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: HamburgerMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                HamburgerMenuComponent: {
                                    component: HamburgerMenuComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [HamburgerMenuComponent],
                    exports: [HamburgerMenuComponent],
                }]
        }] });
