// @spartacus/storefront/esm2015/shared/components/list-navigation/list-navigation.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgSelectModule } from '@ng-select/ng-select';
import { IconModule } from '../../../cms-components/misc/icon/icon.module';
import { PaginationComponent, PaginationModule } from './pagination/index';
import { SortingComponent } from './sorting/sorting.component';
import * as i0 from "@angular/core";
export class ListNavigationModule {
}
ListNavigationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListNavigationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ListNavigationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListNavigationModule, declarations: [SortingComponent], imports: [CommonModule,
        NgSelectModule,
        FormsModule,
        ReactiveFormsModule,
        IconModule,
        PaginationModule], exports: [SortingComponent, PaginationComponent] });
ListNavigationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListNavigationModule, imports: [[
            CommonModule,
            NgSelectModule,
            FormsModule,
            ReactiveFormsModule,
            IconModule,
            PaginationModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListNavigationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        NgSelectModule,
                        FormsModule,
                        ReactiveFormsModule,
                        IconModule,
                        PaginationModule,
                    ],
                    declarations: [SortingComponent],
                    exports: [SortingComponent, PaginationComponent],
                }]
        }] });
