// @spartacus/storefront/esm2015/cms-components/user/login-route/login-route.module.js
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PageLayoutComponent, PageLayoutModule, } from '../../../cms-structure/page/index';
import { LoginGuard } from './login.guard';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
/**
 * This module enables to quickly switch from Resource Owner Password Flow
 * to Implicit Flow or Authorization Code Flow. The `login` route in this case will be
 * responsible for initalizing the redirect to OAuth server to login.
 *
 * Instead of manually invoking OAuth redirect you only have to redirect to `login` page.
 */
export class LoginRouteModule {
}
LoginRouteModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LoginRouteModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LoginRouteModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LoginRouteModule, imports: [PageLayoutModule, i1.RouterModule] });
LoginRouteModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LoginRouteModule, imports: [[
            PageLayoutModule,
            RouterModule.forChild([
                {
                    path: null,
                    canActivate: [LoginGuard],
                    component: PageLayoutComponent,
                    data: { cxRoute: 'login' },
                },
            ]),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LoginRouteModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        PageLayoutModule,
                        RouterModule.forChild([
                            {
                                path: null,
                                canActivate: [LoginGuard],
                                component: PageLayoutComponent,
                                data: { cxRoute: 'login' },
                            },
                        ]),
                    ],
                }]
        }] });
