// @spartacus/storefront/esm2015/cms-components/cart/cart.module.js
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { CartModule } from '@spartacus/core';
import { CmsPageGuard } from '../../cms-structure/guards/cms-page.guard';
import { PAGE_LAYOUT_HANDLER } from '../../cms-structure/page/page-layout/page-layout-handler';
import { PageLayoutComponent } from '../../cms-structure/page/page-layout/page-layout.component';
import { AddToCartModule } from './add-to-cart/add-to-cart.module';
import { AddToWishListModule } from './add-to-wishlist/add-to-wish-list.module';
import { CartDetailsModule } from './cart-details/cart-details.module';
import { CartPageLayoutHandler } from './cart-page-layout-handler';
import { ActiveCartOrderEntriesContext } from './cart-page/active-cart-order-entries-context';
import { CartSharedModule } from './cart-shared/cart-shared.module';
import { CartTotalsModule } from './cart-totals/cart-totals.module';
import { MiniCartModule } from './mini-cart/mini-cart.module';
import { ORDER_ENTRIES_CONTEXT } from './order-entries-context/order-entires.context';
import { SaveForLaterModule } from './save-for-later/save-for-later.module';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class CartComponentModule {
}
CartComponentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartComponentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CartComponentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartComponentModule, imports: [NgbModule,
        CartDetailsModule,
        CartTotalsModule,
        CartSharedModule,
        SaveForLaterModule, i1.RouterModule], exports: [AddToWishListModule,
        CartDetailsModule,
        CartTotalsModule,
        CartSharedModule,
        AddToCartModule,
        MiniCartModule,
        CartModule,
        SaveForLaterModule] });
CartComponentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartComponentModule, providers: [
        {
            provide: PAGE_LAYOUT_HANDLER,
            useExisting: CartPageLayoutHandler,
            multi: true,
        },
    ], imports: [[
            NgbModule,
            CartDetailsModule,
            CartTotalsModule,
            CartSharedModule,
            SaveForLaterModule,
            RouterModule.forChild([
                {
                    // @ts-ignore
                    path: null,
                    canActivate: [CmsPageGuard],
                    component: PageLayoutComponent,
                    data: {
                        cxRoute: 'cart',
                        cxContext: {
                            [ORDER_ENTRIES_CONTEXT]: ActiveCartOrderEntriesContext,
                        },
                    },
                },
            ]),
        ], AddToWishListModule,
        CartDetailsModule,
        CartTotalsModule,
        CartSharedModule,
        AddToCartModule,
        MiniCartModule,
        CartModule,
        SaveForLaterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartComponentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        NgbModule,
                        CartDetailsModule,
                        CartTotalsModule,
                        CartSharedModule,
                        SaveForLaterModule,
                        RouterModule.forChild([
                            {
                                // @ts-ignore
                                path: null,
                                canActivate: [CmsPageGuard],
                                component: PageLayoutComponent,
                                data: {
                                    cxRoute: 'cart',
                                    cxContext: {
                                        [ORDER_ENTRIES_CONTEXT]: ActiveCartOrderEntriesContext,
                                    },
                                },
                            },
                        ]),
                    ],
                    exports: [
                        AddToWishListModule,
                        CartDetailsModule,
                        CartTotalsModule,
                        CartSharedModule,
                        AddToCartModule,
                        MiniCartModule,
                        CartModule,
                        SaveForLaterModule,
                    ],
                    declarations: [],
                    providers: [
                        {
                            provide: PAGE_LAYOUT_HANDLER,
                            useExisting: CartPageLayoutHandler,
                            multi: true,
                        },
                    ],
                }]
        }] });
