// @spartacus/checkout/esm2015/core/checkout-core.module.js
import { NgModule } from '@angular/core';
import { PageMetaResolver } from '@spartacus/core';
import { CheckoutConnector } from './connectors/checkout/checkout.connector';
import { CheckoutCostCenterConnector } from './connectors/cost-center/checkout-cost-center.connector';
import { CheckoutDeliveryConnector } from './connectors/delivery/checkout-delivery.connector';
import { PaymentTypeConnector } from './connectors/payment-type/payment-type.connector';
import { CheckoutPaymentConnector } from './connectors/payment/checkout-payment.connector';
import { CheckoutReplenishmentOrderConnector } from './connectors/replenishment-order/checkout-replenishment-order.connector';
import { CheckoutEventBuilder } from './events/checkout-event.builder';
import { CheckoutEventModule } from './events/checkout-event.module';
import { facadeProviders } from './facade/facade-providers';
import { CheckoutPageMetaResolver } from './services/checkout-page-meta.resolver';
import { CheckoutStoreModule } from './store/checkout-store.module';
import * as i0 from "@angular/core";
export class CheckoutCoreModule {
}
CheckoutCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CheckoutCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutCoreModule, imports: [CheckoutStoreModule, CheckoutEventModule] });
CheckoutCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutCoreModule, providers: [
        ...facadeProviders,
        CheckoutDeliveryConnector,
        CheckoutCostCenterConnector,
        CheckoutConnector,
        CheckoutPaymentConnector,
        PaymentTypeConnector,
        CheckoutReplenishmentOrderConnector,
        CheckoutEventBuilder,
        CheckoutPageMetaResolver,
        {
            provide: PageMetaResolver,
            useExisting: CheckoutPageMetaResolver,
            multi: true,
        },
    ], imports: [[CheckoutStoreModule, CheckoutEventModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CheckoutStoreModule, CheckoutEventModule],
                    providers: [
                        ...facadeProviders,
                        CheckoutDeliveryConnector,
                        CheckoutCostCenterConnector,
                        CheckoutConnector,
                        CheckoutPaymentConnector,
                        PaymentTypeConnector,
                        CheckoutReplenishmentOrderConnector,
                        CheckoutEventBuilder,
                        CheckoutPageMetaResolver,
                        {
                            provide: PageMetaResolver,
                            useExisting: CheckoutPageMetaResolver,
                            multi: true,
                        },
                    ],
                }]
        }] });
