// @spartacus/storefront/esm2015/cms-components/product/product-tabs/product-attributes/product-attributes.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { ProductAttributesComponent } from './product-attributes.component';
import * as i0 from "@angular/core";
export class ProductAttributesModule {
}
ProductAttributesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductAttributesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductAttributesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductAttributesModule, declarations: [ProductAttributesComponent], imports: [CommonModule, I18nModule], exports: [ProductAttributesComponent] });
ProductAttributesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductAttributesModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                ProductSpecsTabComponent: {
                    component: ProductAttributesComponent,
                },
            },
        }),
    ], imports: [[CommonModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductAttributesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ProductSpecsTabComponent: {
                                    component: ProductAttributesComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [ProductAttributesComponent],
                    exports: [ProductAttributesComponent],
                }]
        }] });
