// @spartacus/order/esm2015/occ/order-occ.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';
import { OrderAdapter, ORDER_RETURN_REQUEST_NORMALIZER, ReplenishmentOrderAdapter, } from '@spartacus/order/core';
import { OccReturnRequestNormalizer } from './adapters/converters/occ-return-request-normalizer';
import { OccOrderAdapter } from './adapters/occ-order.adapter';
import { OccReplenishmentOrderAdapter } from './adapters/occ-replenishment-order.adapter';
import { defaultOccOrderConfig } from './config/default-occ-order-config';
import * as i0 from "@angular/core";
export class OrderOccModule {
}
OrderOccModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrderOccModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOccModule, imports: [CommonModule] });
OrderOccModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOccModule, providers: [
        provideDefaultConfig(defaultOccOrderConfig),
        { provide: OrderAdapter, useClass: OccOrderAdapter },
        {
            provide: ReplenishmentOrderAdapter,
            useClass: OccReplenishmentOrderAdapter,
        },
        {
            provide: ORDER_RETURN_REQUEST_NORMALIZER,
            useExisting: OccReturnRequestNormalizer,
            multi: true,
        },
    ], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccOrderConfig),
                        { provide: OrderAdapter, useClass: OccOrderAdapter },
                        {
                            provide: ReplenishmentOrderAdapter,
                            useClass: OccReplenishmentOrderAdapter,
                        },
                        {
                            provide: ORDER_RETURN_REQUEST_NORMALIZER,
                            useExisting: OccReturnRequestNormalizer,
                            multi: true,
                        },
                    ],
                }]
        }] });
