// @spartacus/storefront/esm2015/cms-components/cart/validation/guards/cart-validation.guard.js
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { GlobalMessageType, CartValidationStatusCode, } from '@spartacus/core';
import { map, withLatestFrom } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@angular/router";
import * as i3 from "../cart-validation-state.service";
export class CartValidationGuard {
    constructor(cartValidationService, semanticPathService, router, globalMessageService, activeCartService, cartValidationStateService, cartConfigService) {
        this.cartValidationService = cartValidationService;
        this.semanticPathService = semanticPathService;
        this.router = router;
        this.globalMessageService = globalMessageService;
        this.activeCartService = activeCartService;
        this.cartValidationStateService = cartValidationStateService;
        this.cartConfigService = cartConfigService;
        this.GLOBAL_MESSAGE_TIMEOUT = 10000;
    }
    canActivate() {
        return !this.cartConfigService.isCartValidationEnabled()
            ? of(true)
            : this.cartValidationService.validateCart().pipe(withLatestFrom(this.activeCartService.getEntries()), map(([cartModificationList, cartEntries]) => {
                var _a;
                this.cartValidationStateService.updateValidationResultAndRoutingId(cartModificationList === null || cartModificationList === void 0 ? void 0 : cartModificationList.cartModifications);
                if (((_a = cartModificationList === null || cartModificationList === void 0 ? void 0 : cartModificationList.cartModifications) === null || _a === void 0 ? void 0 : _a.length) !== 0) {
                    let validationResultMessage;
                    if (cartEntries.length === 1 &&
                        cartEntries[0].product.code ===
                            (cartModificationList === null || cartModificationList === void 0 ? void 0 : cartModificationList.cartModifications[0].entry.product.code) &&
                        (cartModificationList === null || cartModificationList === void 0 ? void 0 : cartModificationList.cartModifications[0].statusCode) ===
                            CartValidationStatusCode.NO_STOCK) {
                        validationResultMessage = {
                            key: 'validation.cartEntryRemoved',
                            params: {
                                name: cartModificationList === null || cartModificationList === void 0 ? void 0 : cartModificationList.cartModifications[0].entry.product.name,
                            },
                        };
                    }
                    else {
                        validationResultMessage = {
                            key: 'validation.cartEntriesChangeDuringCheckout',
                        };
                    }
                    this.globalMessageService.add(validationResultMessage, GlobalMessageType.MSG_TYPE_ERROR, this.GLOBAL_MESSAGE_TIMEOUT);
                    this.activeCartService.reloadActiveCart();
                    return this.router.parseUrl(this.semanticPathService.get('cart'));
                }
                return true;
            }));
    }
}
CartValidationGuard.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationGuard, deps: [{ token: i1.CartValidationService }, { token: i1.SemanticPathService }, { token: i2.Router }, { token: i1.GlobalMessageService }, { token: i1.ActiveCartService }, { token: i3.CartValidationStateService }, { token: i1.CartConfigService }], target: i0.ɵɵFactoryTarget.Injectable });
CartValidationGuard.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationGuard, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationGuard, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CartValidationService }, { type: i1.SemanticPathService }, { type: i2.Router }, { type: i1.GlobalMessageService }, { type: i1.ActiveCartService }, { type: i3.CartValidationStateService }, { type: i1.CartConfigService }]; } });
