// @spartacus/checkout/bundles/spartacus-checkout-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/router'), require('@spartacus/core'), require('@spartacus/storefront'), require('@angular/common/http'), require('rxjs'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('@spartacus/checkout/root', ['exports', '@angular/core', '@angular/router', '@spartacus/core', '@spartacus/storefront', '@angular/common/http', 'rxjs', 'rxjs/operators'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.checkout = global.spartacus.checkout || {}, global.spartacus.checkout.root = {}), global.ng.core, global.ng.router, global.core, global.storefront, global.ng.common.http, global.rxjs, global.rxjs.operators));
}(this, (function (exports, i0, i1$1, i1, storefront, http, rxjs, operators) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || from);
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    exports.CheckoutStepType = void 0;
    (function (CheckoutStepType) {
        CheckoutStepType["SHIPPING_ADDRESS"] = "shippingAddress";
        CheckoutStepType["DELIVERY_MODE"] = "deliveryMode";
        CheckoutStepType["PAYMENT_DETAILS"] = "paymentDetails";
        CheckoutStepType["REVIEW_ORDER"] = "reviewOrder";
        CheckoutStepType["PAYMENT_TYPE"] = "paymentType";
    })(exports.CheckoutStepType || (exports.CheckoutStepType = {}));
    var checkoutShippingSteps = [
        exports.CheckoutStepType.SHIPPING_ADDRESS,
        exports.CheckoutStepType.DELIVERY_MODE,
    ];
    var checkoutPaymentSteps = [
        exports.CheckoutStepType.PAYMENT_DETAILS,
        exports.CheckoutStepType.PAYMENT_TYPE,
        exports.CheckoutStepType.SHIPPING_ADDRESS,
    ];

    exports.DeliveryModePreferences = void 0;
    (function (DeliveryModePreferences) {
        DeliveryModePreferences["FREE"] = "FREE";
        DeliveryModePreferences["LEAST_EXPENSIVE"] = "LEAST_EXPENSIVE";
        DeliveryModePreferences["MOST_EXPENSIVE"] = "MOST_EXPENSIVE";
    })(exports.DeliveryModePreferences || (exports.DeliveryModePreferences = {}));
    var CheckoutConfig = /** @class */ (function () {
        function CheckoutConfig() {
        }
        return CheckoutConfig;
    }());
    CheckoutConfig.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutConfig, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CheckoutConfig.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutConfig, providedIn: 'root', useExisting: i1.Config });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutConfig, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useExisting: i1.Config,
                    }]
            }] });

    var defaultCheckoutConfig = {
        checkout: {
            steps: [
                {
                    id: 'shippingAddress',
                    name: 'checkoutProgress.shippingAddress',
                    routeName: 'checkoutShippingAddress',
                    type: [exports.CheckoutStepType.SHIPPING_ADDRESS],
                },
                {
                    id: 'deliveryMode',
                    name: 'checkoutProgress.deliveryMode',
                    routeName: 'checkoutDeliveryMode',
                    type: [exports.CheckoutStepType.DELIVERY_MODE],
                },
                {
                    id: 'paymentDetails',
                    name: 'checkoutProgress.paymentDetails',
                    routeName: 'checkoutPaymentDetails',
                    type: [exports.CheckoutStepType.PAYMENT_DETAILS],
                },
                {
                    id: 'reviewOrder',
                    name: 'checkoutProgress.reviewOrder',
                    routeName: 'checkoutReviewOrder',
                    type: [exports.CheckoutStepType.REVIEW_ORDER],
                },
            ],
            express: false,
            defaultDeliveryMode: [exports.DeliveryModePreferences.FREE],
            guest: false,
        },
    };

    var defaultCheckoutRoutingConfig = {
        routing: {
            routes: {
                checkoutLogin: { paths: ['checkout-login'], authFlow: true },
                checkout: { paths: ['checkout'] },
                checkoutPaymentType: { paths: ['checkout/payment-type'] },
                checkoutShippingAddress: { paths: ['checkout/shipping-address'] },
                checkoutDeliveryMode: { paths: ['checkout/delivery-mode'] },
                checkoutPaymentDetails: { paths: ['checkout/payment-details'] },
                checkoutReviewOrder: { paths: ['checkout/review-order'] },
                orderConfirmation: { paths: ['order-confirmation'] },
                replenishmentConfirmation: { paths: ['replenishment/confirmation'] },
            },
        },
    };

    var CHECKOUT_FEATURE = 'checkout';
    var CHECKOUT_CORE_FEATURE = 'checkoutCore';

    /**
     * Interceptor that handles "Cart not found" errors while a user is in a checkout step.
     *
     * When a user doing a checkout has a "Cart not found" error, he is redirected to checkout and the cart is reloaded.
     * If a "Cart not found" error happens and the user is not on checkout, this interceptor does not perform any actions.
     */
    var CheckoutCartInterceptor = /** @class */ (function () {
        function CheckoutCartInterceptor(routingService, multiCartService) {
            this.routingService = routingService;
            this.multiCartService = multiCartService;
        }
        CheckoutCartInterceptor.prototype.intercept = function (request, next) {
            var _this = this;
            return this.routingService.getRouterState().pipe(operators.take(1), operators.switchMap(function (state) {
                return next.handle(request).pipe(operators.catchError(function (response) {
                    var _a;
                    if (response instanceof http.HttpErrorResponse &&
                        _this.isUserInCheckoutRoute((_a = state.state) === null || _a === void 0 ? void 0 : _a.semanticRoute)) {
                        if (_this.isCartNotFoundError(response)) {
                            _this.routingService.go({ cxRoute: 'cart' });
                            var cartCode = _this.getCartIdFromError(response);
                            if (cartCode) {
                                _this.multiCartService.reloadCart(cartCode);
                            }
                        }
                    }
                    return rxjs.throwError(response);
                }));
            }));
        };
        /**
         * Returns true if the parameter semantic route is part of "checkout"
         * Checkout semantic routes:
         * checkout
         * checkoutPaymentType
         * CheckoutShippingAddress
         * checkoutDeliveryMode
         * checkoutPaymentDetails
         * checkoutReviewOrder
         * checkoutLogin
         * @param semanticRoute
         */
        CheckoutCartInterceptor.prototype.isUserInCheckoutRoute = function (semanticRoute) {
            var _a;
            return (_a = semanticRoute === null || semanticRoute === void 0 ? void 0 : semanticRoute.toLowerCase().startsWith('checkout')) !== null && _a !== void 0 ? _a : false;
        };
        /**
         * Checks of the error is for a cart not found, i.e. the cart doesn't exist anymore
         *
         * @param response
         */
        CheckoutCartInterceptor.prototype.isCartNotFoundError = function (response) {
            var _a, _b, _c, _d, _e, _f;
            return (response.status === 400 &&
                ((_c = (_b = (_a = response.error) === null || _a === void 0 ? void 0 : _a.errors) === null || _b === void 0 ? void 0 : _b[0]) === null || _c === void 0 ? void 0 : _c.type) === 'CartError' &&
                ((_f = (_e = (_d = response.error) === null || _d === void 0 ? void 0 : _d.errors) === null || _e === void 0 ? void 0 : _e[0]) === null || _f === void 0 ? void 0 : _f.reason) === 'notFound');
        };
        CheckoutCartInterceptor.prototype.getCartIdFromError = function (response) {
            var _a, _b, _c;
            return (_c = (_b = (_a = response.error) === null || _a === void 0 ? void 0 : _a.errors) === null || _b === void 0 ? void 0 : _b[0]) === null || _c === void 0 ? void 0 : _c.subject;
        };
        return CheckoutCartInterceptor;
    }());
    CheckoutCartInterceptor.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutCartInterceptor, deps: [{ token: i1__namespace.RoutingService }, { token: i1__namespace.MultiCartService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CheckoutCartInterceptor.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutCartInterceptor, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutCartInterceptor, decorators: [{
                type: i0.Injectable,
                args: [{ providedIn: 'root' }]
            }], ctorParameters: function () { return [{ type: i1__namespace.RoutingService }, { type: i1__namespace.MultiCartService }]; } });

    var interceptors = [
        {
            provide: http.HTTP_INTERCEPTORS,
            useExisting: CheckoutCartInterceptor,
            multi: true,
        },
    ];

    var CheckoutFacade = /** @class */ (function () {
        function CheckoutFacade() {
        }
        return CheckoutFacade;
    }());
    CheckoutFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CheckoutFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutFacade, providedIn: 'root', useFactory: function () { return i1.facadeFactory({
            facade: CheckoutFacade,
            feature: CHECKOUT_CORE_FEATURE,
            methods: [
                'placeOrder',
                'scheduleReplenishmentOrder',
                'getPlaceOrderLoading',
                'getPlaceOrderSuccess',
                'getPlaceOrderError',
                'clearPlaceOrderState',
                'clearCheckoutData',
                'clearCheckoutStep',
                'loadCheckoutDetails',
                'getCheckoutDetailsLoaded',
                'isLoading',
                'getOrderDetails',
                'setOrderType',
                'getCurrentOrderType',
            ],
            async: true,
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return i1.facadeFactory({
                            facade: CheckoutFacade,
                            feature: CHECKOUT_CORE_FEATURE,
                            methods: [
                                'placeOrder',
                                'scheduleReplenishmentOrder',
                                'getPlaceOrderLoading',
                                'getPlaceOrderSuccess',
                                'getPlaceOrderError',
                                'clearPlaceOrderState',
                                'clearCheckoutData',
                                'clearCheckoutStep',
                                'loadCheckoutDetails',
                                'getCheckoutDetailsLoaded',
                                'isLoading',
                                'getOrderDetails',
                                'setOrderType',
                                'getCurrentOrderType',
                            ],
                            async: true,
                        }); },
                    }]
            }] });

    var OrderConfirmationOrderEntriesContext = /** @class */ (function () {
        function OrderConfirmationOrderEntriesContext(checkoutService) {
            this.checkoutService = checkoutService;
            this.type = storefront.OrderEntriesSource.ORDER_CONFIRMATION;
        }
        OrderConfirmationOrderEntriesContext.prototype.getEntries = function () {
            return this.checkoutService
                .getOrderDetails()
                .pipe(operators.map(function (order) { var _a; return (_a = order === null || order === void 0 ? void 0 : order.entries) !== null && _a !== void 0 ? _a : []; }));
        };
        return OrderConfirmationOrderEntriesContext;
    }());
    OrderConfirmationOrderEntriesContext.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OrderConfirmationOrderEntriesContext, deps: [{ token: CheckoutFacade }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    OrderConfirmationOrderEntriesContext.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OrderConfirmationOrderEntriesContext, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OrderConfirmationOrderEntriesContext, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                    }]
            }], ctorParameters: function () { return [{ type: CheckoutFacade }]; } });

    var _a, _b;
    function defaultCheckoutComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[CHECKOUT_FEATURE] = {
                    cmsComponents: [
                        'CheckoutOrchestrator',
                        'CheckoutOrderSummary',
                        'CheckoutProgress',
                        'CheckoutProgressMobileBottom',
                        'CheckoutProgressMobileTop',
                        'CheckoutCostCenterComponent',
                        'CheckoutDeliveryMode',
                        'CheckoutPaymentDetails',
                        'CheckoutPaymentType',
                        'CheckoutPlaceOrder',
                        'CheckoutReviewOrder',
                        'CheckoutScheduleReplenishmentOrder',
                        'CheckoutShippingAddress',
                        'GuestCheckoutLoginComponent',
                        'OrderConfirmationThankMessageComponent',
                        'OrderConfirmationItemsComponent',
                        'OrderConfirmationTotalsComponent',
                        'OrderConfirmationOverviewComponent',
                        'ReplenishmentConfirmationMessageComponent',
                        'ReplenishmentConfirmationOverviewComponent',
                        'ReplenishmentConfirmationItemsComponent',
                        'ReplenishmentConfirmationTotalsComponent',
                    ],
                },
                // by default core is bundled together with components
                _a[CHECKOUT_CORE_FEATURE] = CHECKOUT_FEATURE,
                _a),
        };
        return config;
    }
    var CheckoutRootModule = /** @class */ (function () {
        function CheckoutRootModule() {
        }
        return CheckoutRootModule;
    }());
    CheckoutRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    CheckoutRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutRootModule, imports: [i1__namespace$1.RouterModule] });
    CheckoutRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutRootModule, providers: __spreadArray(__spreadArray([], __read(interceptors)), [
            i1.provideDefaultConfig(defaultCheckoutRoutingConfig),
            i1.provideDefaultConfig(defaultCheckoutConfig),
            i1.provideDefaultConfigFactory(defaultCheckoutComponentsConfig),
        ]), imports: [[
                i1$1.RouterModule.forChild([
                    {
                        // @ts-ignore
                        path: null,
                        canActivate: [storefront.CmsPageGuard],
                        component: storefront.PageLayoutComponent,
                        data: {
                            cxRoute: 'orderConfirmation',
                            cxContext: (_a = {},
                                _a[storefront.ORDER_ENTRIES_CONTEXT] = OrderConfirmationOrderEntriesContext,
                                _a),
                        },
                    },
                ]),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i1$1.RouterModule.forChild([
                                {
                                    // @ts-ignore
                                    path: null,
                                    canActivate: [storefront.CmsPageGuard],
                                    component: storefront.PageLayoutComponent,
                                    data: {
                                        cxRoute: 'orderConfirmation',
                                        cxContext: (_b = {},
                                            _b[storefront.ORDER_ENTRIES_CONTEXT] = OrderConfirmationOrderEntriesContext,
                                            _b),
                                    },
                                },
                            ]),
                        ],
                        providers: __spreadArray(__spreadArray([], __read(interceptors)), [
                            i1.provideDefaultConfig(defaultCheckoutRoutingConfig),
                            i1.provideDefaultConfig(defaultCheckoutConfig),
                            i1.provideDefaultConfigFactory(defaultCheckoutComponentsConfig),
                        ]),
                    }]
            }] });

    /**
     * Indicates that a user has successfully placed an order
     */
    var OrderPlacedEvent = /** @class */ (function (_super) {
        __extends(OrderPlacedEvent, _super);
        function OrderPlacedEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return OrderPlacedEvent;
    }(i1.CxEvent));
    /**
     * Event's type
     */
    OrderPlacedEvent.type = 'OrderPlacedEvent';

    var CheckoutCostCenterFacade = /** @class */ (function () {
        function CheckoutCostCenterFacade() {
        }
        return CheckoutCostCenterFacade;
    }());
    CheckoutCostCenterFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutCostCenterFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CheckoutCostCenterFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutCostCenterFacade, providedIn: 'root', useFactory: function () { return i1.facadeFactory({
            facade: CheckoutCostCenterFacade,
            feature: CHECKOUT_CORE_FEATURE,
            methods: ['setCostCenter', 'getCostCenter'],
            async: true,
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutCostCenterFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return i1.facadeFactory({
                            facade: CheckoutCostCenterFacade,
                            feature: CHECKOUT_CORE_FEATURE,
                            methods: ['setCostCenter', 'getCostCenter'],
                            async: true,
                        }); },
                    }]
            }] });

    var CheckoutDeliveryFacade = /** @class */ (function () {
        function CheckoutDeliveryFacade() {
        }
        return CheckoutDeliveryFacade;
    }());
    CheckoutDeliveryFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutDeliveryFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CheckoutDeliveryFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutDeliveryFacade, providedIn: 'root', useFactory: function () { return i1.facadeFactory({
            facade: CheckoutDeliveryFacade,
            feature: CHECKOUT_CORE_FEATURE,
            methods: [
                'getSupportedDeliveryModes',
                'getSelectedDeliveryMode',
                'getSelectedDeliveryModeCode',
                'getDeliveryAddress',
                'getSetDeliveryAddressProcess',
                'resetSetDeliveryAddressProcess',
                'getSetDeliveryModeProcess',
                'resetSetDeliveryModeProcess',
                'resetLoadSupportedDeliveryModesProcess',
                'getLoadSupportedDeliveryModeProcess',
                'clearCheckoutDeliveryModes',
                'createAndSetAddress',
                'loadSupportedDeliveryModes',
                'setDeliveryMode',
                'setDeliveryAddress',
                'clearCheckoutDeliveryAddress',
                'clearCheckoutDeliveryMode',
                'clearCheckoutDeliveryDetails',
            ],
            async: true,
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutDeliveryFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return i1.facadeFactory({
                            facade: CheckoutDeliveryFacade,
                            feature: CHECKOUT_CORE_FEATURE,
                            methods: [
                                'getSupportedDeliveryModes',
                                'getSelectedDeliveryMode',
                                'getSelectedDeliveryModeCode',
                                'getDeliveryAddress',
                                'getSetDeliveryAddressProcess',
                                'resetSetDeliveryAddressProcess',
                                'getSetDeliveryModeProcess',
                                'resetSetDeliveryModeProcess',
                                'resetLoadSupportedDeliveryModesProcess',
                                'getLoadSupportedDeliveryModeProcess',
                                'clearCheckoutDeliveryModes',
                                'createAndSetAddress',
                                'loadSupportedDeliveryModes',
                                'setDeliveryMode',
                                'setDeliveryAddress',
                                'clearCheckoutDeliveryAddress',
                                'clearCheckoutDeliveryMode',
                                'clearCheckoutDeliveryDetails',
                            ],
                            async: true,
                        }); },
                    }]
            }] });

    var CheckoutPaymentFacade = /** @class */ (function () {
        function CheckoutPaymentFacade() {
        }
        return CheckoutPaymentFacade;
    }());
    CheckoutPaymentFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutPaymentFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    CheckoutPaymentFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutPaymentFacade, providedIn: 'root', useFactory: function () { return i1.facadeFactory({
            facade: CheckoutPaymentFacade,
            feature: CHECKOUT_CORE_FEATURE,
            methods: [
                'getCardTypes',
                'getPaymentDetails',
                'getSetPaymentDetailsResultProcess',
                'resetSetPaymentDetailsProcess',
                'loadSupportedCardTypes',
                'createPaymentDetails',
                'setPaymentDetails',
                'paymentProcessSuccess',
            ],
            async: true,
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: CheckoutPaymentFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return i1.facadeFactory({
                            facade: CheckoutPaymentFacade,
                            feature: CHECKOUT_CORE_FEATURE,
                            methods: [
                                'getCardTypes',
                                'getPaymentDetails',
                                'getSetPaymentDetailsResultProcess',
                                'resetSetPaymentDetailsProcess',
                                'loadSupportedCardTypes',
                                'createPaymentDetails',
                                'setPaymentDetails',
                                'paymentProcessSuccess',
                            ],
                            async: true,
                        }); },
                    }]
            }] });

    var ClearCheckoutFacade = /** @class */ (function () {
        function ClearCheckoutFacade() {
        }
        return ClearCheckoutFacade;
    }());
    ClearCheckoutFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ClearCheckoutFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    ClearCheckoutFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ClearCheckoutFacade, providedIn: 'root', useFactory: function () { return i1.facadeFactory({
            facade: ClearCheckoutFacade,
            feature: CHECKOUT_CORE_FEATURE,
            methods: ['resetCheckoutProcesses'],
            async: true,
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ClearCheckoutFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return i1.facadeFactory({
                            facade: ClearCheckoutFacade,
                            feature: CHECKOUT_CORE_FEATURE,
                            methods: ['resetCheckoutProcesses'],
                            async: true,
                        }); },
                    }]
            }] });

    var PaymentTypeFacade = /** @class */ (function () {
        function PaymentTypeFacade() {
        }
        return PaymentTypeFacade;
    }());
    PaymentTypeFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: PaymentTypeFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    PaymentTypeFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: PaymentTypeFacade, providedIn: 'root', useFactory: function () { return i1.facadeFactory({
            facade: PaymentTypeFacade,
            feature: CHECKOUT_CORE_FEATURE,
            methods: [
                'getPaymentTypes',
                'loadPaymentTypes',
                'setPaymentType',
                'getSelectedPaymentType',
                'isAccountPayment',
                'getPoNumber',
            ],
            async: true,
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: PaymentTypeFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return i1.facadeFactory({
                            facade: PaymentTypeFacade,
                            feature: CHECKOUT_CORE_FEATURE,
                            methods: [
                                'getPaymentTypes',
                                'loadPaymentTypes',
                                'setPaymentType',
                                'getSelectedPaymentType',
                                'isAccountPayment',
                                'getPoNumber',
                            ],
                            async: true,
                        }); },
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CHECKOUT_CORE_FEATURE = CHECKOUT_CORE_FEATURE;
    exports.CHECKOUT_FEATURE = CHECKOUT_FEATURE;
    exports.CheckoutConfig = CheckoutConfig;
    exports.CheckoutCostCenterFacade = CheckoutCostCenterFacade;
    exports.CheckoutDeliveryFacade = CheckoutDeliveryFacade;
    exports.CheckoutFacade = CheckoutFacade;
    exports.CheckoutPaymentFacade = CheckoutPaymentFacade;
    exports.CheckoutRootModule = CheckoutRootModule;
    exports.ClearCheckoutFacade = ClearCheckoutFacade;
    exports.OrderConfirmationOrderEntriesContext = OrderConfirmationOrderEntriesContext;
    exports.OrderPlacedEvent = OrderPlacedEvent;
    exports.PaymentTypeFacade = PaymentTypeFacade;
    exports.checkoutPaymentSteps = checkoutPaymentSteps;
    exports.checkoutShippingSteps = checkoutShippingSteps;
    exports.defaultCheckoutComponentsConfig = defaultCheckoutComponentsConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
