// @spartacus/core/esm2015/src/global-message/global-message.module.js
import { NgModule } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';
import { errorHandlers, httpErrorInterceptors, } from './http-interceptors/index';
import { GlobalMessageStoreModule } from './store/global-message-store.module';
import { GlobalMessageEffect } from './store/effects/global-message.effect';
import { defaultGlobalMessageConfigFactory } from './config/default-global-message-config';
import { provideDefaultConfigFactory } from '../config/config-providers';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/effects";
export class GlobalMessageModule {
    static forRoot() {
        return {
            ngModule: GlobalMessageModule,
            providers: [...errorHandlers, ...httpErrorInterceptors],
        };
    }
}
GlobalMessageModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
GlobalMessageModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageModule, imports: [GlobalMessageStoreModule, i1.EffectsFeatureModule] });
GlobalMessageModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageModule, providers: [provideDefaultConfigFactory(defaultGlobalMessageConfigFactory)], imports: [[
            GlobalMessageStoreModule,
            EffectsModule.forFeature([GlobalMessageEffect]),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        GlobalMessageStoreModule,
                        EffectsModule.forFeature([GlobalMessageEffect]),
                    ],
                    providers: [provideDefaultConfigFactory(defaultGlobalMessageConfigFactory)],
                }]
        }] });
