// @spartacus/storefront/esm2015/shared/components/modal/modal.directive.js
import { Directive, HostListener, Injectable, Input, Optional, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@ng-bootstrap/ng-bootstrap";
/**
 * Service to control the behavior of the DismissModalDirective
 */
export class ModalDirectiveService {
    /**
     * Handler for the click event on the directive
     *
     * @param options.type    Action to perform after click - close or dismiss
     * @param options.reason  Reason for dismissing or closing the modal
     * @param url             url (optional) - exists only when the directive is also a `routerLink`.
     */
    onClick(options, activeModal, url) {
        let reason = options.reason;
        if (!reason && url) {
            reason = `Link click: ${url}`;
        }
        if (options.type === 'dismiss') {
            activeModal.dismiss(reason);
        }
        else if (options.type === 'close') {
            activeModal.close(reason);
        }
    }
}
ModalDirectiveService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ModalDirectiveService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
ModalDirectiveService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ModalDirectiveService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ModalDirectiveService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
/**
 * Directive to dismiss the modal on click
 */
export class ModalDirective {
    constructor(service, router, 
    // active modal can be injected only when the directive is projected inside modal
    activeModal, routerLink, routerLinkWithHref) {
        this.service = service;
        this.router = router;
        this.activeModal = activeModal;
        this.routerLink = routerLink;
        this.routerLinkWithHref = routerLinkWithHref;
    }
    /**
     * Returns URL in case when the directive is in the scope of a `routerLink` directive at the same time.
     */
    getUrl() {
        var _a;
        const routerLink = (_a = this.routerLink) !== null && _a !== void 0 ? _a : this.routerLinkWithHref;
        return routerLink
            ? this.router.serializeUrl(routerLink.urlTree)
            : undefined;
    }
    onClick() {
        if (this.activeModal) {
            this.service.onClick(this.options, this.activeModal, this.getUrl());
        }
    }
    /**
     * Converts the directive's inputs to the shape of modal directive options
     */
    get options() {
        return { type: this.cxModal, reason: this.cxModalReason };
    }
}
ModalDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ModalDirective, deps: [{ token: ModalDirectiveService }, { token: i1.Router }, { token: i2.NgbActiveModal, optional: true }, { token: i1.RouterLink, optional: true }, { token: i1.RouterLinkWithHref, optional: true }], target: i0.ɵɵFactoryTarget.Directive });
ModalDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: ModalDirective, selector: "[cxModal]", inputs: { cxModal: "cxModal", cxModalReason: "cxModalReason" }, host: { listeners: { "click": "onClick()" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ModalDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxModal]',
                }]
        }], ctorParameters: function () { return [{ type: ModalDirectiveService }, { type: i1.Router }, { type: i2.NgbActiveModal, decorators: [{
                    type: Optional
                }] }, { type: i1.RouterLink, decorators: [{
                    type: Optional
                }] }, { type: i1.RouterLinkWithHref, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { cxModal: [{
                type: Input
            }], cxModalReason: [{
                type: Input
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }] } });
