// @spartacus/tracking/esm2015/personalization/root/personalization-root.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { defaultPersonalizationConfig } from './config/default-personalization-config';
import { PERSONALIZATION_FEATURE } from './feature-name';
import { interceptors } from './http-interceptors/index';
import * as i0 from "@angular/core";
// TODO: Inline this factory when we start releasing Ivy compiled libraries
export function defaultPersonalizationComponentsConfig() {
    const config = {
        featureModules: {
            [PERSONALIZATION_FEATURE]: {
                cmsComponents: ['PersonalizationScriptComponent'],
            },
        },
    };
    return config;
}
export class PersonalizationRootModule {
}
PersonalizationRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PersonalizationRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PersonalizationRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PersonalizationRootModule });
PersonalizationRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PersonalizationRootModule, providers: [
        ...interceptors,
        provideDefaultConfig(defaultPersonalizationConfig),
        provideDefaultConfigFactory(defaultPersonalizationComponentsConfig),
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PersonalizationRootModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        ...interceptors,
                        provideDefaultConfig(defaultPersonalizationConfig),
                        provideDefaultConfigFactory(defaultPersonalizationComponentsConfig),
                    ],
                }]
        }] });
