// @spartacus/smartedit/esm2015/core/decorators/smart-edit-slot-decorator.js
import { Injectable } from '@angular/core';
import { SlotDecorator } from '@spartacus/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/smart-edit.service";
export class SmartEditSlotDecorator extends SlotDecorator {
    constructor(smartEditService) {
        super();
        this.smartEditService = smartEditService;
    }
    decorate(element, renderer, slot) {
        if (slot) {
            this.smartEditService.addSmartEditContract(element, renderer, slot.properties);
        }
    }
}
SmartEditSlotDecorator.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SmartEditSlotDecorator, deps: [{ token: i1.SmartEditService }], target: i0.ɵɵFactoryTarget.Injectable });
SmartEditSlotDecorator.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SmartEditSlotDecorator, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SmartEditSlotDecorator, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.SmartEditService }]; } });
