// @spartacus/smartedit/esm2015/core/decorators/smart-edit-component-decorator.js
import { Injectable } from '@angular/core';
import { ComponentDecorator } from '@spartacus/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/smart-edit.service";
export class SmartEditComponentDecorator extends ComponentDecorator {
    constructor(smartEditService) {
        super();
        this.smartEditService = smartEditService;
    }
    decorate(element, renderer, component) {
        if (component) {
            this.smartEditService.addSmartEditContract(element, renderer, component.properties);
        }
    }
}
SmartEditComponentDecorator.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SmartEditComponentDecorator, deps: [{ token: i1.SmartEditService }], target: i0.ɵɵFactoryTarget.Injectable });
SmartEditComponentDecorator.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SmartEditComponentDecorator, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SmartEditComponentDecorator, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.SmartEditService }]; } });
