// @spartacus/storefront/esm2015/cms-components/myaccount/order/order-details/order-detail-shipping/order-detail-shipping.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../order-details.service";
import * as i2 from "../../../../../shared/components/order-overview/order-overview.component";
import * as i3 from "@angular/common";
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class OrderDetailShippingComponent {
    constructor(orderDetailsService) {
        this.orderDetailsService = orderDetailsService;
    }
    ngOnInit() {
        this.order$ = this.orderDetailsService.getOrderDetails();
    }
}
OrderDetailShippingComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailShippingComponent, deps: [{ token: i1.OrderDetailsService }], target: i0.ɵɵFactoryTarget.Component });
OrderDetailShippingComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: OrderDetailShippingComponent, selector: "cx-order-details-shipping", ngImport: i0, template: "<ng-container *ngIf=\"order$ | async as order\">\n  <cx-order-overview [order]=\"order\"></cx-order-overview>\n</ng-container>\n", components: [{ type: i2.OrderOverviewComponent, selector: "cx-order-overview", inputs: ["order"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i3.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailShippingComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-order-details-shipping',
                    templateUrl: './order-detail-shipping.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.OrderDetailsService }]; } });
