// @spartacus/storefront/esm2015/cms-components/anonymous-consent-management/anonymous-consent-management.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { DeferLoadingStrategy, FeaturesConfigModule, I18nModule, provideConfig, provideDefaultConfig, } from '@spartacus/core';
import { KeyboardFocusModule } from '../../layout/a11y/keyboard-focus/index';
import { AnonymousConsentManagementBannerComponent } from './banner/anonymous-consent-management-banner.component';
import { defaultAnonymousConsentLayoutConfig } from './default-anonymous-consent-layout.config';
import { AnonymousConsentOpenDialogComponent } from './open-dialog/anonymous-consent-open-dialog.component';
import * as i0 from "@angular/core";
export class AnonymousConsentManagementBannerModule {
}
AnonymousConsentManagementBannerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentManagementBannerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AnonymousConsentManagementBannerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentManagementBannerModule, declarations: [AnonymousConsentManagementBannerComponent,
        AnonymousConsentOpenDialogComponent], imports: [CommonModule,
        I18nModule,
        FeaturesConfigModule,
        KeyboardFocusModule], exports: [AnonymousConsentManagementBannerComponent,
        AnonymousConsentOpenDialogComponent] });
AnonymousConsentManagementBannerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentManagementBannerModule, providers: [
        provideConfig(defaultAnonymousConsentLayoutConfig),
        provideDefaultConfig({
            cmsComponents: {
                AnonymousConsentManagementBannerComponent: {
                    component: AnonymousConsentManagementBannerComponent,
                    deferLoading: DeferLoadingStrategy.INSTANT,
                },
                AnonymousConsentOpenDialogComponent: {
                    component: AnonymousConsentOpenDialogComponent,
                },
            },
        }),
    ], imports: [[
            CommonModule,
            I18nModule,
            FeaturesConfigModule,
            KeyboardFocusModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentManagementBannerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        I18nModule,
                        FeaturesConfigModule,
                        KeyboardFocusModule,
                    ],
                    providers: [
                        provideConfig(defaultAnonymousConsentLayoutConfig),
                        provideDefaultConfig({
                            cmsComponents: {
                                AnonymousConsentManagementBannerComponent: {
                                    component: AnonymousConsentManagementBannerComponent,
                                    deferLoading: DeferLoadingStrategy.INSTANT,
                                },
                                AnonymousConsentOpenDialogComponent: {
                                    component: AnonymousConsentOpenDialogComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [
                        AnonymousConsentManagementBannerComponent,
                        AnonymousConsentOpenDialogComponent,
                    ],
                    exports: [
                        AnonymousConsentManagementBannerComponent,
                        AnonymousConsentOpenDialogComponent,
                    ],
                }]
        }] });
