// @spartacus/storefront/esm2015/cms-structure/outlet/outlet-ref/outlet-ref.directive.js
import { Directive, Input, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../outlet.service";
export class OutletRefDirective {
    constructor(tpl, outletService) {
        this.tpl = tpl;
        this.outletService = outletService;
    }
    ngOnInit() {
        this.outletService.add(this.cxOutletRef, this.tpl, this.cxOutletPos);
    }
    ngOnDestroy() {
        this.outletService.remove(this.cxOutletRef, this.cxOutletPos, this.tpl);
    }
}
OutletRefDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OutletRefDirective, deps: [{ token: i0.TemplateRef }, { token: i1.OutletService }], target: i0.ɵɵFactoryTarget.Directive });
OutletRefDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: OutletRefDirective, selector: "[cxOutletRef]", inputs: { cxOutletRef: "cxOutletRef", cxOutletPos: "cxOutletPos" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OutletRefDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxOutletRef]',
                }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }, { type: i1.OutletService }]; }, propDecorators: { cxOutletRef: [{
                type: Input
            }], cxOutletPos: [{
                type: Input
            }] } });
