// @spartacus/storefront/esm2015/cms-structure/outlet/outlet-ref/outlet-ref.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { OutletRefDirective } from './outlet-ref.directive';
import * as i0 from "@angular/core";
export class OutletRefModule {
}
OutletRefModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OutletRefModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OutletRefModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OutletRefModule, declarations: [OutletRefDirective], imports: [CommonModule], exports: [OutletRefDirective] });
OutletRefModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OutletRefModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OutletRefModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [OutletRefDirective],
                    exports: [OutletRefDirective],
                }]
        }] });
