// @spartacus/storefront/esm2015/cms-components/cart/cart-totals/cart-totals.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, FeaturesConfigModule, } from '@spartacus/core';
import { CartCouponModule } from '../cart-coupon/cart-coupon.module';
import { CartSharedModule } from '../cart-shared/cart-shared.module';
import { CartTotalsComponent } from './cart-totals.component';
import { ProgressButtonModule } from '../../../shared/components/progress-button/progress-button.module';
import * as i0 from "@angular/core";
export class CartTotalsModule {
}
CartTotalsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartTotalsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CartTotalsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartTotalsModule, declarations: [CartTotalsComponent], imports: [CommonModule,
        RouterModule,
        UrlModule,
        CartSharedModule,
        I18nModule,
        CartCouponModule,
        FeaturesConfigModule,
        ProgressButtonModule], exports: [CartTotalsComponent] });
CartTotalsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartTotalsModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                CartTotalsComponent: {
                    component: CartTotalsComponent,
                },
            },
        }),
    ], imports: [[
            CommonModule,
            RouterModule,
            UrlModule,
            CartSharedModule,
            I18nModule,
            CartCouponModule,
            FeaturesConfigModule,
            ProgressButtonModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartTotalsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        UrlModule,
                        CartSharedModule,
                        I18nModule,
                        CartCouponModule,
                        FeaturesConfigModule,
                        ProgressButtonModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CartTotalsComponent: {
                                    component: CartTotalsComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [CartTotalsComponent],
                    exports: [CartTotalsComponent],
                }]
        }] });
