// @spartacus/storefront/esm2015/cms-components/cart/cart-totals/cart-totals.component.js
import { ChangeDetectionStrategy, Component, } from '@angular/core';
import { Subscription } from 'rxjs';
import { filter } from 'rxjs/operators';
import { NavigationCancel, NavigationEnd, } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@angular/router";
import * as i3 from "../cart-shared/order-summary/order-summary.component";
import * as i4 from "../../../shared/components/progress-button/progress-button.component";
import * as i5 from "@angular/common";
export class CartTotalsComponent {
    constructor(activeCartService, router) {
        this.activeCartService = activeCartService;
        this.router = router;
        this.cartValidationInProgress = false;
        this.subscription = new Subscription();
    }
    ngOnInit() {
        var _a;
        this.cart$ = this.activeCartService.getActive();
        this.entries$ = this.activeCartService
            .getEntries()
            .pipe(filter((entries) => entries.length > 0));
        this.subscription.add((_a = this.router) === null || _a === void 0 ? void 0 : _a.events.subscribe((event) => {
            if (event instanceof NavigationEnd ||
                event instanceof NavigationCancel) {
                this.cartValidationInProgress = false;
            }
        }));
    }
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
    disableButtonWhileNavigation() {
        this.cartValidationInProgress = true;
    }
}
CartTotalsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartTotalsComponent, deps: [{ token: i1.ActiveCartService }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Component });
CartTotalsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: CartTotalsComponent, selector: "cx-cart-totals", ngImport: i0, template: "<ng-container *ngIf=\"cart$ | async as cart\">\n  <ng-container *ngIf=\"entries$ | async as entries\">\n    <cx-order-summary [cart]=\"cart\"></cx-order-summary>\n    <ng-container *cxFeatureLevel=\"'!4.2'\">\n      <button\n        [routerLink]=\"{ cxRoute: 'checkout' } | cxUrl\"\n        *ngIf=\"entries.length\"\n        class=\"btn btn-primary btn-block\"\n        type=\"button\"\n      >\n        {{ 'cartDetails.proceedToCheckout' | cxTranslate }}\n      </button>\n    </ng-container>\n\n    <ng-container *cxFeatureLevel=\"'4.2'\">\n      <cx-progress-button\n        *ngIf=\"entries.length\"\n        [routerLink]=\"{ cxRoute: 'checkout' } | cxUrl\"\n        [loading]=\"cartValidationInProgress\"\n        [disabled]=\"cartValidationInProgress\"\n        [class]=\"'btn btn-primary btn-block'\"\n        (clikEvent)=\"disableButtonWhileNavigation()\"\n      >\n        {{\n          (!cartValidationInProgress ? 'cartDetails.proceedToCheckout' : '')\n            | cxTranslate\n        }}\n      </cx-progress-button>\n    </ng-container>\n  </ng-container>\n</ng-container>\n", components: [{ type: i3.OrderSummaryComponent, selector: "cx-order-summary", inputs: ["cart"] }, { type: i4.ProgressButtonComponent, selector: "cx-progress-button", inputs: ["ariaLabel", "class", "disabled", "loading"], outputs: ["clikEvent"] }], directives: [{ type: i5.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.FeatureLevelDirective, selector: "[cxFeatureLevel]", inputs: ["cxFeatureLevel"] }, { type: i2.RouterLink, selector: ":not(a):not(area)[routerLink]", inputs: ["routerLink", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }], pipes: { "async": i5.AsyncPipe, "cxUrl": i1.UrlPipe, "cxTranslate": i1.TranslatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartTotalsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-cart-totals',
                    templateUrl: './cart-totals.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.ActiveCartService }, { type: i2.Router }]; } });
