// @spartacus/order/esm2015/core/store/reducers/index.js
import { InjectionToken } from '@angular/core';
import { StateUtils, } from '@spartacus/core';
import { ORDERS, ORDER_DETAILS, REPLENISHMENT_ORDERS, REPLENISHMENT_ORDER_DETAILS, RETURN_REQUESTS, RETURN_REQUEST_DETAILS, } from '../order-state';
import * as fromConsignmentTrackingReducer from './consignment-tracking.reducer';
import * as fromOrderDetailsReducer from './order-details.reducer';
import * as fromOrderReturnRequestReducer from './order-return-request.reducer';
import * as fromUserOrdersReducer from './orders.reducer';
import * as fromReplenishmentOrderDetailsReducer from './replenishment-order-details.reducer';
import * as fromUserReplenishmentOrdersReducer from './replenishment-orders.reducer';
export function getReducers() {
    return {
        orders: StateUtils.loaderReducer(ORDERS, fromUserOrdersReducer.reducer),
        orderDetail: StateUtils.loaderReducer(ORDER_DETAILS, fromOrderDetailsReducer.reducer),
        replenishmentOrders: StateUtils.loaderReducer(REPLENISHMENT_ORDERS, fromUserReplenishmentOrdersReducer.reducer),
        orderReturn: StateUtils.loaderReducer(RETURN_REQUEST_DETAILS),
        orderReturnList: StateUtils.loaderReducer(RETURN_REQUESTS, fromOrderReturnRequestReducer.reducer),
        consignmentTracking: fromConsignmentTrackingReducer.reducer,
        replenishmentOrder: StateUtils.loaderReducer(REPLENISHMENT_ORDER_DETAILS, fromReplenishmentOrderDetailsReducer.reducer),
    };
}
export const reducerToken = new InjectionToken('OrderReducers');
export const reducerProvider = {
    provide: reducerToken,
    useFactory: getReducers,
};
