// @spartacus/core/esm2015/src/user/events/user-event.builder.js
import { Injectable } from '@angular/core';
import { UserActions } from '../store/actions/index';
import { AddUserAddressEvent, DeleteUserAddressEvent, UpdateUserAddressEvent, } from './user.events';
import * as i0 from "@angular/core";
import * as i1 from "../../state/event/state-event.service";
export class UserEventBuilder {
    constructor(stateEventService) {
        this.stateEventService = stateEventService;
        this.register();
    }
    /**
     * Registers user events
     */
    register() {
        this.updateUserAddressEvent();
        this.deleteUserAddressEvent();
        this.addUserAddressEvent();
    }
    /**
     * Register an address successfully updated event
     */
    updateUserAddressEvent() {
        this.stateEventService.register({
            action: UserActions.UPDATE_USER_ADDRESS,
            event: UpdateUserAddressEvent,
        });
    }
    addUserAddressEvent() {
        this.stateEventService.register({
            action: UserActions.ADD_USER_ADDRESS,
            event: AddUserAddressEvent,
        });
    }
    deleteUserAddressEvent() {
        this.stateEventService.register({
            action: UserActions.DELETE_USER_ADDRESS,
            event: DeleteUserAddressEvent,
        });
    }
}
UserEventBuilder.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserEventBuilder, deps: [{ token: i1.StateEventService }], target: i0.ɵɵFactoryTarget.Injectable });
UserEventBuilder.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserEventBuilder });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserEventBuilder, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.StateEventService }]; } });
