// @spartacus/storefront/esm2015/cms-components/cart/cart-coupon/cart-coupon.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { IconModule } from '../../../cms-components/misc/icon/icon.module';
import { AppliedCouponsComponent } from './applied-coupons/applied-coupons.component';
import { CartCouponComponent } from './cart-coupon.component';
import { NgSelectModule } from '@ng-select/ng-select';
import { FormErrorsModule } from '../../../shared/index';
import * as i0 from "@angular/core";
export class CartCouponModule {
}
CartCouponModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartCouponModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CartCouponModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartCouponModule, declarations: [CartCouponComponent, AppliedCouponsComponent], imports: [CommonModule,
        NgSelectModule,
        FormsModule,
        ReactiveFormsModule,
        I18nModule,
        IconModule,
        FormErrorsModule], exports: [CartCouponComponent, AppliedCouponsComponent] });
CartCouponModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartCouponModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                CartApplyCouponComponent: {
                    component: CartCouponComponent,
                },
            },
        }),
    ], imports: [[
            CommonModule,
            NgSelectModule,
            FormsModule,
            ReactiveFormsModule,
            I18nModule,
            IconModule,
            FormErrorsModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartCouponModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CartCouponComponent, AppliedCouponsComponent],
                    exports: [CartCouponComponent, AppliedCouponsComponent],
                    imports: [
                        CommonModule,
                        NgSelectModule,
                        FormsModule,
                        ReactiveFormsModule,
                        I18nModule,
                        IconModule,
                        FormErrorsModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CartApplyCouponComponent: {
                                    component: CartCouponComponent,
                                },
                            },
                        }),
                    ],
                }]
        }] });
