// @spartacus/storefront/esm2015/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule } from '@spartacus/core';
import { IconModule } from '../../../../../cms-components/misc/icon/icon.module';
import { KeyboardFocusModule } from '../../../../../layout/a11y/keyboard-focus/keyboard-focus.module';
import { ActiveFacetsComponent } from './active-facets.component';
import * as i0 from "@angular/core";
export class ActiveFacetsModule {
}
ActiveFacetsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ActiveFacetsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ActiveFacetsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ActiveFacetsModule, declarations: [ActiveFacetsComponent], imports: [CommonModule,
        RouterModule,
        I18nModule,
        IconModule,
        KeyboardFocusModule], exports: [ActiveFacetsComponent] });
ActiveFacetsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ActiveFacetsModule, imports: [[
            CommonModule,
            RouterModule,
            I18nModule,
            IconModule,
            KeyboardFocusModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ActiveFacetsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        I18nModule,
                        IconModule,
                        KeyboardFocusModule,
                    ],
                    declarations: [ActiveFacetsComponent],
                    exports: [ActiveFacetsComponent],
                }]
        }] });
