// @spartacus/storefront/esm2015/cms-components/user/logout/logout.module.js
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { CmsPageGuard } from '../../../cms-structure/guards/cms-page.guard';
import { PageLayoutComponent, PageLayoutModule, } from '../../../cms-structure/page/index';
import { LogoutGuard } from './logout.guard';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class LogoutModule {
}
LogoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LogoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LogoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LogoutModule, imports: [PageLayoutModule, i1.RouterModule] });
LogoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LogoutModule, imports: [[
            PageLayoutModule,
            RouterModule.forChild([
                {
                    path: null,
                    canActivate: [LogoutGuard, CmsPageGuard],
                    component: PageLayoutComponent,
                    data: { cxRoute: 'logout' },
                },
            ]),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LogoutModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        PageLayoutModule,
                        RouterModule.forChild([
                            {
                                path: null,
                                canActivate: [LogoutGuard, CmsPageGuard],
                                component: PageLayoutComponent,
                                data: { cxRoute: 'logout' },
                            },
                        ]),
                    ],
                }]
        }] });
