// @spartacus/cart/esm2015/import-export/components/import-to-cart/import-order-entries.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { ConfigModule, I18nModule, provideConfig, } from '@spartacus/core';
import { FileUploadModule, FormErrorsModule, IconModule, KeyboardFocusModule, } from '@spartacus/storefront';
import { defaultImportEntriesLayoutConfig } from './default-import-entries-layout.config';
import { ImportEntriesDialogComponent } from './import-entries-dialog/import-entries-dialog.component';
import { ImportEntriesFormComponent } from './import-entries-dialog/import-entries-form/import-entries-form.component';
import { ImportEntriesSummaryComponent } from './import-entries-dialog/import-entries-summary/import-entries-summary.component';
import { ImportToNewSavedCartFormComponent } from './import-entries-dialog/import-to-new-saved-cart-form/import-to-new-saved-cart-form.component';
import { ImportOrderEntriesComponent } from './import-entries/import-order-entries.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class ImportOrderEntriesModule {
}
ImportOrderEntriesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportOrderEntriesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ImportOrderEntriesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportOrderEntriesModule, declarations: [ImportOrderEntriesComponent,
        ImportEntriesDialogComponent,
        ImportEntriesFormComponent,
        ImportEntriesSummaryComponent,
        ImportToNewSavedCartFormComponent], imports: [CommonModule,
        FormsModule,
        ReactiveFormsModule,
        FormErrorsModule,
        IconModule,
        KeyboardFocusModule,
        FileUploadModule,
        I18nModule, i1.ConfigModule], exports: [ImportOrderEntriesComponent,
        ImportEntriesDialogComponent,
        ImportEntriesFormComponent,
        ImportEntriesSummaryComponent,
        ImportToNewSavedCartFormComponent] });
ImportOrderEntriesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportOrderEntriesModule, providers: [provideConfig(defaultImportEntriesLayoutConfig)], imports: [[
            CommonModule,
            FormsModule,
            ReactiveFormsModule,
            FormErrorsModule,
            IconModule,
            KeyboardFocusModule,
            FileUploadModule,
            I18nModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    ImportOrderEntriesComponent: {
                        component: ImportOrderEntriesComponent,
                    },
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportOrderEntriesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        FormErrorsModule,
                        IconModule,
                        KeyboardFocusModule,
                        FileUploadModule,
                        I18nModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                ImportOrderEntriesComponent: {
                                    component: ImportOrderEntriesComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [
                        ImportOrderEntriesComponent,
                        ImportEntriesDialogComponent,
                        ImportEntriesFormComponent,
                        ImportEntriesSummaryComponent,
                        ImportToNewSavedCartFormComponent,
                    ],
                    exports: [
                        ImportOrderEntriesComponent,
                        ImportEntriesDialogComponent,
                        ImportEntriesFormComponent,
                        ImportEntriesSummaryComponent,
                        ImportToNewSavedCartFormComponent,
                    ],
                    providers: [provideConfig(defaultImportEntriesLayoutConfig)],
                }]
        }] });
