// @spartacus/core/esm2015/src/cart/facade/cart-validation.service.js
import { Injectable } from '@angular/core';
import { combineLatest } from 'rxjs';
import { filter, switchMap, take } from 'rxjs/operators';
import { CommandStrategy, } from '../../util/command-query/command.service';
import * as i0 from "@angular/core";
import * as i1 from "../connectors/validation/cart-validation.connector";
import * as i2 from "../../util/command-query/command.service";
import * as i3 from "../../auth/user-auth/facade/user-id.service";
import * as i4 from "./active-cart.service";
export class CartValidationService {
    constructor(cartValidationConnector, command, userIdService, activeCartService) {
        this.cartValidationConnector = cartValidationConnector;
        this.command = command;
        this.userIdService = userIdService;
        this.activeCartService = activeCartService;
        this.validateCartCommand = this.command.create(() => combineLatest([
            this.activeCartService.getActiveCartId(),
            this.userIdService.takeUserId(),
            this.activeCartService.isStable(),
        ]).pipe(filter(([_, __, loaded]) => loaded), take(1), switchMap(([cartId, userId]) => this.cartValidationConnector.validate(cartId, userId))), {
            strategy: CommandStrategy.CancelPrevious,
        });
    }
    /**
     * Returns cart modification list.
     *
     * @param cartId
     * @param userId
     */
    validateCart() {
        return this.validateCartCommand.execute();
    }
}
CartValidationService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationService, deps: [{ token: i1.CartValidationConnector }, { token: i2.CommandService }, { token: i3.UserIdService }, { token: i4.ActiveCartService }], target: i0.ɵɵFactoryTarget.Injectable });
CartValidationService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CartValidationConnector }, { type: i2.CommandService }, { type: i3.UserIdService }, { type: i4.ActiveCartService }]; } });
