// @spartacus/storefront/esm2015/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.js
import { Injectable } from '@angular/core';
import { NavigationStart } from '@angular/router';
import { filter, take } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../config";
import * as i2 from "@angular/router";
import * as i3 from "../../../../layout/breakpoint/breakpoint.service";
import * as i4 from "@spartacus/core";
/**
 * @deprecated since 4.2, refer to spartacus issues (#13762)
 * Shared service for keyboard focus features called when the browser navigates.
 */
export class OnNavigateFocusService {
    constructor(config, router, breakpointService, winRef) {
        this.config = config;
        this.router = router;
        this.breakpointService = breakpointService;
        this.winRef = winRef;
    }
    ngOnDestroy() {
        var _a, _b;
        (_a = this.resetFocusOnNavigate) === null || _a === void 0 ? void 0 : _a.unsubscribe();
        (_b = this.resetViewOnNavigate) === null || _b === void 0 ? void 0 : _b.unsubscribe();
    }
    /**
     * Reads configuration and enables features based on flags set.
     */
    initializeWithConfig() {
        var _a, _b, _c, _d;
        if (!this.winRef.isBrowser()) {
            return;
        }
        if ((_b = (_a = this.config) === null || _a === void 0 ? void 0 : _a.keyboardFocus) === null || _b === void 0 ? void 0 : _b.enableResetFocusOnNavigate) {
            this.setResetFocusOnNavigate(this.config.keyboardFocus.enableResetFocusOnNavigate);
        }
        if ((_d = (_c = this.config) === null || _c === void 0 ? void 0 : _c.keyboardFocus) === null || _d === void 0 ? void 0 : _d.enableResetViewOnNavigate) {
            this.setResetViewOnNavigate(this.config.keyboardFocus.enableResetViewOnNavigate);
        }
    }
    /**
     * Resets focus back to body element in the DOM tree when a navigation is started.
     * @param enable Enable or disable this feature. Set this to an array of BREAKPOINTS to enable for specified screen widths.
     */
    setResetFocusOnNavigate(enable) {
        var _a;
        (_a = this.resetFocusOnNavigate) === null || _a === void 0 ? void 0 : _a.unsubscribe();
        if (enable) {
            this.resetFocusOnNavigate = this.router.events
                .pipe(filter((event) => event instanceof NavigationStart))
                .subscribe(() => {
                if (Array.isArray(enable)) {
                    this.breakpointService.breakpoint$
                        .pipe(take(1))
                        .subscribe((breakpoint) => {
                        if (enable.includes(breakpoint)) {
                            this.winRef.document.body.focus();
                        }
                    });
                }
                else if (typeof enable === 'boolean') {
                    this.winRef.document.body.focus();
                }
            });
        }
    }
    /**
     * Resets view back to body element in the DOM tree when a navigation is started.
     * @param enable Enable or disable this feature. Set this to an array of BREAKPOINTS to enable for specified screen widths.
     */
    setResetViewOnNavigate(enable) {
        var _a;
        (_a = this.resetViewOnNavigate) === null || _a === void 0 ? void 0 : _a.unsubscribe();
        if (enable) {
            this.resetViewOnNavigate = this.router.events
                .pipe(filter((event) => event instanceof NavigationStart))
                .subscribe(() => {
                if (Array.isArray(enable)) {
                    this.breakpointService.breakpoint$
                        .pipe(take(1))
                        .subscribe((breakpoint) => {
                        if (enable.includes(breakpoint)) {
                            this.winRef.document.body.scrollIntoView();
                        }
                    });
                }
                else if (typeof enable === 'boolean') {
                    this.winRef.document.body.scrollIntoView();
                }
            });
        }
    }
}
OnNavigateFocusService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OnNavigateFocusService, deps: [{ token: i1.KeyboardFocusConfig }, { token: i2.Router }, { token: i3.BreakpointService }, { token: i4.WindowRef }], target: i0.ɵɵFactoryTarget.Injectable });
OnNavigateFocusService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OnNavigateFocusService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OnNavigateFocusService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.KeyboardFocusConfig }, { type: i2.Router }, { type: i3.BreakpointService }, { type: i4.WindowRef }]; } });
