// @spartacus/core/esm2015/src/process/process.module.js
import { NgModule } from '@angular/core';
import { ProcessStoreModule } from './store/process-store.module';
import * as i0 from "@angular/core";
export class ProcessModule {
    static forRoot() {
        return {
            ngModule: ProcessModule,
            providers: [],
        };
    }
}
ProcessModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProcessModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProcessModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProcessModule, imports: [ProcessStoreModule] });
ProcessModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProcessModule, imports: [[ProcessStoreModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProcessModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [ProcessStoreModule],
                }]
        }] });
