// @spartacus/storefront/esm2015/cms-components/cart/validation/cart-warnings/cart-validation-warnings.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { CartValidationWarningsComponent } from './cart-validation-warnings.component';
import { IconModule } from '../../../misc/icon/index';
import * as i0 from "@angular/core";
export class CartValidationWarningsModule {
}
CartValidationWarningsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationWarningsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CartValidationWarningsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationWarningsModule, declarations: [CartValidationWarningsComponent], imports: [CommonModule, RouterModule, I18nModule, UrlModule, IconModule], exports: [CartValidationWarningsComponent] });
CartValidationWarningsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationWarningsModule, imports: [[CommonModule, RouterModule, I18nModule, UrlModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartValidationWarningsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RouterModule, I18nModule, UrlModule, IconModule],
                    exports: [CartValidationWarningsComponent],
                    declarations: [CartValidationWarningsComponent],
                }]
        }] });
