// @spartacus/storefront/esm2015/cms-components/myaccount/address-book/address-book.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AuthGuard, I18nModule, provideDefaultConfig, UserAddressService, } from '@spartacus/core';
import { CardModule } from '../../../shared/components/card/card.module';
import { SpinnerModule } from '../../../shared/components/spinner/spinner.module';
import { AddressBookComponent } from './address-book.component';
import { AddressFormModule } from './address-form/address-form.module';
import * as i0 from "@angular/core";
export class AddressBookModule {
}
AddressBookModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddressBookModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AddressBookModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddressBookModule, declarations: [AddressBookComponent], imports: [CommonModule,
        CardModule,
        AddressFormModule,
        SpinnerModule,
        I18nModule], exports: [AddressBookComponent] });
AddressBookModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddressBookModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                AccountAddressBookComponent: {
                    component: AddressBookComponent,
                    guards: [AuthGuard],
                },
            },
        }),
        UserAddressService,
    ], imports: [[
            CommonModule,
            CardModule,
            AddressFormModule,
            SpinnerModule,
            I18nModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddressBookModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        CardModule,
                        AddressFormModule,
                        SpinnerModule,
                        I18nModule,
                    ],
                    declarations: [AddressBookComponent],
                    exports: [AddressBookComponent],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                AccountAddressBookComponent: {
                                    component: AddressBookComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                        UserAddressService,
                    ],
                }]
        }] });
