// @spartacus/core/esm2015/src/cart/services/cart-config.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../config/cart-config";
export class CartConfigService {
    constructor(config) {
        this.config = config;
    }
    isSelectiveCartEnabled() {
        var _a, _b, _c;
        return Boolean((_c = (_b = (_a = this.config) === null || _a === void 0 ? void 0 : _a.cart) === null || _b === void 0 ? void 0 : _b.selectiveCart) === null || _c === void 0 ? void 0 : _c.enabled);
    }
    isCartValidationEnabled() {
        var _a, _b, _c;
        return Boolean((_c = (_b = (_a = this.config) === null || _a === void 0 ? void 0 : _a.cart) === null || _b === void 0 ? void 0 : _b.validation) === null || _c === void 0 ? void 0 : _c.enabled);
    }
}
CartConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartConfigService, deps: [{ token: i1.CartConfig }], target: i0.ɵɵFactoryTarget.Injectable });
CartConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartConfigService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartConfigService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CartConfig }]; } });
