// @spartacus/core/esm2015/src/user/store/selectors/order-return-request.selectors.js
import { createSelector } from '@ngrx/store';
import { StateUtils } from '../../../state/utils/index';
import { getUserState } from './feature.selector';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const getOrderReturnRequestState = createSelector(getUserState, (state) => state.orderReturn);
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const getOrderReturnRequest = createSelector(getOrderReturnRequestState, (state) => StateUtils.loaderValueSelector(state));
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const getOrderReturnRequestLoading = createSelector(getOrderReturnRequestState, (state) => StateUtils.loaderLoadingSelector(state));
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const getOrderReturnRequestSuccess = createSelector(getOrderReturnRequestState, (state) => StateUtils.loaderSuccessSelector(state) &&
    !StateUtils.loaderLoadingSelector(state));
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const getOrderReturnRequestListState = createSelector(getUserState, (state) => state.orderReturnList);
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const getOrderReturnRequestList = createSelector(getOrderReturnRequestListState, (state) => StateUtils.loaderValueSelector(state));
