// @spartacus/core/esm2015/src/user/store/selectors/regions.selectors.js
import { createSelector } from '@ngrx/store';
import { StateUtils } from '../../../state/utils/index';
import { getUserState } from './feature.selector';
export const getRegionsLoaderState = createSelector(getUserState, (state) => state.regions);
export const getAllRegions = createSelector(getRegionsLoaderState, (state) => {
    return StateUtils.loaderValueSelector(state).entities;
});
export const getRegionsDataAndLoading = createSelector(getRegionsLoaderState, (state) => ({
    loaded: StateUtils.loaderSuccessSelector(state),
    loading: StateUtils.loaderLoadingSelector(state),
    regions: StateUtils.loaderValueSelector(state).entities,
    country: StateUtils.loaderValueSelector(state).country,
}));
export const getRegionsCountry = createSelector(getRegionsLoaderState, (state) => StateUtils.loaderValueSelector(state).country);
export const getRegionsLoading = createSelector(getRegionsLoaderState, (state) => StateUtils.loaderLoadingSelector(state));
export const getRegionsLoaded = createSelector(getRegionsLoaderState, (state) => StateUtils.loaderSuccessSelector(state));
