// @spartacus/product/schematics/add-product/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addImageZoom = exports.addVariantsFeature = exports.addBulkPricingFeature = exports.addSpartacusProduct = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const schematics_2 = require("@spartacus/schematics");
const package_json_1 = require("../../package.json");
const constants_1 = require("../constants");
function addSpartacusProduct(options) {
    return (tree, _context) => {
        const packageJson = schematics_2.readPackageJson(tree);
        schematics_2.validateSpartacusInstallation(packageJson);
        return schematics_1.chain([
            schematics_2.addPackageJsonDependenciesForLibrary(package_json_1.peerDependencies, options),
            schematics_2.shouldAddFeature(schematics_2.CLI_PRODUCT_BULK_PRICING_FEATURE, options.features)
                ? addBulkPricingFeature(options)
                : schematics_1.noop(),
            schematics_2.shouldAddFeature(schematics_2.CLI_PRODUCT_VARIANTS_FEATURE, options.features)
                ? addVariantsFeature(options)
                : schematics_1.noop(),
            schematics_2.shouldAddFeature(schematics_2.CLI_PRODUCT_IMAGE_ZOOM_FEATURE, options.features)
                ? addImageZoom(options)
                : schematics_1.noop(),
        ]);
    };
}
exports.addSpartacusProduct = addSpartacusProduct;
function addBulkPricingFeature(options) {
    return schematics_2.addLibraryFeature(options, {
        folderName: constants_1.PRODUCT_FOLDER_NAME,
        moduleName: constants_1.BULK_PRICING_MODULE_NAME,
        featureModule: {
            name: constants_1.BULK_PRICING_MODULE,
            importPath: constants_1.SPARTACUS_BULK_PRICING,
        },
        rootModule: {
            name: constants_1.BULK_PRICING_ROOT_MODULE,
            importPath: constants_1.SPARTACUS_BULK_PRICING_ROOT,
        },
        lazyLoadingChunk: {
            moduleSpecifier: constants_1.SPARTACUS_BULK_PRICING_ROOT,
            namedImports: [constants_1.BULK_PRICING_FEATURE_NAME_CONSTANT],
        },
        i18n: {
            resources: constants_1.BULK_PRICING_TRANSLATIONS,
            chunks: constants_1.BULK_PRICING_TRANSLATION_CHUNKS_CONFIG,
            importPath: constants_1.SPARTACUS_BULK_PRICING_ASSETS,
        },
        styles: {
            scssFileName: constants_1.PRODUCT_SCSS_FILE_NAME,
            importStyle: schematics_2.SPARTACUS_PRODUCT,
        },
    });
}
exports.addBulkPricingFeature = addBulkPricingFeature;
function addVariantsFeature(options) {
    return schematics_2.addLibraryFeature(options, {
        folderName: constants_1.PRODUCT_FOLDER_NAME,
        moduleName: constants_1.VARIANTS_MODULE_NAME,
        featureModule: {
            name: constants_1.VARIANTS_MODULE,
            importPath: constants_1.SPARTACUS_VARIANTS,
        },
        rootModule: {
            name: constants_1.VARIANTS_ROOT_MODULE,
            importPath: constants_1.SPARTACUS_VARIANTS_ROOT,
        },
        lazyLoadingChunk: {
            moduleSpecifier: constants_1.SPARTACUS_VARIANTS_ROOT,
            namedImports: [constants_1.VARIANTS_FEATURE_NAME_CONSTANT],
        },
        i18n: {
            resources: constants_1.VARIANTS_TRANSLATIONS,
            chunks: constants_1.VARIANTS_TRANSLATION_CHUNKS_CONFIG,
            importPath: constants_1.SPARTACUS_VARIANTS_ASSETS,
        },
        styles: {
            scssFileName: constants_1.PRODUCT_SCSS_FILE_NAME,
            importStyle: schematics_2.SPARTACUS_PRODUCT,
        },
    });
}
exports.addVariantsFeature = addVariantsFeature;
function addImageZoom(options) {
    return schematics_2.addLibraryFeature(options, {
        folderName: constants_1.PRODUCT_FOLDER_NAME,
        moduleName: constants_1.IMAGE_ZOOM_MODULE_NAME,
        featureModule: {
            name: constants_1.IMAGE_ZOOM_MODULE,
            importPath: constants_1.SPARTACUS_IMAGE_ZOOM,
        },
        rootModule: {
            name: constants_1.IMAGE_ZOOM_ROOT_MODULE,
            importPath: constants_1.SPARTACUS_IMAGE_ZOOM_ROOT,
        },
        lazyLoadingChunk: {
            moduleSpecifier: constants_1.SPARTACUS_IMAGE_ZOOM_ROOT,
            namedImports: [constants_1.IMAGE_ZOOM_FEATURE_NAME_CONSTANT],
        },
        i18n: {
            resources: constants_1.IMAGE_ZOOM_TRANSLATIONS,
            chunks: constants_1.IMAGE_ZOOM_TRANSLATION_CHUNKS_CONFIG,
            importPath: constants_1.SPARTACUS_IMAGE_ZOOM_ASSETS,
        },
        styles: {
            scssFileName: constants_1.PRODUCT_SCSS_FILE_NAME,
            importStyle: schematics_2.SPARTACUS_PRODUCT,
        },
    });
}
exports.addImageZoom = addImageZoom;
