// @spartacus/product/bundles/spartacus-product-image-zoom-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core'), require('@spartacus/storefront')) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/image-zoom/root', ['exports', '@angular/core', '@spartacus/core', '@spartacus/storefront'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product['image-zoom'] = global.spartacus.product['image-zoom'] || {}, global.spartacus.product['image-zoom'].root = {}), global.ng.core, global.core));
}(this, (function (exports, i0, core) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var PRODUCT_IMAGE_ZOOM_FEATURE = 'productImageZoom';

    function defaultImageZoomComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[PRODUCT_IMAGE_ZOOM_FEATURE] = {
                    cmsComponents: ['ProductImagesComponent'],
                },
                _a),
        };
        return config;
    }
    var ProductImageZoomRootModule = /** @class */ (function () {
        function ProductImageZoomRootModule() {
        }
        return ProductImageZoomRootModule;
    }());
    ProductImageZoomRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductImageZoomRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomRootModule });
    ProductImageZoomRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomRootModule, providers: [core.provideDefaultConfigFactory(defaultImageZoomComponentsConfig)], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        providers: [core.provideDefaultConfigFactory(defaultImageZoomComponentsConfig)],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.PRODUCT_IMAGE_ZOOM_FEATURE = PRODUCT_IMAGE_ZOOM_FEATURE;
    exports.ProductImageZoomRootModule = ProductImageZoomRootModule;
    exports.defaultImageZoomComponentsConfig = defaultImageZoomComponentsConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
