// @spartacus/product/bundles/spartacus-product-bulk-pricing-components.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@spartacus/core'), require('@spartacus/storefront'), require('rxjs/operators'), require('@spartacus/product/bulk-pricing/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/bulk-pricing/components', ['exports', '@angular/common', '@angular/core', '@spartacus/core', '@spartacus/storefront', 'rxjs/operators', '@spartacus/product/bulk-pricing/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product['bulk-pricing'] = global.spartacus.product['bulk-pricing'] || {}, global.spartacus.product['bulk-pricing'].components = {}), global.ng.common, global.ng.core, global.core, global.storefront, global.rxjs.operators, global.spartacus.product['bulk-pricing'].core));
}(this, (function (exports, i3, i0, i1, storefront, operators, i2) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);

    var BulkPricingTableComponent = /** @class */ (function () {
        function BulkPricingTableComponent(routingService, bulkPricingService) {
            this.routingService = routingService;
            this.bulkPricingService = bulkPricingService;
            this.PRODUCT_KEY = 'productCode';
        }
        BulkPricingTableComponent.prototype.ngOnInit = function () {
            this.priceTiers$ = this.getPrices();
        };
        BulkPricingTableComponent.prototype.formatQuantity = function (tier) {
            var formattedQuantityRange = '';
            if (!tier.maxQuantity) {
                formattedQuantityRange = tier.minQuantity + '+';
            }
            else {
                formattedQuantityRange = tier.minQuantity + ' - ' + tier.maxQuantity;
            }
            return formattedQuantityRange;
        };
        BulkPricingTableComponent.prototype.getPrices = function () {
            var _this = this;
            return this.routingService.getRouterState().pipe(operators.switchMap(function (state) {
                var productCode = state.state.params[_this.PRODUCT_KEY];
                return _this.bulkPricingService.getBulkPrices(productCode);
            }));
        };
        return BulkPricingTableComponent;
    }());
    BulkPricingTableComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingTableComponent, deps: [{ token: i1__namespace.RoutingService }, { token: i2__namespace.BulkPricingService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    BulkPricingTableComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: BulkPricingTableComponent, selector: "cx-bulk-pricing-table", ngImport: i0__namespace, template: "<ng-container *ngIf=\"priceTiers$ | async as priceTiers\">\n  <div class=\"container\" *ngIf=\"priceTiers.length > 0\">\n    <table class=\"table\">\n      <thead class=\"thead-light\">\n        <tr>\n          <th scope=\"col\">{{ 'bulkPricingTable.quantity' | cxTranslate }}</th>\n          <th scope=\"col\">{{ 'bulkPricingTable.price' | cxTranslate }}</th>\n          <th scope=\"col\">{{ 'bulkPricingTable.discount' | cxTranslate }}</th>\n        </tr>\n      </thead>\n      <tbody>\n        <tr *ngFor=\"let tier of priceTiers\">\n          <td>{{ formatQuantity(tier) }}</td>\n          <td>{{ tier.formattedValue }}</td>\n          <td>{{ tier.formattedDiscount }}</td>\n        </tr>\n      </tbody>\n    </table>\n  </div>\n</ng-container>\n", directives: [{ type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3__namespace.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "async": i3__namespace.AsyncPipe, "cxTranslate": i1__namespace.TranslatePipe } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingTableComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'cx-bulk-pricing-table',
                        templateUrl: './bulk-pricing-table.component.html',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.RoutingService }, { type: i2__namespace.BulkPricingService }]; } });

    var BulkPricingTableModule = /** @class */ (function () {
        function BulkPricingTableModule() {
        }
        return BulkPricingTableModule;
    }());
    BulkPricingTableModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingTableModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    BulkPricingTableModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingTableModule, declarations: [BulkPricingTableComponent], imports: [i3.CommonModule,
            i1.I18nModule,
            storefront.SpinnerModule, i1__namespace.ConfigModule], exports: [BulkPricingTableComponent] });
    BulkPricingTableModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingTableModule, imports: [[
                i3.CommonModule,
                i1.I18nModule,
                storefront.SpinnerModule,
                i1.ConfigModule.withConfig({
                    cmsComponents: {
                        BulkPricingTableComponent: {
                            component: BulkPricingTableComponent,
                        },
                    },
                }),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingTableModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i3.CommonModule,
                            i1.I18nModule,
                            storefront.SpinnerModule,
                            i1.ConfigModule.withConfig({
                                cmsComponents: {
                                    BulkPricingTableComponent: {
                                        component: BulkPricingTableComponent,
                                    },
                                },
                            }),
                        ],
                        declarations: [BulkPricingTableComponent],
                        exports: [BulkPricingTableComponent],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BulkPricingTableComponent = BulkPricingTableComponent;
    exports.BulkPricingTableModule = BulkPricingTableModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
