// @spartacus/product/bundles/spartacus-product-variants-assets.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/variants/assets', ['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product.variants = global.spartacus.product.variants || {}, global.spartacus.product.variants.assets = {})));
}(this, (function (exports) { 'use strict';

    var productVariants = {
        productVariants: {
            style: 'Style',
            selectedStyle: 'Selected style',
            size: 'Size',
            color: 'Color',
            sizeGuideLabel: 'Style guide',
        },
    };

    var en = {
        productVariants: productVariants,
    };

    var productVariantsTranslations = {
        en: en,
    };
    // expose all translation chunk mapping for variants sub features
    var productVariantsTranslationChunksConfig = {
        productVariants: ['productVariants'],
    };

    /**
     * Generated bundle index. Do not edit.
     */

    exports.productVariantsTranslationChunksConfig = productVariantsTranslationChunksConfig;
    exports.productVariantsTranslations = productVariantsTranslations;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
