// @spartacus/product/bundles/spartacus-product-variants-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core'), require('@spartacus/storefront'), require('rxjs'), require('@angular/common'), require('@angular/router')) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/variants/root', ['exports', '@angular/core', '@spartacus/core', '@spartacus/storefront', 'rxjs', '@angular/common', '@angular/router'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product.variants = global.spartacus.product.variants || {}, global.spartacus.product.variants.root = {}), global.ng.core, global.core, global.storefront, global.rxjs, global.ng.common, global.ng.router));
}(this, (function (exports, i0, i1, i2, rxjs, i3, router) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3);

    var PRODUCT_VARIANTS_FEATURE = 'productVariants';

    var ProductVariantStyleIconsComponent = /** @class */ (function () {
        function ProductVariantStyleIconsComponent(config, productListItemContext) {
            var _a, _b;
            this.config = config;
            this.productListItemContext = productListItemContext;
            this.subscriptions = new rxjs.Subscription();
            this.ProductListOutlets = i2.ProductListOutlets;
            this.product$ = (_b = (_a = this.productListItemContext) === null || _a === void 0 ? void 0 : _a.product$) !== null && _b !== void 0 ? _b : rxjs.EMPTY;
            this.variantNames = {};
        }
        ProductVariantStyleIconsComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.setVariantsNames();
            this.subscriptions.add(this.product$.subscribe(function (product) {
                if (product.variantOptions && product.variantOptions.length) {
                    _this.variants = product.variantOptions;
                    _this.setVariantsNames();
                }
            }));
        };
        ProductVariantStyleIconsComponent.prototype.setVariantsNames = function () {
            var _this = this;
            var _a;
            (_a = this.variants) === null || _a === void 0 ? void 0 : _a.forEach(function (variant) {
                if (variant.code && variant.variantOptionQualifiers) {
                    _this.variantNames[variant.code] =
                        _this.getVariantName(variant.variantOptionQualifiers) || '';
                }
            });
        };
        ProductVariantStyleIconsComponent.prototype.getVariantThumbnailUrl = function (variantOptionQualifiers) {
            var _a, _b, _c, _d;
            var thumbnail = variantOptionQualifiers.find(function (item) { return item.qualifier === i1.VariantQualifier.THUMBNAIL; });
            return thumbnail
                ? "" + ((_c = (_b = (_a = this.config) === null || _a === void 0 ? void 0 : _a.backend) === null || _b === void 0 ? void 0 : _b.occ) === null || _c === void 0 ? void 0 : _c.baseUrl) + ((_d = thumbnail.image) === null || _d === void 0 ? void 0 : _d.url)
                : '';
        };
        ProductVariantStyleIconsComponent.prototype.getVariantName = function (variantOptionQualifiers) {
            var rollupProperty = variantOptionQualifiers.find(function (item) { return item.qualifier === i1.VariantQualifier.ROLLUP_PROPERTY; });
            var property = rollupProperty
                ? variantOptionQualifiers.find(function (item) { return item.qualifier === rollupProperty.value; })
                : null;
            return property ? property.value : '';
        };
        ProductVariantStyleIconsComponent.prototype.ngOnDestroy = function () {
            this.subscriptions.unsubscribe();
        };
        return ProductVariantStyleIconsComponent;
    }());
    ProductVariantStyleIconsComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantStyleIconsComponent, deps: [{ token: i1__namespace.OccConfig }, { token: i2__namespace.ProductListItemContext, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductVariantStyleIconsComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ProductVariantStyleIconsComponent, selector: "cx-variant-style-icons", inputs: { variants: "variants" }, ngImport: i0__namespace, template: "<ul class=\"variant-list\" *ngIf=\"variants && variants.length\">\n  <li *ngFor=\"let v of variants\">\n    <img\n      [attr.src]=\"getVariantThumbnailUrl(v.variantOptionQualifiers)\"\n      [attr.title]=\"variantNames[v.code]\"\n      [attr.alt]=\"variantNames[v.code]\"\n    />\n  </li>\n</ul>\n", styles: ["ul{padding:0;white-space:nowrap;overflow:hidden}ul li{display:inline}ul li img{width:50px}"], directives: [{ type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3__namespace.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantStyleIconsComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'cx-variant-style-icons',
                        templateUrl: './product-variant-style-icons.component.html',
                        styleUrls: ['./product-variant-style-icons.component.scss'],
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.OccConfig }, { type: i2__namespace.ProductListItemContext, decorators: [{
                            type: i0.Optional
                        }] }];
        }, propDecorators: { variants: [{
                    type: i0.Input
                }] } });

    var ProductVariantStyleIconsModule = /** @class */ (function () {
        function ProductVariantStyleIconsModule() {
        }
        return ProductVariantStyleIconsModule;
    }());
    ProductVariantStyleIconsModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantStyleIconsModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductVariantStyleIconsModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantStyleIconsModule, declarations: [ProductVariantStyleIconsComponent], imports: [i3.CommonModule, router.RouterModule, i1.UrlModule, i1.I18nModule], exports: [ProductVariantStyleIconsComponent] });
    ProductVariantStyleIconsModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantStyleIconsModule, imports: [[i3.CommonModule, router.RouterModule, i1.UrlModule, i1.I18nModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantStyleIconsModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [i3.CommonModule, router.RouterModule, i1.UrlModule, i1.I18nModule],
                        declarations: [ProductVariantStyleIconsComponent],
                        exports: [ProductVariantStyleIconsComponent],
                    }]
            }] });

    // TODO: Inline this factory when we start releasing Ivy compiled libraries
    function defaultProductVariantsComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[PRODUCT_VARIANTS_FEATURE] = {
                    cmsComponents: ['ProductVariantSelectorComponent'],
                },
                _a),
        };
        return config;
    }
    var ProductVariantsRootModule = /** @class */ (function () {
        function ProductVariantsRootModule() {
        }
        return ProductVariantsRootModule;
    }());
    ProductVariantsRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductVariantsRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsRootModule, imports: [ProductVariantStyleIconsModule] });
    ProductVariantsRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsRootModule, providers: [
            i1.provideDefaultConfigFactory(defaultProductVariantsComponentsConfig),
            i2.provideOutlet({
                id: i2.ProductListOutlets.ITEM_DETAILS,
                position: i2.OutletPosition.AFTER,
                component: ProductVariantStyleIconsComponent,
            }),
        ], imports: [[ProductVariantStyleIconsModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [ProductVariantStyleIconsModule],
                        providers: [
                            i1.provideDefaultConfigFactory(defaultProductVariantsComponentsConfig),
                            i2.provideOutlet({
                                id: i2.ProductListOutlets.ITEM_DETAILS,
                                position: i2.OutletPosition.AFTER,
                                component: ProductVariantStyleIconsComponent,
                            }),
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.PRODUCT_VARIANTS_FEATURE = PRODUCT_VARIANTS_FEATURE;
    exports.ProductVariantStyleIconsComponent = ProductVariantStyleIconsComponent;
    exports.ProductVariantStyleIconsModule = ProductVariantStyleIconsModule;
    exports.ProductVariantsRootModule = ProductVariantsRootModule;
    exports.defaultProductVariantsComponentsConfig = defaultProductVariantsComponentsConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
