// @spartacus/product/bundles/spartacus-product-bulk-pricing-assets.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/bulk-pricing/assets', ['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product['bulk-pricing'] = global.spartacus.product['bulk-pricing'] || {}, global.spartacus.product['bulk-pricing'].assets = {})));
}(this, (function (exports) { 'use strict';

    var bulkPricingTable = {
        quantity: 'Quantity',
        price: 'Price',
        discount: 'Discount',
    };
    var bulkPricing = {
        bulkPricingTable: bulkPricingTable,
    };

    var en = {
        bulkPricing: bulkPricing,
    };

    var bulkPricingTranslations = {
        en: en,
    };
    // expose all translation chunk mapping for bulkPricing feature
    var bulkPricingTranslationChunksConfig = {
        bulkPricing: ['bulkPricingTable'],
    };

    /**
     * Generated bundle index. Do not edit.
     */

    exports.bulkPricingTranslationChunksConfig = bulkPricingTranslationChunksConfig;
    exports.bulkPricingTranslations = bulkPricingTranslations;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
