// @spartacus/product/bundles/spartacus-product-bulk-pricing-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/bulk-pricing/root', ['exports', '@angular/core', '@spartacus/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product['bulk-pricing'] = global.spartacus.product['bulk-pricing'] || {}, global.spartacus.product['bulk-pricing'].root = {}), global.ng.core, global.core));
}(this, (function (exports, i0, core) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var PRODUCT_BULK_PRICING_FEATURE = 'productBulkPricing';

    // TODO: Inline this factory when we start releasing Ivy compiled libraries
    function defaultProductBulkPricingComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[PRODUCT_BULK_PRICING_FEATURE] = {
                    cmsComponents: ['BulkPricingTableComponent'],
                },
                _a),
        };
        return config;
    }
    var BulkPricingRootModule = /** @class */ (function () {
        function BulkPricingRootModule() {
        }
        return BulkPricingRootModule;
    }());
    BulkPricingRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    BulkPricingRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingRootModule });
    BulkPricingRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingRootModule, providers: [
            core.provideDefaultConfigFactory(defaultProductBulkPricingComponentsConfig),
        ], imports: [[]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [],
                        providers: [
                            core.provideDefaultConfigFactory(defaultProductBulkPricingComponentsConfig),
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BulkPricingRootModule = BulkPricingRootModule;
    exports.PRODUCT_BULK_PRICING_FEATURE = PRODUCT_BULK_PRICING_FEATURE;
    exports.defaultProductBulkPricingComponentsConfig = defaultProductBulkPricingComponentsConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
