// @spartacus/product/bundles/spartacus-product-image-zoom-components.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/router'), require('@spartacus/core'), require('@spartacus/storefront'), require('rxjs'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/image-zoom/components', ['exports', '@angular/core', '@angular/common', '@angular/router', '@spartacus/core', '@spartacus/storefront', 'rxjs', 'rxjs/operators'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product['image-zoom'] = global.spartacus.product['image-zoom'] || {}, global.spartacus.product['image-zoom'].components = {}), global.ng.core, global.ng.common, global.ng.router, global.core, global.storefront, global.rxjs, global.rxjs.operators));
}(this, (function (exports, i0, i3$1, router, i3, i1, rxjs, operators) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i3__namespace = /*#__PURE__*/_interopNamespace(i3$1);
    var i3__namespace$1 = /*#__PURE__*/_interopNamespace(i3);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || from);
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var ProductImageZoomThumbnailsComponent = /** @class */ (function () {
        function ProductImageZoomThumbnailsComponent() {
            this.mainMediaContainer = new rxjs.BehaviorSubject({});
            this.productImage = new i0.EventEmitter();
            this.subscription = new rxjs.Subscription();
        }
        ProductImageZoomThumbnailsComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.subscription.add(this.activeThumb.subscribe(function (image) {
                _this.mainMediaContainer.next(image);
            }));
        };
        ProductImageZoomThumbnailsComponent.prototype.openImage = function (image) {
            var _a;
            this.mainMediaContainer.next(image);
            if (typeof ((_a = image.zoom) === null || _a === void 0 ? void 0 : _a.galleryIndex) === 'number') {
                this.productImage.emit({ image: image, index: image.zoom.galleryIndex });
            }
        };
        ProductImageZoomThumbnailsComponent.prototype.isActive = function (thumbnail) {
            return this.mainMediaContainer.asObservable().pipe(operators.filter(i3.isNotNullable), operators.map(function (container) {
                var _a, _b;
                return (((_a = container.zoom) === null || _a === void 0 ? void 0 : _a.url) &&
                    ((_b = thumbnail.zoom) === null || _b === void 0 ? void 0 : _b.url) &&
                    container.zoom.url === thumbnail.zoom.url);
            }));
        };
        ProductImageZoomThumbnailsComponent.prototype.ngOnDestroy = function () {
            this.subscription.unsubscribe();
        };
        return ProductImageZoomThumbnailsComponent;
    }());
    ProductImageZoomThumbnailsComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomThumbnailsComponent, deps: [], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductImageZoomThumbnailsComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ProductImageZoomThumbnailsComponent, selector: "cx-product-image-zoom-thumbnails", inputs: { thumbs$: "thumbs$", activeThumb: "activeThumb" }, outputs: { productImage: "productImage" }, ngImport: i0__namespace, template: "<ng-container *ngIf=\"thumbs$ | async as thumbs\">\n  <cx-carousel\n    *ngIf=\"thumbs.length\"\n    class=\"thumbs\"\n    [items]=\"thumbs\"\n    itemWidth=\"70px\"\n    [hideIndicators]=\"false\"\n    [template]=\"thumb\"\n  ></cx-carousel>\n</ng-container>\n\n<ng-template #thumb let-item=\"item\">\n  <cx-media\n    [container]=\"item.container\"\n    tabindex=\"0\"\n    (focus)=\"openImage(item.container)\"\n    [class.is-active]=\"isActive(item.container) | async\"\n  >\n  </cx-media>\n</ng-template>\n", components: [{ type: i1__namespace.CarouselComponent, selector: "cx-carousel", inputs: ["title", "items", "template", "itemWidth", "hideIndicators", "indicatorIcon", "previousIcon", "nextIcon"] }, { type: i1__namespace.MediaComponent, selector: "cx-media", inputs: ["container", "format", "alt", "role", "loading"], outputs: ["loaded"] }], directives: [{ type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i3__namespace.AsyncPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomThumbnailsComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'cx-product-image-zoom-thumbnails',
                        templateUrl: './product-image-zoom-thumbnails.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return []; }, propDecorators: { productImage: [{
                    type: i0.Output
                }], thumbs$: [{
                    type: i0.Input
                }], activeThumb: [{
                    type: i0.Input
                }] } });

    var ProductImageZoomViewComponent = /** @class */ (function () {
        function ProductImageZoomViewComponent(currentProductService, renderer, cdRef, breakpointService) {
            var _this = this;
            this.currentProductService = currentProductService;
            this.renderer = renderer;
            this.cdRef = cdRef;
            this.breakpointService = breakpointService;
            this.iconType = i1.ICON_TYPE;
            this.mainMediaContainer = new rxjs.BehaviorSubject(null);
            this.defaultImageReady = new rxjs.BehaviorSubject(false);
            this.zoomReady = new rxjs.BehaviorSubject(false);
            this.subscription = new rxjs.Subscription();
            this.mainMediaContainer$ = this.mainMediaContainer.asObservable();
            this.defaultImageReady$ = this.defaultImageReady.asObservable();
            this.zoomReady$ = this.zoomReady.asObservable();
            this.activeThumb = new i0.EventEmitter();
            this.defaultImageClickHandler$ = this.defaultImageReady$.pipe(operators.filter(Boolean), operators.switchMap(function (_) { return rxjs.merge.apply(void 0, __spreadArray([], __read(_this.clickOrDoubleClick(_this.defaultImage)))).pipe(operators.tap(function () { return _this.zoom(); })); }));
            this.zoomImageClickHandler$ = this.zoomReady$.pipe(operators.filter(Boolean), operators.switchMap(function (_) { return rxjs.merge.apply(void 0, __spreadArray([], __read(_this.clickOrDoubleClick(_this.zoomImage)))).pipe(operators.tap(function () { return _this.zoom(); })); }));
            this.startCoords = null;
            this.left = 0;
            this.top = 0;
            this.isZoomed = false;
            this.product$ = this.currentProductService
                .getProduct()
                .pipe(operators.filter(i3.isNotNullable), operators.distinctUntilChanged(), operators.tap(function (p) {
                var _a, _b, _c, _d;
                if (_this.galleryIndex) {
                    var image = Array.isArray((_a = p.images) === null || _a === void 0 ? void 0 : _a.GALLERY)
                        ? (_b = p.images) === null || _b === void 0 ? void 0 : _b.GALLERY.find(function (img) { var _a; return ((_a = img.zoom) === null || _a === void 0 ? void 0 : _a.galleryIndex) === _this.galleryIndex; })
                        : (_c = p.images) === null || _c === void 0 ? void 0 : _c.GALLERY;
                    _this.mainMediaContainer.next(image || null);
                }
                else {
                    _this.mainMediaContainer.next(((_d = p.images) === null || _d === void 0 ? void 0 : _d.PRIMARY) ? p.images.PRIMARY : {});
                }
            }), operators.shareReplay(1));
            this.thumbnails$ = this.product$.pipe(operators.map(function (p) { return _this.createThumbs(p); }), operators.shareReplay(1));
            this.mainImage$ = rxjs.combineLatest([
                this.product$,
                this.mainMediaContainer$,
            ]).pipe(operators.map(function (_e) {
                var _f = __read(_e, 2), container = _f[1];
                return container;
            }));
        }
        Object.defineProperty(ProductImageZoomViewComponent.prototype, "defaultImage", {
            get: function () {
                return this._defaultImage;
            },
            set: function (el) {
                if (el) {
                    this._defaultImage = el;
                    this.defaultImageReady.next(true);
                }
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(ProductImageZoomViewComponent.prototype, "zoomImage", {
            get: function () {
                return this._zoomImage;
            },
            set: function (el) {
                if (el) {
                    this._zoomImage = el;
                    this.zoomReady.next(true);
                }
            },
            enumerable: false,
            configurable: true
        });
        ProductImageZoomViewComponent.prototype.ngOnInit = function () {
            this.subscription.add(this.defaultImageClickHandler$.subscribe());
            this.subscription.add(this.zoomImageClickHandler$.subscribe());
        };
        ProductImageZoomViewComponent.prototype.openImage = function (item) {
            this.mainMediaContainer.next(item);
            this.activeThumb.emit(item);
        };
        /** find the index of the main media in the list of media */
        ProductImageZoomViewComponent.prototype.getActive = function () {
            var _a, _b, _c, _d;
            if (Array.isArray(this.mainMediaContainer.value)) {
                return ((_a = this.mainMediaContainer.value[0].thumbnail) === null || _a === void 0 ? void 0 : _a.galleryIndex) || 0;
            }
            return ((_d = (_c = (_b = this.mainMediaContainer) === null || _b === void 0 ? void 0 : _b.value) === null || _c === void 0 ? void 0 : _c.thumbnail) === null || _d === void 0 ? void 0 : _d.galleryIndex) || 0;
        };
        ProductImageZoomViewComponent.prototype.getPreviousProduct = function (thumbs) {
            var active = this.getActive();
            if (active === 0) {
                return thumbs[active];
            }
            return thumbs[active - 1];
        };
        ProductImageZoomViewComponent.prototype.getNextProduct = function (thumbs) {
            var active = this.getActive();
            if (active === thumbs.length - 1) {
                return thumbs[active];
            }
            return thumbs[active + 1];
        };
        /**
         * Zoom in or out of the image
         */
        ProductImageZoomViewComponent.prototype.zoom = function () {
            this.isZoomed = !this.isZoomed;
            this.startCoords = null;
            this.left = 0;
            this.top = 0;
            this.cdRef.markForCheck();
        };
        /**
         * Touch screen image pan
         *
         * @param event
         */
        ProductImageZoomViewComponent.prototype.touchMove = function (event) {
            var _a, _b, _c, _d;
            var touch = event.touches[0] || event.changedTouches[0];
            var boundingRect = (_b = (_a = this.zoomedImage) === null || _a === void 0 ? void 0 : _a.nativeElement) === null || _b === void 0 ? void 0 : _b.getBoundingClientRect();
            var imageElement = (_d = (_c = this.zoomedImage) === null || _c === void 0 ? void 0 : _c.nativeElement) === null || _d === void 0 ? void 0 : _d.firstChild;
            if (!this.startCoords)
                this.startCoords = { x: touch.clientX, y: touch.clientY };
            this.left += touch.clientX - this.startCoords.x;
            this.top += touch.clientY - this.startCoords.y;
            this.moveImage(this.left, this.top, boundingRect, imageElement);
            this.startCoords = { x: touch.clientX, y: touch.clientY };
        };
        /**
         * Clears touch location
         */
        ProductImageZoomViewComponent.prototype.clearTouch = function () {
            this.startCoords = null;
        };
        /**
         * Pointer image pan
         *
         * @param event
         */
        ProductImageZoomViewComponent.prototype.pointerMove = function (event) {
            var boundingRect = this.zoomedImage.nativeElement.getBoundingClientRect();
            var imageElement = this.zoomedImage.nativeElement.firstChild;
            var _e = this.calculatePointerMovePosition(this.zoomedImage, event.clientX, event.clientY), positionX = _e.positionX, positionY = _e.positionY;
            this.moveImage(positionX, positionY, boundingRect, imageElement);
        };
        ProductImageZoomViewComponent.prototype.changeImage = function (event) {
            this.mainMediaContainer.next(event.image);
        };
        /**
         * Applies the offset from touchMove or pointerMove to the image element
         *
         * @param positionX
         * @param positionY
         * @param boundingRect
         * @param imageElement
         */
        ProductImageZoomViewComponent.prototype.moveImage = function (positionX, positionY, boundingRect, imageElement) {
            var _e = this.handleOutOfBounds(positionX, positionY, imageElement, boundingRect), x = _e.x, y = _e.y;
            if (imageElement) {
                this.renderer.setStyle(imageElement, 'left', x + 'px');
                this.renderer.setStyle(imageElement, 'top', y + 'px');
            }
        };
        ProductImageZoomViewComponent.prototype.ngOnDestroy = function () {
            this.subscription.unsubscribe();
        };
        /**
         * Returns click and dblclick event mapping for the given element
         *
         * @param element
         */
        ProductImageZoomViewComponent.prototype.clickOrDoubleClick = function (element) {
            return [
                rxjs.fromEvent(element.nativeElement, 'click').pipe(operators.switchMapTo(this.breakpointService.isUp(i1.BREAKPOINT.md)), operators.filter(Boolean)),
                rxjs.fromEvent(element.nativeElement, 'dblclick').pipe(operators.switchMapTo(this.breakpointService.isDown(i1.BREAKPOINT.lg)), operators.filter(Boolean)),
            ];
        };
        /**
         * Return an array of CarouselItems for the product thumbnails.
         * In case there are less then 2 thumbs, we return null.
         */
        ProductImageZoomViewComponent.prototype.createThumbs = function (product) {
            if (!product.images ||
                !product.images.GALLERY ||
                product.images.GALLERY.length < 2) {
                return [];
            }
            var images = product.images.GALLERY;
            return images.map(function (c) { return rxjs.of({ container: c }); });
        };
        /**
         * Keeps the zoom image from leaving the bounding container
         *
         * @param positionX
         * @param positionY
         * @param imageElement
         * @param boundingRect
         */
        ProductImageZoomViewComponent.prototype.handleOutOfBounds = function (positionX, positionY, imageElement, boundingRect) {
            var paddingX = 60;
            var paddingY = 60;
            if (positionY <= -(imageElement === null || imageElement === void 0 ? void 0 : imageElement.height) + paddingY) {
                positionY = -(imageElement === null || imageElement === void 0 ? void 0 : imageElement.height) + paddingY;
            }
            if (positionY >= (boundingRect === null || boundingRect === void 0 ? void 0 : boundingRect.height) - paddingY) {
                positionY = (boundingRect === null || boundingRect === void 0 ? void 0 : boundingRect.height) - paddingY;
            }
            if (positionX <=
                -(imageElement === null || imageElement === void 0 ? void 0 : imageElement.width) - (boundingRect === null || boundingRect === void 0 ? void 0 : boundingRect.width) / 2 + paddingX) {
                positionX = -(imageElement === null || imageElement === void 0 ? void 0 : imageElement.width) - (boundingRect === null || boundingRect === void 0 ? void 0 : boundingRect.width) / 2 + paddingX;
            }
            if (positionX >= (imageElement === null || imageElement === void 0 ? void 0 : imageElement.width) + (boundingRect === null || boundingRect === void 0 ? void 0 : boundingRect.width) / 2 - paddingX) {
                positionX = (imageElement === null || imageElement === void 0 ? void 0 : imageElement.width) + (boundingRect === null || boundingRect === void 0 ? void 0 : boundingRect.width) / 2 - paddingX;
            }
            return { x: positionX, y: positionY };
        };
        /**
         * Returns the position of the image based on the cursor pointer
         *
         * @param element
         * @param clientX
         * @param clientY
         */
        ProductImageZoomViewComponent.prototype.calculatePointerMovePosition = function (element, clientX, clientY) {
            var boundingRect = element.nativeElement.getBoundingClientRect();
            var x = clientX - boundingRect.left;
            var y = clientY - boundingRect.top;
            var positionX = -x + element.nativeElement.clientWidth / 2;
            var positionY = -y + element.nativeElement.clientHeight / 2;
            return { positionX: positionX, positionY: positionY };
        };
        return ProductImageZoomViewComponent;
    }());
    ProductImageZoomViewComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomViewComponent, deps: [{ token: i1__namespace.CurrentProductService }, { token: i0__namespace.Renderer2 }, { token: i0__namespace.ChangeDetectorRef }, { token: i1__namespace.BreakpointService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductImageZoomViewComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ProductImageZoomViewComponent, selector: "cx-product-image-zoom-view", inputs: { galleryIndex: "galleryIndex" }, viewQueries: [{ propertyName: "defaultImage", first: true, predicate: ["defaultImage"], descendants: true, read: i0.ElementRef }, { propertyName: "zoomImage", first: true, predicate: ["zoomContainer"], descendants: true, read: i0.ElementRef }, { propertyName: "zoomedImage", first: true, predicate: ["zoomedImage"], descendants: true, read: i0.ElementRef }], ngImport: i0__namespace, template: "<ng-container *ngIf=\"mainImage$ | async as main\">\n  <div class=\"cx-main-image-group\" *ngIf=\"thumbnails$ | async as thumbs\">\n    <div class=\"cx-navigate-image\">\n      <button\n        class=\"btn btn-link\"\n        *ngIf=\"getPreviousProduct(thumbs) | async as previousProduct\"\n        (click)=\"openImage(previousProduct.container)\"\n      >\n        <cx-icon [type]=\"iconType.CARET_LEFT\"></cx-icon>\n      </button>\n    </div>\n    <cx-media\n      #defaultImage\n      class=\"cx-default-image-zoom\"\n      *ngIf=\"!isZoomed\"\n      [container]=\"main\"\n    >\n    </cx-media>\n    <div #zoomContainer class=\"cx-zoom-container\" *ngIf=\"isZoomed\">\n      <cx-media\n        #zoomedImage\n        class=\"cx-image-zoomed\"\n        [container]=\"main\"\n        (mousemove)=\"pointerMove($event)\"\n        (touchmove)=\"touchMove($event)\"\n        (touchend)=\"clearTouch()\"\n      >\n      </cx-media>\n    </div>\n    <div class=\"cx-navigate-image\">\n      <button\n        class=\"btn btn-link\"\n        *ngIf=\"getNextProduct(thumbs) | async as nextProduct\"\n        (click)=\"openImage(nextProduct.container)\"\n      >\n        <cx-icon [type]=\"iconType.CARET_RIGHT\"></cx-icon>\n      </button>\n    </div>\n  </div>\n</ng-container>\n\n<cx-product-image-zoom-thumbnails\n  [thumbs$]=\"thumbnails$\"\n  [activeThumb]=\"activeThumb\"\n  (productImage)=\"changeImage($event)\"\n></cx-product-image-zoom-thumbnails>\n", components: [{ type: i1__namespace.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }, { type: i1__namespace.MediaComponent, selector: "cx-media", inputs: ["container", "format", "alt", "role", "loading"], outputs: ["loaded"] }, { type: ProductImageZoomThumbnailsComponent, selector: "cx-product-image-zoom-thumbnails", inputs: ["thumbs$", "activeThumb"], outputs: ["productImage"] }], directives: [{ type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i3__namespace.AsyncPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomViewComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'cx-product-image-zoom-view',
                        templateUrl: './product-image-zoom-view.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.CurrentProductService }, { type: i0__namespace.Renderer2 }, { type: i0__namespace.ChangeDetectorRef }, { type: i1__namespace.BreakpointService }]; }, propDecorators: { galleryIndex: [{
                    type: i0.Input
                }], defaultImage: [{
                    type: i0.ViewChild,
                    args: ['defaultImage', { read: i0.ElementRef }]
                }], zoomImage: [{
                    type: i0.ViewChild,
                    args: ['zoomContainer', { read: i0.ElementRef }]
                }], zoomedImage: [{
                    type: i0.ViewChild,
                    args: ['zoomedImage', { read: i0.ElementRef }]
                }] } });

    var ProductImageZoomDialogComponent = /** @class */ (function () {
        function ProductImageZoomDialogComponent(launchDialogService, el) {
            this.launchDialogService = launchDialogService;
            this.el = el;
            this.iconType = i1.ICON_TYPE;
            this.focusConfig = {
                trap: true,
                block: true,
                autofocus: 'button',
                focusOnEscape: true,
            };
        }
        ProductImageZoomDialogComponent.prototype.handleClick = function (event) {
            // Close on click outside the dialog window
            if (event.target.tagName === this.el.nativeElement.tagName) {
                this.close('Cross click');
            }
        };
        ProductImageZoomDialogComponent.prototype.close = function (reason) {
            if (reason === void 0) { reason = ''; }
            this.launchDialogService.closeDialog(reason);
        };
        return ProductImageZoomDialogComponent;
    }());
    ProductImageZoomDialogComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomDialogComponent, deps: [{ token: i1__namespace.LaunchDialogService }, { token: i0__namespace.ElementRef }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductImageZoomDialogComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ProductImageZoomDialogComponent, selector: "cx-product-image-zoom-dialog", inputs: { galleryIndex: "galleryIndex" }, host: { listeners: { "click": "handleClick($event)" } }, ngImport: i0__namespace, template: "<div class=\"cx-image-zoom-dialog\" [cxFocus]=\"focusConfig\">\n  <div class=\"cx-dialog-content\">\n    <div class=\"cx-dialog-header\">\n      <button\n        type=\"button\"\n        class=\"close\"\n        [attr.aria-label]=\"'productImageZoomDialog.close' | cxTranslate\"\n        (click)=\"close('cross click')\"\n      >\n        <span aria-hidden=\"true\">\n          <cx-icon [type]=\"iconType.CLOSE\"></cx-icon>\n        </span>\n      </button>\n    </div>\n    <div class=\"cx-dialog-body\">\n      <cx-product-image-zoom-view\n        [galleryIndex]=\"galleryIndex\"\n      ></cx-product-image-zoom-view>\n    </div>\n  </div>\n</div>\n", components: [{ type: i1__namespace.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }, { type: ProductImageZoomViewComponent, selector: "cx-product-image-zoom-view", inputs: ["galleryIndex"] }], directives: [{ type: i1__namespace.FocusDirective, selector: "[cxFocus]", inputs: ["cxFocus"] }], pipes: { "cxTranslate": i3__namespace$1.TranslatePipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomDialogComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'cx-product-image-zoom-dialog',
                        templateUrl: 'product-image-zoom-dialog.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.LaunchDialogService }, { type: i0__namespace.ElementRef }]; }, propDecorators: { galleryIndex: [{
                    type: i0.Input
                }], handleClick: [{
                    type: i0.HostListener,
                    args: ['click', ['$event']]
                }] } });

    var defaultProductImageZoomLayoutConfig = {
        launch: {
            PRODUCT_IMAGE_ZOOM: {
                inline: true,
                component: ProductImageZoomDialogComponent,
                dialogType: i1.DIALOG_TYPE.DIALOG,
            },
        },
    };

    var ProductImageZoomTriggerComponent = /** @class */ (function () {
        function ProductImageZoomTriggerComponent(launchDialogService, vcr) {
            this.launchDialogService = launchDialogService;
            this.vcr = vcr;
            this.iconType = i1.ICON_TYPE;
            this.subscriptions = new rxjs.Subscription();
            this.dialogClose = new i0.EventEmitter();
        }
        Object.defineProperty(ProductImageZoomTriggerComponent.prototype, "expandImage", {
            set: function (expand) {
                if (expand) {
                    this.triggerZoom();
                }
            },
            enumerable: false,
            configurable: true
        });
        ProductImageZoomTriggerComponent.prototype.triggerZoom = function () {
            var _this = this;
            var component = this.launchDialogService.launch("PRODUCT_IMAGE_ZOOM" /* PRODUCT_IMAGE_ZOOM */, this.vcr);
            if (component) {
                this.subscriptions.add(rxjs.combineLatest([component, this.launchDialogService.dialogClose])
                    .pipe(operators.tap(function (_a) {
                    var _b = __read(_a, 1), comp = _b[0];
                    if (_this.galleryIndex) {
                        comp.instance.galleryIndex = _this.galleryIndex;
                    }
                }), operators.filter(function (_a) {
                    var _b = __read(_a, 2), close = _b[1];
                    return Boolean(close);
                }), operators.tap(function (_a) {
                    var _b = __read(_a, 1), comp = _b[0];
                    _this.launchDialogService.clear("PRODUCT_IMAGE_ZOOM" /* PRODUCT_IMAGE_ZOOM */);
                    comp === null || comp === void 0 ? void 0 : comp.destroy();
                    _this.dialogClose.emit();
                    _this.expandButton.nativeElement.focus();
                }))
                    .subscribe());
            }
        };
        ProductImageZoomTriggerComponent.prototype.ngOnDestroy = function () {
            this.subscriptions.unsubscribe();
        };
        return ProductImageZoomTriggerComponent;
    }());
    ProductImageZoomTriggerComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomTriggerComponent, deps: [{ token: i1__namespace.LaunchDialogService }, { token: i0__namespace.ViewContainerRef }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductImageZoomTriggerComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ProductImageZoomTriggerComponent, selector: "cx-product-image-zoom-trigger", inputs: { galleryIndex: "galleryIndex", expandImage: "expandImage" }, outputs: { dialogClose: "dialogClose" }, viewQueries: [{ propertyName: "expandButton", first: true, predicate: ["expandButton"], descendants: true }], ngImport: i0__namespace, template: "<button\n  #expandButton\n  class=\"btn btn-link cx-action-link\"\n  (click)=\"triggerZoom()\"\n>\n  <span>\n    {{ 'productImageZoomTrigger.expand' | cxTranslate }}\n    <cx-icon [type]=\"iconType.EXPAND_ARROWS\"></cx-icon\n  ></span>\n</button>\n", components: [{ type: i1__namespace.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }], pipes: { "cxTranslate": i3__namespace$1.TranslatePipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomTriggerComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'cx-product-image-zoom-trigger',
                        templateUrl: 'product-image-zoom-trigger.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.LaunchDialogService }, { type: i0__namespace.ViewContainerRef }]; }, propDecorators: { expandButton: [{
                    type: i0.ViewChild,
                    args: ['expandButton']
                }], galleryIndex: [{
                    type: i0.Input
                }], expandImage: [{
                    type: i0.Input
                }], dialogClose: [{
                    type: i0.Output
                }] } });

    var ProductImageZoomProductImagesComponent = /** @class */ (function (_super) {
        __extends(ProductImageZoomProductImagesComponent, _super);
        function ProductImageZoomProductImagesComponent(currentProductService) {
            var _this = _super.call(this, currentProductService) || this;
            _this.currentProductService = currentProductService;
            _this.expandImage = new rxjs.BehaviorSubject(false);
            return _this;
        }
        ProductImageZoomProductImagesComponent.prototype.openImage = function (item) {
            var _a, _b;
            this.mainMediaContainer.next(item);
            this.selectedIndex = (_b = (_a = this.mainMediaContainer.value) === null || _a === void 0 ? void 0 : _a.zoom) === null || _b === void 0 ? void 0 : _b.galleryIndex;
        };
        /**
         * Opens image zoom dialog.
         */
        ProductImageZoomProductImagesComponent.prototype.triggerZoom = function (value) {
            this.expandImage.next(value);
        };
        return ProductImageZoomProductImagesComponent;
    }(i1.ProductImagesComponent));
    ProductImageZoomProductImagesComponent.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomProductImagesComponent, deps: [{ token: i1__namespace.CurrentProductService }], target: i0__namespace.ɵɵFactoryTarget.Component });
    ProductImageZoomProductImagesComponent.ɵcmp = i0__namespace.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ProductImageZoomProductImagesComponent, selector: "cx-product-images", usesInheritance: true, ngImport: i0__namespace, template: "<ng-container *ngIf=\"mainImage$ | async as main\">\n  <cx-media [container]=\"main\" (click)=\"triggerZoom(true)\"></cx-media>\n  <cx-product-image-zoom-trigger\n    [expandImage]=\"expandImage.value\"\n    [galleryIndex]=\"selectedIndex\"\n    (dialogClose)=\"triggerZoom(false)\"\n  ></cx-product-image-zoom-trigger>\n</ng-container>\n\n<ng-container *ngIf=\"thumbs$ | async as thumbs\">\n  <cx-carousel\n    *ngIf=\"thumbs.length\"\n    class=\"thumbs\"\n    [items]=\"thumbs\"\n    itemWidth=\"120px\"\n    [hideIndicators]=\"false\"\n    [template]=\"thumb\"\n  ></cx-carousel>\n</ng-container>\n\n<ng-template #thumb let-item=\"item\">\n  <cx-media\n    [container]=\"item.container\"\n    tabindex=\"0\"\n    (focus)=\"openImage(item.container)\"\n    [class.is-active]=\"isActive(item.container) | async\"\n    format=\"product\"\n  >\n  </cx-media>\n</ng-template>\n", components: [{ type: i1__namespace.MediaComponent, selector: "cx-media", inputs: ["container", "format", "alt", "role", "loading"], outputs: ["loaded"] }, { type: ProductImageZoomTriggerComponent, selector: "cx-product-image-zoom-trigger", inputs: ["galleryIndex", "expandImage"], outputs: ["dialogClose"] }, { type: i1__namespace.CarouselComponent, selector: "cx-carousel", inputs: ["title", "items", "template", "itemWidth", "hideIndicators", "indicatorIcon", "previousIcon", "nextIcon"] }], directives: [{ type: i3__namespace.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i3__namespace.AsyncPipe }, changeDetection: i0__namespace.ChangeDetectionStrategy.OnPush });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomProductImagesComponent, decorators: [{
                type: i0.Component,
                args: [{
                        selector: 'cx-product-images',
                        templateUrl: './product-image-zoom-product-images.component.html',
                        changeDetection: i0.ChangeDetectionStrategy.OnPush,
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.CurrentProductService }]; } });

    var ProductImageZoomModule = /** @class */ (function () {
        function ProductImageZoomModule() {
        }
        return ProductImageZoomModule;
    }());
    ProductImageZoomModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductImageZoomModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomModule, declarations: [ProductImageZoomDialogComponent,
            ProductImageZoomProductImagesComponent,
            ProductImageZoomThumbnailsComponent,
            ProductImageZoomTriggerComponent,
            ProductImageZoomViewComponent], imports: [i1.CarouselModule,
            i3$1.CommonModule,
            i3.I18nModule,
            i1.IconModule,
            i1.KeyboardFocusModule,
            i1.MediaModule,
            i1.OutletModule,
            router.RouterModule], exports: [ProductImageZoomDialogComponent,
            ProductImageZoomProductImagesComponent,
            ProductImageZoomThumbnailsComponent,
            ProductImageZoomTriggerComponent,
            ProductImageZoomViewComponent] });
    ProductImageZoomModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomModule, providers: [
            i3.provideConfig(defaultProductImageZoomLayoutConfig),
            i3.provideDefaultConfig({
                cmsComponents: {
                    ProductImagesComponent: {
                        component: ProductImageZoomProductImagesComponent,
                    },
                },
            }),
        ], imports: [[
                i1.CarouselModule,
                i3$1.CommonModule,
                i3.I18nModule,
                i1.IconModule,
                i1.KeyboardFocusModule,
                i1.MediaModule,
                i1.OutletModule,
                router.RouterModule,
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i1.CarouselModule,
                            i3$1.CommonModule,
                            i3.I18nModule,
                            i1.IconModule,
                            i1.KeyboardFocusModule,
                            i1.MediaModule,
                            i1.OutletModule,
                            router.RouterModule,
                        ],
                        providers: [
                            i3.provideConfig(defaultProductImageZoomLayoutConfig),
                            i3.provideDefaultConfig({
                                cmsComponents: {
                                    ProductImagesComponent: {
                                        component: ProductImageZoomProductImagesComponent,
                                    },
                                },
                            }),
                        ],
                        declarations: [
                            ProductImageZoomDialogComponent,
                            ProductImageZoomProductImagesComponent,
                            ProductImageZoomThumbnailsComponent,
                            ProductImageZoomTriggerComponent,
                            ProductImageZoomViewComponent,
                        ],
                        exports: [
                            ProductImageZoomDialogComponent,
                            ProductImageZoomProductImagesComponent,
                            ProductImageZoomThumbnailsComponent,
                            ProductImageZoomTriggerComponent,
                            ProductImageZoomViewComponent,
                        ],
                    }]
            }] });

    var ProductImageZoomComponentsModule = /** @class */ (function () {
        function ProductImageZoomComponentsModule() {
        }
        return ProductImageZoomComponentsModule;
    }());
    ProductImageZoomComponentsModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomComponentsModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductImageZoomComponentsModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomComponentsModule, imports: [ProductImageZoomModule] });
    ProductImageZoomComponentsModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomComponentsModule, imports: [[ProductImageZoomModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductImageZoomComponentsModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [ProductImageZoomModule],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ProductImageZoomComponentsModule = ProductImageZoomComponentsModule;
    exports.ProductImageZoomDialogComponent = ProductImageZoomDialogComponent;
    exports.ProductImageZoomModule = ProductImageZoomModule;
    exports.ProductImageZoomProductImagesComponent = ProductImageZoomProductImagesComponent;
    exports.ProductImageZoomThumbnailsComponent = ProductImageZoomThumbnailsComponent;
    exports.ProductImageZoomTriggerComponent = ProductImageZoomTriggerComponent;
    exports.ProductImageZoomViewComponent = ProductImageZoomViewComponent;
    exports.defaultProductImageZoomLayoutConfig = defaultProductImageZoomLayoutConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
