// @spartacus/product/bundles/spartacus-product-image-zoom-assets.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/image-zoom/assets', ['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product['image-zoom'] = global.spartacus.product['image-zoom'] || {}, global.spartacus.product['image-zoom'].assets = {})));
}(this, (function (exports) { 'use strict';

    var productImageZoomTrigger = {
        expand: 'Expand image',
    };
    var productImageZoomDialog = {
        close: 'Close',
    };
    var productImageZoom = {
        productImageZoomTrigger: productImageZoomTrigger,
        productImageZoomDialog: productImageZoomDialog,
    };

    var en = {
        productImageZoom: productImageZoom,
    };

    var productImageZoomTranslations = {
        en: en,
    };
    // expose all translation chunk mapping for imageZoom feature
    var productImageZoomTranslationChunksConfig = {
        productImageZoom: ['productImageZoomTrigger', 'productImageZoomDialog'],
    };

    /**
     * Generated bundle index. Do not edit.
     */

    exports.productImageZoomTranslationChunksConfig = productImageZoomTranslationChunksConfig;
    exports.productImageZoomTranslations = productImageZoomTranslations;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
