// @spartacus/product/bundles/spartacus-product-bulk-pricing-core.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@spartacus/core'), require('@angular/core'), require('rxjs'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/bulk-pricing/core', ['exports', '@spartacus/core', '@angular/core', 'rxjs', 'rxjs/operators'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product['bulk-pricing'] = global.spartacus.product['bulk-pricing'] || {}, global.spartacus.product['bulk-pricing'].core = {}), global.core, global.ng.core, global.rxjs, global.rxjs.operators));
}(this, (function (exports, i1, i0, rxjs, operators) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);
    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var BulkPricingService = /** @class */ (function () {
        function BulkPricingService(productService) {
            this.productService = productService;
            this.PRODUCT_SCOPE = "bulkPrices" /* BULK_PRICES */;
        }
        BulkPricingService.prototype.getBulkPrices = function (productCode) {
            var _this = this;
            return this.productService.get(productCode, this.PRODUCT_SCOPE).pipe(operators.switchMap(function (productPriceScope) {
                return rxjs.of(_this.convert(productPriceScope));
            }));
        };
        BulkPricingService.prototype.convert = function (productPriceScope) {
            var _this = this;
            var _a;
            var bulkPrices = [];
            if (productPriceScope) {
                var basePrice_1 = (_a = productPriceScope.price) === null || _a === void 0 ? void 0 : _a.value;
                var volumePrices = productPriceScope.volumePrices;
                bulkPrices = volumePrices === null || volumePrices === void 0 ? void 0 : volumePrices.map(function (volumePrice) { return _this.parsePrice(volumePrice, basePrice_1); });
            }
            return bulkPrices;
        };
        BulkPricingService.prototype.parsePrice = function (priceTier, basePrice) {
            var bulkPriceTemplate = {
                currencyIso: priceTier.currencyIso,
                formattedValue: priceTier.formattedValue,
                maxQuantity: priceTier.maxQuantity,
                minQuantity: priceTier.minQuantity,
                priceType: priceTier.priceType,
                value: priceTier.value,
                formattedDiscount: '',
                discount: 0,
            };
            return this.calculateDiscount(bulkPriceTemplate, basePrice);
        };
        BulkPricingService.prototype.calculateDiscount = function (bulkPriceTemplate, basePrice) {
            var bulkPrice = Object.assign({}, bulkPriceTemplate);
            var tierPrice = bulkPriceTemplate.value;
            if (tierPrice && basePrice) {
                var discount = Math.round(100.0 - (tierPrice / basePrice) * 100);
                var formatted = discount === 0 ? discount + "%" : "-" + discount + "%";
                bulkPrice.formattedDiscount = formatted;
                bulkPrice.discount = discount;
            }
            return bulkPrice;
        };
        return BulkPricingService;
    }());
    BulkPricingService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingService, deps: [{ token: i1__namespace.ProductService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    BulkPricingService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: BulkPricingService, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.ProductService }]; } });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BulkPricingService = BulkPricingService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
