// @spartacus/product/bundles/spartacus-product-variants-occ.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@spartacus/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/product/variants/occ', ['exports', '@angular/common', '@angular/core', '@spartacus/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.product = global.spartacus.product || {}, global.spartacus.product.variants = global.spartacus.product.variants || {}, global.spartacus.product.variants.occ = {}), global.ng.common, global.ng.core, global.core));
}(this, (function (exports, common, i0, core) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var defaultOccProductVariantsConfig = {
        backend: {
            occ: {
                endpoints: {
                    product: {
                        variants: 'products/${productCode}?fields=name,purchasable,baseOptions(DEFAULT),baseProduct,variantOptions(DEFAULT),variantType',
                    },
                },
            },
        },
    };

    var ProductVariantsOccModule = /** @class */ (function () {
        function ProductVariantsOccModule() {
        }
        return ProductVariantsOccModule;
    }());
    ProductVariantsOccModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsOccModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ProductVariantsOccModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsOccModule, imports: [common.CommonModule] });
    ProductVariantsOccModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsOccModule, providers: [core.provideDefaultConfig(defaultOccProductVariantsConfig)], imports: [[common.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ProductVariantsOccModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [common.CommonModule],
                        providers: [core.provideDefaultConfig(defaultOccProductVariantsConfig)],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ProductVariantsOccModule = ProductVariantsOccModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
