// @spartacus/cart/esm2015/saved-cart/root/events/saved-cart.events.js
import { CartEvent } from '@spartacus/core';
/**
 * Base saved cart event. Most cart events should have these properties.
 */
export class SavedCartEvent extends CartEvent {
}
// =====================================================================
export class DeleteSavedCartEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
DeleteSavedCartEvent.type = 'DeleteSavedCartEvent';
export class DeleteSavedCartSuccessEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
DeleteSavedCartSuccessEvent.type = 'DeleteSavedCartSuccessEvent';
export class DeleteSavedCartFailEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
DeleteSavedCartFailEvent.type = 'DeleteSavedCartFailEvent';
export class SaveCartEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
SaveCartEvent.type = 'SaveCartEvent';
export class SaveCartSuccessEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
SaveCartSuccessEvent.type = 'SaveCartSuccessEvent';
export class SaveCartFailEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
SaveCartFailEvent.type = 'SaveCartEvent';
export class RestoreSavedCartEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
RestoreSavedCartEvent.type = 'RestoreSavedCartEvent';
export class RestoreSavedCartSuccessEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
RestoreSavedCartSuccessEvent.type = 'RestoreSavedCartSuccessEvent';
export class RestoreSavedCartFailEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
RestoreSavedCartFailEvent.type = 'RestoreSavedCartFailEvent';
export class EditSavedCartEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
EditSavedCartEvent.type = 'EditSavedCartEvent';
export class EditSavedCartSuccessEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
EditSavedCartSuccessEvent.type = 'EditSavedCartSuccessEvent';
export class EditSavedCartFailEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
EditSavedCartFailEvent.type = 'EditSavedCartFailEvent';
export class CloneSavedCartEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
CloneSavedCartEvent.type = 'CloneSavedCartEvent';
export class CloneSavedCartSuccessEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
CloneSavedCartSuccessEvent.type = 'CloneSavedCartSuccessEvent';
export class CloneSavedCartFailEvent extends SavedCartEvent {
}
/**
 * Event's type
 */
CloneSavedCartFailEvent.type = 'CloneSavedCartFailEvent';
