// @spartacus/cart/bundles/spartacus-cart-import-export-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/cart/import-export/root', ['exports', '@angular/core', '@spartacus/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.cart = global.spartacus.cart || {}, global.spartacus.cart['import-export'] = global.spartacus.cart['import-export'] || {}, global.spartacus.cart['import-export'].root = {}), global.ng.core, global.core));
}(this, (function (exports, i0, core) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var CART_IMPORT_EXPORT_FEATURE = 'cartImportExport';

    var _a, _b;
    var ImportExportRootModule = /** @class */ (function () {
        function ImportExportRootModule() {
        }
        return ImportExportRootModule;
    }());
    ImportExportRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ImportExportRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ImportExportRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ImportExportRootModule });
    ImportExportRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ImportExportRootModule, providers: [
            core.provideDefaultConfig({
                featureModules: (_a = {},
                    _a[CART_IMPORT_EXPORT_FEATURE] = {
                        cmsComponents: [
                            'ImportExportOrderEntriesComponent',
                            'ImportOrderEntriesComponent',
                            'ExportOrderEntriesComponent',
                        ],
                    },
                    _a),
            }),
        ] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: ImportExportRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        providers: [
                            core.provideDefaultConfig({
                                featureModules: (_b = {},
                                    _b[CART_IMPORT_EXPORT_FEATURE] = {
                                        cmsComponents: [
                                            'ImportExportOrderEntriesComponent',
                                            'ImportOrderEntriesComponent',
                                            'ExportOrderEntriesComponent',
                                        ],
                                    },
                                    _b),
                            }),
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CART_IMPORT_EXPORT_FEATURE = CART_IMPORT_EXPORT_FEATURE;
    exports.ImportExportRootModule = ImportExportRootModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
