// @spartacus/cart/bundles/spartacus-cart-saved-cart-occ.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core'), require('rxjs/operators'), require('@angular/common/http'), require('@angular/common'), require('@spartacus/cart/saved-cart/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/cart/saved-cart/occ', ['exports', '@angular/core', '@spartacus/core', 'rxjs/operators', '@angular/common/http', '@angular/common', '@spartacus/cart/saved-cart/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.cart = global.spartacus.cart || {}, global.spartacus.cart['saved-cart'] = global.spartacus.cart['saved-cart'] || {}, global.spartacus.cart['saved-cart'].occ = {}), global.ng.core, global.core, global.rxjs.operators, global.ng.common.http, global.ng.common, global.spartacus.cart['saved-cart'].core));
}(this, (function (exports, i0, i2, operators, i1, common, core) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var OccSavedCartAdapter = /** @class */ (function () {
        function OccSavedCartAdapter(http, occEndpoints, converter) {
            this.http = http;
            this.occEndpoints = occEndpoints;
            this.converter = converter;
        }
        OccSavedCartAdapter.prototype.load = function (userId, cartId) {
            return this.http
                .get(this.getSavedCartEndpoint(userId, cartId))
                .pipe(operators.pluck('savedCartData'), this.converter.pipeable(i2.CART_NORMALIZER));
        };
        OccSavedCartAdapter.prototype.loadList = function (userId) {
            return this.http
                .get(this.getSavedCartListEndpoint(userId))
                .pipe(operators.pluck('carts'), operators.map(function (carts) { return carts !== null && carts !== void 0 ? carts : []; }), this.converter.pipeableMany(i2.CART_NORMALIZER));
        };
        OccSavedCartAdapter.prototype.restoreSavedCart = function (userId, cartId) {
            return this.http
                .patch(this.getRestoreSavedCartEndpoint(userId, cartId), cartId)
                .pipe(operators.pluck('savedCartData'), this.converter.pipeable(i2.CART_NORMALIZER));
        };
        OccSavedCartAdapter.prototype.saveCart = function (userId, cartId, saveCartName, saveCartDescription) {
            return this.http
                .patch(this.getSaveCartEndpoint(userId, cartId, saveCartName, saveCartDescription), cartId)
                .pipe(operators.pluck('savedCartData'), this.converter.pipeable(i2.CART_NORMALIZER));
        };
        OccSavedCartAdapter.prototype.cloneSavedCart = function (userId, cartId, saveCartName) {
            return this.http
                .post(this.getCloneSavedCartEndpoint(userId, cartId, saveCartName), cartId)
                .pipe(operators.pluck('savedCartData'), this.converter.pipeable(i2.CART_NORMALIZER));
        };
        OccSavedCartAdapter.prototype.getSavedCartEndpoint = function (userId, cartId) {
            return this.occEndpoints.buildUrl('savedCart', {
                urlParams: { userId: userId, cartId: cartId },
            });
        };
        OccSavedCartAdapter.prototype.getSavedCartListEndpoint = function (userId) {
            return this.occEndpoints.buildUrl('savedCarts', { urlParams: { userId: userId } });
        };
        OccSavedCartAdapter.prototype.getRestoreSavedCartEndpoint = function (userId, cartId) {
            return this.occEndpoints.buildUrl('restoreSavedCart', {
                urlParams: { userId: userId, cartId: cartId },
            });
        };
        OccSavedCartAdapter.prototype.getSaveCartEndpoint = function (userId, cartId, saveCartName, saveCartDescription) {
            return this.occEndpoints.buildUrl('saveCart', {
                urlParams: {
                    userId: userId,
                    cartId: cartId,
                    saveCartName: saveCartName,
                    saveCartDescription: saveCartDescription,
                },
            });
        };
        OccSavedCartAdapter.prototype.getCloneSavedCartEndpoint = function (userId, cartId, saveCartName) {
            return this.occEndpoints.buildUrl('cloneSavedCart', {
                urlParams: { userId: userId, cartId: cartId, saveCartName: saveCartName },
            });
        };
        return OccSavedCartAdapter;
    }());
    OccSavedCartAdapter.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccSavedCartAdapter, deps: [{ token: i1__namespace.HttpClient }, { token: i2__namespace.OccEndpointsService }, { token: i2__namespace.ConverterService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    OccSavedCartAdapter.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccSavedCartAdapter });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccSavedCartAdapter, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.HttpClient }, { type: i2__namespace.OccEndpointsService }, { type: i2__namespace.ConverterService }]; } });

    // Imported for side effects (module augmentation)

    var defaultOccSavedCartConfig = {
        backend: {
            occ: {
                endpoints: {
                    savedCarts: '/users/${userId}/carts?savedCartsOnly=true&fields=carts(DEFAULT,potentialProductPromotions,appliedProductPromotions,potentialOrderPromotions,appliedOrderPromotions,entries(totalPrice(formattedValue),product(images(FULL),stock(FULL)),basePrice(formattedValue,value),updateable),totalPrice(formattedValue),totalItems,totalPriceWithTax(formattedValue),totalDiscounts(value,formattedValue),subTotal(formattedValue),deliveryItemsQuantity,deliveryCost(formattedValue),totalTax(formattedValue, value),pickupItemsQuantity,net,appliedVouchers,productDiscounts(formattedValue),saveTime,user,name,description)',
                    saveCart: '/users/${userId}/carts/${cartId}/save?saveCartName=${saveCartName}&saveCartDescription=${saveCartDescription}',
                    savedCart: '/users/${userId}/carts/${cartId}/savedcart',
                    restoreSavedCart: '/users/${userId}/carts/${cartId}/restoresavedcart',
                    cloneSavedCart: '/users/${userId}/carts/${cartId}/clonesavedcart?name=${saveCartName}',
                },
            },
        },
    };

    var SavedCartOccModule = /** @class */ (function () {
        function SavedCartOccModule() {
        }
        return SavedCartOccModule;
    }());
    SavedCartOccModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartOccModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    SavedCartOccModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartOccModule, imports: [common.CommonModule] });
    SavedCartOccModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartOccModule, providers: [
            i2.provideDefaultConfig(defaultOccSavedCartConfig),
            {
                provide: core.SavedCartAdapter,
                useClass: OccSavedCartAdapter,
            },
        ], imports: [[common.CommonModule]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartOccModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [common.CommonModule],
                        providers: [
                            i2.provideDefaultConfig(defaultOccSavedCartConfig),
                            {
                                provide: core.SavedCartAdapter,
                                useClass: OccSavedCartAdapter,
                            },
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.OccSavedCartAdapter = OccSavedCartAdapter;
    exports.SavedCartOccModule = SavedCartOccModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
