// @spartacus/cart/bundles/spartacus-cart-saved-cart-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/router'), require('@spartacus/core'), require('@spartacus/storefront'), require('rxjs'), require('rxjs/operators'), require('@ngrx/store')) :
    typeof define === 'function' && define.amd ? define('@spartacus/cart/saved-cart/root', ['exports', '@angular/core', '@angular/router', '@spartacus/core', '@spartacus/storefront', 'rxjs', 'rxjs/operators', '@ngrx/store'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.cart = global.spartacus.cart || {}, global.spartacus.cart['saved-cart'] = global.spartacus.cart['saved-cart'] || {}, global.spartacus.cart['saved-cart'].root = {}), global.ng.core, global.ng.router, global.core, global.storefront, global.rxjs, global.rxjs.operators, global.i1));
}(this, (function (exports, i0, i1$1, i2, storefront, rxjs, operators, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace$1 = /*#__PURE__*/_interopNamespace(i1$1);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var CART_SAVED_CART_FEATURE = 'cartSavedCart';
    var CART_SAVED_CART_CORE_FEATURE = 'cartSavedCartCore';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || from);
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var SavedCartFacade = /** @class */ (function () {
        function SavedCartFacade() {
        }
        return SavedCartFacade;
    }());
    SavedCartFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    SavedCartFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartFacade, providedIn: 'root', useFactory: function () { return i2.facadeFactory({
            facade: SavedCartFacade,
            feature: CART_SAVED_CART_CORE_FEATURE,
            methods: [
                'editSavedCart',
                'deleteSavedCart',
                'getSavedCart',
                'getSavedCartList',
                'loadSavedCart',
                'clearCloneSavedCart',
                'clearRestoreSavedCart',
                'clearSaveCart',
                'clearSavedCarts',
                'get',
                'getList',
                'getCloneSavedCartProcessError',
                'getCloneSavedCartProcessLoading',
                'getCloneSavedCartProcessSuccess',
                'getRestoreSavedCartProcessError',
                'getRestoreSavedCartProcessLoading',
                'getRestoreSavedCartProcessSuccess',
                'getSaveCartProcessError',
                'getSaveCartProcessLoading',
                'getSaveCartProcessSuccess',
                'getSavedCartListProcess',
                'getSavedCartListProcessLoading',
                'isStable',
                'cloneSavedCart',
                'loadSavedCarts',
                'restoreSavedCart',
                'saveCart',
            ],
            async: true,
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return i2.facadeFactory({
                            facade: SavedCartFacade,
                            feature: CART_SAVED_CART_CORE_FEATURE,
                            methods: [
                                'editSavedCart',
                                'deleteSavedCart',
                                'getSavedCart',
                                'getSavedCartList',
                                'loadSavedCart',
                                'clearCloneSavedCart',
                                'clearRestoreSavedCart',
                                'clearSaveCart',
                                'clearSavedCarts',
                                'get',
                                'getList',
                                'getCloneSavedCartProcessError',
                                'getCloneSavedCartProcessLoading',
                                'getCloneSavedCartProcessSuccess',
                                'getRestoreSavedCartProcessError',
                                'getRestoreSavedCartProcessLoading',
                                'getRestoreSavedCartProcessSuccess',
                                'getSaveCartProcessError',
                                'getSaveCartProcessLoading',
                                'getSaveCartProcessSuccess',
                                'getSavedCartListProcess',
                                'getSavedCartListProcessLoading',
                                'isStable',
                                'cloneSavedCart',
                                'loadSavedCarts',
                                'restoreSavedCart',
                                'saveCart',
                            ],
                            async: true,
                        }); },
                    }]
            }] });

    var SavedCartOrderEntriesContext = /** @class */ (function (_super) {
        __extends(SavedCartOrderEntriesContext, _super);
        function SavedCartOrderEntriesContext(actionsSubject, userIdService, multiCartService, savedCartService, routingService) {
            var _this = _super.call(this, actionsSubject) || this;
            _this.actionsSubject = actionsSubject;
            _this.userIdService = userIdService;
            _this.multiCartService = multiCartService;
            _this.savedCartService = savedCartService;
            _this.routingService = routingService;
            _this.type = storefront.OrderEntriesSource.SAVED_CART;
            _this.savedCartId$ = _this.routingService.getRouterState().pipe(operators.map(function (routingData) { return routingData.state.params.savedCartId; }), operators.distinctUntilChanged());
            return _this;
        }
        SavedCartOrderEntriesContext.prototype.getEntries = function () {
            var _this = this;
            return this.savedCartId$.pipe(operators.switchMap(function (cartId) { return _this.savedCartService.get(cartId); }), operators.map(function (cart) { var _a; return (_a = cart === null || cart === void 0 ? void 0 : cart.entries) !== null && _a !== void 0 ? _a : []; }));
        };
        SavedCartOrderEntriesContext.prototype.add = function (products) {
            var _this = this;
            return rxjs.combineLatest([
                this.userIdService.takeUserId(),
                this.savedCartId$,
            ]).pipe(operators.tap(function (_b) {
                var _c = __read(_b, 2), userId = _c[0], cartId = _c[1];
                return _this.multiCartService.addEntries(userId, cartId, products);
            }), operators.map(function (_b) {
                var _c = __read(_b, 2), _userId = _c[0], cartId = _c[1];
                return cartId;
            }));
        };
        return SavedCartOrderEntriesContext;
    }(storefront.CartOrderEntriesContext));
    SavedCartOrderEntriesContext.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartOrderEntriesContext, deps: [{ token: i1__namespace.ActionsSubject }, { token: i2__namespace.UserIdService }, { token: i2__namespace.MultiCartService }, { token: SavedCartFacade }, { token: i2__namespace.RoutingService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    SavedCartOrderEntriesContext.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartOrderEntriesContext, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartOrderEntriesContext, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.ActionsSubject }, { type: i2__namespace.UserIdService }, { type: i2__namespace.MultiCartService }, { type: SavedCartFacade }, { type: i2__namespace.RoutingService }]; } });

    var NewSavedCartOrderEntriesContext = /** @class */ (function (_super) {
        __extends(NewSavedCartOrderEntriesContext, _super);
        function NewSavedCartOrderEntriesContext(actionsSubject, userIdService, multiCartService, savedCartService) {
            var _this = _super.call(this, actionsSubject) || this;
            _this.actionsSubject = actionsSubject;
            _this.userIdService = userIdService;
            _this.multiCartService = multiCartService;
            _this.savedCartService = savedCartService;
            _this.type = storefront.OrderEntriesSource.NEW_SAVED_CART;
            return _this;
        }
        NewSavedCartOrderEntriesContext.prototype.addEntries = function (products, savedCartInfo) {
            var _this = this;
            return this.add(products, savedCartInfo).pipe(operators.switchMap(function (cartId) { return _this.getResults(cartId); }), operators.take(products.length));
        };
        NewSavedCartOrderEntriesContext.prototype.add = function (products, savedCartInfo) {
            var _this = this;
            return this.userIdService.takeUserId().pipe(operators.switchMap(function (userId) { return _this.multiCartService
                .createCart({
                userId: userId,
                extraData: { active: false },
            })
                .pipe(operators.filter(function (cartData) { var _a; return Boolean((_a = cartData.value) === null || _a === void 0 ? void 0 : _a.code); }), operators.map(function (cartData) { var _a; return (_a = cartData.value) === null || _a === void 0 ? void 0 : _a.code; }), operators.tap(function (cartId) {
                _this.savedCartService.saveCart({
                    cartId: cartId,
                    saveCartName: savedCartInfo === null || savedCartInfo === void 0 ? void 0 : savedCartInfo.name,
                    saveCartDescription: savedCartInfo === null || savedCartInfo === void 0 ? void 0 : savedCartInfo.description,
                });
                _this.savedCartService.loadSavedCarts();
            }), operators.observeOn(rxjs.queueScheduler), operators.delayWhen(function () { return _this.savedCartService
                .getSaveCartProcessLoading()
                .pipe(operators.filter(function (loading) { return !loading; })); }), operators.tap(function (cartId) { return _this.multiCartService.addEntries(userId, cartId, products); })); }));
        };
        return NewSavedCartOrderEntriesContext;
    }(storefront.CartOrderEntriesContext));
    NewSavedCartOrderEntriesContext.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: NewSavedCartOrderEntriesContext, deps: [{ token: i1__namespace.ActionsSubject }, { token: i2__namespace.UserIdService }, { token: i2__namespace.MultiCartService }, { token: SavedCartFacade }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    NewSavedCartOrderEntriesContext.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: NewSavedCartOrderEntriesContext, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: NewSavedCartOrderEntriesContext, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                    }]
            }], ctorParameters: function () { return [{ type: i1__namespace.ActionsSubject }, { type: i2__namespace.UserIdService }, { type: i2__namespace.MultiCartService }, { type: SavedCartFacade }]; } });

    var _a, _b, _c, _d;
    // TODO: Inline this factory when we start releasing Ivy compiled libraries
    function defaultCartSavedCartComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[CART_SAVED_CART_FEATURE] = {
                    cmsComponents: [
                        'AddToSavedCartsComponent',
                        'AccountSavedCartHistoryComponent',
                        'SavedCartDetailsOverviewComponent',
                        'SavedCartDetailsItemsComponent',
                        'SavedCartDetailsActionComponent',
                    ],
                },
                // by default core is bundled together with components
                _a[CART_SAVED_CART_CORE_FEATURE] = CART_SAVED_CART_FEATURE,
                _a),
        };
        return config;
    }
    var SavedCartRootModule = /** @class */ (function () {
        function SavedCartRootModule() {
        }
        return SavedCartRootModule;
    }());
    SavedCartRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    SavedCartRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartRootModule, imports: [i1__namespace$1.RouterModule] });
    SavedCartRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartRootModule, providers: [
            i2.provideDefaultConfigFactory(defaultCartSavedCartComponentsConfig),
            i2.provideDefaultConfig({
                routing: {
                    routes: {
                        savedCarts: {
                            paths: ['my-account/saved-carts'],
                        },
                        savedCartsDetails: {
                            paths: ['my-account/saved-cart/:savedCartId'],
                            paramsMapping: { savedCartId: 'savedCartId' },
                        },
                    },
                },
            }),
        ], imports: [[
                i1$1.RouterModule.forChild([
                    {
                        // @ts-ignore
                        path: null,
                        canActivate: [i2.AuthGuard, storefront.CmsPageGuard],
                        component: storefront.PageLayoutComponent,
                        data: {
                            cxRoute: 'savedCartsDetails',
                            cxContext: (_a = {},
                                _a[storefront.ORDER_ENTRIES_CONTEXT] = SavedCartOrderEntriesContext,
                                _a),
                        },
                    },
                    {
                        // @ts-ignore
                        path: null,
                        canActivate: [i2.AuthGuard, storefront.CmsPageGuard],
                        component: storefront.PageLayoutComponent,
                        data: {
                            cxRoute: 'savedCarts',
                            cxContext: (_b = {},
                                _b[storefront.ORDER_ENTRIES_CONTEXT] = NewSavedCartOrderEntriesContext,
                                _b),
                        },
                    },
                ]),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: SavedCartRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i1$1.RouterModule.forChild([
                                {
                                    // @ts-ignore
                                    path: null,
                                    canActivate: [i2.AuthGuard, storefront.CmsPageGuard],
                                    component: storefront.PageLayoutComponent,
                                    data: {
                                        cxRoute: 'savedCartsDetails',
                                        cxContext: (_c = {},
                                            _c[storefront.ORDER_ENTRIES_CONTEXT] = SavedCartOrderEntriesContext,
                                            _c),
                                    },
                                },
                                {
                                    // @ts-ignore
                                    path: null,
                                    canActivate: [i2.AuthGuard, storefront.CmsPageGuard],
                                    component: storefront.PageLayoutComponent,
                                    data: {
                                        cxRoute: 'savedCarts',
                                        cxContext: (_d = {},
                                            _d[storefront.ORDER_ENTRIES_CONTEXT] = NewSavedCartOrderEntriesContext,
                                            _d),
                                    },
                                },
                            ]),
                        ],
                        providers: [
                            i2.provideDefaultConfigFactory(defaultCartSavedCartComponentsConfig),
                            i2.provideDefaultConfig({
                                routing: {
                                    routes: {
                                        savedCarts: {
                                            paths: ['my-account/saved-carts'],
                                        },
                                        savedCartsDetails: {
                                            paths: ['my-account/saved-cart/:savedCartId'],
                                            paramsMapping: { savedCartId: 'savedCartId' },
                                        },
                                    },
                                },
                            }),
                        ],
                    }]
            }] });

    /**
     * Base saved cart event. Most cart events should have these properties.
     */
    var SavedCartEvent = /** @class */ (function (_super) {
        __extends(SavedCartEvent, _super);
        function SavedCartEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return SavedCartEvent;
    }(i2.CartEvent));
    // =====================================================================
    var DeleteSavedCartEvent = /** @class */ (function (_super) {
        __extends(DeleteSavedCartEvent, _super);
        function DeleteSavedCartEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return DeleteSavedCartEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    DeleteSavedCartEvent.type = 'DeleteSavedCartEvent';
    var DeleteSavedCartSuccessEvent = /** @class */ (function (_super) {
        __extends(DeleteSavedCartSuccessEvent, _super);
        function DeleteSavedCartSuccessEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return DeleteSavedCartSuccessEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    DeleteSavedCartSuccessEvent.type = 'DeleteSavedCartSuccessEvent';
    var DeleteSavedCartFailEvent = /** @class */ (function (_super) {
        __extends(DeleteSavedCartFailEvent, _super);
        function DeleteSavedCartFailEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return DeleteSavedCartFailEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    DeleteSavedCartFailEvent.type = 'DeleteSavedCartFailEvent';
    var SaveCartEvent = /** @class */ (function (_super) {
        __extends(SaveCartEvent, _super);
        function SaveCartEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return SaveCartEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    SaveCartEvent.type = 'SaveCartEvent';
    var SaveCartSuccessEvent = /** @class */ (function (_super) {
        __extends(SaveCartSuccessEvent, _super);
        function SaveCartSuccessEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return SaveCartSuccessEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    SaveCartSuccessEvent.type = 'SaveCartSuccessEvent';
    var SaveCartFailEvent = /** @class */ (function (_super) {
        __extends(SaveCartFailEvent, _super);
        function SaveCartFailEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return SaveCartFailEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    SaveCartFailEvent.type = 'SaveCartEvent';
    var RestoreSavedCartEvent = /** @class */ (function (_super) {
        __extends(RestoreSavedCartEvent, _super);
        function RestoreSavedCartEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return RestoreSavedCartEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    RestoreSavedCartEvent.type = 'RestoreSavedCartEvent';
    var RestoreSavedCartSuccessEvent = /** @class */ (function (_super) {
        __extends(RestoreSavedCartSuccessEvent, _super);
        function RestoreSavedCartSuccessEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return RestoreSavedCartSuccessEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    RestoreSavedCartSuccessEvent.type = 'RestoreSavedCartSuccessEvent';
    var RestoreSavedCartFailEvent = /** @class */ (function (_super) {
        __extends(RestoreSavedCartFailEvent, _super);
        function RestoreSavedCartFailEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return RestoreSavedCartFailEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    RestoreSavedCartFailEvent.type = 'RestoreSavedCartFailEvent';
    var EditSavedCartEvent = /** @class */ (function (_super) {
        __extends(EditSavedCartEvent, _super);
        function EditSavedCartEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return EditSavedCartEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    EditSavedCartEvent.type = 'EditSavedCartEvent';
    var EditSavedCartSuccessEvent = /** @class */ (function (_super) {
        __extends(EditSavedCartSuccessEvent, _super);
        function EditSavedCartSuccessEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return EditSavedCartSuccessEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    EditSavedCartSuccessEvent.type = 'EditSavedCartSuccessEvent';
    var EditSavedCartFailEvent = /** @class */ (function (_super) {
        __extends(EditSavedCartFailEvent, _super);
        function EditSavedCartFailEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return EditSavedCartFailEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    EditSavedCartFailEvent.type = 'EditSavedCartFailEvent';
    var CloneSavedCartEvent = /** @class */ (function (_super) {
        __extends(CloneSavedCartEvent, _super);
        function CloneSavedCartEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CloneSavedCartEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    CloneSavedCartEvent.type = 'CloneSavedCartEvent';
    var CloneSavedCartSuccessEvent = /** @class */ (function (_super) {
        __extends(CloneSavedCartSuccessEvent, _super);
        function CloneSavedCartSuccessEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CloneSavedCartSuccessEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    CloneSavedCartSuccessEvent.type = 'CloneSavedCartSuccessEvent';
    var CloneSavedCartFailEvent = /** @class */ (function (_super) {
        __extends(CloneSavedCartFailEvent, _super);
        function CloneSavedCartFailEvent() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CloneSavedCartFailEvent;
    }(SavedCartEvent));
    /**
     * Event's type
     */
    CloneSavedCartFailEvent.type = 'CloneSavedCartFailEvent';

    exports.SavedCartFormType = void 0;
    (function (SavedCartFormType) {
        SavedCartFormType["EDIT"] = "edit";
        SavedCartFormType["DELETE"] = "delete";
        SavedCartFormType["SAVE"] = "save";
        SavedCartFormType["RESTORE"] = "restore";
    })(exports.SavedCartFormType || (exports.SavedCartFormType = {}));

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CART_SAVED_CART_CORE_FEATURE = CART_SAVED_CART_CORE_FEATURE;
    exports.CART_SAVED_CART_FEATURE = CART_SAVED_CART_FEATURE;
    exports.CloneSavedCartEvent = CloneSavedCartEvent;
    exports.CloneSavedCartFailEvent = CloneSavedCartFailEvent;
    exports.CloneSavedCartSuccessEvent = CloneSavedCartSuccessEvent;
    exports.DeleteSavedCartEvent = DeleteSavedCartEvent;
    exports.DeleteSavedCartFailEvent = DeleteSavedCartFailEvent;
    exports.DeleteSavedCartSuccessEvent = DeleteSavedCartSuccessEvent;
    exports.EditSavedCartEvent = EditSavedCartEvent;
    exports.EditSavedCartFailEvent = EditSavedCartFailEvent;
    exports.EditSavedCartSuccessEvent = EditSavedCartSuccessEvent;
    exports.NewSavedCartOrderEntriesContext = NewSavedCartOrderEntriesContext;
    exports.RestoreSavedCartEvent = RestoreSavedCartEvent;
    exports.RestoreSavedCartFailEvent = RestoreSavedCartFailEvent;
    exports.RestoreSavedCartSuccessEvent = RestoreSavedCartSuccessEvent;
    exports.SaveCartEvent = SaveCartEvent;
    exports.SaveCartFailEvent = SaveCartFailEvent;
    exports.SaveCartSuccessEvent = SaveCartSuccessEvent;
    exports.SavedCartEvent = SavedCartEvent;
    exports.SavedCartFacade = SavedCartFacade;
    exports.SavedCartOrderEntriesContext = SavedCartOrderEntriesContext;
    exports.SavedCartRootModule = SavedCartRootModule;
    exports.defaultCartSavedCartComponentsConfig = defaultCartSavedCartComponentsConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
