// @spartacus/storefinder/esm2015/core/facade/store-finder.service.js
import { isPlatformBrowser } from '@angular/common';
import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
import { select } from '@ngrx/store';
import { GlobalMessageType, } from '@spartacus/core';
import { Subscription } from 'rxjs';
import { filter, map, withLatestFrom } from 'rxjs/operators';
import { StoreFinderActions } from '../store/actions/index';
import { StoreFinderSelectors } from '../store/selectors/index';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/store";
import * as i2 from "@spartacus/core";
export class StoreFinderService {
    constructor(store, winRef, globalMessageService, routingService, platformId) {
        this.store = store;
        this.winRef = winRef;
        this.globalMessageService = globalMessageService;
        this.routingService = routingService;
        this.platformId = platformId;
        this.geolocationWatchId = null;
        this.subscription = new Subscription();
        this.reloadStoreEntitiesOnContextChange();
    }
    /**
     * Returns boolean observable for store's loading state
     */
    getStoresLoading() {
        return this.store.pipe(select(StoreFinderSelectors.getStoresLoading));
    }
    /**
     * Returns boolean observable for store's success state
     */
    getStoresLoaded() {
        return this.store.pipe(select(StoreFinderSelectors.getStoresSuccess));
    }
    /**
     * Returns observable for store's entities
     */
    getFindStoresEntities() {
        return this.store.pipe(select(StoreFinderSelectors.getFindStoresEntities), map((data) => data.findStoresEntities));
    }
    /**
     * Returns observable for a single store by Id
     */
    getFindStoreEntityById() {
        return this.store.pipe(select(StoreFinderSelectors.getFindStoresEntities), map((data) => data.findStoreEntityById));
    }
    /**
     * Returns boolean observable for view all store's loading state
     */
    getViewAllStoresLoading() {
        return this.store.pipe(select(StoreFinderSelectors.getViewAllStoresLoading));
    }
    /**
     * Returns observable for view all store's entities
     */
    getViewAllStoresEntities() {
        return this.store.pipe(select(StoreFinderSelectors.getViewAllStoresEntities), map((data) => data.viewAllStoresEntities));
    }
    /**
     * Store finding action functionality
     * @param queryText text query
     * @param searchConfig search configuration
     * @param longitudeLatitude longitude and latitude coordinates
     * @param countryIsoCode country ISO code
     * @param useMyLocation current location coordinates
     * @param radius radius of the scope from the center point
     */
    findStoresAction(queryText, searchConfig, longitudeLatitude, countryIsoCode, useMyLocation, radius) {
        if (useMyLocation && this.winRef.nativeWindow) {
            this.clearWatchGeolocation(new StoreFinderActions.FindStoresOnHold());
            this.geolocationWatchId =
                this.winRef.nativeWindow.navigator.geolocation.watchPosition((pos) => {
                    const position = {
                        longitude: pos.coords.longitude,
                        latitude: pos.coords.latitude,
                    };
                    this.clearWatchGeolocation(new StoreFinderActions.FindStores({
                        queryText: queryText,
                        searchConfig: searchConfig,
                        longitudeLatitude: position,
                        countryIsoCode: countryIsoCode,
                        radius: radius,
                    }));
                }, () => {
                    this.globalMessageService.add({ key: 'storeFinder.geolocationNotEnabled' }, GlobalMessageType.MSG_TYPE_ERROR);
                    this.routingService.go(['/store-finder']);
                });
        }
        else {
            this.clearWatchGeolocation(new StoreFinderActions.FindStores({
                queryText: queryText,
                searchConfig: searchConfig,
                longitudeLatitude: longitudeLatitude,
                countryIsoCode: countryIsoCode,
                radius: radius,
            }));
        }
    }
    /**
     * View all stores
     */
    viewAllStores() {
        this.clearWatchGeolocation(new StoreFinderActions.ViewAllStores());
    }
    /**
     * View all stores by id
     * @param storeId store id
     */
    viewStoreById(storeId) {
        this.clearWatchGeolocation(new StoreFinderActions.FindStoreById({ storeId }));
    }
    clearWatchGeolocation(callbackAction) {
        var _a;
        if (this.geolocationWatchId !== null) {
            (_a = this.winRef.nativeWindow) === null || _a === void 0 ? void 0 : _a.navigator.geolocation.clearWatch(this.geolocationWatchId);
            this.geolocationWatchId = null;
        }
        this.store.dispatch(callbackAction);
    }
    isEmpty(store) {
        return (!store || (typeof store === 'object' && Object.keys(store).length === 0));
    }
    /**
     * Reload store data when store entities are empty because of the context change
     */
    reloadStoreEntitiesOnContextChange() {
        if (isPlatformBrowser(this.platformId) || !this.platformId) {
            this.subscription = this.getFindStoresEntities()
                .pipe(filter((data) => this.isEmpty(data)), withLatestFrom(this.getStoresLoading(), this.getStoresLoaded(), this.routingService.getParams()))
                .subscribe(([, loading, loaded, routeParams]) => {
                if (!loading && !loaded) {
                    if (routeParams.country && !routeParams.store) {
                        this.callFindStoresAction(routeParams);
                    }
                    if (routeParams.store) {
                        this.viewStoreById(routeParams.store);
                    }
                }
            });
        }
    }
    callFindStoresAction(routeParams) {
        this.findStoresAction('', {
            pageSize: -1,
        }, undefined, routeParams.country);
    }
    ngOnDestroy() {
        var _a;
        (_a = this.subscription) === null || _a === void 0 ? void 0 : _a.unsubscribe();
    }
    /**
     * Returns store latitude
     * @param location store location
     */
    getStoreLatitude(location) {
        var _a;
        return (_a = location === null || location === void 0 ? void 0 : location.geoPoint) === null || _a === void 0 ? void 0 : _a.latitude;
    }
    /**
     * Returns store longitude
     * @param location store location
     */
    getStoreLongitude(location) {
        var _a;
        return (_a = location === null || location === void 0 ? void 0 : location.geoPoint) === null || _a === void 0 ? void 0 : _a.longitude;
    }
}
StoreFinderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderService, deps: [{ token: i1.Store }, { token: i2.WindowRef }, { token: i2.GlobalMessageService }, { token: i2.RoutingService }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Injectable });
StoreFinderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.Store }, { type: i2.WindowRef }, { type: i2.GlobalMessageService }, { type: i2.RoutingService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }]; } });
