// @spartacus/cart/esm2015/import-export/components/export-entries/export-order-entries.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ConfigModule, I18nModule, UrlModule, } from '@spartacus/core';
import { ExportOrderEntriesComponent } from './export-order-entries.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class ExportOrderEntriesModule {
}
ExportOrderEntriesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExportOrderEntriesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ExportOrderEntriesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExportOrderEntriesModule, declarations: [ExportOrderEntriesComponent], imports: [CommonModule,
        RouterModule,
        I18nModule,
        UrlModule, i1.ConfigModule], exports: [ExportOrderEntriesComponent] });
ExportOrderEntriesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExportOrderEntriesModule, imports: [[
            CommonModule,
            RouterModule,
            I18nModule,
            UrlModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    ExportOrderEntriesComponent: {
                        component: ExportOrderEntriesComponent,
                    },
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExportOrderEntriesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        I18nModule,
                        UrlModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                ExportOrderEntriesComponent: {
                                    component: ExportOrderEntriesComponent,
                                },
                            },
                        }),
                    ],
                    exports: [ExportOrderEntriesComponent],
                    declarations: [ExportOrderEntriesComponent],
                    entryComponents: [ExportOrderEntriesComponent],
                }]
        }] });
