// @spartacus/core/esm2015/src/user/connectors/cost-center/user-cost-center.connector.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./user-cost-center.adapter";
export class UserCostCenterConnector {
    constructor(adapter) {
        this.adapter = adapter;
    }
    getActiveList(userId) {
        return this.adapter.loadActiveList(userId);
    }
}
UserCostCenterConnector.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserCostCenterConnector, deps: [{ token: i1.UserCostCenterAdapter }], target: i0.ɵɵFactoryTarget.Injectable });
UserCostCenterConnector.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserCostCenterConnector, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserCostCenterConnector, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.UserCostCenterAdapter }]; } });
