// @spartacus/product/esm2015/variants/components/guards/product-variants.guard.js
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { filter, map, switchMap, take } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@angular/router";
/**
 * Guard that will redirect to purchasable variant of product, if the navigation
 * is for the non-purchasable one
 */
export class ProductVariantsGuard {
    constructor(productService, semanticPathService, router) {
        this.productService = productService;
        this.semanticPathService = semanticPathService;
        this.router = router;
    }
    canActivate(activatedRoute) {
        var _a;
        const productCode = (_a = activatedRoute.params) === null || _a === void 0 ? void 0 : _a.productCode;
        if (!productCode) {
            return of(true);
        }
        return this.productService.get(productCode, "variants" /* VARIANTS */).pipe(filter((p) => !!p), switchMap((product) => {
            if (!product.purchasable) {
                const purchasableCode = this.findPurchasableProductCode(product);
                if (purchasableCode) {
                    return this.productService
                        .get(purchasableCode, "list" /* LIST */)
                        .pipe(filter((p) => !!p), take(1), map((_product) => {
                        return this.router.createUrlTree(this.semanticPathService.transform({
                            cxRoute: 'product',
                            params: _product,
                        }));
                    }));
                }
            }
            return of(true);
        }));
    }
    /**
     * Finds a purchasable product code looking at variant options, if any
     *
     * @param product
     */
    findPurchasableProductCode(product) {
        var _a, _b, _c;
        if ((_a = product.variantOptions) === null || _a === void 0 ? void 0 : _a.length) {
            const results = product.variantOptions.filter((variant) => {
                return variant.stock && variant.stock.stockLevel ? variant : false;
            });
            return results && results.length
                ? (_b = results[0]) === null || _b === void 0 ? void 0 : _b.code
                : (_c = product.variantOptions[0]) === null || _c === void 0 ? void 0 : _c.code;
        }
        return undefined;
    }
}
ProductVariantsGuard.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantsGuard, deps: [{ token: i1.ProductService }, { token: i1.SemanticPathService }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Injectable });
ProductVariantsGuard.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantsGuard, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantsGuard, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.ProductService }, { type: i1.SemanticPathService }, { type: i2.Router }]; } });
