// @spartacus/product/esm2015/image-zoom/root/product-image-zoom-root.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfigFactory } from '@spartacus/core';
import { PRODUCT_IMAGE_ZOOM_FEATURE } from './feature-name';
import * as i0 from "@angular/core";
export function defaultImageZoomComponentsConfig() {
    const config = {
        featureModules: {
            [PRODUCT_IMAGE_ZOOM_FEATURE]: {
                cmsComponents: ['ProductImagesComponent'],
            },
        },
    };
    return config;
}
export class ProductImageZoomRootModule {
}
ProductImageZoomRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductImageZoomRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomRootModule });
ProductImageZoomRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomRootModule, providers: [provideDefaultConfigFactory(defaultImageZoomComponentsConfig)], imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    providers: [provideDefaultConfigFactory(defaultImageZoomComponentsConfig)],
                }]
        }] });
