// @spartacus/storefront/esm2015/cms-structure/services/cms-routes.service.js
import { Injectable } from '@angular/core';
import { CmsRoutesImplService } from './cms-routes-impl.service';
import * as i0 from "@angular/core";
// Public injection token for the private implementation of the service `CmsRoutesImplService`.
// After #7070, this class should be replaced with a real implementation.
export class CmsRoutesService {
}
CmsRoutesService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CmsRoutesService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
CmsRoutesService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CmsRoutesService, providedIn: 'root', useExisting: CmsRoutesImplService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CmsRoutesService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useExisting: CmsRoutesImplService,
                }]
        }] });
