// @spartacus/tracking/schematics/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SPARTACUS_TMS_AEP = exports.TMS_AEP_MODULE = exports.SPARTACUS_TMS_GTM = exports.TMS_GTM_MODULE = exports.SPARTACUS_TMS_CORE = exports.TMS_BASE_MODULE = exports.TMS_CONFIG = exports.TMS_MODULE_NAME = exports.SPARTACUS_PERSONALIZATION_ROOT = exports.SPARTACUS_PERSONALIZATION = exports.PERSONALIZATION_ROOT_MODULE = exports.PERSONALIZATION_MODULE = exports.PERSONALIZATION_FEATURE_NAME_CONSTANT = exports.PERSONALIZATION_MODULE_NAME = exports.TRACKING_FOLDER_NAME = void 0;
const schematics_1 = require("@spartacus/schematics");
exports.TRACKING_FOLDER_NAME = 'tracking';
exports.PERSONALIZATION_MODULE_NAME = 'Personalization';
exports.PERSONALIZATION_FEATURE_NAME_CONSTANT = 'PERSONALIZATION_FEATURE';
exports.PERSONALIZATION_MODULE = 'PersonalizationModule';
exports.PERSONALIZATION_ROOT_MODULE = 'PersonalizationRootModule';
exports.SPARTACUS_PERSONALIZATION = `${schematics_1.SPARTACUS_TRACKING}/personalization`;
exports.SPARTACUS_PERSONALIZATION_ROOT = `${exports.SPARTACUS_PERSONALIZATION}/root`;
exports.TMS_MODULE_NAME = 'TagManagement';
exports.TMS_CONFIG = 'TmsConfig';
exports.TMS_BASE_MODULE = 'BaseTmsModule';
exports.SPARTACUS_TMS_CORE = `${schematics_1.SPARTACUS_TRACKING}/tms/core`;
exports.TMS_GTM_MODULE = 'GtmModule';
exports.SPARTACUS_TMS_GTM = `${schematics_1.SPARTACUS_TRACKING}/tms/gtm`;
exports.TMS_AEP_MODULE = 'AepModule';
exports.SPARTACUS_TMS_AEP = `${schematics_1.SPARTACUS_TRACKING}/tms/aep`;
