// @spartacus/cart/esm2015/import-export/root/import-export-root.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';
import { CART_IMPORT_EXPORT_FEATURE } from './feature-name';
import * as i0 from "@angular/core";
export class ImportExportRootModule {
}
ImportExportRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ImportExportRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule });
ImportExportRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule, providers: [
        provideDefaultConfig({
            featureModules: {
                [CART_IMPORT_EXPORT_FEATURE]: {
                    cmsComponents: [
                        'ImportExportOrderEntriesComponent',
                        'ImportOrderEntriesComponent',
                        'ExportOrderEntriesComponent',
                    ],
                },
            },
        }),
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfig({
                            featureModules: {
                                [CART_IMPORT_EXPORT_FEATURE]: {
                                    cmsComponents: [
                                        'ImportExportOrderEntriesComponent',
                                        'ImportOrderEntriesComponent',
                                        'ExportOrderEntriesComponent',
                                    ],
                                },
                            },
                        }),
                    ],
                }]
        }] });
