// @nguniversal/builders/src/prerender/worker.js
/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define("@nguniversal/builders/src/prerender/worker", ["require", "exports", "@nguniversal/common/engine", "fs", "path"], factory);
    }
})(function (require, exports) {
    "use strict";
    var __syncRequire = typeof module === "object" && typeof module.exports === "object";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.render = exports.setup = void 0;
    const engine_1 = require("@nguniversal/common/engine");
    const fs = require("fs");
    const path = require("path");
    let workerArgs;
    let inlineCriticalCssProcessor;
    function setup(options) {
        workerArgs = options;
        if (workerArgs.inlineCriticalCss) {
            inlineCriticalCssProcessor = new engine_1.ɵInlineCriticalCssProcessor({
                deployUrl: workerArgs.deployUrl,
                minify: workerArgs.minifyCss,
            });
        }
    }
    exports.setup = setup;
    /**
     * Renders each route in routes and writes them to <outputPath>/<route>/index.html.
     */
    function render(outputPath, serverBundlePath, route) {
        return __awaiter(this, void 0, void 0, function* () {
            const result = {};
            const browserIndexOutputPath = path.join(outputPath, workerArgs.indexFile);
            const outputFolderPath = path.join(outputPath, route);
            const outputIndexPath = path.join(outputFolderPath, 'index.html');
            const { renderModule, AppServerModule } = yield __syncRequire ? Promise.resolve().then(() => require(serverBundlePath)) : new Promise((resolve_1, reject_1) => { require([serverBundlePath], resolve_1, reject_1); });
            const indexBaseName = fs.existsSync(path.join(outputPath, 'index.original.html'))
                ? 'index.original.html'
                : workerArgs.indexFile;
            const browserIndexInputPath = path.join(outputPath, indexBaseName);
            let indexHtml = yield fs.promises.readFile(browserIndexInputPath, 'utf8');
            indexHtml = indexHtml.replace('</html>', '<!-- This page was prerendered with Angular Universal -->\n</html>');
            if (workerArgs.inlineCriticalCss) {
                // Workaround for https://github.com/GoogleChromeLabs/critters/issues/64
                indexHtml = indexHtml.replace(/ media=\"print\" onload=\"this\.media='all'"><noscript><link .+?><\/noscript>/g, '>');
            }
            let html = yield renderModule(AppServerModule, {
                document: indexHtml,
                url: route,
            });
            if (inlineCriticalCssProcessor) {
                const { content, warnings, errors } = yield inlineCriticalCssProcessor.process(html, {
                    outputPath,
                });
                result.errors = errors;
                result.warnings = warnings;
                html = content;
            }
            // This case happens when we are prerendering "/".
            if (browserIndexOutputPath === outputIndexPath) {
                const browserIndexOutputPathOriginal = path.join(outputPath, 'index.original.html');
                fs.renameSync(browserIndexOutputPath, browserIndexOutputPathOriginal);
            }
            fs.mkdirSync(outputFolderPath, { recursive: true });
            fs.writeFileSync(outputIndexPath, html);
            return result;
        });
    }
    exports.render = render;
});
