// @spartacus/storefront/esm2015/shared/components/progress-button/progress-button.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { I18nModule } from '@spartacus/core';
import { ProgressButtonComponent } from './progress-button.component';
import * as i0 from "@angular/core";
export class ProgressButtonModule {
}
ProgressButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProgressButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProgressButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProgressButtonModule, declarations: [ProgressButtonComponent], imports: [CommonModule, I18nModule], exports: [ProgressButtonComponent] });
ProgressButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProgressButtonModule, imports: [[CommonModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProgressButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule],
                    declarations: [ProgressButtonComponent],
                    exports: [ProgressButtonComponent],
                }]
        }] });
