// @spartacus/storefront/esm2015/shared/components/progress-button/progress-button.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@spartacus/core";
export class ProgressButtonComponent {
    constructor() {
        this.ariaLabel = '';
        this.class = '';
        this.disabled = false;
        this.loading = false;
        this.clikEvent = new EventEmitter();
    }
}
ProgressButtonComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProgressButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ProgressButtonComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ProgressButtonComponent, selector: "cx-progress-button", inputs: { ariaLabel: "ariaLabel", class: "class", disabled: "disabled", loading: "loading" }, outputs: { clikEvent: "clikEvent" }, ngImport: i0, template: "<button\n  (click)=\"clikEvent.emit()\"\n  [attr.aria-label]=\"ariaLabel\"\n  [disabled]=\"disabled || loading\"\n  [ngClass]=\"class\"\n  class=\"btn btn-primary\"\n>\n  <div class=\"cx-progress-button-container\">\n    <div *ngIf=\"loading\" class=\"loader-container\">\n      <div class=\"loader\">{{ 'spinner.loading' | cxTranslate }}</div>\n    </div>\n\n    <ng-content></ng-content>\n  </div>\n</button>\n", directives: [{ type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "cxTranslate": i2.TranslatePipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProgressButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-progress-button',
                    templateUrl: './progress-button.component.html',
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { ariaLabel: [{
                type: Input
            }], class: [{
                type: Input
            }], disabled: [{
                type: Input
            }], loading: [{
                type: Input
            }], clikEvent: [{
                type: Output
            }] } });
