// @spartacus/product/esm2015/image-zoom/components/product-image-zoom/product-image-zoom.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { provideConfig, I18nModule, provideDefaultConfig, } from '@spartacus/core';
import { CarouselModule, IconModule, KeyboardFocusModule, MediaModule, OutletModule, } from '@spartacus/storefront';
import { defaultProductImageZoomLayoutConfig } from './default-product-image-zoom-layout.config';
import { ProductImageZoomDialogComponent } from './product-image-zoom-dialog/product-image-zoom-dialog.component';
import { ProductImageZoomProductImagesComponent } from './product-image-zoom-product-images/product-image-zoom-product-images.component';
import { ProductImageZoomThumbnailsComponent } from './product-image-zoom-thumbnails/product-image-zoom-thumbnails.component';
import { ProductImageZoomTriggerComponent } from './product-image-zoom-trigger/product-image-zoom-trigger.component';
import { ProductImageZoomViewComponent } from './product-image-zoom-view/product-image-zoom-view.component';
import * as i0 from "@angular/core";
export class ProductImageZoomModule {
}
ProductImageZoomModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductImageZoomModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomModule, declarations: [ProductImageZoomDialogComponent,
        ProductImageZoomProductImagesComponent,
        ProductImageZoomThumbnailsComponent,
        ProductImageZoomTriggerComponent,
        ProductImageZoomViewComponent], imports: [CarouselModule,
        CommonModule,
        I18nModule,
        IconModule,
        KeyboardFocusModule,
        MediaModule,
        OutletModule,
        RouterModule], exports: [ProductImageZoomDialogComponent,
        ProductImageZoomProductImagesComponent,
        ProductImageZoomThumbnailsComponent,
        ProductImageZoomTriggerComponent,
        ProductImageZoomViewComponent] });
ProductImageZoomModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomModule, providers: [
        provideConfig(defaultProductImageZoomLayoutConfig),
        provideDefaultConfig({
            cmsComponents: {
                ProductImagesComponent: {
                    component: ProductImageZoomProductImagesComponent,
                },
            },
        }),
    ], imports: [[
            CarouselModule,
            CommonModule,
            I18nModule,
            IconModule,
            KeyboardFocusModule,
            MediaModule,
            OutletModule,
            RouterModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductImageZoomModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CarouselModule,
                        CommonModule,
                        I18nModule,
                        IconModule,
                        KeyboardFocusModule,
                        MediaModule,
                        OutletModule,
                        RouterModule,
                    ],
                    providers: [
                        provideConfig(defaultProductImageZoomLayoutConfig),
                        provideDefaultConfig({
                            cmsComponents: {
                                ProductImagesComponent: {
                                    component: ProductImageZoomProductImagesComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [
                        ProductImageZoomDialogComponent,
                        ProductImageZoomProductImagesComponent,
                        ProductImageZoomThumbnailsComponent,
                        ProductImageZoomTriggerComponent,
                        ProductImageZoomViewComponent,
                    ],
                    exports: [
                        ProductImageZoomDialogComponent,
                        ProductImageZoomProductImagesComponent,
                        ProductImageZoomThumbnailsComponent,
                        ProductImageZoomTriggerComponent,
                        ProductImageZoomViewComponent,
                    ],
                }]
        }] });
