// @spartacus/storefront/esm2015/cms-structure/seo/structured-data/json-ld.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./json-ld-script.factory";
import * as i2 from "@angular/platform-browser";
/**
 * Low level directive that adds a json-ld script tag to the component.
 * This code bypasses the strict XSS security, as otherwise we're not able
 * to append a script tag with JS inside.
 *
 * This helper directive is actually not used in Spartacus, as Spartacus
 * appends json-ld the data to the document body.
 *
 * This directive can however be used by merchants to write static schema data
 * to the DOM in a save way.
 */
export class JsonLdDirective {
    constructor(jsonLdScriptFactory, sanitizer) {
        this.jsonLdScriptFactory = jsonLdScriptFactory;
        this.sanitizer = sanitizer;
    }
    /**
     * Writes the schema data to a json-ld script element.
     */
    set cxJsonLd(schema) {
        this.jsonLD = this.generateJsonLdScript(schema);
    }
    /**
     * Returns the json-ld script tag with the schema data. The script is
     * _bypassing_ sanitization explicitly.
     */
    generateJsonLdScript(schema) {
        if (schema && this.jsonLdScriptFactory.isJsonLdRequired()) {
            const sanitizedSchema = this.jsonLdScriptFactory.sanitize(schema);
            const html = `<script type="application/ld+json">${sanitizedSchema}</script>`;
            return this.sanitizer.bypassSecurityTrustHtml(html);
        }
    }
}
JsonLdDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: JsonLdDirective, deps: [{ token: i1.JsonLdScriptFactory }, { token: i2.DomSanitizer }], target: i0.ɵɵFactoryTarget.Directive });
JsonLdDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: JsonLdDirective, selector: "[cxJsonLd]", inputs: { cxJsonLd: "cxJsonLd" }, host: { properties: { "innerHTML": "this.jsonLD" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: JsonLdDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxJsonLd]',
                }]
        }], ctorParameters: function () { return [{ type: i1.JsonLdScriptFactory }, { type: i2.DomSanitizer }]; }, propDecorators: { cxJsonLd: [{
                type: Input
            }], jsonLD: [{
                type: HostBinding,
                args: ['innerHTML']
            }] } });
