// @spartacus/storefront/esm2015/cms-components/anonymous-consent-management/open-dialog/anonymous-consent-open-dialog.component.js
import { Component, ViewChild, } from '@angular/core';
import { take } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../../layout/launch-dialog/services/launch-dialog.service";
import * as i2 from "@spartacus/core";
export class AnonymousConsentOpenDialogComponent {
    constructor(vcr, launchDialogService) {
        this.vcr = vcr;
        this.launchDialogService = launchDialogService;
    }
    openDialog() {
        const dialog = this.launchDialogService.openDialog("ANONYMOUS_CONSENT" /* ANONYMOUS_CONSENT */, this.openElement, this.vcr);
        if (dialog) {
            dialog.pipe(take(1)).subscribe();
        }
    }
}
AnonymousConsentOpenDialogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentOpenDialogComponent, deps: [{ token: i0.ViewContainerRef }, { token: i1.LaunchDialogService }], target: i0.ɵɵFactoryTarget.Component });
AnonymousConsentOpenDialogComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: AnonymousConsentOpenDialogComponent, selector: "cx-anonymous-consent-open-dialog", viewQueries: [{ propertyName: "openElement", first: true, predicate: ["open"], descendants: true }], ngImport: i0, template: "<button #open class=\"btn btn-link\" (click)=\"openDialog()\">\n  {{ 'anonymousConsents.preferences' | cxTranslate }}\n</button>\n", pipes: { "cxTranslate": i2.TranslatePipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentOpenDialogComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-anonymous-consent-open-dialog',
                    templateUrl: './anonymous-consent-open-dialog.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i1.LaunchDialogService }]; }, propDecorators: { openElement: [{
                type: ViewChild,
                args: ['open']
            }] } });
