// @spartacus/core/esm2015/src/occ/adapters/user/converters/occ-address-list-normalizer.js
import { Injectable } from '@angular/core';
import { ADDRESS_NORMALIZER } from '../../../../user/connectors/address/converters';
import * as i0 from "@angular/core";
import * as i1 from "../../../../util/converter.service";
export class OccAddressListNormalizer {
    constructor(converter) {
        this.converter = converter;
    }
    convert(source, target) {
        if (target === undefined) {
            target = Object.assign({}, source);
        }
        target.values = source.addresses.map((address) => (Object.assign({}, this.converter.convert(address, ADDRESS_NORMALIZER))));
        return target;
    }
}
OccAddressListNormalizer.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccAddressListNormalizer, deps: [{ token: i1.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable });
OccAddressListNormalizer.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccAddressListNormalizer, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccAddressListNormalizer, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.ConverterService }]; } });
