// @spartacus/checkout/schematics/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CHECKOUT_TRANSLATION_CHUNKS_CONFIG = exports.CHECKOUT_TRANSLATIONS = exports.SPARTACUS_CHECKOUT_ASSETS = exports.SPARTACUS_CHECKOUT_ROOT = exports.CHECKOUT_ROOT_MODULE = exports.CHECKOUT_MODULE = exports.CHECKOUT_FEATURE_NAME_CONSTANT = exports.SCSS_FILE_NAME = exports.CHECKOUT_MODULE_NAME = exports.CHECKOUT_FOLDER_NAME = void 0;
const schematics_1 = require("@spartacus/schematics");
exports.CHECKOUT_FOLDER_NAME = 'checkout';
exports.CHECKOUT_MODULE_NAME = 'Checkout';
exports.SCSS_FILE_NAME = 'checkout.scss';
exports.CHECKOUT_FEATURE_NAME_CONSTANT = 'CHECKOUT_FEATURE';
exports.CHECKOUT_MODULE = 'CheckoutModule';
exports.CHECKOUT_ROOT_MODULE = 'CheckoutRootModule';
exports.SPARTACUS_CHECKOUT_ROOT = `${schematics_1.SPARTACUS_CHECKOUT}/root`;
exports.SPARTACUS_CHECKOUT_ASSETS = `${schematics_1.SPARTACUS_CHECKOUT}/assets`;
exports.CHECKOUT_TRANSLATIONS = 'checkoutTranslations';
exports.CHECKOUT_TRANSLATION_CHUNKS_CONFIG = 'checkoutTranslationChunksConfig';
