// @spartacus/storefinder/esm2015/core/store/actions/view-all-stores.action.js
import { STORE_FINDER_DATA } from '../store-finder-state';
import { StateUtils } from '@spartacus/core';
export const VIEW_ALL_STORES = '[StoreFinder] View All Stores';
export const VIEW_ALL_STORES_FAIL = '[StoreFinder] View All Stores Fail';
export const VIEW_ALL_STORES_SUCCESS = '[StoreFinder] View All Stores Success';
export const CLEAR_STORE_FINDER_DATA = '[StoreFinder] Clear Data';
export class ViewAllStores extends StateUtils.LoaderLoadAction {
    constructor() {
        super(STORE_FINDER_DATA);
        this.type = VIEW_ALL_STORES;
    }
}
export class ViewAllStoresFail extends StateUtils.LoaderFailAction {
    constructor(payload) {
        super(STORE_FINDER_DATA, payload);
        this.payload = payload;
        this.type = VIEW_ALL_STORES_FAIL;
    }
}
export class ViewAllStoresSuccess extends StateUtils.LoaderSuccessAction {
    constructor(payload) {
        super(STORE_FINDER_DATA);
        this.payload = payload;
        this.type = VIEW_ALL_STORES_SUCCESS;
    }
}
export class ClearStoreFinderData {
    constructor() {
        this.type = CLEAR_STORE_FINDER_DATA;
    }
}
