// @spartacus/cart/esm2015/quick-order/root/pages/quick-order-order-entries-context.js
import { Injectable, isDevMode } from '@angular/core';
import { forkJoin, from, of, Subject } from 'rxjs';
import { catchError, filter, switchMap, take, tap } from 'rxjs/operators';
import { OrderEntriesSource, ProductImportStatus, } from '@spartacus/storefront';
import * as i0 from "@angular/core";
import * as i1 from "../facade/quick-order.facade";
import * as i2 from "@spartacus/core";
export class QuickOrderOrderEntriesContext {
    constructor(quickOrderService, productConnector) {
        this.quickOrderService = quickOrderService;
        this.productConnector = productConnector;
        this.type = OrderEntriesSource.QUICK_ORDER;
    }
    getEntries() {
        return this.quickOrderService.getEntries();
    }
    addEntries(productsData) {
        const results$ = new Subject();
        forkJoin(productsData.map((productData) => this.productConnector.get(productData.productCode).pipe(take(1), catchError((response) => {
            this.handleErrors(response, productData.productCode, results$);
            return of(null);
        }))))
            .pipe(switchMap((products) => from(products).pipe(filter((product) => !!product), switchMap((product) => this.quickOrderService.canAdd(product.code).pipe(take(1), tap((canAdd) => {
            const productData = productsData.find((p) => p.productCode === product.code);
            if (canAdd) {
                this.handleResults(product, productData, results$);
                this.quickOrderService.addProduct(product, productData.quantity);
            }
            else {
                results$.next({
                    productCode: productData.productCode,
                    statusCode: ProductImportStatus.LIMIT_EXCEEDED,
                });
            }
        }))))))
            .subscribe();
        return results$.pipe(take(productsData.length));
    }
    handleResults(product, productData, results$) {
        var _a, _b;
        if (((_a = product.stock) === null || _a === void 0 ? void 0 : _a.stockLevel) &&
            productData.quantity >= product.stock.stockLevel) {
            results$.next({
                productCode: productData.productCode,
                productName: product === null || product === void 0 ? void 0 : product.name,
                statusCode: ProductImportStatus.LOW_STOCK,
                quantity: productData.quantity,
                quantityAdded: product.stock.stockLevel,
            });
        }
        else if (((_b = product.stock) === null || _b === void 0 ? void 0 : _b.stockLevelStatus) === 'outOfStock') {
            results$.next({
                productCode: productData.productCode,
                statusCode: ProductImportStatus.NO_STOCK,
                productName: product === null || product === void 0 ? void 0 : product.name,
            });
        }
        else {
            results$.next({
                productCode: productData.productCode,
                statusCode: ProductImportStatus.SUCCESS,
            });
        }
    }
    handleErrors(response, productCode, results$) {
        var _a;
        if (((_a = response === null || response === void 0 ? void 0 : response.error) === null || _a === void 0 ? void 0 : _a.errors[0].type) === 'UnknownIdentifierError') {
            results$.next({
                productCode,
                statusCode: ProductImportStatus.UNKNOWN_IDENTIFIER,
            });
        }
        else {
            if (isDevMode()) {
                console.warn('Unrecognized cart add entry action type while mapping messages', response);
            }
            results$.next({
                productCode,
                statusCode: ProductImportStatus.UNKNOWN_ERROR,
            });
        }
    }
}
QuickOrderOrderEntriesContext.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderOrderEntriesContext, deps: [{ token: i1.QuickOrderFacade }, { token: i2.ProductConnector }], target: i0.ɵɵFactoryTarget.Injectable });
QuickOrderOrderEntriesContext.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderOrderEntriesContext, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderOrderEntriesContext, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.QuickOrderFacade }, { type: i2.ProductConnector }]; } });
