// @spartacus/cart/esm2015/quick-order/core/quick-order-core.module.js
import { NgModule } from '@angular/core';
import { facadeProviders } from './facade/facade-providers';
import { QuickOrderStatePersistenceService } from './services/quick-order-state-persistance.service';
import * as i0 from "@angular/core";
export class QuickOrderCoreModule {
    static forRoot() {
        return {
            ngModule: QuickOrderCoreModule,
        };
    }
}
QuickOrderCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
QuickOrderCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderCoreModule });
QuickOrderCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderCoreModule, providers: [...facadeProviders, QuickOrderStatePersistenceService] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [...facadeProviders, QuickOrderStatePersistenceService],
                }]
        }] });
