// @spartacus/storefront/esm2015/cms-structure/pwa/components/add-to-home-screen-btn/add-to-home-screen-btn.component.js
import { Component } from '@angular/core';
import { AddToHomeScreenComponent } from '../add-to-home-screen.component';
import * as i0 from "@angular/core";
import * as i1 from "../../services/add-to-home-screen.service";
import * as i2 from "@angular/common";
export class AddToHomeScreenBtnComponent extends AddToHomeScreenComponent {
    constructor(addToHomeScreenService) {
        super(addToHomeScreenService);
        this.addToHomeScreenService = addToHomeScreenService;
    }
}
AddToHomeScreenBtnComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToHomeScreenBtnComponent, deps: [{ token: i1.AddToHomeScreenService }], target: i0.ɵɵFactoryTarget.Component });
AddToHomeScreenBtnComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: AddToHomeScreenBtnComponent, selector: "cx-add-to-home-screen-btn", usesInheritance: true, ngImport: i0, template: "<span (click)=\"prompt()\">\n  <ng-content *ngIf=\"canPrompt$ | async\"></ng-content>\n</span>\n", directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i2.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToHomeScreenBtnComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-add-to-home-screen-btn',
                    templateUrl: './add-to-home-screen-btn.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.AddToHomeScreenService }]; } });
