// @spartacus/core/esm2015/src/cart/cart.module.js
import { NgModule } from '@angular/core';
import { CartPersistenceModule } from './cart-persistence.module';
import { CartEventModule } from './event/cart-event.module';
import { MultiCartStoreModule } from './store/multi-cart-store.module';
import * as i0 from "@angular/core";
import * as i1 from "./cart-persistence.module";
export class CartModule {
    static forRoot() {
        return {
            ngModule: CartModule,
        };
    }
}
CartModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CartModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartModule, imports: [MultiCartStoreModule,
        CartEventModule, i1.CartPersistenceModule] });
CartModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartModule, imports: [[
            MultiCartStoreModule,
            CartEventModule,
            CartPersistenceModule.forRoot(),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        MultiCartStoreModule,
                        CartEventModule,
                        CartPersistenceModule.forRoot(),
                    ],
                }]
        }] });
