// @spartacus/storefront/esm2015/shared/services/file/csv/import-csv-file.service.js
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { catchError, map, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../file-reader.service";
export class ImportCsvFileService {
    constructor(fileReaderService) {
        this.fileReaderService = fileReaderService;
    }
    /**
     * Load CSV file.
     *
     * @param file File we want to load as CSV.
     * @param separator Separator for CSV data.
     * @return {Observable<string[][]>} Imported file
     */
    loadFile(file, separator) {
        return this.fileReaderService
            .loadTextFile(file)
            .pipe(map((res) => this.parse(res, separator)));
    }
    /**
     * Combined csv validation
     *
     * @param file File we want to load as CSV.
     * @param separator Separator for CSV data.
     * @param isDataParsable (optional) Callback for verify that structure type is proper.
     * @param maxEntries (optional) Limitation for maximum entries count.
     * @return {Observable<CsvFileValidationErrors | null>} Result of validation
     */
    validateFile(file, { separator, isDataParsable, maxEntries, }) {
        const errors = {};
        return this.fileReaderService.loadTextFile(file).pipe(tap((data) => {
            this.validateEmpty(data, errors);
        }), map((res) => this.parse(res, separator)), tap((data) => {
            this.validateNotParsable(data, errors, isDataParsable);
            this.validateTooManyEntries(data, errors, maxEntries);
        }), catchError((errors) => of(errors)), map(() => (Object.keys(errors).length === 0 ? null : errors)));
    }
    /**
     * Processes the CSV data
     *
     * @param csvString raw extracted data from CSV
     * @param separator for csv data
     * @param ignoreHeader (optional) flag allows for ignore headers row while reading
     * @returns {string[][]} Parsed file
     */
    parse(csvString, separator, ignoreHeader = true) {
        return csvString
            .split('\n')
            .map((row) => row.split(separator).map((cell) => cell.replace(/"/g, '')))
            .filter((value, index) => !(ignoreHeader && index === 0) && value[0] !== '');
    }
    validateEmpty(data, errors) {
        if (data.toString().length === 0) {
            errors.empty = true;
            throw errors;
        }
    }
    validateTooManyEntries(data, errors, maxEntries) {
        if (maxEntries && data.length > maxEntries) {
            errors.tooManyEntries = { maxEntries };
            throw errors;
        }
    }
    validateNotParsable(data, errors, isDataParsable) {
        if (isDataParsable && !isDataParsable(data)) {
            errors.notParsable = true;
            throw errors;
        }
    }
}
ImportCsvFileService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportCsvFileService, deps: [{ token: i1.FileReaderService }], target: i0.ɵɵFactoryTarget.Injectable });
ImportCsvFileService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportCsvFileService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportCsvFileService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.FileReaderService }]; } });
