// @spartacus/storefront/esm2015/cms-components/cart/order-entries-context/cart-order-entries.context.js
import { isDevMode } from '@angular/core';
import { ofType } from '@ngrx/effects';
import { CartActions } from '@spartacus/core';
import { filter, map, switchMap, take } from 'rxjs/operators';
import { ProductImportStatus, } from './import-to-cart.model';
/**
 * An abstract class for contexts based on cart (active cart, saved cart) which share part of logic - mainly handle cart actions and map results.
 */
export class CartOrderEntriesContext {
    constructor(actionsSubject) {
        this.actionsSubject = actionsSubject;
    }
    addEntries(products) {
        return this.add(products).pipe(switchMap((cartId) => this.getResults(cartId)), take(products.length));
    }
    /**
     * Get emission of add entry results from actions subject
     *
     * @param {string} cartId
     * @returns {Observable<ProductImportInfo>}
     */
    getResults(cartId) {
        return this.actionsSubject.pipe(ofType(CartActions.CART_ADD_ENTRY_SUCCESS, CartActions.CART_ADD_ENTRY_FAIL), filter((action) => action.payload.cartId === cartId), map((action) => this.mapMessages(action)));
    }
    /**
     * Map actions to summary messages
     *
     * @param {CartActions.CartAddEntrySuccess | CartActions.CartAddEntryFail} action
     * @returns ProductImportInfo
     */
    mapMessages(action) {
        var _a, _b, _c;
        const { productCode } = action.payload;
        if (action instanceof CartActions.CartAddEntrySuccess) {
            const { quantity, quantityAdded, entry, statusCode } = action.payload;
            if (statusCode === ProductImportStatus.LOW_STOCK) {
                return {
                    productCode,
                    statusCode,
                    productName: (_a = entry === null || entry === void 0 ? void 0 : entry.product) === null || _a === void 0 ? void 0 : _a.name,
                    quantity,
                    quantityAdded,
                };
            }
            if (statusCode === ProductImportStatus.SUCCESS ||
                statusCode === ProductImportStatus.NO_STOCK) {
                return { productCode, statusCode, productName: (_b = entry === null || entry === void 0 ? void 0 : entry.product) === null || _b === void 0 ? void 0 : _b.name };
            }
        }
        else if (action instanceof CartActions.CartAddEntryFail) {
            const { error } = action.payload;
            if (((_c = error === null || error === void 0 ? void 0 : error.details[0]) === null || _c === void 0 ? void 0 : _c.type) === 'UnknownIdentifierError') {
                return {
                    productCode,
                    statusCode: ProductImportStatus.UNKNOWN_IDENTIFIER,
                };
            }
        }
        if (isDevMode()) {
            console.warn('Unrecognized cart add entry action type while mapping messages', action);
        }
        return { productCode, statusCode: ProductImportStatus.UNKNOWN_ERROR };
    }
}
