// @spartacus/user/esm2015/account/root/user-account-root.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfigFactory, UserAccountFacadeTransitionalToken, } from '@spartacus/core';
import { UserAccountFacade } from './facade/user-account.facade';
import { USER_ACCOUNT_CORE_FEATURE, USER_ACCOUNT_FEATURE, } from './feature-name';
import * as i0 from "@angular/core";
// TODO: Inline this factory when we start releasing Ivy compiled libraries
export function defaultUserAccountComponentsConfig() {
    const config = {
        featureModules: {
            [USER_ACCOUNT_FEATURE]: {
                cmsComponents: [
                    'LoginComponent',
                    'ReturningCustomerLoginComponent',
                    'ReturningCustomerRegisterComponent',
                ],
            },
            // by default core is bundled together with components
            [USER_ACCOUNT_CORE_FEATURE]: USER_ACCOUNT_FEATURE,
        },
    };
    return config;
}
export class UserAccountRootModule {
}
UserAccountRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserAccountRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountRootModule });
UserAccountRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountRootModule, providers: [
        provideDefaultConfigFactory(defaultUserAccountComponentsConfig),
        {
            provide: UserAccountFacadeTransitionalToken,
            useExisting: UserAccountFacade,
        },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountRootModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfigFactory(defaultUserAccountComponentsConfig),
                        {
                            provide: UserAccountFacadeTransitionalToken,
                            useExisting: UserAccountFacade,
                        },
                    ],
                }]
        }] });
