// @spartacus/storefront/esm2015/cms-components/cart/add-to-wishlist/add-to-wish-list.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { IconModule } from '../../misc/icon/icon.module';
import { AddToWishListComponent } from './add-to-wish-list.component';
import * as i0 from "@angular/core";
export class AddToWishListModule {
}
AddToWishListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToWishListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AddToWishListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToWishListModule, declarations: [AddToWishListComponent], imports: [CommonModule, I18nModule, IconModule, RouterModule, UrlModule], exports: [AddToWishListComponent] });
AddToWishListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToWishListModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                AddToWishListComponent: {
                    component: AddToWishListComponent,
                },
            },
        }),
    ], imports: [[CommonModule, I18nModule, IconModule, RouterModule, UrlModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToWishListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule, IconModule, RouterModule, UrlModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                AddToWishListComponent: {
                                    component: AddToWishListComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [AddToWishListComponent],
                    exports: [AddToWishListComponent],
                }]
        }] });
