// @spartacus/core/esm2015/src/model/cart.model.js
export var PromotionLocation;
(function (PromotionLocation) {
    PromotionLocation["ActiveCart"] = "CART";
    PromotionLocation["Checkout"] = "CHECKOUT";
    PromotionLocation["Order"] = "ORDER";
    PromotionLocation["SaveForLater"] = "SAVE_FOR_LATER";
    PromotionLocation["SavedCart"] = "SAVED_CART";
})(PromotionLocation || (PromotionLocation = {}));
export var B2BPaymentTypeEnum;
(function (B2BPaymentTypeEnum) {
    B2BPaymentTypeEnum["ACCOUNT_PAYMENT"] = "ACCOUNT";
    B2BPaymentTypeEnum["CARD_PAYMENT"] = "CARD";
})(B2BPaymentTypeEnum || (B2BPaymentTypeEnum = {}));
export var CartValidationStatusCode;
(function (CartValidationStatusCode) {
    CartValidationStatusCode["NO_STOCK"] = "noStock";
    CartValidationStatusCode["LOW_STOCK"] = "lowStock";
    CartValidationStatusCode["REVIEW_CONFIGURATION"] = "reviewConfiguration";
    CartValidationStatusCode["PRICING_ERROR"] = "pricingError";
    CartValidationStatusCode["UNRESOLVABLE_ISSUES"] = "unresolvableIssues";
})(CartValidationStatusCode || (CartValidationStatusCode = {}));
