// @spartacus/checkout/esm2015/components/components/schedule-replenishment-order/schedule-replenishment-order.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { IconModule } from '@spartacus/storefront';
import { CartNotEmptyGuard } from '../../guards/cart-not-empty.guard';
import { CheckoutAuthGuard } from '../../guards/checkout-auth.guard';
import { ScheduleReplenishmentOrderComponent } from './schedule-replenishment-order.component';
import * as i0 from "@angular/core";
export class ScheduleReplenishmentOrderModule {
}
ScheduleReplenishmentOrderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ScheduleReplenishmentOrderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ScheduleReplenishmentOrderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ScheduleReplenishmentOrderModule, declarations: [ScheduleReplenishmentOrderComponent], imports: [CommonModule, RouterModule, I18nModule, IconModule], exports: [ScheduleReplenishmentOrderComponent] });
ScheduleReplenishmentOrderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ScheduleReplenishmentOrderModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                CheckoutScheduleReplenishmentOrder: {
                    component: ScheduleReplenishmentOrderComponent,
                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                },
            },
        }),
    ], imports: [[CommonModule, RouterModule, I18nModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ScheduleReplenishmentOrderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RouterModule, I18nModule, IconModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutScheduleReplenishmentOrder: {
                                    component: ScheduleReplenishmentOrderComponent,
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [ScheduleReplenishmentOrderComponent],
                    exports: [ScheduleReplenishmentOrderComponent],
                }]
        }] });
