// @spartacus/cart/esm2015/import-export/core/import-export-core.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';
import { defaultImportExportConfig } from './config/default-import-export-config';
import * as i0 from "@angular/core";
export class ImportExportCoreModule {
    static forRoot() {
        return {
            ngModule: ImportExportCoreModule,
            providers: [provideDefaultConfig(defaultImportExportConfig)],
        };
    }
}
ImportExportCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ImportExportCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportCoreModule });
ImportExportCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportCoreModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportCoreModule, decorators: [{
            type: NgModule,
            args: [{}]
        }] });
