// @spartacus/checkout/esm2015/core/events/checkout-event.listener.js
import { Injectable } from '@angular/core';
import { RestoreSavedCartSuccessEvent, SaveCartSuccessEvent, SavedCartEvent, } from '@spartacus/cart/saved-cart/root';
import { DeleteUserAddressEvent, UpdateUserAddressEvent, UserAddressEvent, } from '@spartacus/core';
import { Subscription } from 'rxjs';
import { filter } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/checkout/root";
import * as i2 from "@spartacus/core";
export class CheckoutEventListener {
    constructor(checkoutDeliveryFacade, clearCheckoutFacade, eventService) {
        this.checkoutDeliveryFacade = checkoutDeliveryFacade;
        this.clearCheckoutFacade = clearCheckoutFacade;
        this.eventService = eventService;
        this.subscription = new Subscription();
        this.onUserAddressChange();
        this.onSavedCartChange();
    }
    /**
     * Clears the checkout delivery info when an address in the address book is
     * changed ( updated, set as default or deleted )
     *
     * Listens for UpdateUserAddressEvent or DeleteUserAddressEvent
     */
    onUserAddressChange() {
        this.subscription.add(this.eventService
            .get(UserAddressEvent)
            .pipe(filter((event) => {
            return (event instanceof UpdateUserAddressEvent ||
                event instanceof DeleteUserAddressEvent);
        }))
            .subscribe((_event) => {
            this.checkoutDeliveryFacade.clearCheckoutDeliveryDetails();
        }));
    }
    /**
     * What te active cart is saved for later or when a saved card is restored,
     * the whole checkout state is reset.
     *
     * Listens for SaveCartSuccessEvent or RestoreSavedCartSuccessEvent
     */
    onSavedCartChange() {
        this.subscription.add(this.eventService
            .get(SavedCartEvent)
            .pipe(filter((event) => {
            return (event instanceof SaveCartSuccessEvent ||
                event instanceof RestoreSavedCartSuccessEvent);
        }))
            .subscribe((_event) => {
            this.clearCheckoutFacade.resetCheckoutProcesses();
        }));
    }
    ngOnDestroy() {
        var _a;
        (_a = this.subscription) === null || _a === void 0 ? void 0 : _a.unsubscribe();
    }
}
CheckoutEventListener.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutEventListener, deps: [{ token: i1.CheckoutDeliveryFacade }, { token: i1.ClearCheckoutFacade }, { token: i2.EventService }], target: i0.ɵɵFactoryTarget.Injectable });
CheckoutEventListener.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutEventListener, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutEventListener, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CheckoutDeliveryFacade }, { type: i1.ClearCheckoutFacade }, { type: i2.EventService }]; } });
