// @spartacus/checkout/esm2015/core/events/checkout-event.builder.js
import { Injectable } from '@angular/core';
import { CheckoutActions } from '../store/actions/index';
import { OrderPlacedEvent } from '@spartacus/checkout/root';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class CheckoutEventBuilder {
    constructor(stateEventService) {
        this.stateEventService = stateEventService;
        this.register();
    }
    /**
     * Registers checkout events
     */
    register() {
        this.orderPlacedEvent();
    }
    /**
     * Register an order successfully placed event
     */
    orderPlacedEvent() {
        this.stateEventService.register({
            action: CheckoutActions.PLACE_ORDER_SUCCESS,
            event: OrderPlacedEvent,
        });
    }
}
CheckoutEventBuilder.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutEventBuilder, deps: [{ token: i1.StateEventService }], target: i0.ɵɵFactoryTarget.Injectable });
CheckoutEventBuilder.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutEventBuilder });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutEventBuilder, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.StateEventService }]; } });
