// @spartacus/storefinder/esm2015/root/store-finder-root.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { defaultStoreFinderLayoutConfig } from './config/default-store-finder-layout-config';
import { STORE_FINDER_FEATURE } from './feature-name';
import * as i0 from "@angular/core";
// TODO: Inline this factory when we start releasing Ivy compiled libraries
export function defaultStoreFinderComponentsConfig() {
    const config = {
        featureModules: {
            [STORE_FINDER_FEATURE]: {
                cmsComponents: ['StoreFinderComponent'],
            },
        },
    };
    return config;
}
export class StoreFinderRootModule {
}
StoreFinderRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StoreFinderRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule });
StoreFinderRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule, providers: [
        provideDefaultConfig(defaultStoreFinderLayoutConfig),
        provideDefaultConfigFactory(defaultStoreFinderComponentsConfig),
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    providers: [
                        provideDefaultConfig(defaultStoreFinderLayoutConfig),
                        provideDefaultConfigFactory(defaultStoreFinderComponentsConfig),
                    ],
                }]
        }] });
