// @spartacus/cart/esm2015/saved-cart/components/saved-cart-form-dialog/saved-cart-form-dialog.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { I18nModule } from '@spartacus/core';
import { FormErrorsModule, IconModule, KeyboardFocusModule, } from '@spartacus/storefront';
import { SavedCartFormDialogComponent } from './saved-cart-form-dialog.component';
import * as i0 from "@angular/core";
export class SavedCartFormDialogModule {
}
SavedCartFormDialogModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartFormDialogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SavedCartFormDialogModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartFormDialogModule, declarations: [SavedCartFormDialogComponent], imports: [CommonModule,
        FormsModule,
        ReactiveFormsModule,
        FormErrorsModule,
        I18nModule,
        IconModule,
        KeyboardFocusModule], exports: [SavedCartFormDialogComponent] });
SavedCartFormDialogModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartFormDialogModule, imports: [[
            CommonModule,
            FormsModule,
            ReactiveFormsModule,
            FormErrorsModule,
            I18nModule,
            IconModule,
            KeyboardFocusModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SavedCartFormDialogModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        FormErrorsModule,
                        I18nModule,
                        IconModule,
                        KeyboardFocusModule,
                    ],
                    declarations: [SavedCartFormDialogComponent],
                    exports: [SavedCartFormDialogComponent],
                }]
        }] });
