// @spartacus/storefront/esm2015/cms-structure/page/page-layout/page-layout.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PageSlotModule } from '../../../cms-structure/page/slot/page-slot.module';
import { OutletModule } from '../../outlet/outlet.module';
import { PageLayoutComponent } from './page-layout.component';
import { PageTemplateDirective } from './page-template.directive';
import * as i0 from "@angular/core";
export class PageLayoutModule {
}
PageLayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageLayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PageLayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageLayoutModule, declarations: [PageLayoutComponent, PageTemplateDirective], imports: [CommonModule, OutletModule, PageSlotModule], exports: [PageLayoutComponent, PageTemplateDirective] });
PageLayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageLayoutModule, imports: [[CommonModule, OutletModule, PageSlotModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageLayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, OutletModule, PageSlotModule],
                    declarations: [PageLayoutComponent, PageTemplateDirective],
                    exports: [PageLayoutComponent, PageTemplateDirective],
                }]
        }] });
