// @spartacus/storefront/esm2015/cms-structure/routing/routing.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfig, RoutingModule as CoreRoutingModule, } from '@spartacus/core';
import { CmsRouteModule } from './cms-route/cms-route.module';
import { defaultRoutingConfig } from './default-routing-config';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class RoutingModule {
    static forRoot() {
        return {
            ngModule: RoutingModule,
            providers: [provideDefaultConfig(defaultRoutingConfig)],
        };
    }
}
RoutingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: RoutingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
RoutingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: RoutingModule, imports: [i1.RoutingModule, CmsRouteModule] });
RoutingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: RoutingModule, imports: [[CoreRoutingModule.forRoot(), CmsRouteModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: RoutingModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CoreRoutingModule.forRoot(), CmsRouteModule],
                }]
        }] });
