// @spartacus/storefront/esm2015/shared/components/anonymous-consents-dialog/anonymous-consents-dialog.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule } from '@spartacus/core';
import { IconModule } from '../../../cms-components/misc/icon/icon.module';
import { ConsentManagementModule } from '../../../cms-components/myaccount/consent-management/consent-management.module';
import { KeyboardFocusModule } from '../../../layout/a11y/keyboard-focus/index';
import { SpinnerModule } from '../spinner/spinner.module';
import { AnonymousConsentDialogComponent } from './anonymous-consent-dialog.component';
import * as i0 from "@angular/core";
export class AnonymousConsentsDialogModule {
}
AnonymousConsentsDialogModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsDialogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AnonymousConsentsDialogModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsDialogModule, declarations: [AnonymousConsentDialogComponent], imports: [CommonModule,
        I18nModule,
        IconModule,
        SpinnerModule,
        ConsentManagementModule,
        KeyboardFocusModule], exports: [AnonymousConsentDialogComponent] });
AnonymousConsentsDialogModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsDialogModule, imports: [[
            CommonModule,
            I18nModule,
            IconModule,
            SpinnerModule,
            ConsentManagementModule,
            KeyboardFocusModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsDialogModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        I18nModule,
                        IconModule,
                        SpinnerModule,
                        ConsentManagementModule,
                        KeyboardFocusModule,
                    ],
                    declarations: [AnonymousConsentDialogComponent],
                    exports: [AnonymousConsentDialogComponent],
                }]
        }] });
