// @spartacus/core/esm2015/src/features-config/features-config.module.js
import { NgModule } from '@angular/core';
import { FeatureLevelDirective } from './directives/feature-level.directive';
import { FeatureDirective } from './directives/feature.directive';
import { provideDefaultConfig } from '../config/config-providers';
import * as i0 from "@angular/core";
export class FeaturesConfigModule {
    static forRoot(defaultLevel = '3.0') {
        return {
            ngModule: FeaturesConfigModule,
            providers: [
                provideDefaultConfig({
                    features: {
                        level: defaultLevel || '*',
                    },
                }),
            ],
        };
    }
}
FeaturesConfigModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FeaturesConfigModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FeaturesConfigModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FeaturesConfigModule, declarations: [FeatureLevelDirective, FeatureDirective], exports: [FeatureLevelDirective, FeatureDirective] });
FeaturesConfigModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FeaturesConfigModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FeaturesConfigModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [FeatureLevelDirective, FeatureDirective],
                    exports: [FeatureLevelDirective, FeatureDirective],
                }]
        }] });
