// @spartacus/order/esm2015/core/store/actions/orders.action.js
import { StateUtils } from '@spartacus/core';
import { ORDERS } from '../order-state';
export const LOAD_USER_ORDERS = '[Order] Load User Orders';
export const LOAD_USER_ORDERS_FAIL = '[Order] Load User Orders Fail';
export const LOAD_USER_ORDERS_SUCCESS = '[Order] Load User Orders Success';
export const CLEAR_USER_ORDERS = '[Order] Clear User Orders';
export class LoadUserOrders extends StateUtils.LoaderLoadAction {
    constructor(payload) {
        super(ORDERS);
        this.payload = payload;
        this.type = LOAD_USER_ORDERS;
    }
}
export class LoadUserOrdersFail extends StateUtils.LoaderFailAction {
    constructor(payload) {
        super(ORDERS, payload);
        this.payload = payload;
        this.type = LOAD_USER_ORDERS_FAIL;
    }
}
export class LoadUserOrdersSuccess extends StateUtils.LoaderSuccessAction {
    constructor(payload) {
        super(ORDERS);
        this.payload = payload;
        this.type = LOAD_USER_ORDERS_SUCCESS;
    }
}
export class ClearUserOrders extends StateUtils.LoaderResetAction {
    constructor() {
        super(ORDERS);
        this.type = CLEAR_USER_ORDERS;
    }
}
