// @spartacus/tracking/bundles/spartacus-tracking-tms-aep.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/tracking/tms/aep', ['exports', '@angular/core', '@spartacus/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.tracking = global.spartacus.tracking || {}, global.spartacus.tracking.tms = global.spartacus.tracking.tms || {}, global.spartacus.tracking.tms.aep = {}), global.ng.core, global.core));
}(this, (function (exports, i0, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /**
     * Default Adobe Experience Platform Launch collector.
     */
    var AepCollectorService = /** @class */ (function () {
        function AepCollectorService(scriptLoader) {
            this.scriptLoader = scriptLoader;
        }
        /**
         * If the `TmsCollectorConfig.dataLayerProperty` is not specified, it uses the default `digitalData`
         */
        AepCollectorService.prototype.init = function (config, windowObject) {
            var _a, _b;
            var dataLayerProperty = (_a = config.dataLayerProperty) !== null && _a !== void 0 ? _a : 'digitalData';
            windowObject[dataLayerProperty] = (_b = windowObject[dataLayerProperty]) !== null && _b !== void 0 ? _b : {};
            if (config.scriptUrl) {
                this.scriptLoader.embedScript({ src: config.scriptUrl });
            }
        };
        AepCollectorService.prototype.pushEvent = function (config, windowObject, event) {
            var _a;
            var dataLayerProperty = (_a = config.dataLayerProperty) !== null && _a !== void 0 ? _a : 'digitalData';
            windowObject[dataLayerProperty] = Object.assign(Object.assign({}, windowObject[dataLayerProperty]), event);
        };
        return AepCollectorService;
    }());
    AepCollectorService.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: AepCollectorService, deps: [{ token: i1__namespace.ScriptLoader }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    AepCollectorService.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: AepCollectorService, providedIn: 'root' });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: AepCollectorService, decorators: [{
                type: i0.Injectable,
                args: [{ providedIn: 'root' }]
            }], ctorParameters: function () { return [{ type: i1__namespace.ScriptLoader }]; } });

    var defaultAdobeExperiencePlatformConfig = {
        tagManager: {
            aep: {
                collector: AepCollectorService,
            },
        },
    };

    var AepModule = /** @class */ (function () {
        function AepModule() {
        }
        return AepModule;
    }());
    AepModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: AepModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    AepModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: AepModule });
    AepModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: AepModule, providers: [i1.provideDefaultConfig(defaultAdobeExperiencePlatformConfig)] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: AepModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        providers: [i1.provideDefaultConfig(defaultAdobeExperiencePlatformConfig)],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AepCollectorService = AepCollectorService;
    exports.AepModule = AepModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
