// @spartacus/storefront/esm2015/cms-components/product/product-intro/product-intro.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { StarRatingModule } from '../../../shared/components/star-rating/star-rating.module';
import { ProductIntroComponent } from './product-intro.component';
import * as i0 from "@angular/core";
export class ProductIntroModule {
}
ProductIntroModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductIntroModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductIntroModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductIntroModule, declarations: [ProductIntroComponent], imports: [CommonModule, I18nModule, StarRatingModule], exports: [ProductIntroComponent] });
ProductIntroModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductIntroModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                ProductIntroComponent: {
                    component: ProductIntroComponent,
                },
            },
        }),
    ], imports: [[CommonModule, I18nModule, StarRatingModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductIntroModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule, StarRatingModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ProductIntroComponent: {
                                    component: ProductIntroComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [ProductIntroComponent],
                    exports: [ProductIntroComponent],
                }]
        }] });
