// @spartacus/storefront/esm2015/cms-components/content/paragraph/paragraph.component.js
import { ChangeDetectionStrategy, Component, HostListener, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../../cms-structure/page/model/cms-component-data";
import * as i2 from "@angular/router";
import * as i3 from "@angular/common";
import * as i4 from "../../../shared/pipes/suplement-hash-anchors/supplement-hash-anchors.pipe";
export class ParagraphComponent {
    constructor(component, router) {
        this.component = component;
        this.router = router;
    }
    handleClick(event) {
        if (this.router && event.target instanceof HTMLAnchorElement) {
            const element = event.target;
            const href = element === null || element === void 0 ? void 0 : element.getAttribute('href');
            // Use router for internal link navigation
            if ((href === null || href === void 0 ? void 0 : href.indexOf('/')) === 0) {
                event.preventDefault();
                this.router.navigate([`/${href}`]);
            }
        }
    }
}
ParagraphComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ParagraphComponent, deps: [{ token: i1.CmsComponentData }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Component });
ParagraphComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ParagraphComponent, selector: "cx-paragraph", host: { listeners: { "click": "handleClick($event)" } }, ngImport: i0, template: "<p\n  *ngIf=\"component.data$ | async as data\"\n  [innerHTML]=\"data.content | cxSupplementHashAnchors\"\n></p>\n", directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "cxSupplementHashAnchors": i4.SupplementHashAnchorsPipe, "async": i3.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ParagraphComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-paragraph',
                    templateUrl: './paragraph.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.CmsComponentData }, { type: i2.Router }]; }, propDecorators: { handleClick: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
