// @spartacus/storefront/esm2015/cms-components/cart/save-for-later/save-for-later.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule, provideDefaultConfig, } from '@spartacus/core';
import { CartSharedModule } from '../cart-shared/cart-shared.module';
import { SaveForLaterComponent } from './save-for-later.component';
import * as i0 from "@angular/core";
export class SaveForLaterModule {
}
SaveForLaterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SaveForLaterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SaveForLaterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SaveForLaterModule, declarations: [SaveForLaterComponent], imports: [CommonModule, I18nModule, CartSharedModule], exports: [SaveForLaterComponent] });
SaveForLaterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SaveForLaterModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                SaveForLaterComponent: {
                    component: SaveForLaterComponent,
                },
            },
        }),
    ], imports: [[CommonModule, I18nModule, CartSharedModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SaveForLaterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule, CartSharedModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                SaveForLaterComponent: {
                                    component: SaveForLaterComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [SaveForLaterComponent],
                    exports: [SaveForLaterComponent],
                }]
        }] });
