// @spartacus/schematics/src/shared/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SPARTACUS_USER_PROFILE_COMPONENTS = exports.SPARTACUS_USER_PROFILE_CORE = exports.SPARTACUS_USER_PROFILE_OCC = exports.SPARTACUS_USER_PROFILE = exports.SPARTACUS_CHECKOUT_COMPONENTS = exports.SPARTACUS_CHECKOUT_ROOT = exports.SPARTACUS_CHECKOUT_CORE = exports.SPARTACUS_CHECKOUT_OCC = exports.SPARTACUS_CHECKOUT = exports.SPARTACUS_USER = exports.SPARTACUS_TRACKING = exports.SPARTACUS_STOREFINDER = exports.SPARTACUS_SMARTEDIT = exports.SPARTACUS_QUALTRICS_COMPONENTS = exports.SPARTACUS_QUALTRICS = exports.SPARTACUS_PRODUCT_CONFIGURATOR_RULEBASED = exports.SPARTACUS_PRODUCT_CONFIGURATOR_COMMON = exports.SPARTACUS_PRODUCT_CONFIGURATOR = exports.SPARTACUS_PRODUCT = exports.SPARTACUS_CART_SAVED_CART_ROOT = exports.SPARTACUS_CART_SAVED_CART_COMPONENTS = exports.SPARTACUS_CART = exports.SPARTACUS_ASM = exports.SPARTACUS_ORGANIZATION_ADMINISTRATION_COMPONENTS = exports.SPARTACUS_ORGANIZATION_ADMINISTRATION_CORE = exports.SPARTACUS_ORGANIZATION_ADMINISTRATION_ROOT = exports.SPARTACUS_ORGANIZATION = exports.SPARTACUS_STYLES = exports.SPARTACUS_ASSETS = exports.SPARTACUS_EPD_VISUALIZATION = exports.SPARTACUS_DIGITAL_PAYMENTS = exports.SPARTACUS_CDC = exports.SPARTACUS_CDS = exports.SPARTACUS_SETUP_SSR = exports.SPARTACUS_SETUP = exports.SPARTACUS_STOREFRONTLIB = exports.SPARTACUS_CORE = exports.SPARTACUS_SCHEMATICS = exports.SPARTACUS_SCOPE = exports.ANGULAR_OAUTH2_OIDC = exports.PLATFORM_ID = exports.ANGULAR_PLATFORM_BROWSER = exports.ANGULAR_COMMON = exports.RXJS = exports.ANGULAR_HTTP = exports.ANGULAR_ROUTER = exports.ANGULAR_FORMS = exports.ANGULAR_CORE = exports.ANGULAR_SCHEMATICS = exports.UTF_8 = void 0;
exports.FEATURES_CONFIG = exports.OCC_CONFIG = exports.I18N_CONFIG = exports.SITE_CONTEXT_CONFIG = exports.PROVIDE_CONFIG_FUNCTION = exports.CMS_COMPONENT_DATA_CLASS = exports.CONFIG_MODULE_CLASS = exports.EVENTS_MODULE = exports.OCC_MODULE = exports.VIEW_CONFIG_MODULE = exports.STOREFRONT_FOUNDATION_MODULE = exports.MAIN_MODULE = exports.B2B_STOREFRONT_MODULE = exports.B2C_STOREFRONT_MODULE = exports.STOREFRONT_MODULE = exports.BASE_STOREFRONT_MODULE = exports.NGB_MODAL = exports.HTTP_CLIENT = exports.COMPONENT_FACTORY_RESOLVER = exports.LOCATION = exports.APPLICATION_REF = exports.TEMPLATE_REF = exports.VIEW_CONTAINER_REF = exports.WINDOW_REF = exports.ROUTER = exports.ACTIVATED_ROUTE = exports.CHANGE_DETECTOR_REF = exports.JSONLD_PRODUCT_REVIEW_BUILDER = exports.FORM_BUILDER = exports.ACTION_SUBJECT = exports.ACTIONS = exports.STORE = exports.COMPILER = exports.INJECTOR = exports.ELEMENT_REF = exports.RENDERER_2 = exports.OBSERVABLE_CLASS = exports.SPARTACUS_PRODUCT_VARIANTS_ROOT = exports.SPARTACUS_PRODUCT_VARIANTS_COMPONENTS = exports.NG_BOOTSTRAP = exports.NGUNIVERSAL_EXPRESS_ENGINE = exports.NGRX_ROUTER_STORE = exports.NGRX_EFFECTS = exports.NGRX_STORE = exports.SPARTACUS_ORDER = exports.SPARTACUS_USER_ACCOUNT_COMPONENTS = exports.SPARTACUS_USER_ACCOUNT_CORE = exports.SPARTACUS_USER_ACCOUNT_OCC = exports.SPARTACUS_USER_ACCOUNT = exports.SPARTACUS_USER_PROFILE_ROOT = void 0;
exports.CHECKOUT_PAYMENT_SERVICE = exports.OCC_CHECKOUT_ADAPTER = exports.CHECKOUT_ADAPTER = exports.USER_REPLENISHMENT_ORDER_SERVICE = exports.REPLENISHMENT_ORDER_CONNECTOR = exports.USER_ORDER_CONNECTOR = exports.CHECKOUT_CONNECTOR = exports.CHECKOUT_FACADE = exports.CLEAR_CHECKOUT_SERVICE = exports.CLEAR_CHECKOUT_FACADE = exports.CHECKOUT_SERVICE = exports.PRODUCT_SEARCH_SERVICE = exports.ORGANIZATION_PAGE_META_RESOLVER = exports.CHECKOUT_PAGE_META_RESOLVER = exports.CART_PAGE_META_RESOLVER = exports.CATEGORY_PAGE_META_RESOLVER = exports.SEARCH_PAGE_META_RESOLVER = exports.PRODUCT_PAGE_META_RESOLVER = exports.PAGE_LINK_SERVICE = exports.BASE_PAGE_META_RESOLVER = exports.PAGE_META_RESOLVER = exports.CONFIGURATOR_CART_SERVICE = exports.SAVED_CART_FORM_DIALOG_COMPONENT = exports.SAVED_CART_LIST_COMPONENT = exports.SPLIT_VIEW_SERVICE = exports.STORE_DATA_SERVICE = exports.PAGE_META_CONFIG = exports.PAGE_META_SERVICE = exports.CMS_SERVICE = exports.SEMANTIC_PATH_SERVICE = exports.CMS_FEATURES_SERVICE = exports.CHECKOUT_EVENT_LISTENER = exports.CHECKOUT_EVENT_BUILDER = exports.CHECKOUT_EVENT_MODULE = exports.FEATURE_MODULES_SERVICE = exports.FEATURE_CONFIG_SERVICE = exports.AUTH_SERVICE = exports.COMMAND_SERVICE = exports.USER_ADDRESS_CONNECTOR = exports.USER_ADDRESS_SERVICE = exports.CMS_GET_COMPONENT_FROM_PAGE = exports.LOAD_CMS_COMPONENT_SUCCESS_CLASS = exports.LOAD_CMS_COMPONENT_FAIL_CLASS = exports.LOAD_CMS_COMPONENT_CLASS = exports.STOCK_NOTIFICATION_COMPONENT = exports.CMS_ACTIONS = exports.CMS_SELECTORS = exports.CMS_CONFIG = exports.DEFAULT_B2B_OCC_CONFIG = exports.PROVIDE_DEFAULT_CONFIG = void 0;
exports.CHECKOUT_GUARD = exports.ADDRESS_BOOK_COMPONENT_SERVICE = exports.SMART_EDIT_CONFIG = exports.CDS_CONFIG = exports.EVENT_SERVICE = exports.CONSENT_SERVICE = exports.CDS_SPARTACUS_EVENT_SERVICE = exports.SELECTIVE_CART_SERVICE = exports.CART_PAGE_LAYOUT_HANDLER = exports.CHECKOUT_AUTH_GUARD = exports.GLOBAL_MESSAGE_SERVICE = exports.USER_PAYMENT_SERVICE = exports.PAYMENT_METHOD_COMPONENT = exports.ROUTING_CONFIG_SERVICE = exports.ROUTING_CONFIG = exports.URL_PARSING_SERVICE = exports.SEO_CONFIG = exports.CHECKOUT_CONFIG = exports.CHECKOUT_PROGRESS_MOBILE_TOP_COMPONENT = exports.CHECKOUT_ORDER_SUMMARY_COMPONENT = exports.CHECKOUT_REPLENISHMENT_FORM_SERVICE = exports.MINI_CART_COMPONENT = exports.CART_TOTALS_COMPONENT = exports.ROUTING_PAGE_META_RESOLVER = exports.CONTENT_PAGE_META_RESOLVER = exports.CART_NOT_EMPTY_GUARD = exports.CURRENT_PRODUCT_SERVICE = exports.MODAL_SERVICE = exports.ADDED_TO_CART_DIALOG_COMPONENT = exports.ADD_TO_CART_COMPONENT = exports.TRANSLATION_SERVICE = exports.CHECKOUT_CONFIG_SERVICE = exports.SHIPPING_ADDRESS_COMPONENT = exports.ROUTING_SERVICE = exports.NOT_CHECKOUT_AUTH_GUARD = exports.CHECKOUT_DETAILS_SERVICE = exports.AUTH_REDIRECT_SERVICE = exports.ORDER_OVERVIEW_COMPONENT = exports.ORDER_HISTORY_COMPONENT = exports.ORDER_CONFIRMATION_OVERVIEW_COMPONENT = exports.CHECKOUT_LOGIN_COMPONENT = exports.ORDER_RETURN_REQUEST_SERVICE = exports.ORDER_DETAILS_SERVICE = exports.PROMOTION_SERVICE = exports.USER_ORDERS_EFFECT = exports.USER_ORDER_SERVICE = exports.ACTIVE_CART_SERVICE = exports.CART_SERVICE = exports.CART_DATA_SERVICE = exports.CHECKOUT_DELIVERY_SERVICE = void 0;
exports.NAVIGATION_SERVICE = exports.FOOTER_NAVIGATION_COMPONENT = exports.VIEW_CONFIG = exports.PAGE_LAYOUT_SERVICE = exports.CART_ITEM_CONTEXT = exports.CART_ITEM_CONTEXT_SOURCE = exports.PRODUCT_GRID_ITEM_COMPONENT = exports.PRODUCT_LIST_ITEM_CONTEXT_SOURCE = exports.PRODUCT_LIST_ITEM_COMPONENT = exports.PRODUCT_LIST_COMPONENT_SERVICE = exports.PRODUCT_LIST_COMPONENT = exports.PRODUCT_SCROLL_COMPONENT = exports.PRODUCT_CAROUSEL_COMPONENT = exports.PRODUCT_LOADING_SERVICE = exports.PRODUCT_SERVICE = exports.STAR_RATING_COMPONENT = exports.ANONYMOUS_CONSENTS_CONFIG = exports.ANONYMOUS_CONSENTS_SERVICE = exports.REGISTER_COMPONENT = exports.LOGIN_FORM_COMPONENT = exports.PROTECTED_ROUTES_GUARD = exports.PROTECTED_ROUTES_SERVICE = exports.CART_ITEM_COMPONENT = exports.CART_ITEM_LIST_COMPONENT = exports.CUSTOMER_COUPON_SERVICE = exports.CART_COUPON_COMPONENT = exports.CART_VOUCHER_SERVICE = exports.REVIEW_SUBMIT_COMPONENT = exports.ORDER_CONFIRMATION_ITEMS_COMPONENT = exports.ORDER_DETAIL_ITEMS_COMPONENT = exports.CART_DETAILS_COMPONENT = exports.USER_SERVICE = exports.USER_CONSENT_SERVICE = exports.PRODUCT_IMAGES_COMPONENT = exports.ANONYMOUS_CONSENT_DIALOG_COMPONENT = exports.CONSENT_MANAGEMENT_COMPONENT = exports.CONVERTER_SERVICE = exports.OCC_USER_PROFILE_ADAPTER = exports.OCC_USER_ACCOUNT_ADAPTER = exports.OCC_USER_ADAPTER = exports.OCC_USER_ORDER_ADAPTER = exports.OCC_CART_ADAPTER = exports.OCC_CART_ENTRY_ADAPTER = exports.AUTH_CONFIG = exports.OCC_ENDPOINTS_SERVICE = exports.USER_AUTHENTICATION_TOKEN_SERVICE = exports.CLIENT_AUTHENTICATION_TOKEN_SERVICE = exports.CONSENT_MANAGEMENT_FORM_COMPONENT = exports.EXPRESS_CHECKOUT_SERVICE = exports.LOGOUT_GUARD = void 0;
exports.BREAKPOINT_SERVICE = exports.PRODUCT_FACET_NAVIGATION_COMPONENT = exports.QUALTRICS_MODULE = exports.QUALTRICS_EVENT_NAME = exports.QUALTRICS_COMPONENT = exports.QUALTRICS_CONFIG = exports.RENDERER_FACTORY_2 = exports.QUALTRICS_LOADER_SERVICE = exports.BASE_SITE_SERVICE = exports.MULTI_CART_SERVICE = exports.CART_CONFIG_SERVICE = exports.STORE_FINDER_LIST_ITEM_COMPONENT = exports.NAVIGATION_UI_COMPONENT = exports.TAB_PARAGRAPH_CONTAINER_COMPONENT = exports.SEARCH_BOX_COMPONENT_SERVICE = exports.PRODUCT_REVIEW_SERVICE = exports.STORE_FINDER_CONFIG = exports.STORE_FINDER_SERVICE = exports.STORE_FINDER_SEARCH_RESULT_COMPONENT = exports.SMART_EDIT_MODULE = exports.SMART_EDIT_SERVICE = exports.CMS_INJECTOR_SERVICE = exports.COMPONENT_HANDLER_SERVICE = exports.CMS_MAPPING_SERVICE = exports.POSITIONING_SERVICE = exports.POPOVER_SERVICE = exports.POPOVER_DIRECTIVE = exports.POPOVER_COMPONENT = exports.COMPONENT_WRAPPER_DIRECTIVE = exports.AMEND_ORDER_ACTIONS_COMPONENT = exports.PAYMENT_FORM_COMPONENT = exports.ADDRESS_FORM_COMPONENT = exports.ADDRESS_BOOK_COMPONENT = exports.SEARCH_BOX_COMPONENT = exports.PRODUCT_REVIEWS_COMPONENT = exports.USER_NOTIFICATION_PREFERENCE_SERVICE = exports.USER_INTERESTS_SERVICE = exports.PLACE_ORDER_COMPONENT = exports.LAUNCH_DIALOG_SERVICE = exports.LAUNCH_COMPONENT_SERVICE = exports.DEFER_LOADER_SERVICE = exports.OUTLET_SERVICE = exports.OUTLET_DIRECTIVE = exports.DYNAMIC_ATTRIBUTE_SERVICE = exports.PAGE_SLOT_SERVICE = exports.PAGE_SLOT_COMPONENT = exports.EXTERNAL_ROUTES_CONFIG = exports.EXTERNAL_ROUTES_SERVICE = exports.CONFIGURABLE_ROUTES_SERVICE = exports.URL_MATCHER_SERVICE = void 0;
exports.AUTH_GUARD = exports.NOT_AUTH_GUARD = exports.TEST_CONFIG_MODULE = exports.TEST_OUTLET_MODULE = exports.EPD_VISUALIZATION_CONFIG = exports.GUEST_REGISTER_FORM_COMPONENT = exports.UNIT_ADDRESS_FORM_SERVICE = exports.USER_REGISTER_FACADE = exports.USER_PROFILE_FACADE = exports.CONFIGURATOR_ISSUES_NOTIFICATION_COMPONENT = exports.USER_REGISTER_EFFECT = exports.USER_CONNECTOR = exports.PRODUCT_VARIANT_GUARD = exports.ORDER_RETURN_SERVICE = exports.ORDER_RETURN_GUARD = exports.ORDER_CANCELLATION_SERVICE = exports.ORDER_CANCELLATION_GUARD = exports.PRODUCT_REFERENCE_SERVICE = exports.PRODUCT_CAROUSEL_SERVICE = exports.USER_EMAIL_FACADE = exports.UPDATE_EMAIL_COMPONENT_SERVICE = exports.UPDATE_EMAIL_COMPONENT = exports.ITEM_COUNTER_COMPONENT = exports.ORDER_DETAIL_SHIPPING_COMPONENT = exports.PAYMENT_DETAILS_SET_GUARD = exports.DELIVERY_MODE_SET_GUARD = exports.SHIPPING_ADDRESS_SET_GUARD = exports.DELIVERY_MODE_COMPONENT = exports.CHECKOUT_PROGRESS_COMPONENT = exports.CHECKOUT_PROGRESS_MOBILE_BOTTOM_COMPONENT = exports.WISH_LIST_SERVICE = exports.INTERSECTION_SERVICE = exports.MERCHANDISING_CAROUSEL_COMPONENT_SERVICE = exports.MERCHANDISING_CAROUSEL_COMPONENT = exports.ANONYMOUS_CONSENT_OPEN_DIALOG_COMPONENT = exports.ANONYMOUS_CONSENT_LAUNCH_DIALOG_SERVICE = exports.ANONYMOUS_CONSENT_MANAGEMENT_BANNER_COMPONENT = exports.UNIFIED_INJECTOR = exports.LAZY_MODULES_SERVICE = exports.CONFIGURATION_SERVICE = exports.CONFIG_INITIALIZER_SERVICE = exports.ROUTING_PARAMS_SERVICE = exports.HAMBURGER_MENU_SERVICE = exports.STOREFRONT_COMPONENT = exports.KEYBOARD_FOCUS_SERVICE = exports.SKIP_LINK_CONFIG = exports.SKIP_LINK_SERVICE = exports.OUTLET_RENDERER_SERVICE = exports.STORE_FINDER_GRID_COMPONENT = exports.MEDIA_SERVICE = void 0;
exports.CONFIGURATOR_STOREFRONT_UTILS_SERVICE = exports.CONFIGURATOR_UI_SETTINGS_CONFIG = exports.CONFIGURATOR_ATTRIBUTE_QUANTITY_SERVICE = exports.CONFIGURATOR_UTILS_SERVICE = exports.COMMON_CONFIGURATOR_UTILS_SERVICE = exports.REPLENISHMENT_ORDER_HISTORY_COMPONENT = exports.REPLENISHMENT_ORDER_CANCELLATION_COMPONENT = exports.REPLENISHMENT_ORDER_CANCELLATION_LAUNCH_DIALOG_SERVICE = exports.SAVED_CART_FACADE = exports.SAVED_CART_DETAILS_SERVICE = exports.SAVED_CART_DETAILS_OVERVIEW_COMPONENT = exports.SAVED_CART_DETAILS_ACTION_COMPONENT = exports.ADD_TO_SAVED_CART_COMPONENT = exports.SAVED_CART_FORM_LAUNCH_DIALOG_SERVICE = exports.DISABLE_INFO_SERVICE = exports.ITEM_SERVICE = exports.DELETE_ITEM_COMPONENT = exports.TOGGLE_STATUS_COMPONENT = exports.USER_GROUP_USER_LIST_SERVICE = exports.USER_GROUP_USER_LIST_COMPONENT = exports.UNIT_FORM_COMPONENT = exports.UNIT_USER_LIST_COMPONENT = exports.UNIT_COST_CENTER_LIST_COMPONENT = exports.UNIT_CHILDREN_COMPONENT = exports.CURRENT_USER_GROUP_SERVICE = exports.CURRENT_UNIT_SERVICE = exports.MESSAGE_SERVICE = exports.OCC_CMS_COMPONENT_ADAPTER = exports.SPLIT_VIEW_COMPONENT = exports.VIEW_COMPONENT = exports.ANONYMOUS_CONSENT_TEMPLATES_CONNECTOR = exports.ANONYMOUS_CONSENT_TEMPLATES_ADAPTER = exports.DOM_SANITIZER = exports.JSON_LD_SCRIPT_FACTORY = exports.OUTLET_REF_DIRECTIVE = exports.CLOSE_ACCOUNT_MODAL_COMPONENT = exports.AUTH_CONFIG_SERVICE = exports.FORGOT_PASSWORD_COMPONENT = exports.SEARCH_BOX_SERVICE = exports.CURRENCY_SERVICE = exports.LANGUAGE_SERVICE = exports.MULTI_CART_STATE_PERSISTENCE_SERVICE = exports.FORBIDDEN_HANDLER = exports.COLLAPSE_MENU_IF_CLICK_OUTSIDE = exports.CDC_LOGOUT_GUARD = exports.CDC_AUTH_SERVICE = exports.AUTH_STORAGE_SERVICE = exports.OAUTH_LIB_WRAPPER_SERVICE = exports.USER_ID_SERVICE = exports.AUTH_REDIRECT_STORAGE_SERVICE = void 0;
exports.CART_COMBINED_EFFECTS = exports.CART_ENTRY_EFFECTS = exports.CART_VOUCHER_EFFECTS = exports.WISHLIST_EFFECTS = exports.CART_EFFECTS = exports.KEYBOARD_FOCUS_CONFIG = exports.ON_NAVIGATE_FOCUS_SERVICE = exports.CHECKOUT_PAYMENT_FACADE = exports.CHECKOUT_DELIVERY_FACADE = exports.ON_RESTORE_COMPLETE = exports.RESTORE_SAVED_CART = exports.OPEN_DIALOG = exports.SAVED_CART_FORM_TYPE = exports.SCRIPT_LOADER = exports.EXTERNAL_JS_FILE_LOADER = exports.FORM$ = exports.FORM_GROUP = exports.INITIALIZE = exports.SCRIPT_LOADER_SERVICE = exports.PARAMS = exports.URL = exports.SEMANTIC_ROUTE = exports.CONTEXT = exports.NAVIGATION_EVENT = exports.PAGE_EVENT = exports.PAGE_EVENT_MODULE = exports.PAGE_EVENT_BUILDER = exports.PRODUCT_PAGE_EVENT_BUILDER = exports.HOME_PAGE_EVENT_BUILDER = exports.CART_PAGE_EVENT_BUILDER = exports.DIRECTION_SERVICE = exports.CONFIGURATOR_OVERVIEW_ATTRIBUTE_COMPONENT = exports.CONFIGURATOR_FORM_COMPONENT = exports.CONFIGURATOR_MESSAGE_CONFIG = exports.MESSAGE_CONFIG = exports.CONFIGURATOR_GROUP_MENU_SERVICE = exports.CONFIGURATOR_GROUP_MENU_COMPONENT = exports.CONFIGURATOR_PRODUCT_TITLE_COMPONENT = exports.CONFIGURATOR_ATTRIBUTE_RADIO_BUTTON_COMPONENT = exports.CONFIGURATOR_ATTRIBUTE_INPUT_FIELD_COMPONENT = exports.CONFIGURATOR_ATTRIBUTE_DROP_DOWN_COMPONENT = exports.CONFIGURATOR_ATTRIBUTE_CHECKBOX_LIST_COMPONENT = exports.CONFIGURATOR_ATTRIBUTE_NUMERIC_INPUT_FIELD_SERVICE = exports.CONFIGURATOR_ATTRIBUTE_NUMERIC_INPUT_FIELD_COMPONENT = exports.CONFIGURATOR_ATTRIBUTE_SINGLE_SELECTION_BASE_COMPONENT = exports.CONFIGURATOR_UPDATE_MESSAGE_COMPONENT = exports.CONFIGURATOR_CART_ENTRY_INFO_COMPONENT = exports.CONFIGURATOR_ROUTER_EXTRACTOR_SERVICE = exports.CONFIGURATOR_COMMONS_SERVICE = exports.CONFIGURATOR_GROUPS_SERVICE = void 0;
exports.ENTITY_PROCESSES_LOADER_REDUCER = exports.ENTITY_LOADER_REDUCER = exports.ENTITY_REDUCER = exports.PROCESSES_LOADER_REDUCER = exports.LOADER_REDUCER = exports.STATE_ENTITY_PROCESSESS_LOADER_SELECTORS = exports.STATE_ENTITY_LOADER_SELECTORS = exports.STATE_ENTITY_SELECTORS = exports.STATE_PROCESSES_LOADER_SELECTORS = exports.STATE_LOADER_SELECTORS = exports.STATE_ENTITY_PROCESSESS_LOADER_ACTIONS = exports.STATE_ENTITY_LOADER_ACTIONS = exports.STATE_ENTITY_ACTIONS = exports.STATE_PROCESSES_LOADER_ACTIONS = exports.STATE_LOADER_ACTIONS = exports.ENTITY_PROCESSES_LOADER_STATE = exports.ENTITY_LOADER_STATE = exports.ENTITY_STATE = exports.PROCESSES_LOADER_STATE = exports.LOADER_STATE = exports.BILLING_ADDRESS_FORM_MODULE = exports.BILLING_ADDRESS_FORM_COMPONENT = exports.FORM_UTILS = exports.SKIP_LINK_FACTORY = exports.GET_STRUCTURED_DATA_FACTORY = exports.PWA_FACTORY = exports.PWA_CONFIGURATION_FACTORY = exports.ANONYMOUS_USERID_CONST = exports.CMS_COMPONENTS_SERVICE = exports.URL_MATCHER_FACTORY_SERVICE = exports.CONTEXT_SERVICE_PROVIDERS = exports.INITITIALIZE_CONTEXT = exports.INIT_SITE_CONTEXT_ROUTES_HANDLER = exports.SITE_CONTEXT_PARAMS_PROVIDERS = exports.MULTI_CART_REDUCER_PROVIDER = exports.GET_MULTI_CART_REDUCERS = exports.MULTI_CART_REDUCER_TOKEN = exports.MULTI_CART_META_REDUCERS = exports.CLEAR_MULTI_CART_STATE = exports.META_REDUCERS = exports.CLEAR_CART_STATE = exports.REDUCER_PROVIDER = exports.REDUCER_TOKEN = exports.GET_REDUCERS = exports.CART_STATE = exports.CART_FEATURE_CONST = exports.CARTS_STATE = exports.STATE_WITH_CART = exports.CART_DATA_CONST = exports.CART_SELECTORS = void 0;
exports.TOKEN_TARGET = exports.AUTH_HTTP_HEADER_SERVICE = exports.ASM_AUTH_HTTP_HEADER_SERVICE = exports.AUTH_FLOW_ROUTES_SERVICE = exports.ASM_UI = exports.CUSTOMER_SEARCH_OPTIONS = exports.CUSTOMER_SEARCH_PAGE = exports.CS_AGENT_AUTH_SERVICE = exports.ASM_SERVICE = exports.CUSTOMER_SEARCH_PAGE_NORMALIZER = exports.ASM_CONNECTOR = exports.ASM_ADAPTER = exports.ASM_CONFIG = exports.OCC_ASM_ADAPTER = exports.ASM_OCC_MODULE = exports.ASM_MODULE = exports.CUSTOMER_SUPPORT_AGENT_TOKEN_INTERCEPTOR = exports.CSAGENT_TOKEN_DATA = exports.ASM_ACTIONS = exports.ASM_SELECTORS = exports.KYMA_CONFIG = exports.KYMA_SERVICE = exports.KYMA_MODULE = exports.OPEN_ID_TOKEN = exports.OPEN_ID_AUTHENTICATION_TOKEN_SERVICE = exports.KYMA_ACTIONS = exports.KYMA_STATE = exports.STATE_WITH_KYMA = exports.OPEN_ID_TOKEN_DATA = exports.KYMA_FEATURE = exports.KYMA_SELECTORS = exports.AUTHENTICATION_TOKEN = exports.USER_TOKEN = exports.AUTH_ACTIONS = exports.AUTH_FEATURE = exports.USER_TOKEN_STATE = exports.AUTH_STATE = exports.STATE_WITH_AUTH = exports.AUTH_SELECTORS = exports.AUTO_FOCUS_DIRECTIVE_MODULE = exports.AUTO_FOCUS_DIRECTIVE = exports.ONLY_NUMBER_DIRECTIVE = exports.ONLY_NUMBER_DIRECTIVE_MODULE = exports.OF_LOADER_SUCCESS = exports.OF_LOADER_FAIL = exports.OF_LOADER_LOAD = exports.GET_STATE_SLICE = exports.INITIAL_PROCESSES_STATE = exports.INITIAL_LOADER_STATE = exports.INITIAL_ENTITY_STATE = void 0;
exports.ABSTRACT_STORE_ITEM_COMPONENT = exports.STORE_FINDER_SEARCH_PAGE = exports.STORE_COUNT = exports.STORE_FINDER_SELECTORS = exports.VIEW_ALL_STORES_STATE = exports.FIND_STORES_STATE = exports.STORES_STATE = exports.STATE_WITH_STORE_FINDER = exports.STORE_FINDER_DATA = exports.STORE_FINDER_FEATURE = exports.STORE_FINDER_STORE_MODULE = exports.STORE_FINDER_CORE_MODULE = exports.GOOGLE_MAP_RENDERER_SERVICE = exports.STORE_ENTITIES = exports.STORE_FINDER_SEARCH_QUERY = exports.STORE_FINDER_ADAPTER = exports.STORE_FINDER_CONNECTOR = exports.STORE_COUNT_NORMALIZER = exports.STORE_FINDER_SEARCH_PAGE_NORMALIZER = exports.FIND_STORES_CLASS = exports.STORE_FINDER_ACTIONS = exports.STORE_FINDER_SEARCH_CONFIG = exports.SEARCH_CONFIG = exports.STORE_FINDER_OCC_MODULE = exports.OCC_STORE_FINDER_ADAPTER = exports.GET_ASM_STATE = exports.GET_CUSTOMER_SEARCH_RESULTS_LOADING = exports.GET_CUSTOMER_SEARCH_RESULTS = exports.GET_CUSTOMER_SEARCH_RESULTS_LOADER_STATE = exports.GET_ASM_UI = exports.ASM_STATE = exports.STATE_WITH_ASM = exports.CUSTOMER_SEARCH_DATA = exports.ASM_FEATURE = exports.CUSTOMER_ACTION = exports.CUSTOMER_SEARCH_RESET_CLASS = exports.CUSTOMER_SEARCH_RESET = exports.CUSTOMER_SEARCH_SUCCESS_CLASS = exports.CUSTOMER_SEARCH_SUCCESS = exports.CUSTOMER_SEARCH_FAIL_CLASS = exports.CUSTOMER_SEARCH_FAIL = exports.CUSTOMER_SEARCH_CLASS = exports.CUSTOMER_SEARCH = exports.ASM_UI_ACTION = exports.ASM_UI_UPDATE_CLASS = exports.ASM_UI_UPDATE = exports.ASM_STATE_PERSISTENCE_SERVICE = exports.SYNCED_ASM_STATE = exports.ASM_AUTH_SERVICE = exports.ASM_AUTH_STORAGE_SERVICE = void 0;
exports.TRANSLATION_CHUNKS_CONFIG = exports.PERSONALIZATION_CONTEXT = exports.PERSONALIZATION_ACTION = exports.PERSONALIZATION_CONTEXT_SERVICE = exports.PERSONALIZATION_CONFIG = exports.PERSONALIZATION_MODULE = exports.DEFAULT_SESSION_STORAGE_KEY = exports.DEFAULT_LOCAL_STORAGE_KEY = exports.DEFAULT_STATE_CONFIG = exports.PRODUCT_VARIANT_STYLE_ICONS_MODULE = exports.PRODUCT_VARIANT_STYLE_ICONS_COMPONENT = exports.PRODUCT_VARIANTS_MODULE = exports.VARIANT_STYLE_SELECTOR_MODULE = exports.VARIANT_STYLE_SELECTOR_COMPONENT = exports.VARIANT_STYLE_ICONS_MODULE = exports.VARIANT_STYLE_ICONS_COMPONENT = exports.VARIANT_SIZE_SELECTOR_MODULE = exports.VARIANT_SIZE_SELECTOR_COMPONENT = exports.VARIANT_COLOR_SELECTOR_MODULE = exports.VARIANT_COLOR_SELECTOR_COMPONENT = exports.PRODUCT_VARIANT_COMPONENT = exports.GET_RAW_ENDPOINT = exports.GET_END_POINT = exports.GET_URL = exports.GET_BASE_ENDPOINT = exports.GET_OCC_ENDPOINT = exports.SPLIT_VIEW_DEACTIVATE_GUARD = exports.TOKEN_REVOCATION_HEADER = exports.LAYOUT_CONFIG = exports.NG_EXPRESS_ENGINE_DECORATOR = exports.UNAUTHORIZED_ERROR_HANDLER = exports.PAYMENT_TYPE_SERVICE = exports.CHECKOUT_COST_CENTER_SERVICE = exports.USER_COST_CENTER_SERVICE = exports.CHECKOUT_STEP_SERVICE = exports.PLACE_ORDER_CLASS = exports.CHECKOUT_ACTIONS = exports.ON_SUCCESS = exports.STORE_FINDER_SCSS_FILE_NAME = exports.STORE_FINDER_COMPONENT = exports.STORE_FINDER_PAGINATION_DETAILS_COMPONENT = exports.STORE_FINDER_HEADER_COMPONENT = exports.STORE_FINDER_MODULE = exports.STORE_FINDER_STORES_COUNT_COMPONENT = exports.STORE_FINDER_STORE_COMPONENT = exports.STORE_FINDER_STORE_DESCRIPTION_COMPONENT = exports.STORE_FINDER_SEARCH_COMPONENT = exports.STORE_FINDER_LIST_COMPONENT = exports.STORE_FINDER_MAP_COMPONENT = exports.SCHEDULE_COMPONENT = void 0;
exports.LOGIN_MODULE = exports.LOGIN_COMPONENT = exports.RESET_PASSWORD_FORM_COMPONENT = exports.RESET_PASSWORD_COMPONENT = exports.RESET_PASSWORD_MODULE = exports.REGISTER_COMPONENT_MODULE = exports.FORGOT_PASSWORD_MODULE = exports.CLOSE_ACCOUNT_MODULE = exports.CMS_LIB_MODULE = exports.USER_REGISTER_FACADE_TRANSITIONAL_TOKEN = exports.USER_PROFILE_FACADE_TRANSITIONAL_TOKEN = exports.USER_PROFILE_MODULE = exports.USER_ACCOUNT_MODULE = exports.USER_MODULE = exports.USER_STATE_INTERFACE = exports.USER_DETAILS_STATE_INTERFACE = exports.USER_DETAILS_ACTION = exports.UPDATE_PASSWORD_ACTION = exports.EMAIL_ACTIONS = exports.RESET_PASSWORD_ACTION = exports.FORGOT_PASSWORD_EMAIL_ACTION = exports.USER_ACTIONS = exports.USER_SIGN_UP_INTERFACE = exports.USER_PROFILE_CONNECTOR = exports.USER_ACCOUNT_CONNECTOR = exports.USER_PROFILE_ADAPTER = exports.USER_ACCOUNT_ADAPTER = exports.USER_ADAPTER = exports.TITLE_NORMALIZER = exports.USER_SIGN_UP_SERIALIZER = exports.USER_NORMALIZER = exports.USER_SERIALIZER = exports.USER_PROFILE_SERIALIZER = exports.USER_PROFILE_NORMALIZER = exports.USER_ACCOUNT_SERIALIZER = exports.USER_ACCOUNT_NORMALIZER = exports.USER_CLOSE_ACCOUNT_ENDPOINT = exports.USER_UPDATE_PROFILE_ENDPOINT = exports.USER_UPDATE_PASSWORD_ENDPOINT = exports.USER_UPDATE_LOGIN_ID_ENDPOINT = exports.USER_RESET_PASSWORD_ENDPOINT = exports.USER_FORGOT_PASSWORD_ENDPOINT = exports.USER_REGISTER_ENDPOINT = exports.USER_ENDPOINT = exports.TITLES_ENDPOINT = exports.REGISTER_TRANSLATION_CHUNK = exports.LOGIN_FORM_TRANSLATION_CHUNK = exports.FORGOTTEN_PASSWORD_TRANSLATION_CHUNK = exports.UPDATE_EMAIL_FORM_TRANSLATION_CHUNK = exports.MINI_LOGIN_TRANSLATION_CHUNK = void 0;
exports.REVOKE_USER_TOKEN_SUCCESS = exports.REVOKE_USER_TOKEN_FAIL = exports.REVOKE_USER_TOKEN = exports.REFRESH_USER_TOKEN_SUCCESS = exports.REFRESH_USER_TOKEN_FAIL = exports.REFRESH_USER_TOKEN = exports.LOAD_USER_TOKEN_SUCCESS = exports.LOAD_USER_TOKEN_FAIL = exports.LOAD_USER_TOKEN = exports.LOGOUT_CUSTOMER_SUPPORT_AGENT_CLASS = exports.LOGOUT_CUSTOMER_SUPPORT_AGENT = exports.LOAD_CLIENT_TOKEN_FAIL = exports.LOAD_CLIENT_TOKEN_FAIL_CLASS = exports.LOAD_CLIENT_TOKEN_SUCCESS = exports.LOAD_CLIENT_TOKEN_SUCCESS_CLASS = exports.LOAD_CLIENT_TOKEN = exports.LOAD_CLIENT_TOKEN_CLASS = exports.ANONYMOUS_CONSENTS = exports.KYMA_ENABLED = exports.CMS_COMPONENT_DATA_PROPERTY_NAME = exports.B2C_LAYOUT_CONFIG = exports.DEFAULT_USER_ROUTING_CONFIG = exports.DEFAULT_USER_GROUP_ROUTING_CONFIG = exports.DEFAULT_UNITS_ROUTING_CONFIG = exports.DEFAULT_PERMISSION_ROUTING_CONFIG = exports.DEFAULT_COST_CENTER_ROUTING_CONFIG = exports.DEFAULT_BUDGET_ROUTING_CONFIG = exports.USER_ROUTING_CONFIG = exports.USER_GROUP_ROUTING_CONFIG = exports.UNITS_ROUTING_CONFIG = exports.PERMISSION_ROUTING_CONFIG = exports.COST_CENTER_ROUTING_CONFIG = exports.BUDGET_ROUTING_CONFIG = exports.ROUTE_BACK_ACTION = exports.ROUTE_FORWARD_ACTION = exports.ROUTE_GO_BY_URL_ACTION = exports.ROUTE_GO_ACTION = exports.ROUTING_ACTIONS = exports.UPDATE_PROFILE_COMPONENT = exports.UPDATE_PASSWORD_FORM_COMPONENT = exports.UPDATE_PASSWORD_COMPONENT = exports.UPDATE_EMAIL_FORM_COMPONENT = exports.CLOSE_ACCOUNT_COMPONENT = exports.USER_COMPONENT_MODULE = exports.UPDATE_PROFILE_MODULE = exports.UPDATE_PASSWORD_MODULE = exports.UPDATE_EMAIL_MODULE = exports.LOGIN_REGISTER_MODULE = exports.LOGIN_REGISTER_COMPONENT = exports.LOGIN_FORM_MODULE = void 0;
exports.GET_COMPONENTS_STATE_NEW_API = exports.GET_COMPONENT_STATE_OLD_API = exports.LOCATION$ = exports.PROMOTION_LOCATION$ = exports.TABLE_HEADER = exports.SORT_CODE = exports.DEFAULT_PAGE_SIZE = exports.TOGGLE = exports.HANDLE_OPEN = exports.INSIDE_CLICKED = exports.ORDER_PROMOTIONS$ = exports.ORDER_ENTRY = exports.ITEM = exports.INCREMENT = exports.SERVER_REQUEST_ORIGIN_STRING = exports.SERVER_REQUEST_URL_STRING = exports.DOCUMENT_STRING = exports.DOCUMENT = exports.STRING_TYPE = exports.OBJECT_TYPE = exports.ANY_TYPE = exports.CONFIG = exports.DEFAULT_CONFIG = exports.STOREFRONT_CONFIG = exports.ROOT_CONFIG = exports.PLATFORM_ID_STRING = exports.PLATFORM = exports.INJECT_DECORATOR = exports.LEGACY_FLAG = exports.LOAD_ANONYMOUS_CONSENTS = exports.SUB = exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_SUCCESS_CLASS = exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_FAIL_CLASS = exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_CLASS = exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_SUCCESS = exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_FAIL = exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN = exports.GET_CUSTOMER_AGENT_TOKEN_LOADING = exports.GET_CUSTOMER_AGENT_TOKEN = exports.GET_CUSTOMER_AGENT_TOKEN_STATE = exports.CAN_ACTIVATE = exports.REVOKE_USER_TOKEN_SUCCESS_CLASS = exports.REVOKE_USER_TOKEN_FAIL_CLASS = exports.REVOKE_USER_TOKEN_CLASS = exports.REFRESH_USER_TOKEN_SUCCESS_CLASS = exports.REFRESH_USER_TOKEN_FAIL_CLASS = exports.REFRESH_USER_TOKEN_CLASS = exports.LOAD_USER_TOKEN_SUCCESS_CLASS = exports.LOAD_USER_TOKEN_FAIL_CLASS = exports.LOAD_USER_TOKEN_CLASS = void 0;
exports.SET_ACTIVE = exports.INIT_SYNC = exports.SYNC = exports.REDIRECT = exports.LOGOUT = exports.GET_BREAKPOINT = exports.GET_CLOSEST = exports.GET_WINDOW = exports.GET = exports.GET_PRODUCT_REFERENCES = exports.LOAD_PRODUCT_REFERENCES = exports.GET_BASE_SITE_DATA = exports.LOGIN_AS_GUEST = exports.NG_ON_INIT = exports.NG_ON_DESTROY = exports.GET_ALL = exports.GET_INJECTORS = exports.GET_CHILD_ROUTES = exports.STEPS = exports.GET_CURRENT_STEP_INDEX = exports.GET_PREVIOUS_CHECKOUT_STEP_URL = exports.GET_NEXT_CHECKOUT_STEP_URL = exports.GET_FIRST_CHECKOUT_STEP_ROUTE = exports.GET_CHECKOUT_STEP_ROUTE = exports.GET_CHECKOUT_STEP = exports.GET_CONTROL = exports.GET_ORDER_CURRENT_DATE_CARD_CONTENT = exports.GET_SHIPPING_METHOD_CARD_CONTENT = exports.GET_PAYMENT_CARD_CONTENT = exports.GET_BILLING_ADDRESS_CARD_CONTENT = exports.GET_PAYMENT_INFO_CARD_CONTENT = exports.GET_DELIVERY_MODE_CARD_CONTENT = exports.GET_ADDRESS_CARD_CONTENT = exports.GET_ORDER_DETAILS = exports.VIEW_PAGE = exports.SET_QUERY = exports.PLACE_ORDER = exports.GET_GLOB_URL_MATCHER = exports.GET_OPPOSITE_URL_MATCHER = exports.GET_PATH_URL_MATCHER = exports.GET_MULTIPLE_PATHS_URL_MATCHER = exports.GET_FALSY_URL_MATCHER = exports.ADD_DYNAMIC_ATTRIBUTES = exports.IS_LAUNCHED_IN_SMART_EDIT = exports.IS_LAUNCH_IN_SMART_EDIT = exports.COMPONENTS_SELECTOR_FACTORY_NEW_API = exports.COMPONENT_SELECTOR_FACTORY_OLD_API = exports.COMPONENTS_STATE_SELECTOR_FACTORY_NEW_API = exports.COMPONENT_STATE_SELECTOR_FACTORY_OLD_API = exports.GET_COMPONENT_ENTITIES_OLD_API = void 0;
exports.CLI_ORGANIZATION_ADMINISTRATION_FEATURE = exports.CLI_EPD_VISUALIZATION_FEATURE = exports.CLI_DIGITAL_PAYMENTS_FEATURE = exports.CLI_CDS_FEATURE = exports.CLI_CDC_FEATURE = exports.CLI_CART_QUICK_ORDER_FEATURE = exports.CLI_CART_SAVED_CART_FEATURE = exports.CLI_CART_IMPORT_EXPORT_FEATURE = exports.CLI_ASM_FEATURE = exports.SPARTACUS_CONFIGURATION_MODULE = exports.SPARTACUS_FEATURES_NG_MODULE = exports.SPARTACUS_FEATURES_MODULE = exports.SPARTACUS_MODULE = exports.SPARTACUS_ROUTING_MODULE = exports.CSS_V4_DOCS_URL = exports.CSS_V3_DOCS_URL = exports.CSS_V2_DOCS_URL = exports.TODO_SPARTACUS = exports.PRODUCT_CONFIGURATOR_TEXTFIELD_FEATURE = exports.PRODUCT_CONFIGURATOR_RULEBASED_FEATURE = exports.PRODUCT_CONFIGURATOR_TEXTFIELD_FEATURE_OBSOLETE = exports.PRODUCT_CONFIGURATOR_RULEBASED_FEATURE_OBSOLETE = exports.GET_STABLE = exports.GET_STABLE_CONFIG = exports.PREVENT_SCROLLING_ON_SPACE = exports.RESET_CHECKOUT_PROCESSES = exports.NAVIGATE_UP_ON_ENTER = exports.CLICK_ON_ENTER = exports.GET_PRODUCT_IMAGE_ALT = exports.GET_PRODUCT_IMAGE_URL = exports.ON_DESELECT = exports.ON_SELECT = exports.CREATE_EVENT_FROM_INPUT = exports.OCC_CONFIG_LOADER_MODULE = exports.OCC_SITES_CONFIG_LOADER = exports.OCC_LOADED_CONFIG = exports.OCC_LOADED_CONFIG_CONVERTER = exports.OCC_CONFIG_LOADER_SERVICE = exports.BASE_SITES_FOR_CONFIG = exports.OCC_ENDPOINT = exports.OCC_ENDPOINTS = exports.GO = exports.STOREFINDER_MODULE = exports.IS_STABLE = exports.GET_LOADED = exports.FIND_COMPONENTS_BY_IDS_LEGACY = exports.GET_META_RESOLVER = exports.GET_META = exports.SET_RATE = exports.SET_RATE_ON_EVENT = void 0;
exports.CLI_ORDER_FEATURE = exports.CLI_CHECKOUT_FEATURE = exports.CLI_USER_PROFILE_FEATURE = exports.CLI_USER_ACCOUNT_FEATURE = exports.CLI_TRACKING_TMS_AEP_FEATURE = exports.CLI_TRACKING_TMS_GTM_FEATURE = exports.CLI_TRACKING_PERSONALIZATION_FEATURE = exports.CLI_STOREFINDER_FEATURE = exports.CLI_SMARTEDIT_FEATURE = exports.CLI_QUALTRICS_FEATURE = exports.CLI_PRODUCT_CONFIGURATOR_CPQ_FEATURE = exports.CLI_PRODUCT_CONFIGURATOR_TEXTFIELD_FEATURE = exports.CLI_PRODUCT_CONFIGURATOR_VC_FEATURE = exports.CLI_PRODUCT_IMAGE_ZOOM_FEATURE = exports.CLI_PRODUCT_VARIANTS_FEATURE = exports.CLI_PRODUCT_BULK_PRICING_FEATURE = exports.CLI_ORGANIZATION_ORDER_APPROVAL_FEATURE = void 0;
exports.UTF_8 = 'utf-8';
/***** Imports start *****/
exports.ANGULAR_SCHEMATICS = '@schematics/angular';
exports.ANGULAR_CORE = '@angular/core';
exports.ANGULAR_FORMS = '@angular/forms';
exports.ANGULAR_ROUTER = '@angular/router';
exports.ANGULAR_HTTP = '@angular/common/http';
exports.RXJS = 'rxjs';
exports.ANGULAR_COMMON = '@angular/common';
exports.ANGULAR_PLATFORM_BROWSER = '@angular/platform-browser';
exports.PLATFORM_ID = 'PLATFORM_ID';
exports.ANGULAR_OAUTH2_OIDC = 'angular-oauth2-oidc';
exports.SPARTACUS_SCOPE = '@spartacus/';
exports.SPARTACUS_SCHEMATICS = '@spartacus/schematics';
exports.SPARTACUS_CORE = '@spartacus/core';
exports.SPARTACUS_STOREFRONTLIB = '@spartacus/storefront';
exports.SPARTACUS_SETUP = '@spartacus/setup';
exports.SPARTACUS_SETUP_SSR = `${exports.SPARTACUS_SETUP}/ssr`;
exports.SPARTACUS_CDS = '@spartacus/cds';
exports.SPARTACUS_CDC = '@spartacus/cdc';
exports.SPARTACUS_DIGITAL_PAYMENTS = '@spartacus/digital-payments';
exports.SPARTACUS_EPD_VISUALIZATION = '@spartacus/epd-visualization';
exports.SPARTACUS_ASSETS = '@spartacus/assets';
exports.SPARTACUS_STYLES = '@spartacus/styles';
exports.SPARTACUS_ORGANIZATION = '@spartacus/organization';
exports.SPARTACUS_ORGANIZATION_ADMINISTRATION_ROOT = `${exports.SPARTACUS_ORGANIZATION}/administration/root`;
exports.SPARTACUS_ORGANIZATION_ADMINISTRATION_CORE = `${exports.SPARTACUS_ORGANIZATION}/administration/core`;
exports.SPARTACUS_ORGANIZATION_ADMINISTRATION_COMPONENTS = `${exports.SPARTACUS_ORGANIZATION}/administration/components`;
exports.SPARTACUS_ASM = '@spartacus/asm';
exports.SPARTACUS_CART = '@spartacus/cart';
exports.SPARTACUS_CART_SAVED_CART_COMPONENTS = `${exports.SPARTACUS_CART}/saved-cart/components`;
exports.SPARTACUS_CART_SAVED_CART_ROOT = `${exports.SPARTACUS_CART}/saved-cart/root`;
exports.SPARTACUS_PRODUCT = '@spartacus/product';
exports.SPARTACUS_PRODUCT_CONFIGURATOR = '@spartacus/product-configurator';
exports.SPARTACUS_PRODUCT_CONFIGURATOR_COMMON = `${exports.SPARTACUS_PRODUCT_CONFIGURATOR}/common`;
exports.SPARTACUS_PRODUCT_CONFIGURATOR_RULEBASED = `${exports.SPARTACUS_PRODUCT_CONFIGURATOR}/rulebased`;
exports.SPARTACUS_QUALTRICS = '@spartacus/qualtrics';
exports.SPARTACUS_QUALTRICS_COMPONENTS = `${exports.SPARTACUS_QUALTRICS}/components`;
exports.SPARTACUS_SMARTEDIT = '@spartacus/smartedit';
exports.SPARTACUS_STOREFINDER = '@spartacus/storefinder';
exports.SPARTACUS_TRACKING = '@spartacus/tracking';
exports.SPARTACUS_USER = '@spartacus/user';
exports.SPARTACUS_CHECKOUT = '@spartacus/checkout';
exports.SPARTACUS_CHECKOUT_OCC = `${exports.SPARTACUS_CHECKOUT}/occ`;
exports.SPARTACUS_CHECKOUT_CORE = `${exports.SPARTACUS_CHECKOUT}/core`;
exports.SPARTACUS_CHECKOUT_ROOT = `${exports.SPARTACUS_CHECKOUT}/root`;
exports.SPARTACUS_CHECKOUT_COMPONENTS = `${exports.SPARTACUS_CHECKOUT}/components`;
exports.SPARTACUS_USER_PROFILE = `${exports.SPARTACUS_USER}/profile`;
exports.SPARTACUS_USER_PROFILE_OCC = `${exports.SPARTACUS_USER_PROFILE}/occ`;
exports.SPARTACUS_USER_PROFILE_CORE = `${exports.SPARTACUS_USER_PROFILE}/core`;
exports.SPARTACUS_USER_PROFILE_COMPONENTS = `${exports.SPARTACUS_USER_PROFILE}/components`;
exports.SPARTACUS_USER_PROFILE_ROOT = `${exports.SPARTACUS_USER_PROFILE}/root`;
exports.SPARTACUS_USER_ACCOUNT = `${exports.SPARTACUS_USER}/account`;
exports.SPARTACUS_USER_ACCOUNT_OCC = `${exports.SPARTACUS_USER_ACCOUNT}/occ`;
exports.SPARTACUS_USER_ACCOUNT_CORE = `${exports.SPARTACUS_USER_ACCOUNT}/core`;
exports.SPARTACUS_USER_ACCOUNT_COMPONENTS = `${exports.SPARTACUS_USER_ACCOUNT}/components`;
exports.SPARTACUS_ORDER = '@spartacus/order';
exports.NGRX_STORE = '@ngrx/store';
exports.NGRX_EFFECTS = '@ngrx/effects';
exports.NGRX_ROUTER_STORE = '@ngrx/router-store';
exports.NGUNIVERSAL_EXPRESS_ENGINE = '@nguniversal/express-engine';
exports.NG_BOOTSTRAP = '@ng-bootstrap/ng-bootstrap';
exports.SPARTACUS_PRODUCT_VARIANTS_COMPONENTS = '@spartacus/product/variants/components';
exports.SPARTACUS_PRODUCT_VARIANTS_ROOT = '@spartacus/product/variants/root';
/***** Imports end *****/
/***** Classes start *****/
exports.OBSERVABLE_CLASS = 'Observable';
exports.RENDERER_2 = 'Renderer2';
exports.ELEMENT_REF = 'ElementRef';
exports.INJECTOR = 'Injector';
exports.COMPILER = 'Compiler';
exports.STORE = 'Store';
exports.ACTIONS = 'Actions';
exports.ACTION_SUBJECT = 'ActionsSubject';
exports.FORM_BUILDER = 'FormBuilder';
exports.JSONLD_PRODUCT_REVIEW_BUILDER = 'JsonLdProductReviewBuilder';
exports.CHANGE_DETECTOR_REF = 'ChangeDetectorRef';
exports.ACTIVATED_ROUTE = 'ActivatedRoute';
exports.ROUTER = 'Router';
exports.WINDOW_REF = 'WindowRef';
exports.VIEW_CONTAINER_REF = 'ViewContainerRef';
exports.TEMPLATE_REF = 'TemplateRef';
exports.APPLICATION_REF = 'ApplicationRef';
exports.LOCATION = 'Location';
exports.COMPONENT_FACTORY_RESOLVER = 'ComponentFactoryResolver';
exports.HTTP_CLIENT = 'HttpClient';
exports.NGB_MODAL = 'NgbModal';
exports.BASE_STOREFRONT_MODULE = 'BaseStorefrontModule';
exports.STOREFRONT_MODULE = 'StorefrontModule';
exports.B2C_STOREFRONT_MODULE = 'B2cStorefrontModule';
exports.B2B_STOREFRONT_MODULE = 'B2bStorefrontModule';
exports.MAIN_MODULE = 'MainModule';
exports.STOREFRONT_FOUNDATION_MODULE = 'StorefrontFoundationModule';
exports.VIEW_CONFIG_MODULE = 'ViewConfigModule';
exports.OCC_MODULE = 'OccModule';
exports.EVENTS_MODULE = 'EventsModule';
exports.CONFIG_MODULE_CLASS = 'ConfigModule';
exports.CMS_COMPONENT_DATA_CLASS = 'CmsComponentData';
exports.PROVIDE_CONFIG_FUNCTION = 'provideConfig';
exports.SITE_CONTEXT_CONFIG = 'SiteContextConfig';
exports.I18N_CONFIG = 'I18nConfig';
exports.OCC_CONFIG = 'OccConfig';
exports.FEATURES_CONFIG = 'FeaturesConfig';
exports.PROVIDE_DEFAULT_CONFIG = 'provideDefaultConfig';
exports.DEFAULT_B2B_OCC_CONFIG = 'defaultB2bOccConfig';
exports.CMS_CONFIG = 'CmsConfig';
exports.CMS_SELECTORS = 'CmsSelectors';
exports.CMS_ACTIONS = 'CmsActions';
exports.STOCK_NOTIFICATION_COMPONENT = 'StockNotificationComponent';
exports.LOAD_CMS_COMPONENT_CLASS = 'LoadCmsComponent';
exports.LOAD_CMS_COMPONENT_FAIL_CLASS = 'LoadCmsComponentFail';
exports.LOAD_CMS_COMPONENT_SUCCESS_CLASS = 'LoadCmsComponentSuccess';
exports.CMS_GET_COMPONENT_FROM_PAGE = 'CmsGetComponentFromPage';
exports.USER_ADDRESS_SERVICE = 'UserAddressService';
exports.USER_ADDRESS_CONNECTOR = 'UserAddressConnector';
exports.COMMAND_SERVICE = 'CommandService';
exports.AUTH_SERVICE = 'AuthService';
exports.FEATURE_CONFIG_SERVICE = 'FeatureConfigService';
exports.FEATURE_MODULES_SERVICE = 'FeatureModulesService';
exports.CHECKOUT_EVENT_MODULE = 'CheckoutEventModule';
exports.CHECKOUT_EVENT_BUILDER = 'CheckoutEventBuilder';
exports.CHECKOUT_EVENT_LISTENER = 'CheckoutEventListener';
exports.CMS_FEATURES_SERVICE = 'CmsFeaturesService';
exports.SEMANTIC_PATH_SERVICE = 'SemanticPathService';
exports.CMS_SERVICE = 'CmsService';
exports.PAGE_META_SERVICE = 'PageMetaService';
exports.PAGE_META_CONFIG = 'PageMetaConfig';
exports.STORE_DATA_SERVICE = 'StoreDataService';
exports.SPLIT_VIEW_SERVICE = 'SplitViewService';
exports.SAVED_CART_LIST_COMPONENT = 'SavedCartListComponent';
exports.SAVED_CART_FORM_DIALOG_COMPONENT = 'SavedCartFormDialogComponent';
exports.CONFIGURATOR_CART_SERVICE = 'ConfiguratorCartService';
exports.PAGE_META_RESOLVER = 'PageMetaResolver';
exports.BASE_PAGE_META_RESOLVER = 'BasePageMetaResolver';
exports.PAGE_LINK_SERVICE = 'PageLinkService';
exports.PRODUCT_PAGE_META_RESOLVER = 'ProductPageMetaResolver';
exports.SEARCH_PAGE_META_RESOLVER = 'SearchPageMetaResolver';
exports.CATEGORY_PAGE_META_RESOLVER = 'CategoryPageMetaResolver';
exports.CART_PAGE_META_RESOLVER = `CartPageMetaResolver`;
exports.CHECKOUT_PAGE_META_RESOLVER = 'CheckoutPageMetaResolver';
exports.ORGANIZATION_PAGE_META_RESOLVER = 'OrganizationPageMetaResolver';
exports.PRODUCT_SEARCH_SERVICE = 'ProductSearchService';
exports.CHECKOUT_SERVICE = 'CheckoutService';
exports.CLEAR_CHECKOUT_FACADE = 'ClearCheckoutFacade';
exports.CLEAR_CHECKOUT_SERVICE = 'ClearCheckoutService';
exports.CHECKOUT_FACADE = 'CheckoutFacade';
exports.CHECKOUT_CONNECTOR = 'CheckoutConnector';
exports.USER_ORDER_CONNECTOR = 'UserOrderConnector';
exports.REPLENISHMENT_ORDER_CONNECTOR = 'ReplenishmentOrderConnector';
exports.USER_REPLENISHMENT_ORDER_SERVICE = 'UserReplenishmentOrderService';
exports.CHECKOUT_ADAPTER = 'CheckoutAdapter';
exports.OCC_CHECKOUT_ADAPTER = 'OccCheckoutAdapter';
exports.CHECKOUT_PAYMENT_SERVICE = 'CheckoutPaymentService';
exports.CHECKOUT_DELIVERY_SERVICE = 'CheckoutDeliveryService';
exports.CART_DATA_SERVICE = 'CartDataService';
exports.CART_SERVICE = 'CartService';
exports.ACTIVE_CART_SERVICE = 'ActiveCartService';
exports.USER_ORDER_SERVICE = 'UserOrderService';
exports.USER_ORDERS_EFFECT = 'UserOrdersEffect';
exports.PROMOTION_SERVICE = 'PromotionService';
exports.ORDER_DETAILS_SERVICE = 'OrderDetailsService';
exports.ORDER_RETURN_REQUEST_SERVICE = 'OrderReturnRequestService';
exports.CHECKOUT_LOGIN_COMPONENT = 'CheckoutLoginComponent';
exports.ORDER_CONFIRMATION_OVERVIEW_COMPONENT = 'OrderConfirmationOverviewComponent';
exports.ORDER_HISTORY_COMPONENT = 'OrderHistoryComponent';
exports.ORDER_OVERVIEW_COMPONENT = 'OrderOverviewComponent';
exports.AUTH_REDIRECT_SERVICE = 'AuthRedirectService';
exports.CHECKOUT_DETAILS_SERVICE = 'CheckoutDetailsService';
exports.NOT_CHECKOUT_AUTH_GUARD = 'NotCheckoutAuthGuard';
exports.ROUTING_SERVICE = 'RoutingService';
exports.SHIPPING_ADDRESS_COMPONENT = 'ShippingAddressComponent';
exports.CHECKOUT_CONFIG_SERVICE = 'CheckoutConfigService';
exports.TRANSLATION_SERVICE = 'TranslationService';
exports.ADD_TO_CART_COMPONENT = 'AddToCartComponent';
exports.ADDED_TO_CART_DIALOG_COMPONENT = 'AddedToCartDialogComponent';
exports.MODAL_SERVICE = 'ModalService';
exports.CURRENT_PRODUCT_SERVICE = 'CurrentProductService';
exports.CART_NOT_EMPTY_GUARD = 'CartNotEmptyGuard';
exports.CONTENT_PAGE_META_RESOLVER = 'ContentPageMetaResolver';
exports.ROUTING_PAGE_META_RESOLVER = 'RoutingPageMetaResolver';
exports.CART_TOTALS_COMPONENT = 'CartTotalsComponent';
exports.MINI_CART_COMPONENT = 'MiniCartComponent';
exports.CHECKOUT_REPLENISHMENT_FORM_SERVICE = 'CheckoutReplenishmentFormService';
exports.CHECKOUT_ORDER_SUMMARY_COMPONENT = 'CheckoutOrderSummaryComponent';
exports.CHECKOUT_PROGRESS_MOBILE_TOP_COMPONENT = 'CheckoutProgressMobileTopComponent';
exports.CHECKOUT_CONFIG = 'CheckoutConfig';
exports.SEO_CONFIG = 'SeoConfig';
exports.URL_PARSING_SERVICE = 'UrlParsingService';
exports.ROUTING_CONFIG = 'RoutingConfig';
exports.ROUTING_CONFIG_SERVICE = 'RoutingConfigService';
exports.PAYMENT_METHOD_COMPONENT = 'PaymentMethodComponent';
exports.USER_PAYMENT_SERVICE = 'UserPaymentService';
exports.GLOBAL_MESSAGE_SERVICE = 'GlobalMessageService';
exports.CHECKOUT_AUTH_GUARD = 'CheckoutAuthGuard';
exports.CART_PAGE_LAYOUT_HANDLER = 'CartPageLayoutHandler';
exports.SELECTIVE_CART_SERVICE = 'SelectiveCartService';
exports.CDS_SPARTACUS_EVENT_SERVICE = 'SpartacusEventService';
exports.CONSENT_SERVICE = 'ConsentService';
exports.EVENT_SERVICE = 'EventService';
exports.CDS_CONFIG = 'CdsConfig';
exports.SMART_EDIT_CONFIG = 'SmartEditConfig';
exports.ADDRESS_BOOK_COMPONENT_SERVICE = 'AddressBookComponentService';
exports.CHECKOUT_GUARD = 'CheckoutGuard';
exports.LOGOUT_GUARD = 'LogoutGuard';
exports.EXPRESS_CHECKOUT_SERVICE = 'ExpressCheckoutService';
exports.CONSENT_MANAGEMENT_FORM_COMPONENT = 'ConsentManagementFormComponent';
exports.CLIENT_AUTHENTICATION_TOKEN_SERVICE = 'ClientAuthenticationTokenService';
exports.USER_AUTHENTICATION_TOKEN_SERVICE = 'UserAuthenticationTokenService';
exports.OCC_ENDPOINTS_SERVICE = 'OccEndpointsService';
exports.AUTH_CONFIG = 'AuthConfig';
exports.OCC_CART_ENTRY_ADAPTER = 'OccCartEntryAdapter';
exports.OCC_CART_ADAPTER = 'OccCartAdapter';
exports.OCC_USER_ORDER_ADAPTER = 'OccUserOrderAdapter';
exports.OCC_USER_ADAPTER = 'OccUserAdapter';
exports.OCC_USER_ACCOUNT_ADAPTER = 'OccUserAccountAdapter';
exports.OCC_USER_PROFILE_ADAPTER = 'OccUserProfileAdapter';
exports.CONVERTER_SERVICE = 'ConverterService';
exports.CONSENT_MANAGEMENT_COMPONENT = 'ConsentManagementComponent';
exports.ANONYMOUS_CONSENT_DIALOG_COMPONENT = 'AnonymousConsentDialogComponent';
exports.PRODUCT_IMAGES_COMPONENT = 'ProductImagesComponent';
exports.USER_CONSENT_SERVICE = 'UserConsentService';
exports.USER_SERVICE = 'UserService';
exports.CART_DETAILS_COMPONENT = 'CartDetailsComponent';
exports.ORDER_DETAIL_ITEMS_COMPONENT = 'OrderDetailItemsComponent';
exports.ORDER_CONFIRMATION_ITEMS_COMPONENT = 'OrderConfirmationItemsComponent';
exports.REVIEW_SUBMIT_COMPONENT = 'ReviewSubmitComponent';
exports.CART_VOUCHER_SERVICE = 'CartVoucherService';
exports.CART_COUPON_COMPONENT = 'CartCouponComponent';
exports.CUSTOMER_COUPON_SERVICE = 'CustomerCouponService';
exports.CART_ITEM_LIST_COMPONENT = 'CartItemListComponent';
exports.CART_ITEM_COMPONENT = 'CartItemComponent';
exports.PROTECTED_ROUTES_SERVICE = 'ProtectedRoutesService';
exports.PROTECTED_ROUTES_GUARD = 'ProtectedRoutesGuard';
exports.LOGIN_FORM_COMPONENT = 'LoginFormComponent';
exports.REGISTER_COMPONENT = 'RegisterComponent';
exports.ANONYMOUS_CONSENTS_SERVICE = 'AnonymousConsentsService';
exports.ANONYMOUS_CONSENTS_CONFIG = 'AnonymousConsentsConfig';
exports.STAR_RATING_COMPONENT = 'StarRatingComponent';
exports.PRODUCT_SERVICE = 'ProductService';
exports.PRODUCT_LOADING_SERVICE = 'ProductLoadingService';
exports.PRODUCT_CAROUSEL_COMPONENT = 'ProductCarouselComponent';
exports.PRODUCT_SCROLL_COMPONENT = 'ProductScrollComponent';
exports.PRODUCT_LIST_COMPONENT = 'ProductListComponent';
exports.PRODUCT_LIST_COMPONENT_SERVICE = 'ProductListComponentService';
exports.PRODUCT_LIST_ITEM_COMPONENT = 'ProductListItemComponent';
exports.PRODUCT_LIST_ITEM_CONTEXT_SOURCE = 'ProductListItemContextSource';
exports.PRODUCT_GRID_ITEM_COMPONENT = 'ProductGridItemComponent';
exports.CART_ITEM_CONTEXT_SOURCE = 'CartItemContextSource';
exports.CART_ITEM_CONTEXT = 'CartItemContext';
exports.PAGE_LAYOUT_SERVICE = 'PageLayoutService';
exports.VIEW_CONFIG = 'ViewConfig';
exports.FOOTER_NAVIGATION_COMPONENT = 'FooterNavigationComponent';
exports.NAVIGATION_SERVICE = 'NavigationService';
exports.URL_MATCHER_SERVICE = 'UrlMatcherService';
exports.CONFIGURABLE_ROUTES_SERVICE = 'ConfigurableRoutesService';
exports.EXTERNAL_ROUTES_SERVICE = 'ExternalRoutesService';
exports.EXTERNAL_ROUTES_CONFIG = 'ExternalRoutesConfig';
exports.PAGE_SLOT_COMPONENT = 'PageSlotComponent';
exports.PAGE_SLOT_SERVICE = 'PageSlotService';
exports.DYNAMIC_ATTRIBUTE_SERVICE = 'DynamicAttributeService';
exports.OUTLET_DIRECTIVE = 'OutletDirective';
exports.OUTLET_SERVICE = 'OutletService';
exports.DEFER_LOADER_SERVICE = 'DeferLoaderService';
exports.LAUNCH_COMPONENT_SERVICE = 'LaunchComponentService';
exports.LAUNCH_DIALOG_SERVICE = 'LaunchDialogService';
exports.PLACE_ORDER_COMPONENT = 'PlaceOrderComponent';
exports.USER_INTERESTS_SERVICE = 'UserInterestsService';
exports.USER_NOTIFICATION_PREFERENCE_SERVICE = 'UserNotificationPreferenceService';
exports.PRODUCT_REVIEWS_COMPONENT = `ProductReviewsComponent`;
exports.SEARCH_BOX_COMPONENT = `SearchBoxComponent`;
exports.ADDRESS_BOOK_COMPONENT = `AddressBookComponent`;
exports.ADDRESS_FORM_COMPONENT = `AddressFormComponent`;
exports.PAYMENT_FORM_COMPONENT = 'PaymentFormComponent';
exports.AMEND_ORDER_ACTIONS_COMPONENT = 'AmendOrderActionsComponent';
exports.COMPONENT_WRAPPER_DIRECTIVE = 'ComponentWrapperDirective';
exports.POPOVER_COMPONENT = 'PopoverComponent';
exports.POPOVER_DIRECTIVE = 'PopoverDirective';
exports.POPOVER_SERVICE = 'PopoverService';
exports.POSITIONING_SERVICE = 'PositioningService';
exports.CMS_MAPPING_SERVICE = 'CmsMappingService';
exports.COMPONENT_HANDLER_SERVICE = 'ComponentHandlerService';
exports.CMS_INJECTOR_SERVICE = 'CmsInjectorService';
exports.SMART_EDIT_SERVICE = 'SmartEditService';
exports.SMART_EDIT_MODULE = 'SmartEditModule';
exports.STORE_FINDER_SEARCH_RESULT_COMPONENT = 'StoreFinderSearchResultComponent';
exports.STORE_FINDER_SERVICE = 'StoreFinderService';
exports.STORE_FINDER_CONFIG = 'StoreFinderConfig';
exports.PRODUCT_REVIEW_SERVICE = 'ProductReviewService';
exports.SEARCH_BOX_COMPONENT_SERVICE = 'SearchBoxComponentService';
exports.TAB_PARAGRAPH_CONTAINER_COMPONENT = 'TabParagraphContainerComponent';
exports.NAVIGATION_UI_COMPONENT = 'NavigationUIComponent';
exports.STORE_FINDER_LIST_ITEM_COMPONENT = 'StoreFinderListItemComponent';
exports.CART_CONFIG_SERVICE = 'CartConfigService';
exports.MULTI_CART_SERVICE = 'MultiCartService';
exports.BASE_SITE_SERVICE = 'BaseSiteService';
exports.QUALTRICS_LOADER_SERVICE = 'QualtricsLoaderService';
exports.RENDERER_FACTORY_2 = 'RendererFactory2';
exports.QUALTRICS_CONFIG = 'QualtricsConfig';
exports.QUALTRICS_COMPONENT = 'QualtricsComponent';
exports.QUALTRICS_EVENT_NAME = 'QUALTRICS_EVENT_NAME';
exports.QUALTRICS_MODULE = 'QualtricsModule';
exports.PRODUCT_FACET_NAVIGATION_COMPONENT = 'ProductFacetNavigationComponent';
exports.BREAKPOINT_SERVICE = 'BreakpointService';
exports.MEDIA_SERVICE = 'MediaService';
exports.STORE_FINDER_GRID_COMPONENT = 'StoreFinderGridComponent';
exports.OUTLET_RENDERER_SERVICE = 'OutletRendererService';
exports.SKIP_LINK_SERVICE = 'SkipLinkService';
exports.SKIP_LINK_CONFIG = 'SkipLinkConfig';
exports.KEYBOARD_FOCUS_SERVICE = 'KeyboardFocusService';
exports.STOREFRONT_COMPONENT = 'StorefrontComponent';
exports.HAMBURGER_MENU_SERVICE = 'HamburgerMenuService';
exports.ROUTING_PARAMS_SERVICE = 'RoutingParamsService';
exports.CONFIG_INITIALIZER_SERVICE = 'ConfigInitializerService';
exports.CONFIGURATION_SERVICE = 'ConfigurationService';
exports.LAZY_MODULES_SERVICE = 'LazyModulesService';
exports.UNIFIED_INJECTOR = 'UnifiedInjector';
exports.ANONYMOUS_CONSENT_MANAGEMENT_BANNER_COMPONENT = 'AnonymousConsentManagementBannerComponent';
exports.ANONYMOUS_CONSENT_LAUNCH_DIALOG_SERVICE = 'AnonymousConsentLaunchDialogService';
exports.ANONYMOUS_CONSENT_OPEN_DIALOG_COMPONENT = 'AnonymousConsentOpenDialogComponent';
exports.MERCHANDISING_CAROUSEL_COMPONENT = 'MerchandisingCarouselComponent';
exports.MERCHANDISING_CAROUSEL_COMPONENT_SERVICE = 'MerchandisingCarouselComponentService';
exports.INTERSECTION_SERVICE = 'IntersectionService';
exports.WISH_LIST_SERVICE = 'WishListService';
exports.CHECKOUT_PROGRESS_MOBILE_BOTTOM_COMPONENT = 'CheckoutProgressMobileBottomComponent';
exports.CHECKOUT_PROGRESS_COMPONENT = 'CheckoutProgressComponent';
exports.DELIVERY_MODE_COMPONENT = 'DeliveryModeComponent';
exports.SHIPPING_ADDRESS_SET_GUARD = 'ShippingAddressSetGuard';
exports.DELIVERY_MODE_SET_GUARD = 'DeliveryModeSetGuard';
exports.PAYMENT_DETAILS_SET_GUARD = 'PaymentDetailsSetGuard';
exports.ORDER_DETAIL_SHIPPING_COMPONENT = 'OrderDetailShippingComponent';
exports.ITEM_COUNTER_COMPONENT = 'ItemCounterComponent';
exports.UPDATE_EMAIL_COMPONENT = 'UpdateEmailComponent';
exports.UPDATE_EMAIL_COMPONENT_SERVICE = 'UpdateEmailComponentService';
exports.USER_EMAIL_FACADE = 'UserEmailFacade';
exports.PRODUCT_CAROUSEL_SERVICE = 'ProductCarouselService';
exports.PRODUCT_REFERENCE_SERVICE = 'ProductReferenceService';
exports.ORDER_CANCELLATION_GUARD = 'OrderCancellationGuard';
exports.ORDER_CANCELLATION_SERVICE = 'OrderCancellationService';
exports.ORDER_RETURN_GUARD = 'OrderReturnGuard';
exports.ORDER_RETURN_SERVICE = 'OrderReturnService';
exports.PRODUCT_VARIANT_GUARD = 'ProductVariantGuard';
exports.USER_CONNECTOR = 'UserConnector';
exports.USER_REGISTER_EFFECT = 'UserRegisterEffects';
exports.CONFIGURATOR_ISSUES_NOTIFICATION_COMPONENT = 'ConfiguratorIssuesNotificationComponent';
exports.USER_PROFILE_FACADE = 'UserProfileFacade';
exports.USER_REGISTER_FACADE = 'UserRegisterFacade';
exports.UNIT_ADDRESS_FORM_SERVICE = 'UnitAddressFormService';
exports.GUEST_REGISTER_FORM_COMPONENT = 'GuestRegisterFormComponent';
exports.EPD_VISUALIZATION_CONFIG = 'EpdVisualizationConfig';
exports.TEST_OUTLET_MODULE = 'TestOutletModule';
exports.TEST_CONFIG_MODULE = 'TestConfigModule';
exports.NOT_AUTH_GUARD = 'NotAuthGuard';
exports.AUTH_GUARD = 'AuthGuard';
exports.AUTH_REDIRECT_STORAGE_SERVICE = 'AuthRedirectStorageService';
exports.USER_ID_SERVICE = 'UserIdService';
exports.OAUTH_LIB_WRAPPER_SERVICE = 'OAuthLibWrapperService';
exports.AUTH_STORAGE_SERVICE = 'AuthStorageService';
exports.CDC_AUTH_SERVICE = 'CdcAuthService';
exports.CDC_LOGOUT_GUARD = 'CdcLogoutGuard';
exports.COLLAPSE_MENU_IF_CLICK_OUTSIDE = 'collapseMenuIfClickOutside';
exports.FORBIDDEN_HANDLER = 'ForbiddenHandler';
exports.MULTI_CART_STATE_PERSISTENCE_SERVICE = 'MultiCartStatePersistenceService';
exports.LANGUAGE_SERVICE = 'LanguageService';
exports.CURRENCY_SERVICE = 'CurrencyService';
exports.SEARCH_BOX_SERVICE = 'SearchboxService';
exports.FORGOT_PASSWORD_COMPONENT = 'ForgotPasswordComponent';
exports.AUTH_CONFIG_SERVICE = 'AuthConfigService';
exports.CLOSE_ACCOUNT_MODAL_COMPONENT = 'CloseAccountModalComponent';
exports.OUTLET_REF_DIRECTIVE = 'OutletRefDirective';
exports.JSON_LD_SCRIPT_FACTORY = 'JsonLdScriptFactory';
exports.DOM_SANITIZER = 'DomSanitizer';
exports.ANONYMOUS_CONSENT_TEMPLATES_ADAPTER = 'AnonymousConsentTemplatesAdapter';
exports.ANONYMOUS_CONSENT_TEMPLATES_CONNECTOR = 'AnonymousConsentTemplatesConnector';
exports.VIEW_COMPONENT = 'ViewComponent';
exports.SPLIT_VIEW_COMPONENT = 'SplitViewComponent';
exports.OCC_CMS_COMPONENT_ADAPTER = 'OccCmsComponentAdapter';
exports.MESSAGE_SERVICE = 'MessageService';
exports.CURRENT_UNIT_SERVICE = 'CurrentUnitService';
exports.CURRENT_USER_GROUP_SERVICE = 'CurrentUserGroupService';
exports.UNIT_CHILDREN_COMPONENT = 'UnitChildrenComponent';
exports.UNIT_COST_CENTER_LIST_COMPONENT = 'UnitCostCenterListComponent';
exports.UNIT_USER_LIST_COMPONENT = 'UnitUserListComponent';
exports.UNIT_FORM_COMPONENT = 'UnitFormComponent';
exports.USER_GROUP_USER_LIST_COMPONENT = 'UserGroupUserListComponent';
exports.USER_GROUP_USER_LIST_SERVICE = 'UserGroupUserListService';
exports.TOGGLE_STATUS_COMPONENT = 'ToggleStatusComponent';
exports.DELETE_ITEM_COMPONENT = 'DeleteItemComponent';
exports.ITEM_SERVICE = 'ItemService';
exports.DISABLE_INFO_SERVICE = 'DisableInfoService';
exports.SAVED_CART_FORM_LAUNCH_DIALOG_SERVICE = 'SavedCartFormLaunchDialogService';
exports.ADD_TO_SAVED_CART_COMPONENT = 'AddToSavedCartComponent ';
exports.SAVED_CART_DETAILS_ACTION_COMPONENT = 'SavedCartDetailsActionComponent ';
exports.SAVED_CART_DETAILS_OVERVIEW_COMPONENT = 'SavedCartDetailsOverviewComponent ';
exports.SAVED_CART_DETAILS_SERVICE = 'SavedCartDetailsService';
exports.SAVED_CART_FACADE = 'SavedCartFacade';
exports.REPLENISHMENT_ORDER_CANCELLATION_LAUNCH_DIALOG_SERVICE = 'ReplenishmentOrderCancellationLaunchDialogService';
exports.REPLENISHMENT_ORDER_CANCELLATION_COMPONENT = 'ReplenishmentOrderCancellationComponent';
exports.REPLENISHMENT_ORDER_HISTORY_COMPONENT = 'ReplenishmentOrderHistoryComponent';
exports.COMMON_CONFIGURATOR_UTILS_SERVICE = 'CommonConfiguratorUtilsService';
exports.CONFIGURATOR_UTILS_SERVICE = 'ConfiguratorUtilsService';
exports.CONFIGURATOR_ATTRIBUTE_QUANTITY_SERVICE = 'ConfiguratorAttributeQuantityService';
exports.CONFIGURATOR_UI_SETTINGS_CONFIG = 'ConfiguratorUISettingsConfig';
exports.CONFIGURATOR_STOREFRONT_UTILS_SERVICE = 'ConfiguratorStorefrontUtilsService';
exports.CONFIGURATOR_GROUPS_SERVICE = 'ConfiguratorGroupsService';
exports.CONFIGURATOR_COMMONS_SERVICE = 'ConfiguratorCommonsService';
exports.CONFIGURATOR_ROUTER_EXTRACTOR_SERVICE = 'ConfiguratorRouterExtractorService';
exports.CONFIGURATOR_CART_ENTRY_INFO_COMPONENT = 'ConfiguratorCartEntryInfoComponent';
exports.CONFIGURATOR_UPDATE_MESSAGE_COMPONENT = 'ConfiguratorUpdateMessageComponent';
exports.CONFIGURATOR_ATTRIBUTE_SINGLE_SELECTION_BASE_COMPONENT = 'ConfiguratorAttributeSingleSelectionBaseComponent';
exports.CONFIGURATOR_ATTRIBUTE_NUMERIC_INPUT_FIELD_COMPONENT = 'ConfiguratorAttributeNumericInputFieldComponent';
exports.CONFIGURATOR_ATTRIBUTE_NUMERIC_INPUT_FIELD_SERVICE = 'ConfiguratorAttributeNumericInputFieldService';
exports.CONFIGURATOR_ATTRIBUTE_CHECKBOX_LIST_COMPONENT = 'ConfiguratorAttributeCheckBoxListComponent';
exports.CONFIGURATOR_ATTRIBUTE_DROP_DOWN_COMPONENT = 'ConfiguratorAttributeDropDownComponent';
exports.CONFIGURATOR_ATTRIBUTE_INPUT_FIELD_COMPONENT = 'ConfiguratorAttributeInputFieldComponent';
exports.CONFIGURATOR_ATTRIBUTE_RADIO_BUTTON_COMPONENT = 'ConfiguratorAttributeRadioButtonComponent';
exports.CONFIGURATOR_PRODUCT_TITLE_COMPONENT = 'ConfiguratorProductTitleComponent';
exports.CONFIGURATOR_GROUP_MENU_COMPONENT = 'ConfiguratorGroupMenuComponent';
exports.CONFIGURATOR_GROUP_MENU_SERVICE = 'ConfiguratorGroupMenuService';
exports.MESSAGE_CONFIG = 'MessageConfig';
exports.CONFIGURATOR_MESSAGE_CONFIG = 'ConfiguratorMessageConfig';
exports.CONFIGURATOR_FORM_COMPONENT = 'ConfiguratorFormComponent';
exports.CONFIGURATOR_OVERVIEW_ATTRIBUTE_COMPONENT = 'ConfiguratorOverviewAttributeComponent';
exports.DIRECTION_SERVICE = 'DirectionService';
exports.CART_PAGE_EVENT_BUILDER = 'CartPageEventBuilder';
exports.HOME_PAGE_EVENT_BUILDER = 'HomePageEventBuilder';
exports.PRODUCT_PAGE_EVENT_BUILDER = 'ProductPageEventBuilder';
exports.PAGE_EVENT_BUILDER = 'PageEventBuilder';
exports.PAGE_EVENT_MODULE = 'PageEventModule';
exports.PAGE_EVENT = 'PageEvent';
exports.NAVIGATION_EVENT = 'NavigationEvent';
exports.CONTEXT = 'context';
exports.SEMANTIC_ROUTE = 'semanticRoute';
exports.URL = 'url';
exports.PARAMS = 'params';
exports.SCRIPT_LOADER_SERVICE = 'ScriptLoader';
exports.INITIALIZE = 'initialize';
exports.FORM_GROUP = 'formGroup';
exports.FORM$ = 'form$';
exports.EXTERNAL_JS_FILE_LOADER = 'ExternalJsFileLoader';
exports.SCRIPT_LOADER = 'ScriptLoader';
exports.SAVED_CART_FORM_TYPE = 'savedCartFormType';
exports.OPEN_DIALOG = 'openDialog';
exports.RESTORE_SAVED_CART = 'restoreSavedCart';
exports.ON_RESTORE_COMPLETE = 'onRestoreComplete';
exports.CHECKOUT_DELIVERY_FACADE = 'CheckoutDeliveryFacade';
exports.CHECKOUT_PAYMENT_FACADE = 'CheckoutPaymentFacade';
exports.ON_NAVIGATE_FOCUS_SERVICE = 'OnNavigateFocusService';
exports.KEYBOARD_FOCUS_CONFIG = 'KeyboardFocusConfig';
/***** Classes end *****/
/***** Removed public api start *****/
exports.CART_EFFECTS = 'CartEffects';
exports.WISHLIST_EFFECTS = 'WishlistEffects';
exports.CART_VOUCHER_EFFECTS = 'CartVoucherEffects';
exports.CART_ENTRY_EFFECTS = 'CartEntryEffects';
exports.CART_COMBINED_EFFECTS = 'effects';
exports.CART_SELECTORS = 'CartSelectors';
exports.CART_DATA_CONST = 'CART_DATA';
exports.STATE_WITH_CART = 'StateWithCart';
exports.CARTS_STATE = 'CartsState';
exports.CART_FEATURE_CONST = 'CART_FEATURE';
exports.CART_STATE = 'CartState';
exports.GET_REDUCERS = 'getReducers';
exports.REDUCER_TOKEN = 'reducerToken';
exports.REDUCER_PROVIDER = 'reducerProvider';
exports.CLEAR_CART_STATE = 'clearCartState';
exports.META_REDUCERS = 'metaReducers';
exports.CLEAR_MULTI_CART_STATE = 'clearMultiCartState';
exports.MULTI_CART_META_REDUCERS = 'multiCartMetaReducers';
exports.MULTI_CART_REDUCER_TOKEN = 'multiCartReducerToken';
exports.GET_MULTI_CART_REDUCERS = 'getMultiCartReducers';
exports.MULTI_CART_REDUCER_PROVIDER = 'multiCartReducerProvider';
exports.SITE_CONTEXT_PARAMS_PROVIDERS = 'siteContextParamsProviders';
exports.INIT_SITE_CONTEXT_ROUTES_HANDLER = 'initSiteContextRoutesHandler';
exports.INITITIALIZE_CONTEXT = 'inititializeContext';
exports.CONTEXT_SERVICE_PROVIDERS = 'contextServiceProviders';
exports.URL_MATCHER_FACTORY_SERVICE = 'UrlMatcherFactoryService';
exports.CMS_COMPONENTS_SERVICE = 'CmsComponentsService';
exports.ANONYMOUS_USERID_CONST = 'ANONYMOUS_USERID';
exports.PWA_CONFIGURATION_FACTORY = 'pwaConfigurationFactory';
exports.PWA_FACTORY = 'pwaFactory';
exports.GET_STRUCTURED_DATA_FACTORY = 'getStructuredDataFactory';
exports.SKIP_LINK_FACTORY = 'skipLinkFactory';
exports.FORM_UTILS = 'FormUtils';
exports.BILLING_ADDRESS_FORM_COMPONENT = 'BillingAddressFormComponent';
exports.BILLING_ADDRESS_FORM_MODULE = 'BillingAddressFormModule';
exports.LOADER_STATE = 'LoaderState';
exports.PROCESSES_LOADER_STATE = 'ProcessesLoaderState';
exports.ENTITY_STATE = 'EntityState';
exports.ENTITY_LOADER_STATE = 'EntityLoaderState';
exports.ENTITY_PROCESSES_LOADER_STATE = 'EntityProcessesLoaderState';
exports.STATE_LOADER_ACTIONS = 'StateLoaderActions';
exports.STATE_PROCESSES_LOADER_ACTIONS = 'StateProcessesLoaderActions';
exports.STATE_ENTITY_ACTIONS = 'StateEntityActions';
exports.STATE_ENTITY_LOADER_ACTIONS = 'StateEntityLoaderActions';
exports.STATE_ENTITY_PROCESSESS_LOADER_ACTIONS = 'StateEntityProcessesLoaderActions';
exports.STATE_LOADER_SELECTORS = 'StateLoaderSelectors';
exports.STATE_PROCESSES_LOADER_SELECTORS = 'StateProcessesLoaderSelectors';
exports.STATE_ENTITY_SELECTORS = 'StateEntitySelectors';
exports.STATE_ENTITY_LOADER_SELECTORS = 'StateEntityLoaderSelectors';
exports.STATE_ENTITY_PROCESSESS_LOADER_SELECTORS = 'StateEntityProcessesLoaderSelectors';
exports.LOADER_REDUCER = 'loaderReducer';
exports.PROCESSES_LOADER_REDUCER = 'processesLoaderReducer';
exports.ENTITY_REDUCER = 'entityReducer';
exports.ENTITY_LOADER_REDUCER = 'entityLoaderReducer';
exports.ENTITY_PROCESSES_LOADER_REDUCER = 'entityProcessesLoaderReducer';
exports.INITIAL_ENTITY_STATE = 'initialEntityState';
exports.INITIAL_LOADER_STATE = 'initialLoaderState';
exports.INITIAL_PROCESSES_STATE = 'initialProcessesState';
exports.GET_STATE_SLICE = 'getStateSlice';
exports.OF_LOADER_LOAD = 'ofLoaderLoad';
exports.OF_LOADER_FAIL = 'ofLoaderFail';
exports.OF_LOADER_SUCCESS = 'ofLoaderSuccess';
exports.ONLY_NUMBER_DIRECTIVE_MODULE = 'OnlyNumberDirectiveModule';
exports.ONLY_NUMBER_DIRECTIVE = 'OnlyNumberDirective';
exports.AUTO_FOCUS_DIRECTIVE = 'AutoFocusDirective';
exports.AUTO_FOCUS_DIRECTIVE_MODULE = 'AutoFocusDirectiveModule';
exports.AUTH_SELECTORS = 'AuthSelectors';
exports.STATE_WITH_AUTH = 'StateWithAuth';
exports.AUTH_STATE = 'AuthState';
exports.USER_TOKEN_STATE = 'UserTokenState';
exports.AUTH_FEATURE = 'AUTH_FEATURE';
exports.AUTH_ACTIONS = 'AuthActions';
exports.USER_TOKEN = 'UserToken';
exports.AUTHENTICATION_TOKEN = 'AuthenticationToken';
exports.KYMA_SELECTORS = 'KymaSelectors';
exports.KYMA_FEATURE = 'KYMA_FEATURE';
exports.OPEN_ID_TOKEN_DATA = 'OPEN_ID_TOKEN_DATA';
exports.STATE_WITH_KYMA = 'StateWithKyma';
exports.KYMA_STATE = 'KymaState';
exports.KYMA_ACTIONS = 'KymaActions';
exports.OPEN_ID_AUTHENTICATION_TOKEN_SERVICE = 'OpenIdAuthenticationTokenService';
exports.OPEN_ID_TOKEN = 'OpenIdToken';
exports.KYMA_MODULE = 'KymaModule';
exports.KYMA_SERVICE = 'KymaService';
exports.KYMA_CONFIG = 'KymaConfig';
exports.ASM_SELECTORS = 'AsmSelectors';
exports.ASM_ACTIONS = 'AsmActions';
exports.CSAGENT_TOKEN_DATA = 'CSAGENT_TOKEN_DATA';
exports.CUSTOMER_SUPPORT_AGENT_TOKEN_INTERCEPTOR = 'CustomerSupportAgentTokenInterceptor ';
exports.ASM_MODULE = 'AsmModule';
exports.ASM_OCC_MODULE = 'AsmOccModule';
exports.OCC_ASM_ADAPTER = 'OccAsmAdapter';
exports.ASM_CONFIG = 'AsmConfig';
exports.ASM_ADAPTER = 'AsmAdapter';
exports.ASM_CONNECTOR = 'AsmConnector';
exports.CUSTOMER_SEARCH_PAGE_NORMALIZER = 'CUSTOMER_SEARCH_PAGE_NORMALIZER';
exports.ASM_SERVICE = 'AsmService';
exports.CS_AGENT_AUTH_SERVICE = 'CsAgentAuthService';
exports.CUSTOMER_SEARCH_PAGE = 'CustomerSearchPage';
exports.CUSTOMER_SEARCH_OPTIONS = 'CustomerSearchOptions';
exports.ASM_UI = 'AsmUi';
exports.AUTH_FLOW_ROUTES_SERVICE = 'AuthFlowRoutesService';
exports.ASM_AUTH_HTTP_HEADER_SERVICE = 'AsmAuthHttpHeaderService';
exports.AUTH_HTTP_HEADER_SERVICE = 'AuthHttpHeaderService';
exports.TOKEN_TARGET = 'TokenTarget';
exports.ASM_AUTH_STORAGE_SERVICE = 'AsmAuthStorageService';
exports.ASM_AUTH_SERVICE = 'AsmAuthService';
exports.SYNCED_ASM_STATE = 'SyncedAsmState';
exports.ASM_STATE_PERSISTENCE_SERVICE = 'AsmStatePersistenceService';
exports.ASM_UI_UPDATE = 'ASM_UI_UPDATE';
exports.ASM_UI_UPDATE_CLASS = 'AsmUiUpdate';
exports.ASM_UI_ACTION = 'AsmUiAction';
exports.CUSTOMER_SEARCH = 'CUSTOMER_SEARCH';
exports.CUSTOMER_SEARCH_CLASS = 'CustomerSearch';
exports.CUSTOMER_SEARCH_FAIL = 'CUSTOMER_SEARCH_FAIL';
exports.CUSTOMER_SEARCH_FAIL_CLASS = 'CustomerSearchFail';
exports.CUSTOMER_SEARCH_SUCCESS = 'CUSTOMER_SEARCH_SUCCESS';
exports.CUSTOMER_SEARCH_SUCCESS_CLASS = 'CustomerSearchSuccess';
exports.CUSTOMER_SEARCH_RESET = 'CUSTOMER_SEARCH_RESET';
exports.CUSTOMER_SEARCH_RESET_CLASS = 'CustomerSearchReset';
exports.CUSTOMER_ACTION = 'CustomerAction';
exports.ASM_FEATURE = 'ASM_FEATURE';
exports.CUSTOMER_SEARCH_DATA = 'CUSTOMER_SEARCH_DATA';
exports.STATE_WITH_ASM = 'StateWithAsm';
exports.ASM_STATE = 'AsmState';
exports.GET_ASM_UI = 'getAsmUi';
exports.GET_CUSTOMER_SEARCH_RESULTS_LOADER_STATE = 'getCustomerSearchResultsLoaderState';
exports.GET_CUSTOMER_SEARCH_RESULTS = 'getCustomerSearchResults';
exports.GET_CUSTOMER_SEARCH_RESULTS_LOADING = 'getCustomerSearchResultsLoading';
exports.GET_ASM_STATE = 'getAsmState';
exports.OCC_STORE_FINDER_ADAPTER = 'OccStoreFinderAdapter';
exports.STORE_FINDER_OCC_MODULE = 'StoreFinderOccModule';
exports.SEARCH_CONFIG = 'SearchConfig';
exports.STORE_FINDER_SEARCH_CONFIG = 'StoreFinderSearchConfig';
exports.STORE_FINDER_ACTIONS = 'StoreFinderActions';
exports.FIND_STORES_CLASS = 'FindStores';
exports.STORE_FINDER_SEARCH_PAGE_NORMALIZER = 'STORE_FINDER_SEARCH_PAGE_NORMALIZER';
exports.STORE_COUNT_NORMALIZER = 'STORE_COUNT_NORMALIZER';
exports.STORE_FINDER_CONNECTOR = 'StoreFinderConnector';
exports.STORE_FINDER_ADAPTER = 'StoreFinderAdapter';
exports.STORE_FINDER_SEARCH_QUERY = 'StoreFinderSearchQuery';
exports.STORE_ENTITIES = 'StoreEntities';
exports.GOOGLE_MAP_RENDERER_SERVICE = 'GoogleMapRendererService';
exports.STORE_FINDER_CORE_MODULE = 'StoreFinderCoreModule';
exports.STORE_FINDER_STORE_MODULE = 'StoreFinderStoreModule';
exports.STORE_FINDER_FEATURE = 'STORE_FINDER_FEATURE';
exports.STORE_FINDER_DATA = 'STORE_FINDER_DATA';
exports.STATE_WITH_STORE_FINDER = 'StateWithStoreFinder';
exports.STORES_STATE = 'StoresState';
exports.FIND_STORES_STATE = 'FindStoresState';
exports.VIEW_ALL_STORES_STATE = 'ViewAllStoresState';
exports.STORE_FINDER_SELECTORS = 'StoreFinderSelectors';
exports.STORE_COUNT = 'StoreCount';
exports.STORE_FINDER_SEARCH_PAGE = 'StoreFinderSearchPage';
exports.ABSTRACT_STORE_ITEM_COMPONENT = 'AbstractStoreItemComponent';
exports.SCHEDULE_COMPONENT = 'ScheduleComponent';
exports.STORE_FINDER_MAP_COMPONENT = 'StoreFinderMapComponent';
exports.STORE_FINDER_LIST_COMPONENT = 'StoreFinderListComponent';
exports.STORE_FINDER_SEARCH_COMPONENT = 'StoreFinderSearchComponent';
exports.STORE_FINDER_STORE_DESCRIPTION_COMPONENT = 'StoreFinderStoreDescriptionComponent';
exports.STORE_FINDER_STORE_COMPONENT = 'StoreFinderStoreComponent';
exports.STORE_FINDER_STORES_COUNT_COMPONENT = 'StoreFinderStoresCountComponent';
exports.STORE_FINDER_MODULE = 'StoreFinderModule';
exports.STORE_FINDER_HEADER_COMPONENT = 'StoreFinderHeaderComponent';
exports.STORE_FINDER_PAGINATION_DETAILS_COMPONENT = 'StoreFinderPaginationDetailsComponent';
exports.STORE_FINDER_COMPONENT = 'StoreFinderComponent';
exports.STORE_FINDER_SCSS_FILE_NAME = 'storefinder.scss';
exports.ON_SUCCESS = 'onSuccess';
exports.CHECKOUT_ACTIONS = 'CheckoutActions';
exports.PLACE_ORDER_CLASS = 'PlaceOrder';
exports.CHECKOUT_STEP_SERVICE = 'CheckoutStepService';
exports.USER_COST_CENTER_SERVICE = 'UserCostCenterService';
exports.CHECKOUT_COST_CENTER_SERVICE = 'CheckoutCostCenterService';
exports.PAYMENT_TYPE_SERVICE = 'PaymentTypeService';
exports.UNAUTHORIZED_ERROR_HANDLER = 'UnauthorizedErrorHandler';
exports.NG_EXPRESS_ENGINE_DECORATOR = 'NgExpressEngineDecorator';
exports.LAYOUT_CONFIG = 'LayoutConfig';
exports.TOKEN_REVOCATION_HEADER = 'TOKEN_REVOCATION_HEADER';
exports.SPLIT_VIEW_DEACTIVATE_GUARD = 'SplitViewDeactivateGuard';
exports.GET_OCC_ENDPOINT = 'getOccEndpoint';
exports.GET_BASE_ENDPOINT = 'getBaseEndpoint';
exports.GET_URL = 'getUrl';
exports.GET_END_POINT = 'getEndpoint';
exports.GET_RAW_ENDPOINT = 'getRawEndpoint';
exports.PRODUCT_VARIANT_COMPONENT = 'ProductVariantsComponent';
exports.VARIANT_COLOR_SELECTOR_COMPONENT = 'VariantColorSelectorComponent';
exports.VARIANT_COLOR_SELECTOR_MODULE = 'VariantColorSelectorModule';
exports.VARIANT_SIZE_SELECTOR_COMPONENT = 'VariantSizeSelectorComponent';
exports.VARIANT_SIZE_SELECTOR_MODULE = 'VariantSizeSelectorModule';
exports.VARIANT_STYLE_ICONS_COMPONENT = 'VariantStyleIconsComponent';
exports.VARIANT_STYLE_ICONS_MODULE = 'VariantStyleIconsModule';
exports.VARIANT_STYLE_SELECTOR_COMPONENT = 'VariantStyleSelectorComponent';
exports.VARIANT_STYLE_SELECTOR_MODULE = 'VariantStyleSelectorModule';
exports.PRODUCT_VARIANTS_MODULE = 'ProductVariantsModule';
exports.PRODUCT_VARIANT_STYLE_ICONS_COMPONENT = 'ProductVariantStyleIconsComponent';
exports.PRODUCT_VARIANT_STYLE_ICONS_MODULE = 'ProductVariantStyleIconsModule';
exports.DEFAULT_STATE_CONFIG = 'defaultStateConfig';
exports.DEFAULT_LOCAL_STORAGE_KEY = 'DEFAULT_LOCAL_STORAGE_KEY';
exports.DEFAULT_SESSION_STORAGE_KEY = 'DEFAULT_SESSION_STORAGE_KEY';
exports.PERSONALIZATION_MODULE = 'PersonalizationModule';
exports.PERSONALIZATION_CONFIG = 'PersonalizationConfig';
exports.PERSONALIZATION_CONTEXT_SERVICE = 'PersonalizationContextService';
exports.PERSONALIZATION_ACTION = 'PersonalizationAction';
exports.PERSONALIZATION_CONTEXT = 'PersonalizationContext';
exports.TRANSLATION_CHUNKS_CONFIG = 'TranslationChunksConfig';
exports.MINI_LOGIN_TRANSLATION_CHUNK = 'miniLogin';
exports.UPDATE_EMAIL_FORM_TRANSLATION_CHUNK = 'updateEmailForm';
exports.FORGOTTEN_PASSWORD_TRANSLATION_CHUNK = 'forgottenPassword';
exports.LOGIN_FORM_TRANSLATION_CHUNK = 'loginForm';
exports.REGISTER_TRANSLATION_CHUNK = 'register';
exports.TITLES_ENDPOINT = 'titles';
exports.USER_ENDPOINT = 'user';
exports.USER_REGISTER_ENDPOINT = 'userRegister';
exports.USER_FORGOT_PASSWORD_ENDPOINT = 'userForgotPassword';
exports.USER_RESET_PASSWORD_ENDPOINT = 'userResetPassword';
exports.USER_UPDATE_LOGIN_ID_ENDPOINT = 'userUpdateLoginId';
exports.USER_UPDATE_PASSWORD_ENDPOINT = 'userUpdatePassword';
exports.USER_UPDATE_PROFILE_ENDPOINT = 'userUpdateProfile';
exports.USER_CLOSE_ACCOUNT_ENDPOINT = 'userCloseAccount';
exports.USER_ACCOUNT_NORMALIZER = 'USER_ACCOUNT_NORMALIZER';
exports.USER_ACCOUNT_SERIALIZER = 'USER_ACCOUNT_SERIALIZER';
exports.USER_PROFILE_NORMALIZER = 'USER_PROFILE_NORMALIZER';
exports.USER_PROFILE_SERIALIZER = 'USER_PROFILE_SERIALIZER';
exports.USER_SERIALIZER = 'USER_SERIALIZER';
exports.USER_NORMALIZER = 'USER_NORMALIZER';
exports.USER_SIGN_UP_SERIALIZER = 'USER_SIGN_UP_SERIALIZER';
exports.TITLE_NORMALIZER = 'TITLE_NORMALIZER';
exports.USER_ADAPTER = 'UserAdapter';
exports.USER_ACCOUNT_ADAPTER = 'UserAccountAdapter';
exports.USER_PROFILE_ADAPTER = 'UserProfileAdapter';
exports.USER_ACCOUNT_CONNECTOR = 'UserAccountConnector';
exports.USER_PROFILE_CONNECTOR = 'UserProfileConnector';
exports.USER_SIGN_UP_INTERFACE = 'UserSignUp';
exports.USER_ACTIONS = 'UserActions';
exports.FORGOT_PASSWORD_EMAIL_ACTION = 'ForgotPasswordEmailRequestAction';
exports.RESET_PASSWORD_ACTION = 'ResetPasswordAction';
exports.EMAIL_ACTIONS = 'EmailActions';
exports.UPDATE_PASSWORD_ACTION = 'UpdatePasswordAction';
exports.USER_DETAILS_ACTION = 'UserDetailsAction';
exports.USER_DETAILS_STATE_INTERFACE = 'UserDetailsState';
exports.USER_STATE_INTERFACE = 'UserState';
exports.USER_MODULE = 'UserModule';
exports.USER_ACCOUNT_MODULE = 'UserAccountModule';
exports.USER_PROFILE_MODULE = 'UserProfileModule';
exports.USER_PROFILE_FACADE_TRANSITIONAL_TOKEN = 'UserProfileFacadeTransitionalToken';
exports.USER_REGISTER_FACADE_TRANSITIONAL_TOKEN = 'UserRegisterFacadeTransitionalToken';
exports.CMS_LIB_MODULE = 'CmsLibModule';
exports.CLOSE_ACCOUNT_MODULE = 'CloseAccountModule';
exports.FORGOT_PASSWORD_MODULE = 'ForgotPasswordModule';
exports.REGISTER_COMPONENT_MODULE = 'RegisterComponentModule';
exports.RESET_PASSWORD_MODULE = 'ResetPasswordModule';
exports.RESET_PASSWORD_COMPONENT = 'ResetPasswordComponent';
exports.RESET_PASSWORD_FORM_COMPONENT = 'ResetPasswordFormComponent';
exports.LOGIN_COMPONENT = 'LoginComponent';
exports.LOGIN_MODULE = 'LoginModule';
exports.LOGIN_FORM_MODULE = 'LoginFormModule';
exports.LOGIN_REGISTER_COMPONENT = 'LoginRegisterComponent';
exports.LOGIN_REGISTER_MODULE = 'LoginRegisterModule';
exports.UPDATE_EMAIL_MODULE = 'UpdateEmailModule';
exports.UPDATE_PASSWORD_MODULE = 'UpdatePasswordModule';
exports.UPDATE_PROFILE_MODULE = 'UpdateProfileModule';
exports.USER_COMPONENT_MODULE = 'UserComponentModule';
exports.CLOSE_ACCOUNT_COMPONENT = 'CloseAccountComponent';
exports.UPDATE_EMAIL_FORM_COMPONENT = 'UpdateEmailFormComponent';
exports.UPDATE_PASSWORD_COMPONENT = 'UpdatePasswordComponent';
exports.UPDATE_PASSWORD_FORM_COMPONENT = 'UpdatePasswordFormComponent';
exports.UPDATE_PROFILE_COMPONENT = 'UpdateProfileComponent';
exports.ROUTING_ACTIONS = 'RoutingActions';
exports.ROUTE_GO_ACTION = 'RouteGoAction';
exports.ROUTE_GO_BY_URL_ACTION = 'RouteGoByUrlAction';
exports.ROUTE_FORWARD_ACTION = 'RouteForwardAction';
exports.ROUTE_BACK_ACTION = 'RouteBackAction';
exports.BUDGET_ROUTING_CONFIG = 'budgetRoutingConfig';
exports.COST_CENTER_ROUTING_CONFIG = 'costCenterRoutingConfig';
exports.PERMISSION_ROUTING_CONFIG = 'permissionRoutingConfig';
exports.UNITS_ROUTING_CONFIG = 'unitsRoutingConfig';
exports.USER_GROUP_ROUTING_CONFIG = 'userGroupRoutingConfig';
exports.USER_ROUTING_CONFIG = 'userRoutingConfig';
exports.DEFAULT_BUDGET_ROUTING_CONFIG = 'defaultBudgetRoutingConfig';
exports.DEFAULT_COST_CENTER_ROUTING_CONFIG = 'defaultCostCenterRoutingConfig';
exports.DEFAULT_PERMISSION_ROUTING_CONFIG = 'defaultPermissionRoutingConfig';
exports.DEFAULT_UNITS_ROUTING_CONFIG = 'defaultUnitsRoutingConfig';
exports.DEFAULT_USER_GROUP_ROUTING_CONFIG = 'defaultUserGroupRoutingConfig';
exports.DEFAULT_USER_ROUTING_CONFIG = 'defaultUserRoutingConfig';
// uncomment product variants deprecation on 4.0 migration works (#11391)
// export const PRODUCT_VARIANT_COMPONENT = 'ProductVariantsComponent';
// export const VARIANT_COLOR_SELECTOR_COMPONENT = 'VariantColorSelectorComponent';
// export const VARIANT_SIZE_SELECTOR_COMPONENT = 'VariantSizeSelectorComponent';
// export const VARIANT_STYLE_ICONS_COMPONENT = 'VariantStyleIconsComponent';
// export const VARIANT_STYLE_SELECTOR_COMPONENT = 'VariantStyleSelectorComponent';
exports.B2C_LAYOUT_CONFIG = 'b2cLayoutConfig';
/***** Removed public api end *****/
/***** Properties start *****/
exports.CMS_COMPONENT_DATA_PROPERTY_NAME = 'componentData';
exports.KYMA_ENABLED = 'kyma_enabled';
exports.ANONYMOUS_CONSENTS = 'anonymousConsents';
exports.LOAD_CLIENT_TOKEN_CLASS = 'LoadClientToken';
exports.LOAD_CLIENT_TOKEN = 'LOAD_CLIENT_TOKEN';
exports.LOAD_CLIENT_TOKEN_SUCCESS_CLASS = 'LoadClientTokenSuccess';
exports.LOAD_CLIENT_TOKEN_SUCCESS = 'LOAD_CLIENT_TOKEN_SUCCESS';
exports.LOAD_CLIENT_TOKEN_FAIL_CLASS = 'LoadClientTokenFail';
exports.LOAD_CLIENT_TOKEN_FAIL = 'LOAD_CLIENT_TOKEN_FAIL';
exports.LOGOUT_CUSTOMER_SUPPORT_AGENT = 'LOGOUT_CUSTOMER_SUPPORT_AGENT';
exports.LOGOUT_CUSTOMER_SUPPORT_AGENT_CLASS = 'LOGOUT_CUSTOMER_SUPPORT_AGENT_CLASS';
exports.LOAD_USER_TOKEN = 'LOAD_USER_TOKEN';
exports.LOAD_USER_TOKEN_FAIL = 'LOAD_USER_TOKEN_FAIL';
exports.LOAD_USER_TOKEN_SUCCESS = 'LOAD_USER_TOKEN_SUCCESS';
exports.REFRESH_USER_TOKEN = 'REFRESH_USER_TOKEN';
exports.REFRESH_USER_TOKEN_FAIL = 'REFRESH_USER_TOKEN_FAIL';
exports.REFRESH_USER_TOKEN_SUCCESS = 'REFRESH_USER_TOKEN_SUCCESS';
exports.REVOKE_USER_TOKEN = 'REVOKE_USER_TOKEN';
exports.REVOKE_USER_TOKEN_FAIL = 'REVOKE_USER_TOKEN_FAIL';
exports.REVOKE_USER_TOKEN_SUCCESS = 'REVOKE_USER_TOKEN_SUCCESS';
exports.LOAD_USER_TOKEN_CLASS = 'LoadUserToken';
exports.LOAD_USER_TOKEN_FAIL_CLASS = 'LoadUserTokenFail';
exports.LOAD_USER_TOKEN_SUCCESS_CLASS = 'LoadUserTokenSuccess';
exports.REFRESH_USER_TOKEN_CLASS = 'RefreshUserToken';
exports.REFRESH_USER_TOKEN_FAIL_CLASS = 'RefreshUserTokenFail';
exports.REFRESH_USER_TOKEN_SUCCESS_CLASS = 'RefreshUserTokenSuccess';
exports.REVOKE_USER_TOKEN_CLASS = 'RevokeUserToken';
exports.REVOKE_USER_TOKEN_FAIL_CLASS = 'RevokeUserTokenFail';
exports.REVOKE_USER_TOKEN_SUCCESS_CLASS = 'RevokeUserTokenSuccess';
exports.CAN_ACTIVATE = 'canActivate';
exports.GET_CUSTOMER_AGENT_TOKEN_STATE = 'getCustomerSupportAgentTokenState';
exports.GET_CUSTOMER_AGENT_TOKEN = 'getCustomerSupportAgentToken';
exports.GET_CUSTOMER_AGENT_TOKEN_LOADING = 'getCustomerSupportAgentTokenLoading';
exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN = 'LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN';
exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_FAIL = 'LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_FAIL';
exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_SUCCESS = 'LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_SUCCESS';
exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_CLASS = 'LoadCustomerSupportAgentToken';
exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_FAIL_CLASS = 'LoadCustomerSupportAgentTokenFail';
exports.LOAD_CUSTOMER_SUPPORT_AGENT_TOKEN_SUCCESS_CLASS = 'LoadCustomerSupportAgentTokenSuccess';
exports.SUB = 'sub';
exports.LOAD_ANONYMOUS_CONSENTS = 'loadAnonymousConsents';
exports.LEGACY_FLAG = 'legacy';
exports.INJECT_DECORATOR = 'Inject';
exports.PLATFORM = 'Platform';
exports.PLATFORM_ID_STRING = 'PLATFORM_ID';
exports.ROOT_CONFIG = 'RootConfig';
exports.STOREFRONT_CONFIG = 'StorefrontConfig';
exports.DEFAULT_CONFIG = 'DefaultConfig';
exports.CONFIG = 'Config';
exports.ANY_TYPE = 'any';
exports.OBJECT_TYPE = 'Object';
exports.STRING_TYPE = 'string';
exports.DOCUMENT = 'Document';
exports.DOCUMENT_STRING = 'DOCUMENT';
exports.SERVER_REQUEST_URL_STRING = 'SERVER_REQUEST_URL';
exports.SERVER_REQUEST_ORIGIN_STRING = 'SERVER_REQUEST_ORIGIN';
exports.INCREMENT = 'increment';
exports.ITEM = 'Item';
exports.ORDER_ENTRY = 'OrderEntry';
exports.ORDER_PROMOTIONS$ = 'orderPromotions$';
exports.INSIDE_CLICKED = 'insideClicked';
exports.HANDLE_OPEN = 'handleOpen';
exports.TOGGLE = 'toggle';
exports.DEFAULT_PAGE_SIZE = 'defaultPageSize';
exports.SORT_CODE = 'sortCode';
exports.TABLE_HEADER = 'TableHeader';
exports.PROMOTION_LOCATION$ = 'promotionLocation$';
exports.LOCATION$ = 'location$';
/***** Properties end *****/
/***** APIs start *****/
exports.GET_COMPONENT_STATE_OLD_API = 'getComponentState';
exports.GET_COMPONENTS_STATE_NEW_API = 'getComponentsState';
exports.GET_COMPONENT_ENTITIES_OLD_API = 'getComponentEntities';
exports.COMPONENT_STATE_SELECTOR_FACTORY_OLD_API = 'componentStateSelectorFactory';
exports.COMPONENTS_STATE_SELECTOR_FACTORY_NEW_API = 'componentsLoaderStateSelectorFactory';
exports.COMPONENT_SELECTOR_FACTORY_OLD_API = 'componentSelectorFactory';
exports.COMPONENTS_SELECTOR_FACTORY_NEW_API = 'componentsSelectorFactory';
exports.IS_LAUNCH_IN_SMART_EDIT = 'isLaunchInSmartEdit';
exports.IS_LAUNCHED_IN_SMART_EDIT = 'isLaunchedInSmartEdit';
exports.ADD_DYNAMIC_ATTRIBUTES = 'addDynamicAttributes';
exports.GET_FALSY_URL_MATCHER = 'getFalsyUrlMatcher';
exports.GET_MULTIPLE_PATHS_URL_MATCHER = 'getMultiplePathsUrlMatcher';
exports.GET_PATH_URL_MATCHER = 'getPathUrlMatcher';
exports.GET_OPPOSITE_URL_MATCHER = 'getOppositeUrlMatcher';
exports.GET_GLOB_URL_MATCHER = 'getGlobUrlMatcher';
exports.PLACE_ORDER = 'placeOrder';
exports.SET_QUERY = 'setQuery';
exports.VIEW_PAGE = 'viewPage';
exports.GET_ORDER_DETAILS = 'getOrderDetails';
exports.GET_ADDRESS_CARD_CONTENT = 'getAddressCardContent';
exports.GET_DELIVERY_MODE_CARD_CONTENT = 'getDeliveryModeCardContent';
exports.GET_PAYMENT_INFO_CARD_CONTENT = 'getPaymentInfoCardContent';
exports.GET_BILLING_ADDRESS_CARD_CONTENT = 'getBillingAddressCardContent';
exports.GET_PAYMENT_CARD_CONTENT = 'getPaymentCardContent';
exports.GET_SHIPPING_METHOD_CARD_CONTENT = 'getShippingMethodCardContent';
exports.GET_ORDER_CURRENT_DATE_CARD_CONTENT = 'getOrderCurrentDateCardContent';
exports.GET_CONTROL = 'getControl';
exports.GET_CHECKOUT_STEP = 'getCheckoutStep';
exports.GET_CHECKOUT_STEP_ROUTE = 'getCheckoutStepRoute';
exports.GET_FIRST_CHECKOUT_STEP_ROUTE = 'getFirstCheckoutStepRoute';
exports.GET_NEXT_CHECKOUT_STEP_URL = 'getNextCheckoutStepUrl';
exports.GET_PREVIOUS_CHECKOUT_STEP_URL = 'getPreviousCheckoutStepUrl';
exports.GET_CURRENT_STEP_INDEX = 'getCurrentStepIndex';
exports.STEPS = 'steps';
exports.GET_CHILD_ROUTES = 'getChildRoutes';
exports.GET_INJECTORS = 'getInjectors';
exports.GET_ALL = 'getAll';
exports.NG_ON_DESTROY = 'ngOnDestroy';
exports.NG_ON_INIT = 'ngOnInit';
exports.LOGIN_AS_GUEST = 'loginAsGuest';
exports.GET_BASE_SITE_DATA = 'getBaseSiteData';
exports.LOAD_PRODUCT_REFERENCES = 'loadProductReferences';
exports.GET_PRODUCT_REFERENCES = 'getProductReferences';
exports.GET = 'get';
exports.GET_WINDOW = 'window';
exports.GET_CLOSEST = 'getClosest';
exports.GET_BREAKPOINT = 'getBreakpoint';
exports.LOGOUT = 'logout';
exports.REDIRECT = 'redirect';
exports.SYNC = 'sync';
exports.INIT_SYNC = 'initSync';
exports.SET_ACTIVE = 'setActive';
exports.SET_RATE_ON_EVENT = 'setRateOnEvent';
exports.SET_RATE = 'setRate';
exports.GET_META = 'getMeta';
exports.GET_META_RESOLVER = 'getMetaResolver';
exports.FIND_COMPONENTS_BY_IDS_LEGACY = 'findComponentsByIdsLegacy';
exports.GET_LOADED = 'getLoaded';
exports.IS_STABLE = 'isStable';
exports.STOREFINDER_MODULE = 'StoreFinderModule';
exports.GO = 'go';
exports.OCC_ENDPOINTS = 'OccEndpoints';
exports.OCC_ENDPOINT = 'OccEndpoint';
exports.BASE_SITES_FOR_CONFIG = 'baseSitesForConfig';
exports.OCC_CONFIG_LOADER_SERVICE = 'OccConfigLoaderService';
exports.OCC_LOADED_CONFIG_CONVERTER = 'OccLoadedConfigConverter';
exports.OCC_LOADED_CONFIG = 'OccLoadedConfig';
exports.OCC_SITES_CONFIG_LOADER = 'OccSitesConfigLoader';
exports.OCC_CONFIG_LOADER_MODULE = 'OccConfigLoaderModule';
exports.CREATE_EVENT_FROM_INPUT = 'createEventFromInput';
exports.ON_SELECT = 'onSelect';
exports.ON_DESELECT = 'onDeselect';
exports.GET_PRODUCT_IMAGE_URL = 'getProductImageURL';
exports.GET_PRODUCT_IMAGE_ALT = 'getProductImageAlt';
exports.CLICK_ON_ENTER = 'clickOnEnter';
exports.NAVIGATE_UP_ON_ENTER = 'navigateUpOnEnter';
exports.RESET_CHECKOUT_PROCESSES = 'resetCheckoutProcesses';
exports.PREVENT_SCROLLING_ON_SPACE = 'preventScrollingOnSpace';
exports.GET_STABLE_CONFIG = 'getStableConfig';
exports.GET_STABLE = 'getStable';
/***** APIs end *****/
/***** feature keys start *****/
exports.PRODUCT_CONFIGURATOR_RULEBASED_FEATURE_OBSOLETE = 'rulebased';
exports.PRODUCT_CONFIGURATOR_TEXTFIELD_FEATURE_OBSOLETE = 'textfield';
exports.PRODUCT_CONFIGURATOR_RULEBASED_FEATURE = 'productConfiguratorRulebased';
exports.PRODUCT_CONFIGURATOR_TEXTFIELD_FEATURE = 'productConfiguratorTextfield';
/***** feature keys end *****/
/***** Storefinder start *****/
exports.TODO_SPARTACUS = 'TODO:Spartacus -';
exports.CSS_V2_DOCS_URL = 'https://sap.github.io/spartacus-docs/css-changes-in-version-2/';
exports.CSS_V3_DOCS_URL = 'https://sap.github.io/spartacus-docs/css-changes-in-version-3/';
exports.CSS_V4_DOCS_URL = 'https://sap.github.io/spartacus-docs/css-changes-in-version-4/';
/***** Storefinder end *****/
/******* CLI start *******/
exports.SPARTACUS_ROUTING_MODULE = 'app-routing';
exports.SPARTACUS_MODULE = 'spartacus';
exports.SPARTACUS_FEATURES_MODULE = 'spartacus-features';
exports.SPARTACUS_FEATURES_NG_MODULE = 'SpartacusFeaturesModule';
exports.SPARTACUS_CONFIGURATION_MODULE = 'spartacus-configuration';
exports.CLI_ASM_FEATURE = 'ASM';
exports.CLI_CART_IMPORT_EXPORT_FEATURE = 'Import-Export';
exports.CLI_CART_SAVED_CART_FEATURE = 'Saved-Cart';
exports.CLI_CART_QUICK_ORDER_FEATURE = 'Quick-Order';
exports.CLI_CDC_FEATURE = 'CDC';
exports.CLI_CDS_FEATURE = 'CDS';
exports.CLI_DIGITAL_PAYMENTS_FEATURE = 'Digital-Payments';
exports.CLI_EPD_VISUALIZATION_FEATURE = 'EPD-Visualization';
exports.CLI_ORGANIZATION_ADMINISTRATION_FEATURE = 'Administration';
exports.CLI_ORGANIZATION_ORDER_APPROVAL_FEATURE = 'Order-Approval';
exports.CLI_PRODUCT_BULK_PRICING_FEATURE = 'Bulk-Pricing';
exports.CLI_PRODUCT_VARIANTS_FEATURE = 'Product-Variants';
exports.CLI_PRODUCT_IMAGE_ZOOM_FEATURE = 'Image-Zoom';
exports.CLI_PRODUCT_CONFIGURATOR_VC_FEATURE = 'VC-Configurator';
exports.CLI_PRODUCT_CONFIGURATOR_TEXTFIELD_FEATURE = 'Textfield-Configurator';
exports.CLI_PRODUCT_CONFIGURATOR_CPQ_FEATURE = 'CPQ-Configurator';
exports.CLI_QUALTRICS_FEATURE = 'Qualtrics';
exports.CLI_SMARTEDIT_FEATURE = 'SmartEdit';
exports.CLI_STOREFINDER_FEATURE = 'Store-Finder';
exports.CLI_TRACKING_PERSONALIZATION_FEATURE = 'Personalization';
exports.CLI_TRACKING_TMS_GTM_FEATURE = 'TMS-GTM';
exports.CLI_TRACKING_TMS_AEP_FEATURE = 'TMS-AEPL';
exports.CLI_USER_ACCOUNT_FEATURE = 'Account';
exports.CLI_USER_PROFILE_FEATURE = 'Profile';
exports.CLI_CHECKOUT_FEATURE = 'Checkout';
exports.CLI_ORDER_FEATURE = 'Order';
/******* CLI end *******/
