// @spartacus/core/esm2015/src/anonymous-consents/anonymous-consents.module.js
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '../config/config-providers';
import { defaultAnonymousConsentsConfig } from './config/default-anonymous-consents-config';
import { AnonymousConsentsService } from './facade/anonymous-consents.service';
import { interceptors } from './http-interceptors/index';
import { AnonymousConsentsStoreModule } from './store/anonymous-consents-store.module';
import * as i0 from "@angular/core";
export class AnonymousConsentsModule {
    static forRoot() {
        return {
            ngModule: AnonymousConsentsModule,
            providers: [
                ...interceptors,
                AnonymousConsentsService,
                provideDefaultConfig(defaultAnonymousConsentsConfig),
            ],
        };
    }
}
AnonymousConsentsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AnonymousConsentsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsModule, imports: [AnonymousConsentsStoreModule] });
AnonymousConsentsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsModule, imports: [[AnonymousConsentsStoreModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AnonymousConsentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [AnonymousConsentsStoreModule],
                }]
        }] });
