// @spartacus/storefront/esm2015/cms-components/wish-list/wish-list.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AuthGuard, I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { ItemCounterModule, MediaModule, StarRatingModule, } from '../../shared/index';
import { AddToCartModule } from '../cart/index';
import { WishListItemComponent } from './components/wish-list-item/wish-list-item.component';
import { WishListComponent } from './components/wish-list/wish-list.component';
import * as i0 from "@angular/core";
export class WishListModule {
}
WishListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: WishListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
WishListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: WishListModule, declarations: [WishListComponent, WishListItemComponent], imports: [AddToCartModule,
        CommonModule,
        I18nModule,
        MediaModule,
        RouterModule,
        StarRatingModule,
        UrlModule,
        ItemCounterModule], exports: [WishListComponent, WishListItemComponent] });
WishListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: WishListModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                WishListComponent: {
                    component: WishListComponent,
                    guards: [AuthGuard],
                },
            },
        }),
    ], imports: [[
            AddToCartModule,
            CommonModule,
            I18nModule,
            MediaModule,
            RouterModule,
            StarRatingModule,
            UrlModule,
            ItemCounterModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: WishListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AddToCartModule,
                        CommonModule,
                        I18nModule,
                        MediaModule,
                        RouterModule,
                        StarRatingModule,
                        UrlModule,
                        ItemCounterModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                WishListComponent: {
                                    component: WishListComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [WishListComponent, WishListItemComponent],
                    exports: [WishListComponent, WishListItemComponent],
                }]
        }] });
