// @spartacus/cart/esm2015/quick-order/components/cart-quick-order-form/cart-quick-order-form.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { FormErrorsModule } from '@spartacus/storefront';
import { CartQuickOrderFormComponent } from './cart-quick-order-form.component';
import * as i0 from "@angular/core";
export class CartQuickOrderFormModule {
}
CartQuickOrderFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartQuickOrderFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CartQuickOrderFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartQuickOrderFormModule, declarations: [CartQuickOrderFormComponent], imports: [CommonModule, ReactiveFormsModule, I18nModule, FormErrorsModule], exports: [CartQuickOrderFormComponent] });
CartQuickOrderFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartQuickOrderFormModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                CartQuickOrderFormComponent: {
                    component: CartQuickOrderFormComponent,
                },
            },
        }),
    ], imports: [[CommonModule, ReactiveFormsModule, I18nModule, FormErrorsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartQuickOrderFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, I18nModule, FormErrorsModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CartQuickOrderFormComponent: {
                                    component: CartQuickOrderFormComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [CartQuickOrderFormComponent],
                    exports: [CartQuickOrderFormComponent],
                }]
        }] });
