// @spartacus/cart/esm2015/quick-order/components/cart-quick-order-form/cart-quick-order-form.component.js
import { ChangeDetectionStrategy, Component, } from '@angular/core';
import { Validators } from '@angular/forms';
import { CartAddEntryFailEvent, CartAddEntrySuccessEvent, GlobalMessageType, } from '@spartacus/core';
import { Subscription } from 'rxjs';
import { first, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@angular/forms";
import * as i3 from "@spartacus/storefront";
import * as i4 from "@angular/common";
export class CartQuickOrderFormComponent {
    constructor(activeCartService, eventService, formBuilder, globalMessageService) {
        this.activeCartService = activeCartService;
        this.eventService = eventService;
        this.formBuilder = formBuilder;
        this.globalMessageService = globalMessageService;
        this.cartIsLoading$ = this.activeCartService
            .isStable()
            .pipe(map((loaded) => !loaded));
        this.cart$ = this.activeCartService.getActive();
        this.min = 1;
        this.subscription = new Subscription();
        this.cartEventsSubscription = new Subscription();
        this.minQuantityValue = 1;
    }
    ngOnInit() {
        this.buildForm();
        this.watchQuantityChange();
    }
    ngOnDestroy() {
        var _a, _b;
        (_a = this.subscription) === null || _a === void 0 ? void 0 : _a.unsubscribe();
        (_b = this.cartEventsSubscription) === null || _b === void 0 ? void 0 : _b.unsubscribe();
    }
    applyQuickOrder() {
        var _a, _b;
        if (this.quickOrderForm.invalid) {
            this.quickOrderForm.markAllAsTouched();
            return;
        }
        const productCode = (_a = this.quickOrderForm.get('productCode')) === null || _a === void 0 ? void 0 : _a.value;
        const quantity = (_b = this.quickOrderForm.get('quantity')) === null || _b === void 0 ? void 0 : _b.value;
        this.watchAddEntrySuccessEvent();
        this.watchAddEntryFailEvent();
        if (productCode && quantity) {
            this.activeCartService.addEntry(productCode, quantity);
        }
    }
    buildForm() {
        this.quickOrderForm = this.formBuilder.group({
            productCode: ['', [Validators.required]],
            quantity: [
                this.minQuantityValue,
                { updateOn: 'blur', validators: [Validators.required] },
            ],
        });
    }
    watchQuantityChange() {
        var _a;
        this.subscription.add((_a = this.quickOrderForm
            .get('quantity')) === null || _a === void 0 ? void 0 : _a.valueChanges.subscribe((value) => {
            var _a;
            return (_a = this.quickOrderForm
                .get('quantity')) === null || _a === void 0 ? void 0 : _a.setValue(this.getValidCount(value), { emitEvent: false });
        }));
    }
    watchAddEntrySuccessEvent() {
        this.cartEventsSubscription.add(this.eventService
            .get(CartAddEntrySuccessEvent)
            .pipe(first())
            .subscribe((data) => {
            var _a, _b;
            let key = 'quickOrderCartForm.stockLevelReached';
            let productTranslation;
            let messageType = GlobalMessageType.MSG_TYPE_WARNING;
            if (data.quantityAdded) {
                key =
                    data.quantityAdded > 1
                        ? 'quickOrderCartForm.entriesWereAdded'
                        : 'quickOrderCartForm.entryWasAdded';
                productTranslation =
                    data.quantityAdded > 1
                        ? 'quickOrderCartForm.products'
                        : 'quickOrderCartForm.product';
                messageType = GlobalMessageType.MSG_TYPE_CONFIRMATION;
            }
            this.globalMessageService.add({
                key,
                params: {
                    product: ((_b = (_a = data === null || data === void 0 ? void 0 : data.entry) === null || _a === void 0 ? void 0 : _a.product) === null || _b === void 0 ? void 0 : _b.name) || productTranslation,
                    quantity: data.quantityAdded,
                },
            }, messageType);
            this.resetForm();
        }));
    }
    watchAddEntryFailEvent() {
        this.cartEventsSubscription.add(this.eventService
            .get(CartAddEntryFailEvent)
            .pipe(first())
            .subscribe(() => {
            this.globalMessageService.add({
                key: 'quickOrderCartForm.noResults',
            }, GlobalMessageType.MSG_TYPE_ERROR);
        }));
    }
    getValidCount(value) {
        if (value < this.min || !value) {
            value = this.min;
        }
        return value;
    }
    resetForm() {
        this.quickOrderForm.reset();
    }
}
CartQuickOrderFormComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartQuickOrderFormComponent, deps: [{ token: i1.ActiveCartService }, { token: i1.EventService }, { token: i2.FormBuilder }, { token: i1.GlobalMessageService }], target: i0.ɵɵFactoryTarget.Component });
CartQuickOrderFormComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: CartQuickOrderFormComponent, selector: "cx-cart-quick-order-form", ngImport: i0, template: "<ng-container *ngIf=\"cart$ | async as cart\">\n  <div class=\"cx-cart-quick-order-form-title\">\n    {{ 'quickOrderCartForm.title' | cxTranslate }}\n  </div>\n  <div class=\"form-group\">\n    <form (ngSubmit)=\"applyQuickOrder()\" [formGroup]=\"quickOrderForm\">\n      <div class=\"cx-cart-quick-order-form-container\">\n        <span class=\"cx-cart-quick-order-form-productID\">\n          <label class=\"cx-cart-quick-order-form-label\">\n            {{ 'quickOrderCartForm.productCodeLabel' | cxTranslate }}\n          </label>\n          <input\n            aria-required=\"true\"\n            class=\"form-control input-product-code\"\n            formControlName=\"productCode\"\n            placeholder=\"{{\n              'quickOrderCartForm.productCodePlaceholder' | cxTranslate\n            }}\"\n            type=\"text\"\n          />\n        </span>\n\n        <span class=\"cx-cart-quick-order-form-qty\">\n          <label class=\"cx-cart-quick-order-form-label\">\n            {{ 'quickOrderCartForm.quantityLabel' | cxTranslate }}\n          </label>\n          <input\n            aria-required=\"true\"\n            class=\"form-control input-quantity\"\n            formControlName=\"quantity\"\n            type=\"number\"\n          />\n        </span>\n        <button\n          [class.disabled]=\"cartIsLoading$ | async\"\n          [disabled]=\"cartIsLoading$ | async\"\n          class=\"btn btn-block btn-action apply-quick-order-button\"\n          type=\"submit\"\n        >\n          {{ 'quickOrderCartForm.addToCart' | cxTranslate }}\n        </button>\n        <cx-form-errors\n          aria-live=\"assertive\"\n          aria-atomic=\"true\"\n          [control]=\"quickOrderForm.get('productCode')\"\n        ></cx-form-errors>\n        <cx-form-errors\n          aria-live=\"assertive\"\n          aria-atomic=\"true\"\n          [control]=\"quickOrderForm.get('quantity')\"\n        ></cx-form-errors>\n      </div>\n    </form></div\n></ng-container>\n", components: [{ type: i3.FormErrorsComponent, selector: "cx-form-errors", inputs: ["prefix", "translationParams", "control"] }], directives: [{ type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { type: i2.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { type: i2.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { type: i2.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.FormControlName, selector: "[formControlName]", inputs: ["disabled", "formControlName", "ngModel"], outputs: ["ngModelChange"] }, { type: i2.NumberValueAccessor, selector: "input[type=number][formControlName],input[type=number][formControl],input[type=number][ngModel]" }], pipes: { "async": i4.AsyncPipe, "cxTranslate": i1.TranslatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartQuickOrderFormComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-cart-quick-order-form',
                    templateUrl: './cart-quick-order-form.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.ActiveCartService }, { type: i1.EventService }, { type: i2.FormBuilder }, { type: i1.GlobalMessageService }]; } });
