// @spartacus/checkout/esm2015/core/store/actions/payment-types.action.js
import { PROCESS_FEATURE, StateUtils, } from '@spartacus/core';
import { GET_PAYMENT_TYPES_PROCESS_ID } from '../checkout-state';
export const LOAD_PAYMENT_TYPES = '[Checkout] Load Payment Types';
export const LOAD_PAYMENT_TYPES_FAIL = '[Checkout] Load Payment Types Fail';
export const LOAD_PAYMENT_TYPES_SUCCESS = '[Checkout] Load Payment Types Success';
export const RESET_LOAD_PAYMENT_TYPES_PROCESS_ID = '[Checkout] Reset Load Payment Type Process';
export const SET_PAYMENT_TYPE = '[Checkout] Set Payment Type';
export const SET_PAYMENT_TYPE_FAIL = '[Checkout] Set Payment Type Fail';
export const SET_PAYMENT_TYPE_SUCCESS = '[Checkout] Set Payment Type Success';
export class LoadPaymentTypes extends StateUtils.EntityLoadAction {
    constructor() {
        super(PROCESS_FEATURE, GET_PAYMENT_TYPES_PROCESS_ID);
        this.type = LOAD_PAYMENT_TYPES;
    }
}
export class LoadPaymentTypesFail extends StateUtils.EntityFailAction {
    constructor(payload) {
        super(PROCESS_FEATURE, GET_PAYMENT_TYPES_PROCESS_ID);
        this.payload = payload;
        this.type = LOAD_PAYMENT_TYPES_FAIL;
    }
}
export class LoadPaymentTypesSuccess extends StateUtils.EntitySuccessAction {
    constructor(payload) {
        super(PROCESS_FEATURE, GET_PAYMENT_TYPES_PROCESS_ID);
        this.payload = payload;
        this.type = LOAD_PAYMENT_TYPES_SUCCESS;
    }
}
export class ResetLoadPaymentTypesProcess extends StateUtils.EntityLoaderResetAction {
    constructor() {
        super(PROCESS_FEATURE, GET_PAYMENT_TYPES_PROCESS_ID);
        this.type = RESET_LOAD_PAYMENT_TYPES_PROCESS_ID;
    }
}
export class SetPaymentType {
    constructor(payload) {
        this.payload = payload;
        this.type = SET_PAYMENT_TYPE;
    }
}
export class SetPaymentTypeFail {
    constructor(payload) {
        this.payload = payload;
        this.type = SET_PAYMENT_TYPE_FAIL;
    }
}
export class SetPaymentTypeSuccess {
    constructor(payload) {
        this.payload = payload;
        this.type = SET_PAYMENT_TYPE_SUCCESS;
    }
}
