// @spartacus/user/bundles/spartacus-user-account-occ.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core'), require('@spartacus/user/account/core'), require('rxjs'), require('rxjs/operators'), require('@angular/common/http')) :
    typeof define === 'function' && define.amd ? define('@spartacus/user/account/occ', ['exports', '@angular/core', '@spartacus/core', '@spartacus/user/account/core', 'rxjs', 'rxjs/operators', '@angular/common/http'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.user = global.spartacus.user || {}, global.spartacus.user.account = global.spartacus.user.account || {}, global.spartacus.user.account.occ = {}), global.ng.core, global.core, global.spartacus.user.account.core, global.rxjs, global.rxjs.operators, global.ng.common.http));
}(this, (function (exports, i0, i2, core, rxjs, operators, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i2__namespace = /*#__PURE__*/_interopNamespace(i2);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var defaultOccUserAccountConfig = {
        backend: {
            occ: { endpoints: { user: 'users/${userId}' } },
        },
    };

    var OccUserAccountAdapter = /** @class */ (function () {
        function OccUserAccountAdapter(http, occEndpoints, converter) {
            this.http = http;
            this.occEndpoints = occEndpoints;
            this.converter = converter;
        }
        OccUserAccountAdapter.prototype.load = function (userId) {
            var url = this.occEndpoints.buildUrl('user', { urlParams: { userId: userId } });
            return this.http.get(url).pipe(operators.catchError(function (error) { return rxjs.throwError(i2.normalizeHttpError(error)); }), this.converter.pipeable(core.USER_ACCOUNT_NORMALIZER));
        };
        return OccUserAccountAdapter;
    }());
    OccUserAccountAdapter.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccUserAccountAdapter, deps: [{ token: i1__namespace.HttpClient }, { token: i2__namespace.OccEndpointsService }, { token: i2__namespace.ConverterService }], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    OccUserAccountAdapter.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccUserAccountAdapter });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OccUserAccountAdapter, decorators: [{
                type: i0.Injectable
            }], ctorParameters: function () { return [{ type: i1__namespace.HttpClient }, { type: i2__namespace.OccEndpointsService }, { type: i2__namespace.ConverterService }]; } });

    var UserAccountOccModule = /** @class */ (function () {
        function UserAccountOccModule() {
        }
        return UserAccountOccModule;
    }());
    UserAccountOccModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserAccountOccModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    UserAccountOccModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserAccountOccModule });
    UserAccountOccModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserAccountOccModule, providers: [
            i2.provideDefaultConfig(defaultOccUserAccountConfig),
            { provide: core.UserAccountAdapter, useClass: OccUserAccountAdapter },
        ] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserAccountOccModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        providers: [
                            i2.provideDefaultConfig(defaultOccUserAccountConfig),
                            { provide: core.UserAccountAdapter, useClass: OccUserAccountAdapter },
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.OccUserAccountAdapter = OccUserAccountAdapter;
    exports.UserAccountOccModule = UserAccountOccModule;
    exports.defaultOccUserAccountConfig = defaultOccUserAccountConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
