// @spartacus/user/bundles/spartacus-user-profile-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@spartacus/core')) :
    typeof define === 'function' && define.amd ? define('@spartacus/user/profile/root', ['exports', '@angular/core', '@spartacus/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.user = global.spartacus.user || {}, global.spartacus.user.profile = global.spartacus.user.profile || {}, global.spartacus.user.profile.root = {}), global.ng.core, global.core));
}(this, (function (exports, i0, core) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var USER_PROFILE_FEATURE = 'userProfile';
    var USER_PROFILE_CORE_FEATURE = 'userProfileCore';

    var UserProfileFacade = /** @class */ (function () {
        function UserProfileFacade() {
        }
        return UserProfileFacade;
    }());
    UserProfileFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserProfileFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    UserProfileFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserProfileFacade, providedIn: 'root', useFactory: function () { return core.facadeFactory({
            facade: UserProfileFacade,
            feature: USER_PROFILE_CORE_FEATURE,
            methods: ['get', 'update', 'close', 'getTitles'],
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserProfileFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return core.facadeFactory({
                            facade: UserProfileFacade,
                            feature: USER_PROFILE_CORE_FEATURE,
                            methods: ['get', 'update', 'close', 'getTitles'],
                        }); },
                    }]
            }] });

    // TODO: Inline this factory when we start releasing Ivy compiled libraries
    function defaultUserProfileComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[USER_PROFILE_FEATURE] = {
                    cmsComponents: [
                        'RegisterCustomerComponent',
                        'UpdateProfileComponent',
                        'UpdateEmailComponent',
                        'UpdatePasswordComponent',
                        'ForgotPasswordComponent',
                        'ResetPasswordComponent',
                        'CloseAccountComponent',
                    ],
                },
                // by default core is bundled together with components
                _a[USER_PROFILE_CORE_FEATURE] = USER_PROFILE_FEATURE,
                _a),
        };
        return config;
    }
    var UserProfileRootModule = /** @class */ (function () {
        function UserProfileRootModule() {
        }
        return UserProfileRootModule;
    }());
    UserProfileRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserProfileRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    UserProfileRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserProfileRootModule });
    UserProfileRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserProfileRootModule, providers: [
            core.provideDefaultConfigFactory(defaultUserProfileComponentsConfig),
            {
                provide: core.UserProfileFacadeTransitionalToken,
                useExisting: UserProfileFacade,
            },
        ] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserProfileRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        providers: [
                            core.provideDefaultConfigFactory(defaultUserProfileComponentsConfig),
                            {
                                provide: core.UserProfileFacadeTransitionalToken,
                                useExisting: UserProfileFacade,
                            },
                        ],
                    }]
            }] });

    var UserEmailFacade = /** @class */ (function () {
        function UserEmailFacade() {
        }
        return UserEmailFacade;
    }());
    UserEmailFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserEmailFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    UserEmailFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserEmailFacade, providedIn: 'root', useFactory: function () { return core.facadeFactory({
            facade: UserEmailFacade,
            feature: USER_PROFILE_CORE_FEATURE,
            methods: ['update'],
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserEmailFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return core.facadeFactory({
                            facade: UserEmailFacade,
                            feature: USER_PROFILE_CORE_FEATURE,
                            methods: ['update'],
                        }); },
                    }]
            }] });

    var UserPasswordFacade = /** @class */ (function () {
        function UserPasswordFacade() {
        }
        return UserPasswordFacade;
    }());
    UserPasswordFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserPasswordFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    UserPasswordFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserPasswordFacade, providedIn: 'root', useFactory: function () { return core.facadeFactory({
            facade: UserPasswordFacade,
            feature: USER_PROFILE_CORE_FEATURE,
            methods: ['update', 'reset', 'requestForgotPasswordEmail'],
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserPasswordFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return core.facadeFactory({
                            facade: UserPasswordFacade,
                            feature: USER_PROFILE_CORE_FEATURE,
                            methods: ['update', 'reset', 'requestForgotPasswordEmail'],
                        }); },
                    }]
            }] });

    var UserRegisterFacade = /** @class */ (function () {
        function UserRegisterFacade() {
        }
        return UserRegisterFacade;
    }());
    UserRegisterFacade.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserRegisterFacade, deps: [], target: i0__namespace.ɵɵFactoryTarget.Injectable });
    UserRegisterFacade.ɵprov = i0__namespace.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserRegisterFacade, providedIn: 'root', useFactory: function () { return core.facadeFactory({
            facade: UserRegisterFacade,
            feature: USER_PROFILE_CORE_FEATURE,
            methods: ['register', 'registerGuest', 'getTitles'],
        }); } });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: UserRegisterFacade, decorators: [{
                type: i0.Injectable,
                args: [{
                        providedIn: 'root',
                        useFactory: function () { return core.facadeFactory({
                            facade: UserRegisterFacade,
                            feature: USER_PROFILE_CORE_FEATURE,
                            methods: ['register', 'registerGuest', 'getTitles'],
                        }); },
                    }]
            }] });

    /** AUGMENTABLE_TYPES_END */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.USER_PROFILE_CORE_FEATURE = USER_PROFILE_CORE_FEATURE;
    exports.USER_PROFILE_FEATURE = USER_PROFILE_FEATURE;
    exports.UserEmailFacade = UserEmailFacade;
    exports.UserPasswordFacade = UserPasswordFacade;
    exports.UserProfileFacade = UserProfileFacade;
    exports.UserProfileRootModule = UserProfileRootModule;
    exports.UserRegisterFacade = UserRegisterFacade;
    exports.defaultUserProfileComponentsConfig = defaultUserProfileComponentsConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
