// @spartacus/user/bundles/spartacus-user-account-assets.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define('@spartacus/user/account/assets', ['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.user = global.spartacus.user || {}, global.spartacus.user.account = global.spartacus.user.account || {}, global.spartacus.user.account.assets = {})));
}(this, (function (exports) { 'use strict';

    var userAccount = {
        loginForm: {
            forgotPassword: 'Forgot password?',
            signIn: 'Sign In',
            register: 'Register',
            dontHaveAccount: 'Don’t have an account?',
            guestCheckout: 'Guest Checkout',
            emailAddress: {
                label: 'Email address',
                placeholder: 'Enter email',
            },
            password: {
                label: 'Password',
                placeholder: 'Password',
            },
            wrongEmailFormat: 'This is not a valid email format.',
        },
        miniLogin: {
            userGreeting: 'Hi, {{name}}',
            signInRegister: 'Sign In / Register',
        },
    };

    var en = {
        userAccount: userAccount,
    };

    var userAccountTranslations = {
        en: en,
    };
    var userAccountTranslationChunksConfig = {
        userAccount: ['loginForm', 'miniLogin'],
    };

    /**
     * Generated bundle index. Do not edit.
     */

    exports.userAccountTranslationChunksConfig = userAccountTranslationChunksConfig;
    exports.userAccountTranslations = userAccountTranslations;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
