// @spartacus/core/esm2015/src/cart/event/cart.events.js
import { CxEvent } from '../../event/cx-event';
/**
 * Base cart event. Most cart events should have these properties.
 */
export class CartEvent extends CxEvent {
}
// =====================================================================
export class CartAddEntryEvent extends CartEvent {
}
/**
 * Event's type
 */
CartAddEntryEvent.type = 'CartAddEntryEvent';
export class CartAddEntrySuccessEvent extends CartEvent {
}
/**
 * Event's type
 */
CartAddEntrySuccessEvent.type = 'CartAddEntrySuccessEvent';
export class CartAddEntryFailEvent extends CartEvent {
}
/**
 * Event's type
 */
CartAddEntryFailEvent.type = 'CartAddEntryFailEvent';
export class CartRemoveEntryFailEvent extends CartEvent {
}
/**
 * Event's type
 */
CartRemoveEntryFailEvent.type = 'CartRemoveEntryFailEvent';
export class CartRemoveEntrySuccessEvent extends CartEvent {
}
/**
 * Event's type
 */
CartRemoveEntrySuccessEvent.type = 'CartRemoveEntrySuccessEvent';
export class CartUpdateEntrySuccessEvent extends CartEvent {
}
/**
 * Event's type
 */
CartUpdateEntrySuccessEvent.type = 'CartUpdateEntrySuccessEvent';
export class CartUpdateEntryFailEvent extends CartEvent {
}
/**
 * Event's type
 */
CartUpdateEntryFailEvent.type = 'CartUpdateEntryFailEvent';
export class MergeCartSuccessEvent extends CartEvent {
}
/**
 * Event's type
 */
MergeCartSuccessEvent.type = 'MergeCartSuccessEvent';
