// @spartacus/storefront/esm2015/layout/layout.module.js
import { NgModule } from '@angular/core';
import { provideConfig } from '@spartacus/core';
import { OutletRefModule } from '../cms-structure/outlet/outlet-ref/outlet-ref.module';
import { defaultLayoutConfig } from './config/default-layout.config';
import { DirectionModule } from './direction/direction.module';
import { LaunchDialogModule } from './launch-dialog/index';
import { ThemeModule } from './theme/theme.module';
import * as i0 from "@angular/core";
import * as i1 from "./launch-dialog/launch-dialog.module";
export class LayoutModule {
}
LayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LayoutModule, imports: [OutletRefModule, i1.LaunchDialogModule, DirectionModule,
        ThemeModule], exports: [OutletRefModule] });
LayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LayoutModule, providers: [provideConfig(defaultLayoutConfig)], imports: [[
            OutletRefModule,
            LaunchDialogModule.forRoot(),
            DirectionModule,
            ThemeModule,
        ], OutletRefModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: LayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        OutletRefModule,
                        LaunchDialogModule.forRoot(),
                        DirectionModule,
                        ThemeModule,
                    ],
                    providers: [provideConfig(defaultLayoutConfig)],
                    exports: [OutletRefModule],
                }]
        }] });
