// @spartacus/storefront/esm2015/shared/components/order-overview/order-overview.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule } from '@spartacus/core';
import { CardModule } from '../card/card.module';
import { OrderOverviewComponent } from './order-overview.component';
import * as i0 from "@angular/core";
export class OrderOverviewModule {
}
OrderOverviewModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOverviewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrderOverviewModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOverviewModule, declarations: [OrderOverviewComponent], imports: [CommonModule, I18nModule, CardModule], exports: [OrderOverviewComponent] });
OrderOverviewModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOverviewModule, imports: [[CommonModule, I18nModule, CardModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderOverviewModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule, CardModule],
                    declarations: [OrderOverviewComponent],
                    exports: [OrderOverviewComponent],
                }]
        }] });
