// @spartacus/core/esm2015/src/auth/client-auth/client-auth.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { interceptors } from './http-interceptors/index';
import { ClientAuthStoreModule } from './store/client-auth-store.module';
import * as i0 from "@angular/core";
/**
 * Some of the OCC endpoints require Authorization header with the client token (eg. user registration).
 * This pattern should not be used in the frontend apps, but until OCC changes this requirement
 * we provide this module to support using those endpoints.
 *
 * After OCC improvements regarding client authentication this module can be safely removed.
 */
export class ClientAuthModule {
    static forRoot() {
        return {
            ngModule: ClientAuthModule,
            providers: [...interceptors],
        };
    }
}
ClientAuthModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ClientAuthModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ClientAuthModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ClientAuthModule, imports: [CommonModule, ClientAuthStoreModule] });
ClientAuthModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ClientAuthModule, imports: [[CommonModule, ClientAuthStoreModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ClientAuthModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ClientAuthStoreModule],
                }]
        }] });
