// @spartacus/checkout/esm2015/components/components/checkout-order-summary/checkout-order-summary.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/storefront";
import * as i3 from "@angular/common";
export class CheckoutOrderSummaryComponent {
    constructor(activeCartService) {
        this.activeCartService = activeCartService;
        this.cart$ = this.activeCartService.getActive();
    }
}
CheckoutOrderSummaryComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutOrderSummaryComponent, deps: [{ token: i1.ActiveCartService }], target: i0.ɵɵFactoryTarget.Component });
CheckoutOrderSummaryComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: CheckoutOrderSummaryComponent, selector: "cx-checkout-order-summary", ngImport: i0, template: "<cx-order-summary [cart]=\"cart$ | async\"></cx-order-summary>\n", components: [{ type: i2.OrderSummaryComponent, selector: "cx-order-summary", inputs: ["cart"] }], pipes: { "async": i3.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutOrderSummaryComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-checkout-order-summary',
                    templateUrl: './checkout-order-summary.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.ActiveCartService }]; } });
