// @spartacus/checkout/esm2015/components/components/review-submit/review-submit.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { CardModule, CartSharedModule, IconModule, PromotionsModule, } from '@spartacus/storefront';
import { CartNotEmptyGuard } from '../../guards/cart-not-empty.guard';
import { CheckoutAuthGuard } from '../../guards/checkout-auth.guard';
import { ReviewSubmitComponent } from './review-submit.component';
import * as i0 from "@angular/core";
export class ReviewSubmitModule {
}
ReviewSubmitModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ReviewSubmitModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ReviewSubmitModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ReviewSubmitModule, declarations: [ReviewSubmitComponent], imports: [CommonModule,
        CardModule,
        CartSharedModule,
        I18nModule,
        UrlModule,
        RouterModule,
        PromotionsModule,
        IconModule], exports: [ReviewSubmitComponent] });
ReviewSubmitModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ReviewSubmitModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                CheckoutReviewOrder: {
                    component: ReviewSubmitComponent,
                    // TODO(#8880): Shouldn't we keep ShippingAddressSetGuard and others here?
                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                },
            },
        }),
    ], imports: [[
            CommonModule,
            CardModule,
            CartSharedModule,
            I18nModule,
            UrlModule,
            RouterModule,
            PromotionsModule,
            IconModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ReviewSubmitModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        CardModule,
                        CartSharedModule,
                        I18nModule,
                        UrlModule,
                        RouterModule,
                        PromotionsModule,
                        IconModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutReviewOrder: {
                                    component: ReviewSubmitComponent,
                                    // TODO(#8880): Shouldn't we keep ShippingAddressSetGuard and others here?
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [ReviewSubmitComponent],
                    exports: [ReviewSubmitComponent],
                }]
        }] });
