// @spartacus/storefront/esm2015/cms-components/misc/message/message.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule } from '@spartacus/core';
import { IconModule } from '../../../cms-components/misc/icon/icon.module';
import { MessageComponent } from './message.component';
import * as i0 from "@angular/core";
export class MessageComponentModule {
}
MessageComponentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageComponentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MessageComponentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageComponentModule, declarations: [MessageComponent], imports: [CommonModule, IconModule, I18nModule], exports: [MessageComponent] });
MessageComponentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageComponentModule, imports: [[CommonModule, IconModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageComponentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, I18nModule],
                    declarations: [MessageComponent],
                    exports: [MessageComponent],
                }]
        }] });
