// @spartacus/checkout/esm2015/root/facade/checkout-cost-center.facade.js
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { CHECKOUT_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class CheckoutCostCenterFacade {
}
CheckoutCostCenterFacade.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutCostCenterFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
CheckoutCostCenterFacade.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutCostCenterFacade, providedIn: 'root', useFactory: () => facadeFactory({
        facade: CheckoutCostCenterFacade,
        feature: CHECKOUT_CORE_FEATURE,
        methods: ['setCostCenter', 'getCostCenter'],
        async: true,
    }) });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutCostCenterFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: CheckoutCostCenterFacade,
                        feature: CHECKOUT_CORE_FEATURE,
                        methods: ['setCostCenter', 'getCostCenter'],
                        async: true,
                    }),
                }]
        }] });
