// @spartacus/checkout/esm2015/root/facade/checkout-payment.facade.js
import { Injectable } from '@angular/core';
import { facadeFactory, } from '@spartacus/core';
import { CHECKOUT_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class CheckoutPaymentFacade {
}
CheckoutPaymentFacade.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutPaymentFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
CheckoutPaymentFacade.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutPaymentFacade, providedIn: 'root', useFactory: () => facadeFactory({
        facade: CheckoutPaymentFacade,
        feature: CHECKOUT_CORE_FEATURE,
        methods: [
            'getCardTypes',
            'getPaymentDetails',
            'getSetPaymentDetailsResultProcess',
            'resetSetPaymentDetailsProcess',
            'loadSupportedCardTypes',
            'createPaymentDetails',
            'setPaymentDetails',
            'paymentProcessSuccess',
        ],
        async: true,
    }) });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutPaymentFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: CheckoutPaymentFacade,
                        feature: CHECKOUT_CORE_FEATURE,
                        methods: [
                            'getCardTypes',
                            'getPaymentDetails',
                            'getSetPaymentDetailsResultProcess',
                            'resetSetPaymentDetailsProcess',
                            'loadSupportedCardTypes',
                            'createPaymentDetails',
                            'setPaymentDetails',
                            'paymentProcessSuccess',
                        ],
                        async: true,
                    }),
                }]
        }] });
