// @spartacus/checkout/esm2015/root/facade/checkout.facade.js
import { Injectable } from '@angular/core';
import { facadeFactory, } from '@spartacus/core';
import { CHECKOUT_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class CheckoutFacade {
}
CheckoutFacade.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
CheckoutFacade.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutFacade, providedIn: 'root', useFactory: () => facadeFactory({
        facade: CheckoutFacade,
        feature: CHECKOUT_CORE_FEATURE,
        methods: [
            'placeOrder',
            'scheduleReplenishmentOrder',
            'getPlaceOrderLoading',
            'getPlaceOrderSuccess',
            'getPlaceOrderError',
            'clearPlaceOrderState',
            'clearCheckoutData',
            'clearCheckoutStep',
            'loadCheckoutDetails',
            'getCheckoutDetailsLoaded',
            'isLoading',
            'getOrderDetails',
            'setOrderType',
            'getCurrentOrderType',
        ],
        async: true,
    }) });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: CheckoutFacade,
                        feature: CHECKOUT_CORE_FEATURE,
                        methods: [
                            'placeOrder',
                            'scheduleReplenishmentOrder',
                            'getPlaceOrderLoading',
                            'getPlaceOrderSuccess',
                            'getPlaceOrderError',
                            'clearPlaceOrderState',
                            'clearCheckoutData',
                            'clearCheckoutStep',
                            'loadCheckoutDetails',
                            'getCheckoutDetailsLoaded',
                            'isLoading',
                            'getOrderDetails',
                            'setOrderType',
                            'getCurrentOrderType',
                        ],
                        async: true,
                    }),
                }]
        }] });
