// @spartacus/checkout/esm2015/root/facade/checkout-delivery.facade.js
import { Injectable } from '@angular/core';
import { facadeFactory, } from '@spartacus/core';
import { CHECKOUT_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class CheckoutDeliveryFacade {
}
CheckoutDeliveryFacade.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutDeliveryFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
CheckoutDeliveryFacade.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutDeliveryFacade, providedIn: 'root', useFactory: () => facadeFactory({
        facade: CheckoutDeliveryFacade,
        feature: CHECKOUT_CORE_FEATURE,
        methods: [
            'getSupportedDeliveryModes',
            'getSelectedDeliveryMode',
            'getSelectedDeliveryModeCode',
            'getDeliveryAddress',
            'getSetDeliveryAddressProcess',
            'resetSetDeliveryAddressProcess',
            'getSetDeliveryModeProcess',
            'resetSetDeliveryModeProcess',
            'resetLoadSupportedDeliveryModesProcess',
            'getLoadSupportedDeliveryModeProcess',
            'clearCheckoutDeliveryModes',
            'createAndSetAddress',
            'loadSupportedDeliveryModes',
            'setDeliveryMode',
            'setDeliveryAddress',
            'clearCheckoutDeliveryAddress',
            'clearCheckoutDeliveryMode',
            'clearCheckoutDeliveryDetails',
        ],
        async: true,
    }) });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutDeliveryFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: CheckoutDeliveryFacade,
                        feature: CHECKOUT_CORE_FEATURE,
                        methods: [
                            'getSupportedDeliveryModes',
                            'getSelectedDeliveryMode',
                            'getSelectedDeliveryModeCode',
                            'getDeliveryAddress',
                            'getSetDeliveryAddressProcess',
                            'resetSetDeliveryAddressProcess',
                            'getSetDeliveryModeProcess',
                            'resetSetDeliveryModeProcess',
                            'resetLoadSupportedDeliveryModesProcess',
                            'getLoadSupportedDeliveryModeProcess',
                            'clearCheckoutDeliveryModes',
                            'createAndSetAddress',
                            'loadSupportedDeliveryModes',
                            'setDeliveryMode',
                            'setDeliveryAddress',
                            'clearCheckoutDeliveryAddress',
                            'clearCheckoutDeliveryMode',
                            'clearCheckoutDeliveryDetails',
                        ],
                        async: true,
                    }),
                }]
        }] });
