// @spartacus/checkout/esm2015/root/facade/clear-checkout.facade.js
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { CHECKOUT_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class ClearCheckoutFacade {
}
ClearCheckoutFacade.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ClearCheckoutFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
ClearCheckoutFacade.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ClearCheckoutFacade, providedIn: 'root', useFactory: () => facadeFactory({
        facade: ClearCheckoutFacade,
        feature: CHECKOUT_CORE_FEATURE,
        methods: ['resetCheckoutProcesses'],
        async: true,
    }) });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ClearCheckoutFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: ClearCheckoutFacade,
                        feature: CHECKOUT_CORE_FEATURE,
                        methods: ['resetCheckoutProcesses'],
                        async: true,
                    }),
                }]
        }] });
