// @spartacus/checkout/esm2015/root/facade/payment-type.facade.js
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { CHECKOUT_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class PaymentTypeFacade {
}
PaymentTypeFacade.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PaymentTypeFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
PaymentTypeFacade.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PaymentTypeFacade, providedIn: 'root', useFactory: () => facadeFactory({
        facade: PaymentTypeFacade,
        feature: CHECKOUT_CORE_FEATURE,
        methods: [
            'getPaymentTypes',
            'loadPaymentTypes',
            'setPaymentType',
            'getSelectedPaymentType',
            'isAccountPayment',
            'getPoNumber',
        ],
        async: true,
    }) });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PaymentTypeFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: PaymentTypeFacade,
                        feature: CHECKOUT_CORE_FEATURE,
                        methods: [
                            'getPaymentTypes',
                            'loadPaymentTypes',
                            'setPaymentType',
                            'getSelectedPaymentType',
                            'isAccountPayment',
                            'getPoNumber',
                        ],
                        async: true,
                    }),
                }]
        }] });
