// @spartacus/checkout/esm2015/components/components/delivery-mode/delivery-mode.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { SpinnerModule } from '@spartacus/storefront';
import { CartNotEmptyGuard } from '../../guards/cart-not-empty.guard';
import { CheckoutAuthGuard } from '../../guards/checkout-auth.guard';
import { DeliveryModeComponent } from './delivery-mode.component';
import * as i0 from "@angular/core";
export class DeliveryModeModule {
}
DeliveryModeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeliveryModeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DeliveryModeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeliveryModeModule, declarations: [DeliveryModeComponent], imports: [CommonModule, ReactiveFormsModule, I18nModule, SpinnerModule], exports: [DeliveryModeComponent] });
DeliveryModeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeliveryModeModule, providers: [
        provideDefaultConfig({
            cmsComponents: {
                CheckoutDeliveryMode: {
                    component: DeliveryModeComponent,
                    // TODO(#8880): Shouldn't we keep ShippingAddressSetGuard here?
                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                },
            },
        }),
    ], imports: [[CommonModule, ReactiveFormsModule, I18nModule, SpinnerModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeliveryModeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, I18nModule, SpinnerModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutDeliveryMode: {
                                    component: DeliveryModeComponent,
                                    // TODO(#8880): Shouldn't we keep ShippingAddressSetGuard here?
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [DeliveryModeComponent],
                    exports: [DeliveryModeComponent],
                }]
        }] });
