// @spartacus/checkout/esm2015/occ/checkout-occ.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CheckoutAdapter, CheckoutCostCenterAdapter, CheckoutDeliveryAdapter, CheckoutPaymentAdapter, CheckoutReplenishmentOrderAdapter, PaymentTypeAdapter, REPLENISHMENT_ORDER_FORM_SERIALIZER, } from '@spartacus/checkout/core';
import { provideDefaultConfig } from '@spartacus/core';
import { OccReplenishmentOrderFormSerializer } from './adapters/converters/index';
import { OccCheckoutCostCenterAdapter } from './adapters/occ-checkout-cost-center.adapter';
import { OccCheckoutDeliveryAdapter } from './adapters/occ-checkout-delivery.adapter';
import { OccCheckoutPaymentTypeAdapter } from './adapters/occ-checkout-payment-type.adapter';
import { OccCheckoutPaymentAdapter } from './adapters/occ-checkout-payment.adapter';
import { OccCheckoutReplenishmentOrderAdapter } from './adapters/occ-checkout-replenishment-order.adapter';
import { OccCheckoutAdapter } from './adapters/occ-checkout.adapter';
import { defaultOccCheckoutConfig } from './config/default-occ-checkout-config';
import * as i0 from "@angular/core";
export class CheckoutOccModule {
}
CheckoutOccModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CheckoutOccModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutOccModule, imports: [CommonModule] });
CheckoutOccModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutOccModule, providers: [
        provideDefaultConfig(defaultOccCheckoutConfig),
        OccReplenishmentOrderFormSerializer,
        {
            provide: CheckoutAdapter,
            useClass: OccCheckoutAdapter,
        },
        {
            provide: CheckoutDeliveryAdapter,
            useClass: OccCheckoutDeliveryAdapter,
        },
        {
            provide: CheckoutPaymentAdapter,
            useClass: OccCheckoutPaymentAdapter,
        },
        {
            provide: PaymentTypeAdapter,
            useClass: OccCheckoutPaymentTypeAdapter,
        },
        {
            provide: CheckoutCostCenterAdapter,
            useClass: OccCheckoutCostCenterAdapter,
        },
        {
            provide: CheckoutReplenishmentOrderAdapter,
            useClass: OccCheckoutReplenishmentOrderAdapter,
        },
        {
            provide: REPLENISHMENT_ORDER_FORM_SERIALIZER,
            useExisting: OccReplenishmentOrderFormSerializer,
            multi: true,
        },
    ], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccCheckoutConfig),
                        OccReplenishmentOrderFormSerializer,
                        {
                            provide: CheckoutAdapter,
                            useClass: OccCheckoutAdapter,
                        },
                        {
                            provide: CheckoutDeliveryAdapter,
                            useClass: OccCheckoutDeliveryAdapter,
                        },
                        {
                            provide: CheckoutPaymentAdapter,
                            useClass: OccCheckoutPaymentAdapter,
                        },
                        {
                            provide: PaymentTypeAdapter,
                            useClass: OccCheckoutPaymentTypeAdapter,
                        },
                        {
                            provide: CheckoutCostCenterAdapter,
                            useClass: OccCheckoutCostCenterAdapter,
                        },
                        {
                            provide: CheckoutReplenishmentOrderAdapter,
                            useClass: OccCheckoutReplenishmentOrderAdapter,
                        },
                        {
                            provide: REPLENISHMENT_ORDER_FORM_SERIALIZER,
                            useExisting: OccReplenishmentOrderFormSerializer,
                            multi: true,
                        },
                    ],
                }]
        }] });
