// @spartacus/storefinder/fesm2015/spartacus-storefinder-root.js
import * as i0 from '@angular/core';
import { NgModule } from '@angular/core';
import { provideDefaultConfig, provideDefaultConfigFactory } from '@spartacus/core';

const defaultStoreFinderLayoutConfig = {
    layoutSlots: {
        StoreFinderPageTemplate: {
            slots: ['MiddleContent', 'SideContent'],
        },
    },
};

const STORE_FINDER_FEATURE = 'storeFinder';

// TODO: Inline this factory when we start releasing Ivy compiled libraries
function defaultStoreFinderComponentsConfig() {
    const config = {
        featureModules: {
            [STORE_FINDER_FEATURE]: {
                cmsComponents: ['StoreFinderComponent'],
            },
        },
    };
    return config;
}
class StoreFinderRootModule {
}
StoreFinderRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StoreFinderRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule });
StoreFinderRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule, providers: [
        provideDefaultConfig(defaultStoreFinderLayoutConfig),
        provideDefaultConfigFactory(defaultStoreFinderComponentsConfig),
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: StoreFinderRootModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    providers: [
                        provideDefaultConfig(defaultStoreFinderLayoutConfig),
                        provideDefaultConfigFactory(defaultStoreFinderComponentsConfig),
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { STORE_FINDER_FEATURE, StoreFinderRootModule, defaultStoreFinderComponentsConfig, defaultStoreFinderLayoutConfig };
