// @spartacus/checkout/esm2015/core/services/checkout-page-meta.resolver.js
import { Injectable } from '@angular/core';
import { PageMetaResolver, PageType, } from '@spartacus/core';
import { switchMap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
/**
 * Resolves the page data for all Content Pages based on the `PageType.CONTENT_PAGE`
 * and the `MultiStepCheckoutSummaryPageTemplate`. If the checkout page matches this template,
 * the more generic `ContentPageMetaResolver` is overridden by this resolver.
 *
 * The page title and robots are resolved in this implementation only.
 */
export class CheckoutPageMetaResolver extends PageMetaResolver {
    constructor(translation, activeCartService, basePageMetaResolver) {
        super();
        this.translation = translation;
        this.activeCartService = activeCartService;
        this.basePageMetaResolver = basePageMetaResolver;
        this.cart$ = this.activeCartService.getActive();
        this.pageType = PageType.CONTENT_PAGE;
        this.pageTemplate = 'MultiStepCheckoutSummaryPageTemplate';
    }
    /**
     * @override
     * Resolves the page title from the translation `pageMetaResolver.checkout.title`. The
     * cart total item `count` is passed to the translation, so it can be used in the title.
     *
     * The title from the page data is ignored for this page title.
     */
    resolveTitle() {
        return this.cart$.pipe(switchMap((c) => this.translation.translate('pageMetaResolver.checkout.title', {
            count: c.totalItems,
        })));
    }
    resolveDescription() {
        return this.basePageMetaResolver.resolveDescription();
    }
    resolveRobots() {
        return this.basePageMetaResolver.resolveRobots();
    }
}
CheckoutPageMetaResolver.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutPageMetaResolver, deps: [{ token: i1.TranslationService }, { token: i1.ActiveCartService }, { token: i1.BasePageMetaResolver }], target: i0.ɵɵFactoryTarget.Injectable });
CheckoutPageMetaResolver.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutPageMetaResolver, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CheckoutPageMetaResolver, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TranslationService }, { type: i1.ActiveCartService }, { type: i1.BasePageMetaResolver }]; } });
