// @spartacus/storefront/esm2015/cms-components/cart/cart-page/active-cart-order-entries-context.js
import { Injectable } from '@angular/core';
import { CartOrderEntriesContext } from '../order-entries-context/cart-order-entries.context';
import { OrderEntriesSource } from '../order-entries-context/import-export.model';
import * as i0 from "@angular/core";
import * as i1 from "@ngrx/store";
import * as i2 from "@spartacus/core";
export class ActiveCartOrderEntriesContext extends CartOrderEntriesContext {
    constructor(actionsSubject, activeCartService) {
        super(actionsSubject);
        this.actionsSubject = actionsSubject;
        this.activeCartService = activeCartService;
        this.type = OrderEntriesSource.ACTIVE_CART;
    }
    getEntries() {
        return this.activeCartService.getEntries();
    }
    add(products) {
        this.activeCartService.addEntries(this.mapProductsToOrderEntries(products));
        return this.activeCartService.getActiveCartId();
    }
    mapProductsToOrderEntries(products) {
        return products.map((product) => ({
            product: { code: product.productCode },
            quantity: product.quantity,
        }));
    }
}
ActiveCartOrderEntriesContext.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ActiveCartOrderEntriesContext, deps: [{ token: i1.ActionsSubject }, { token: i2.ActiveCartService }], target: i0.ɵɵFactoryTarget.Injectable });
ActiveCartOrderEntriesContext.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ActiveCartOrderEntriesContext, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ActiveCartOrderEntriesContext, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.ActionsSubject }, { type: i2.ActiveCartService }]; } });
