// @spartacus/user/schematics/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.USER_PROFILE_TRANSLATION_CHUNKS_CONFIG = exports.USER_PROFILE_TRANSLATIONS = exports.SPARTACUS_USER_PROFILE_ASSETS = exports.SPARTACUS_USER_PROFILE_ROOT = exports.SPARTACUS_USER_PROFILE = exports.USER_PROFILE_ROOT_MODULE = exports.USER_PROFILE_MODULE = exports.USER_PROFILE_FEATURE_NAME_CONSTANT = exports.USER_ACCOUNT_TRANSLATION_CHUNKS_CONFIG = exports.USER_ACCOUNT_TRANSLATIONS = exports.SPARTACUS_USER_ACCOUNT_ASSETS = exports.SPARTACUS_USER_ACCOUNT_ROOT = exports.SPARTACUS_USER_ACCOUNT = exports.USER_ACCOUNT_ROOT_MODULE = exports.USER_ACCOUNT_MODULE = exports.USER_ACCOUNT_FEATURE_NAME_CONSTANT = exports.USER_MODULE_NAME = exports.SCSS_FILE_NAME = exports.USER_FOLDER_NAME = void 0;
const schematics_1 = require("@spartacus/schematics");
exports.USER_FOLDER_NAME = 'user';
exports.SCSS_FILE_NAME = 'user.scss';
exports.USER_MODULE_NAME = 'User';
exports.USER_ACCOUNT_FEATURE_NAME_CONSTANT = 'USER_ACCOUNT_FEATURE';
exports.USER_ACCOUNT_MODULE = 'UserAccountModule';
exports.USER_ACCOUNT_ROOT_MODULE = 'UserAccountRootModule';
exports.SPARTACUS_USER_ACCOUNT = `${schematics_1.SPARTACUS_USER}/account`;
exports.SPARTACUS_USER_ACCOUNT_ROOT = `${exports.SPARTACUS_USER_ACCOUNT}/root`;
exports.SPARTACUS_USER_ACCOUNT_ASSETS = `${exports.SPARTACUS_USER_ACCOUNT}/assets`;
exports.USER_ACCOUNT_TRANSLATIONS = 'userAccountTranslations';
exports.USER_ACCOUNT_TRANSLATION_CHUNKS_CONFIG = 'userAccountTranslationChunksConfig';
exports.USER_PROFILE_FEATURE_NAME_CONSTANT = 'USER_PROFILE_FEATURE';
exports.USER_PROFILE_MODULE = 'UserProfileModule';
exports.USER_PROFILE_ROOT_MODULE = 'UserProfileRootModule';
exports.SPARTACUS_USER_PROFILE = `${schematics_1.SPARTACUS_USER}/profile`;
exports.SPARTACUS_USER_PROFILE_ROOT = `${exports.SPARTACUS_USER_PROFILE}/root`;
exports.SPARTACUS_USER_PROFILE_ASSETS = `${exports.SPARTACUS_USER_PROFILE}/assets`;
exports.USER_PROFILE_TRANSLATIONS = 'userProfileTranslations';
exports.USER_PROFILE_TRANSLATION_CHUNKS_CONFIG = 'userProfileTranslationChunksConfig';
