// @spartacus/asm/esm2015/root/asm-root.module.js
import { NgModule } from '@angular/core';
import { AuthHttpHeaderService, AuthService, AuthStorageService, } from '@spartacus/core';
import { AsmLoaderModule } from './asm-loader.module';
import { AsmAuthHttpHeaderService } from './services/asm-auth-http-header.service';
import { AsmAuthStorageService } from './services/asm-auth-storage.service';
import { AsmAuthService } from './services/asm-auth.service';
import * as i0 from "@angular/core";
export class AsmRootModule {
}
AsmRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AsmRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AsmRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AsmRootModule, imports: [AsmLoaderModule] });
AsmRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AsmRootModule, providers: [
        {
            provide: AuthStorageService,
            useExisting: AsmAuthStorageService,
        },
        {
            provide: AuthService,
            useExisting: AsmAuthService,
        },
        {
            provide: AuthHttpHeaderService,
            useExisting: AsmAuthHttpHeaderService,
        },
    ], imports: [[AsmLoaderModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AsmRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [AsmLoaderModule],
                    providers: [
                        {
                            provide: AuthStorageService,
                            useExisting: AsmAuthStorageService,
                        },
                        {
                            provide: AuthService,
                            useExisting: AsmAuthService,
                        },
                        {
                            provide: AuthHttpHeaderService,
                            useExisting: AsmAuthHttpHeaderService,
                        },
                    ],
                }]
        }] });
