// @spartacus/storefront/esm2015/router/on-navigate.service.js
import { ApplicationRef, Injectable, } from '@angular/core';
import { Scroll } from '@angular/router';
import { filter, pairwise } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./config";
import * as i2 from "@angular/router";
import * as i3 from "@angular/common";
export class OnNavigateService {
    constructor(config, router, viewportScroller, injector) {
        this.config = config;
        this.router = router;
        this.viewportScroller = viewportScroller;
        this.injector = injector;
    }
    get hostComponent() {
        var _a, _b;
        return (_b = (_a = this.injector.get(ApplicationRef)) === null || _a === void 0 ? void 0 : _a.components) === null || _b === void 0 ? void 0 : _b[0];
    }
    /**
     * Reads configuration and enables features based on flags set.
     */
    initializeWithConfig() {
        var _a, _b;
        if ((_b = (_a = this.config) === null || _a === void 0 ? void 0 : _a.enableResetViewOnNavigate) === null || _b === void 0 ? void 0 : _b.active) {
            this.setResetViewOnNavigate(this.config.enableResetViewOnNavigate.active);
        }
    }
    /**
     * Resets view back to the original position when performing a back navigation and to the top when performing a front navigation
     * and sets the focus back to the top of the page before skiplinks for any type of navigation
     * @param enable Enable or disable this feature
     */
    setResetViewOnNavigate(enable) {
        var _a;
        (_a = this.subscription) === null || _a === void 0 ? void 0 : _a.unsubscribe();
        if (enable) {
            this.subscription = this.router.events
                .pipe(filter((event) => event instanceof Scroll), pairwise())
                .subscribe((event) => {
                var _a, _b, _c;
                const previousRoute = event[0];
                const currentRoute = event[1];
                if (currentRoute.position) {
                    // allow the pages to be repainted before scrolling to proper position
                    setTimeout(() => this.viewportScroller.scrollToPosition(currentRoute.position));
                }
                else {
                    if (((_a = this.config.enableResetViewOnNavigate) === null || _a === void 0 ? void 0 : _a.ignoreQueryString) &&
                        this.isPathEqual(previousRoute, currentRoute)) {
                        return;
                    }
                    if (this.isChildRoute(currentRoute)) {
                        return;
                    }
                    setTimeout(() => this.viewportScroller.scrollToPosition([0, 0]), 100);
                }
                (_c = (_b = this.hostComponent) === null || _b === void 0 ? void 0 : _b.location) === null || _c === void 0 ? void 0 : _c.nativeElement.focus();
            });
        }
    }
    /**
     * Verifies if the current route is a child route from the given ignore config route
     *
     * @param route
     * @returns boolean whether the route is a child route
     */
    isChildRoute(route) {
        var _a, _b, _c;
        return ((_c = (_b = (_a = this.config.enableResetViewOnNavigate) === null || _a === void 0 ? void 0 : _a.ignoreRoutes) === null || _b === void 0 ? void 0 : _b.some((configRoute) => route.routerEvent.urlAfterRedirects.split('/').includes(configRoute))) !== null && _c !== void 0 ? _c : false);
    }
    /**
     * Verifies if the previous and current route are the same without the query string
     *
     * @param previousRoute
     * @param currentRoute
     * @returns boolean depending on the previous and current route are equal without the query strings
     */
    isPathEqual(previousRoute, currentRoute) {
        return (previousRoute.routerEvent.urlAfterRedirects.split('?')[0] ===
            currentRoute.routerEvent.urlAfterRedirects.split('?')[0]);
    }
}
OnNavigateService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OnNavigateService, deps: [{ token: i1.OnNavigateConfig }, { token: i2.Router }, { token: i3.ViewportScroller }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Injectable });
OnNavigateService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OnNavigateService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OnNavigateService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.OnNavigateConfig }, { type: i2.Router }, { type: i3.ViewportScroller }, { type: i0.Injector }]; } });
