// @spartacus/core/esm2015/src/user/user-transitional_4_2.module.js
import { NgModule } from '@angular/core';
import { UserEventBuilder } from './events/user-event.builder';
import { UserEventModule } from './events/user-event.module';
import { UserStoreTransitional_4_2_Module } from './store/user-store-transitional_4_2.module';
import * as i0 from "@angular/core";
export class UserTransitional_4_2_Module {
    static forRoot() {
        return {
            ngModule: UserTransitional_4_2_Module,
        };
    }
}
UserTransitional_4_2_Module.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitional_4_2_Module, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserTransitional_4_2_Module.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitional_4_2_Module, imports: [UserStoreTransitional_4_2_Module, UserEventModule] });
UserTransitional_4_2_Module.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitional_4_2_Module, providers: [UserEventBuilder], imports: [[UserStoreTransitional_4_2_Module, UserEventModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitional_4_2_Module, decorators: [{
            type: NgModule,
            args: [{
                    imports: [UserStoreTransitional_4_2_Module, UserEventModule],
                    providers: [UserEventBuilder],
                }]
        }] });
