// @spartacus/core/esm2015/src/user/user-transitional.module.js
import { NgModule } from '@angular/core';
import { UserEventBuilder } from './events/user-event.builder';
import { UserEventModule } from './events/user-event.module';
import { UserStoreTransitionalModule } from './store/user-store-transitional.module';
import * as i0 from "@angular/core";
/**
 * @deprecated since 4.2 - use UserTransitional_4_2_Module with order lib instead
 */
export class UserTransitionalModule {
    static forRoot() {
        return {
            ngModule: UserTransitionalModule,
        };
    }
}
UserTransitionalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitionalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserTransitionalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitionalModule, imports: [UserStoreTransitionalModule, UserEventModule] });
UserTransitionalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitionalModule, providers: [UserEventBuilder], imports: [[UserStoreTransitionalModule, UserEventModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserTransitionalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [UserStoreTransitionalModule, UserEventModule],
                    providers: [UserEventBuilder],
                }]
        }] });
