// @spartacus/storefront/esm2015/cms-components/myaccount/order/order-details/order-detail-totals/order-detail-totals.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../order-details.service";
import * as i2 from "../../../../cart/cart-shared/order-summary/order-summary.component";
import * as i3 from "@angular/common";
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class OrderDetailTotalsComponent {
    constructor(orderDetailsService) {
        this.orderDetailsService = orderDetailsService;
    }
    ngOnInit() {
        this.order$ = this.orderDetailsService.getOrderDetails();
    }
}
OrderDetailTotalsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailTotalsComponent, deps: [{ token: i1.OrderDetailsService }], target: i0.ɵɵFactoryTarget.Component });
OrderDetailTotalsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: OrderDetailTotalsComponent, selector: "cx-order-details-totals", ngImport: i0, template: "<ng-container *ngIf=\"order$ | async as order\">\n  <div class=\"row justify-content-end\">\n    <div class=\"cx-summary col-sm-12 col-md-6 col-lg-5 col-xl-4\">\n      <cx-order-summary [cart]=\"order\"></cx-order-summary>\n    </div>\n  </div>\n</ng-container>\n", components: [{ type: i2.OrderSummaryComponent, selector: "cx-order-summary", inputs: ["cart"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "async": i3.AsyncPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailTotalsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-order-details-totals',
                    templateUrl: './order-detail-totals.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.OrderDetailsService }]; } });
