// @spartacus/storefront/esm2015/cms-components/misc/global-message/global-message.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule } from '@spartacus/core';
import { IconModule } from '../../../cms-components/misc/icon/icon.module';
import { GlobalMessageComponent } from './global-message.component';
import * as i0 from "@angular/core";
export class GlobalMessageComponentModule {
}
GlobalMessageComponentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageComponentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
GlobalMessageComponentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageComponentModule, declarations: [GlobalMessageComponent], imports: [CommonModule, IconModule, I18nModule], exports: [GlobalMessageComponent] });
GlobalMessageComponentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageComponentModule, imports: [[CommonModule, IconModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: GlobalMessageComponentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, I18nModule],
                    declarations: [GlobalMessageComponent],
                    exports: [GlobalMessageComponent],
                }]
        }] });
