// @spartacus/cart/esm2015/quick-order/root/quick-order-root.module.js
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { CmsPageGuard, ORDER_ENTRIES_CONTEXT, PageLayoutComponent, } from '@spartacus/storefront';
import { defaultQuickOrderConfig } from './config/default-quick-order.config';
import { CART_QUICK_ORDER_CORE_FEATURE, CART_QUICK_ORDER_FEATURE, } from './feature-name';
import { QuickOrderOrderEntriesContext } from './pages/quick-order-order-entries-context';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export function defaultQuickOrderComponentsConfig() {
    const config = {
        featureModules: {
            [CART_QUICK_ORDER_FEATURE]: {
                cmsComponents: ['QuickOrderComponent', 'CartQuickOrderFormComponent'],
            },
            // by default core is bundled together with components
            [CART_QUICK_ORDER_CORE_FEATURE]: CART_QUICK_ORDER_FEATURE,
        },
    };
    return config;
}
export const defaultQuickOrderRoutingConfig = {
    routing: {
        routes: {
            quickOrder: {
                paths: ['my-account/quick-order'],
            },
        },
    },
};
export class QuickOrderRootModule {
}
QuickOrderRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
QuickOrderRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderRootModule, imports: [i1.RouterModule] });
QuickOrderRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderRootModule, providers: [
        provideDefaultConfigFactory(defaultQuickOrderComponentsConfig),
        provideDefaultConfig(defaultQuickOrderRoutingConfig),
        provideDefaultConfig(defaultQuickOrderConfig),
    ], imports: [[
            RouterModule.forChild([
                {
                    // @ts-ignore
                    path: null,
                    canActivate: [CmsPageGuard],
                    component: PageLayoutComponent,
                    data: {
                        cxRoute: 'quickOrder',
                        cxContext: {
                            [ORDER_ENTRIES_CONTEXT]: QuickOrderOrderEntriesContext,
                        },
                    },
                },
            ]),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: QuickOrderRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        RouterModule.forChild([
                            {
                                // @ts-ignore
                                path: null,
                                canActivate: [CmsPageGuard],
                                component: PageLayoutComponent,
                                data: {
                                    cxRoute: 'quickOrder',
                                    cxContext: {
                                        [ORDER_ENTRIES_CONTEXT]: QuickOrderOrderEntriesContext,
                                    },
                                },
                            },
                        ]),
                    ],
                    providers: [
                        provideDefaultConfigFactory(defaultQuickOrderComponentsConfig),
                        provideDefaultConfig(defaultQuickOrderRoutingConfig),
                        provideDefaultConfig(defaultQuickOrderConfig),
                    ],
                }]
        }] });
