// @spartacus/core/esm2015/src/site-context/providers/context-initializer-providers.js
import { APP_INITIALIZER } from '@angular/core';
import { BaseSiteInitializer } from '../services/base-site-initializer';
import { CurrencyInitializer } from '../services/currency-initializer';
import { LanguageInitializer } from '../services/language-initializer';
export function initializeCurrency(currencyInitializer) {
    const result = () => {
        currencyInitializer.initialize();
    };
    return result;
}
export function initializeLanguage(languageInitializer) {
    const result = () => {
        languageInitializer.initialize();
    };
    return result;
}
export function initializeBaseSite(baseSiteInitializer) {
    const result = () => {
        baseSiteInitializer.initialize();
    };
    return result;
}
export const contextInitializerProviders = [
    {
        provide: APP_INITIALIZER,
        useFactory: initializeLanguage,
        deps: [LanguageInitializer],
        multi: true,
    },
    {
        provide: APP_INITIALIZER,
        useFactory: initializeCurrency,
        deps: [CurrencyInitializer],
        multi: true,
    },
    {
        provide: APP_INITIALIZER,
        useFactory: initializeBaseSite,
        deps: [BaseSiteInitializer],
        multi: true,
    },
];
