// @spartacus/storefront/esm2015/shared/components/form/form-errors/form-errors.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormErrorsComponent } from './form-errors.component';
import { I18nModule } from '@spartacus/core';
import * as i0 from "@angular/core";
export class FormErrorsModule {
}
FormErrorsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormErrorsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FormErrorsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormErrorsModule, declarations: [FormErrorsComponent], imports: [CommonModule, I18nModule], exports: [FormErrorsComponent] });
FormErrorsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormErrorsModule, imports: [[CommonModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormErrorsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule],
                    declarations: [FormErrorsComponent],
                    exports: [FormErrorsComponent],
                }]
        }] });
