// @spartacus/core/esm2015/src/routing/external-routes/external-routes.module.js
import { APP_INITIALIZER, NgModule } from '@angular/core';
import { addExternalRoutesFactory } from './external-routes.providers';
import { ExternalRoutesService } from './external-routes.service';
import * as i0 from "@angular/core";
/**
 * Prepends the external route that redirects to a different storefront system for configured URLs
 */
export class ExternalRoutesModule {
    static forRoot() {
        return {
            ngModule: ExternalRoutesModule,
            providers: [
                {
                    provide: APP_INITIALIZER,
                    multi: true,
                    useFactory: addExternalRoutesFactory,
                    deps: [ExternalRoutesService],
                },
            ],
        };
    }
}
ExternalRoutesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExternalRoutesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ExternalRoutesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExternalRoutesModule });
ExternalRoutesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExternalRoutesModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ExternalRoutesModule, decorators: [{
            type: NgModule
        }] });
