// @spartacus/user/fesm2015/spartacus-user-account-occ.js
import * as i0 from '@angular/core';
import { Injectable, NgModule } from '@angular/core';
import * as i2 from '@spartacus/core';
import { normalizeHttpError, provideDefaultConfig } from '@spartacus/core';
import { USER_ACCOUNT_NORMALIZER, UserAccountAdapter } from '@spartacus/user/account/core';
import { throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
import * as i1 from '@angular/common/http';

const defaultOccUserAccountConfig = {
    backend: {
        occ: { endpoints: { user: 'users/${userId}' } },
    },
};

class OccUserAccountAdapter {
    constructor(http, occEndpoints, converter) {
        this.http = http;
        this.occEndpoints = occEndpoints;
        this.converter = converter;
    }
    load(userId) {
        const url = this.occEndpoints.buildUrl('user', { urlParams: { userId } });
        return this.http.get(url).pipe(catchError((error) => throwError(normalizeHttpError(error))), this.converter.pipeable(USER_ACCOUNT_NORMALIZER));
    }
}
OccUserAccountAdapter.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccUserAccountAdapter, deps: [{ token: i1.HttpClient }, { token: i2.OccEndpointsService }, { token: i2.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable });
OccUserAccountAdapter.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccUserAccountAdapter });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccUserAccountAdapter, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: i2.OccEndpointsService }, { type: i2.ConverterService }]; } });

class UserAccountOccModule {
}
UserAccountOccModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserAccountOccModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountOccModule });
UserAccountOccModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountOccModule, providers: [
        provideDefaultConfig(defaultOccUserAccountConfig),
        { provide: UserAccountAdapter, useClass: OccUserAccountAdapter },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAccountOccModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfig(defaultOccUserAccountConfig),
                        { provide: UserAccountAdapter, useClass: OccUserAccountAdapter },
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { OccUserAccountAdapter, UserAccountOccModule, defaultOccUserAccountConfig };
