// @spartacus/core/esm2015/src/user/store/actions/consignment-tracking.action.js
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const LOAD_CONSIGNMENT_TRACKING = '[User] Load Consignment Tracking';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const LOAD_CONSIGNMENT_TRACKING_FAIL = '[User] Load Consignment Tracking Fail';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const LOAD_CONSIGNMENT_TRACKING_SUCCESS = '[User] Load Consignment Tracking Success';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const CLEAR_CONSIGNMENT_TRACKING = '[User] Clear Consignment Tracking';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class LoadConsignmentTracking {
    constructor(payload) {
        this.payload = payload;
        this.type = LOAD_CONSIGNMENT_TRACKING;
    }
}
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class LoadConsignmentTrackingFail {
    constructor(payload) {
        this.payload = payload;
        this.type = LOAD_CONSIGNMENT_TRACKING_FAIL;
    }
}
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class LoadConsignmentTrackingSuccess {
    constructor(payload) {
        this.payload = payload;
        this.type = LOAD_CONSIGNMENT_TRACKING_SUCCESS;
    }
}
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class ClearConsignmentTracking {
    constructor() {
        this.type = CLEAR_CONSIGNMENT_TRACKING;
    }
}
