// @spartacus/core/esm2015/src/user/store/actions/user-replenishment-orders.action.js
import { StateUtils } from '../../../state/utils/index';
import { USER_REPLENISHMENT_ORDERS } from '../user-state';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const LOAD_USER_REPLENISHMENT_ORDERS = '[User] Load User Replenishment Orders';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const LOAD_USER_REPLENISHMENT_ORDERS_FAIL = '[User] Load User Replenishment Orders Fail';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const LOAD_USER_REPLENISHMENT_ORDERS_SUCCESS = '[User] Load User Replenishment Orders Success';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export const CLEAR_USER_REPLENISHMENT_ORDERS = '[User] Clear User Replenishment Orders';
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class LoadUserReplenishmentOrders extends StateUtils.LoaderLoadAction {
    constructor(payload) {
        super(USER_REPLENISHMENT_ORDERS);
        this.payload = payload;
        this.type = LOAD_USER_REPLENISHMENT_ORDERS;
    }
}
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class LoadUserReplenishmentOrdersFail extends StateUtils.LoaderFailAction {
    constructor(payload) {
        super(USER_REPLENISHMENT_ORDERS, payload);
        this.payload = payload;
        this.type = LOAD_USER_REPLENISHMENT_ORDERS_FAIL;
    }
}
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class LoadUserReplenishmentOrdersSuccess extends StateUtils.LoaderSuccessAction {
    constructor(payload) {
        super(USER_REPLENISHMENT_ORDERS);
        this.payload = payload;
        this.type = LOAD_USER_REPLENISHMENT_ORDERS_SUCCESS;
    }
}
/**
 * @deprecated since 4.2 - use order lib instead
 */
export class ClearUserReplenishmentOrders extends StateUtils.LoaderResetAction {
    constructor() {
        super(USER_REPLENISHMENT_ORDERS);
        this.type = CLEAR_USER_REPLENISHMENT_ORDERS;
    }
}
