// @spartacus/storefront/esm2015/cms-structure/pwa/components/add-to-home-screen.component.js
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/add-to-home-screen.service";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class AddToHomeScreenComponent {
    constructor(addToHomeScreenService) {
        this.addToHomeScreenService = addToHomeScreenService;
    }
    ngOnInit() {
        this.canPrompt$ = this.addToHomeScreenService.canPrompt$;
    }
    prompt() {
        this.addToHomeScreenService.firePrompt();
    }
}
AddToHomeScreenComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToHomeScreenComponent, deps: [{ token: i1.AddToHomeScreenService }], target: i0.ɵɵFactoryTarget.Directive });
AddToHomeScreenComponent.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: AddToHomeScreenComponent, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AddToHomeScreenComponent, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.AddToHomeScreenService }]; } });
