// @spartacus/storefront/esm2015/cms-components/cart/validation/cart-item-warning/cart-item-validation-warning.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { CartItemValidationWarningComponent } from './cart-item-validation-warning.component';
import { IconModule } from '../../../misc/icon/index';
import * as i0 from "@angular/core";
export class CartItemValidationWarningModule {
}
CartItemValidationWarningModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartItemValidationWarningModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CartItemValidationWarningModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartItemValidationWarningModule, declarations: [CartItemValidationWarningComponent], imports: [CommonModule, RouterModule, I18nModule, UrlModule, IconModule], exports: [CartItemValidationWarningComponent] });
CartItemValidationWarningModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartItemValidationWarningModule, imports: [[CommonModule, RouterModule, I18nModule, UrlModule, IconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartItemValidationWarningModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RouterModule, I18nModule, UrlModule, IconModule],
                    exports: [CartItemValidationWarningComponent],
                    declarations: [CartItemValidationWarningComponent],
                }]
        }] });
