// @spartacus/product/fesm2015/spartacus-product-bulk-pricing-occ.js
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';

const defaultOccBulkPricingConfig = {
    backend: {
        occ: {
            endpoints: {
                product: {
                    bulkPrices: 'orgProducts/${productCode}?fields=price(DEFAULT),volumePrices(FULL)',
                },
            },
        },
    },
};

class BulkPricingOccModule {
}
BulkPricingOccModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BulkPricingOccModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingOccModule, imports: [CommonModule] });
BulkPricingOccModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingOccModule, providers: [provideDefaultConfig(defaultOccBulkPricingConfig)], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [provideDefaultConfig(defaultOccBulkPricingConfig)],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { BulkPricingOccModule };
