// @spartacus/product/fesm2015/spartacus-product-bulk-pricing-root.js
import * as i0 from '@angular/core';
import { NgModule } from '@angular/core';
import { provideDefaultConfigFactory } from '@spartacus/core';

const PRODUCT_BULK_PRICING_FEATURE = 'productBulkPricing';

// TODO: Inline this factory when we start releasing Ivy compiled libraries
function defaultProductBulkPricingComponentsConfig() {
    const config = {
        featureModules: {
            [PRODUCT_BULK_PRICING_FEATURE]: {
                cmsComponents: ['BulkPricingTableComponent'],
            },
        },
    };
    return config;
}
class BulkPricingRootModule {
}
BulkPricingRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BulkPricingRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule });
BulkPricingRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule, providers: [
        provideDefaultConfigFactory(defaultProductBulkPricingComponentsConfig),
    ], imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    providers: [
                        provideDefaultConfigFactory(defaultProductBulkPricingComponentsConfig),
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { BulkPricingRootModule, PRODUCT_BULK_PRICING_FEATURE, defaultProductBulkPricingComponentsConfig };
