// @spartacus/product/fesm2015/spartacus-product-variants-occ.js
import { CommonModule } from '@angular/common';
import * as i0 from '@angular/core';
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';

const defaultOccProductVariantsConfig = {
    backend: {
        occ: {
            endpoints: {
                product: {
                    variants: 'products/${productCode}?fields=name,purchasable,baseOptions(DEFAULT),baseProduct,variantOptions(DEFAULT),variantType',
                },
            },
        },
    },
};

class ProductVariantsOccModule {
}
ProductVariantsOccModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantsOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductVariantsOccModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantsOccModule, imports: [CommonModule] });
ProductVariantsOccModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantsOccModule, providers: [provideDefaultConfig(defaultOccProductVariantsConfig)], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductVariantsOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [provideDefaultConfig(defaultOccProductVariantsConfig)],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { ProductVariantsOccModule };
