// @spartacus/storefront/esm2015/cms-components/product/product-list/product-facet-navigation/product-facet-navigation.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ConfigModule, I18nModule } from '@spartacus/core';
import { IconModule } from '../../../misc/icon/icon.module';
import { ActiveFacetsModule } from './active-facets/active-facets.module';
import { FacetListModule } from './facet-list/facet-list.module';
import { ProductFacetNavigationComponent } from './product-facet-navigation.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class ProductFacetNavigationModule {
}
ProductFacetNavigationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductFacetNavigationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProductFacetNavigationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductFacetNavigationModule, declarations: [ProductFacetNavigationComponent], imports: [CommonModule,
        FacetListModule,
        ActiveFacetsModule,
        IconModule,
        I18nModule, i1.ConfigModule], exports: [ProductFacetNavigationComponent] });
ProductFacetNavigationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductFacetNavigationModule, imports: [[
            CommonModule,
            FacetListModule,
            ActiveFacetsModule,
            IconModule,
            I18nModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    ProductRefinementComponent: {
                        component: ProductFacetNavigationComponent,
                    },
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ProductFacetNavigationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FacetListModule,
                        ActiveFacetsModule,
                        IconModule,
                        I18nModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                ProductRefinementComponent: {
                                    component: ProductFacetNavigationComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [ProductFacetNavigationComponent],
                    exports: [ProductFacetNavigationComponent],
                }]
        }] });
