// @spartacus/storefront/esm2015/cms-components/cart/cart-shared/cart-item/model/cart-item-context-source.model.js
import { Injectable } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import * as i0 from "@angular/core";
/**
 * Context source for `CartItemComponent`.
 *
 * `CartItemContext` should be injected instead in child components.
 */
export class CartItemContextSource {
    constructor() {
        this.compact$ = new ReplaySubject(1);
        this.readonly$ = new ReplaySubject(1);
        this.item$ = new ReplaySubject(1);
        this.quantityControl$ = new ReplaySubject(1);
        this.location$ = new ReplaySubject(1);
        this.options$ = new ReplaySubject(1);
    }
}
CartItemContextSource.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartItemContextSource, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
CartItemContextSource.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartItemContextSource });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CartItemContextSource, decorators: [{
            type: Injectable
        }] });
