// @spartacus/storefront/esm2015/cms-structure/page/component/inner-components-host.directive.js
import { Directive, } from '@angular/core';
import { map } from 'rxjs/operators';
import { ComponentWrapperDirective } from './component-wrapper.directive';
import * as i0 from "@angular/core";
import * as i1 from "../model/cms-component-data";
import * as i2 from "../../services/cms-components.service";
import * as i3 from "@spartacus/core";
import * as i4 from "./services/component-handler.service";
import * as i5 from "./services/cms-injector.service";
export class InnerComponentsHostDirective {
    constructor(data, vcr, 
    // dependencies required for ComponentWrapper directive
    cmsComponentsService, injector, dynamicAttributeService, renderer, componentHandler, cmsInjector) {
        this.data = data;
        this.vcr = vcr;
        this.cmsComponentsService = cmsComponentsService;
        this.injector = injector;
        this.dynamicAttributeService = dynamicAttributeService;
        this.renderer = renderer;
        this.componentHandler = componentHandler;
        this.cmsInjector = cmsInjector;
        this.innerComponents$ = this.data.data$.pipe(map((data) => { var _a, _b; return (_b = (_a = data === null || data === void 0 ? void 0 : data.composition) === null || _a === void 0 ? void 0 : _a.inner) !== null && _b !== void 0 ? _b : []; }));
        this.componentWrappers = [];
    }
    ngOnInit() {
        this.subscription = this.innerComponents$.subscribe((x) => {
            this.renderComponents(x);
        });
    }
    renderComponents(components) {
        this.clearComponents();
        components.forEach((component) => this.renderComponent(component));
    }
    renderComponent(component) {
        const componentWrapper = new ComponentWrapperDirective(this.vcr, this.cmsComponentsService, this.injector, this.dynamicAttributeService, this.renderer, this.componentHandler, this.cmsInjector);
        componentWrapper.cxComponentWrapper = { flexType: component, uid: '' };
        componentWrapper.ngOnInit();
        this.componentWrappers.push(componentWrapper);
    }
    clearComponents() {
        this.componentWrappers.forEach((wrapper) => wrapper.ngOnDestroy());
        this.componentWrappers = [];
    }
    ngOnDestroy() {
        var _a;
        (_a = this.subscription) === null || _a === void 0 ? void 0 : _a.unsubscribe();
        this.clearComponents();
    }
}
InnerComponentsHostDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: InnerComponentsHostDirective, deps: [{ token: i1.CmsComponentData }, { token: i0.ViewContainerRef }, { token: i2.CmsComponentsService }, { token: i0.Injector }, { token: i3.DynamicAttributeService }, { token: i0.Renderer2 }, { token: i4.ComponentHandlerService }, { token: i5.CmsInjectorService }], target: i0.ɵɵFactoryTarget.Directive });
InnerComponentsHostDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: InnerComponentsHostDirective, selector: "[cxInnerComponentsHost]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: InnerComponentsHostDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxInnerComponentsHost]',
                }]
        }], ctorParameters: function () { return [{ type: i1.CmsComponentData }, { type: i0.ViewContainerRef }, { type: i2.CmsComponentsService }, { type: i0.Injector }, { type: i3.DynamicAttributeService }, { type: i0.Renderer2 }, { type: i4.ComponentHandlerService }, { type: i5.CmsInjectorService }]; } });
