// @spartacus/cart/schematics/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CART_IMPORT_EXPORT_TRANSLATION_CHUNKS_CONFIG = exports.CART_IMPORT_EXPORT_TRANSLATIONS = exports.SPARTACUS_CART_IMPORT_EXPORT_ASSETS = exports.SPARTACUS_CART_IMPORT_EXPORT_ROOT = exports.SPARTACUS_CART_IMPORT_EXPORT = exports.CART_IMPORT_EXPORT_ROOT_MODULE = exports.CART_IMPORT_EXPORT_MODULE = exports.CART_IMPORT_EXPORT_FEATURE_NAME_CONSTANT = exports.CART_IMPORT_EXPORT_MODULE_NAME = exports.QUICK_ORDER_TRANSLATION_CHUNKS_CONFIG = exports.QUICK_ORDER_TRANSLATIONS = exports.SPARTACUS_QUICK_ORDER_ASSETS = exports.SPARTACUS_QUICK_ORDER_ROOT = exports.SPARTACUS_QUICK_ORDER = exports.QUICK_ORDER_ROOT_MODULE = exports.QUICK_ORDER_MODULE = exports.CART_QUICK_ORDER_FEATURE_NAME_CONSTANT = exports.CART_QUICK_ORDER_MODULE_NAME = exports.SAVED_CART_TRANSLATION_CHUNKS_CONFIG = exports.SAVED_CART_TRANSLATIONS = exports.SPARTACUS_SAVED_CART_ASSETS = exports.SPARTACUS_SAVED_CART_ROOT = exports.SPARTACUS_SAVED_CART = exports.SAVED_CART_ROOT_MODULE = exports.SAVED_CART_MODULE = exports.CART_SAVED_CART_FEATURE_NAME_CONSTANT = exports.CART_SAVED_CART_MODULE_NAME = exports.SCSS_FILE_NAME = exports.CART_FOLDER_NAME = void 0;
const schematics_1 = require("@spartacus/schematics");
exports.CART_FOLDER_NAME = 'cart';
exports.SCSS_FILE_NAME = 'cart.scss';
exports.CART_SAVED_CART_MODULE_NAME = 'CartSavedCart';
exports.CART_SAVED_CART_FEATURE_NAME_CONSTANT = 'CART_SAVED_CART_FEATURE';
exports.SAVED_CART_MODULE = 'SavedCartModule';
exports.SAVED_CART_ROOT_MODULE = 'SavedCartRootModule';
exports.SPARTACUS_SAVED_CART = `${schematics_1.SPARTACUS_CART}/saved-cart`;
exports.SPARTACUS_SAVED_CART_ROOT = `${exports.SPARTACUS_SAVED_CART}/root`;
exports.SPARTACUS_SAVED_CART_ASSETS = `${exports.SPARTACUS_SAVED_CART}/assets`;
exports.SAVED_CART_TRANSLATIONS = 'savedCartTranslations';
exports.SAVED_CART_TRANSLATION_CHUNKS_CONFIG = 'savedCartTranslationChunksConfig';
exports.CART_QUICK_ORDER_MODULE_NAME = 'CartQuickOrder';
exports.CART_QUICK_ORDER_FEATURE_NAME_CONSTANT = 'CART_QUICK_ORDER_FEATURE';
exports.QUICK_ORDER_MODULE = 'QuickOrderModule';
exports.QUICK_ORDER_ROOT_MODULE = 'QuickOrderRootModule';
exports.SPARTACUS_QUICK_ORDER = `${schematics_1.SPARTACUS_CART}/quick-order`;
exports.SPARTACUS_QUICK_ORDER_ROOT = `${exports.SPARTACUS_QUICK_ORDER}/root`;
exports.SPARTACUS_QUICK_ORDER_ASSETS = `${exports.SPARTACUS_QUICK_ORDER}/assets`;
exports.QUICK_ORDER_TRANSLATIONS = 'quickOrderTranslations';
exports.QUICK_ORDER_TRANSLATION_CHUNKS_CONFIG = 'quickOrderTranslationChunksConfig';
exports.CART_IMPORT_EXPORT_MODULE_NAME = 'CartImportExport';
exports.CART_IMPORT_EXPORT_FEATURE_NAME_CONSTANT = 'CART_IMPORT_EXPORT_FEATURE';
exports.CART_IMPORT_EXPORT_MODULE = 'ImportExportModule';
exports.CART_IMPORT_EXPORT_ROOT_MODULE = 'ImportExportRootModule';
exports.SPARTACUS_CART_IMPORT_EXPORT = `${schematics_1.SPARTACUS_CART}/import-export`;
exports.SPARTACUS_CART_IMPORT_EXPORT_ROOT = `${exports.SPARTACUS_CART_IMPORT_EXPORT}/root`;
exports.SPARTACUS_CART_IMPORT_EXPORT_ASSETS = `${exports.SPARTACUS_CART_IMPORT_EXPORT}/assets`;
exports.CART_IMPORT_EXPORT_TRANSLATIONS = 'importExportTranslations';
exports.CART_IMPORT_EXPORT_TRANSLATION_CHUNKS_CONFIG = 'importExportTranslationChunksConfig';
