// @spartacus/storefront/esm2015/cms-components/product/product-list/product-facet-navigation/facet/facet.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { IconModule } from '../../../../../cms-components/misc/icon/index';
import { KeyboardFocusModule } from '../../../../../layout/a11y/keyboard-focus/keyboard-focus.module';
import { FacetComponent } from './facet.component';
import * as i0 from "@angular/core";
export class FacetModule {
}
FacetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FacetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FacetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FacetModule, declarations: [FacetComponent], imports: [CommonModule,
        RouterModule,
        UrlModule,
        I18nModule,
        IconModule,
        KeyboardFocusModule], exports: [FacetComponent] });
FacetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FacetModule, imports: [[
            CommonModule,
            RouterModule,
            UrlModule,
            I18nModule,
            IconModule,
            KeyboardFocusModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FacetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        UrlModule,
                        I18nModule,
                        IconModule,
                        KeyboardFocusModule,
                    ],
                    declarations: [FacetComponent],
                    exports: [FacetComponent],
                }]
        }] });
