// @spartacus/cart/fesm2015/spartacus-cart-import-export-root.js
import * as i0 from '@angular/core';
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';

const CART_IMPORT_EXPORT_FEATURE = 'cartImportExport';

class ImportExportRootModule {
}
ImportExportRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ImportExportRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule });
ImportExportRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule, providers: [
        provideDefaultConfig({
            featureModules: {
                [CART_IMPORT_EXPORT_FEATURE]: {
                    cmsComponents: [
                        'ImportExportOrderEntriesComponent',
                        'ImportOrderEntriesComponent',
                        'ExportOrderEntriesComponent',
                    ],
                },
            },
        }),
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportRootModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfig({
                            featureModules: {
                                [CART_IMPORT_EXPORT_FEATURE]: {
                                    cmsComponents: [
                                        'ImportExportOrderEntriesComponent',
                                        'ImportOrderEntriesComponent',
                                        'ExportOrderEntriesComponent',
                                    ],
                                },
                            },
                        }),
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { CART_IMPORT_EXPORT_FEATURE, ImportExportRootModule };
