// @spartacus/product/esm2015/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ConfigModule, I18nModule } from '@spartacus/core';
import { SpinnerModule } from '@spartacus/storefront';
import { BulkPricingTableComponent } from './bulk-pricing-table.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class BulkPricingTableModule {
}
BulkPricingTableModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingTableModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BulkPricingTableModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingTableModule, declarations: [BulkPricingTableComponent], imports: [CommonModule,
        I18nModule,
        SpinnerModule, i1.ConfigModule], exports: [BulkPricingTableComponent] });
BulkPricingTableModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingTableModule, imports: [[
            CommonModule,
            I18nModule,
            SpinnerModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    BulkPricingTableComponent: {
                        component: BulkPricingTableComponent,
                    },
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingTableModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        I18nModule,
                        SpinnerModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                BulkPricingTableComponent: {
                                    component: BulkPricingTableComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [BulkPricingTableComponent],
                    exports: [BulkPricingTableComponent],
                }]
        }] });
