// @spartacus/product/esm2015/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.js
import { Component } from '@angular/core';
import { switchMap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/product/bulk-pricing/core";
import * as i3 from "@angular/common";
export class BulkPricingTableComponent {
    constructor(routingService, bulkPricingService) {
        this.routingService = routingService;
        this.bulkPricingService = bulkPricingService;
        this.PRODUCT_KEY = 'productCode';
    }
    ngOnInit() {
        this.priceTiers$ = this.getPrices();
    }
    formatQuantity(tier) {
        let formattedQuantityRange = '';
        if (!tier.maxQuantity) {
            formattedQuantityRange = tier.minQuantity + '+';
        }
        else {
            formattedQuantityRange = tier.minQuantity + ' - ' + tier.maxQuantity;
        }
        return formattedQuantityRange;
    }
    getPrices() {
        return this.routingService.getRouterState().pipe(switchMap((state) => {
            const productCode = state.state.params[this.PRODUCT_KEY];
            return this.bulkPricingService.getBulkPrices(productCode);
        }));
    }
}
BulkPricingTableComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingTableComponent, deps: [{ token: i1.RoutingService }, { token: i2.BulkPricingService }], target: i0.ɵɵFactoryTarget.Component });
BulkPricingTableComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: BulkPricingTableComponent, selector: "cx-bulk-pricing-table", ngImport: i0, template: "<ng-container *ngIf=\"priceTiers$ | async as priceTiers\">\n  <div class=\"container\" *ngIf=\"priceTiers.length > 0\">\n    <table class=\"table\">\n      <thead class=\"thead-light\">\n        <tr>\n          <th scope=\"col\">{{ 'bulkPricingTable.quantity' | cxTranslate }}</th>\n          <th scope=\"col\">{{ 'bulkPricingTable.price' | cxTranslate }}</th>\n          <th scope=\"col\">{{ 'bulkPricingTable.discount' | cxTranslate }}</th>\n        </tr>\n      </thead>\n      <tbody>\n        <tr *ngFor=\"let tier of priceTiers\">\n          <td>{{ formatQuantity(tier) }}</td>\n          <td>{{ tier.formattedValue }}</td>\n          <td>{{ tier.formattedDiscount }}</td>\n        </tr>\n      </tbody>\n    </table>\n  </div>\n</ng-container>\n", directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "async": i3.AsyncPipe, "cxTranslate": i1.TranslatePipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BulkPricingTableComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-bulk-pricing-table',
                    templateUrl: './bulk-pricing-table.component.html',
                }]
        }], ctorParameters: function () { return [{ type: i1.RoutingService }, { type: i2.BulkPricingService }]; } });
