// @spartacus/checkout/esm2015/root/pages/order-confirmation-order-entries-context.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { OrderEntriesSource, } from '@spartacus/storefront';
import * as i0 from "@angular/core";
import * as i1 from "../facade/checkout.facade";
export class OrderConfirmationOrderEntriesContext {
    constructor(checkoutService) {
        this.checkoutService = checkoutService;
        this.type = OrderEntriesSource.ORDER_CONFIRMATION;
    }
    getEntries() {
        return this.checkoutService
            .getOrderDetails()
            .pipe(map((order) => { var _a; return (_a = order === null || order === void 0 ? void 0 : order.entries) !== null && _a !== void 0 ? _a : []; }));
    }
}
OrderConfirmationOrderEntriesContext.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderConfirmationOrderEntriesContext, deps: [{ token: i1.CheckoutFacade }], target: i0.ɵɵFactoryTarget.Injectable });
OrderConfirmationOrderEntriesContext.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderConfirmationOrderEntriesContext, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderConfirmationOrderEntriesContext, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CheckoutFacade }]; } });
