// @spartacus/storefinder/esm2015/core/store/reducers/index.js
import { InjectionToken } from '@angular/core';
import { STORE_FINDER_DATA } from '../store-finder-state';
import { SiteContextActions, StateUtils } from '@spartacus/core';
import { StoreFinderActions } from '../actions';
import { findStoresReducer } from './find-stores.reducer';
import { viewAllStoresReducer } from './view-all-stores.reducer';
export function getReducers() {
    return {
        findStores: StateUtils.loaderReducer(STORE_FINDER_DATA, findStoresReducer),
        viewAllStores: StateUtils.loaderReducer(STORE_FINDER_DATA, viewAllStoresReducer),
    };
}
export const reducerToken = new InjectionToken('StoreFinderReducers');
export const reducerProvider = {
    provide: reducerToken,
    useFactory: getReducers,
};
export function clearStoreFinderState(reducer) {
    return function (state, action) {
        if (action.type === SiteContextActions.LANGUAGE_CHANGE) {
            state = undefined;
        }
        if (action.type === StoreFinderActions.CLEAR_STORE_FINDER_DATA) {
            state = undefined;
        }
        return reducer(state, action);
    };
}
export const metaReducers = [clearStoreFinderState];
