// @spartacus/storefront/esm2015/cms-structure/page/slot/page-slot.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { OutletModule } from '../../outlet/outlet.module';
import { PageComponentModule } from '../component/page-component.module';
import { PageSlotComponent } from './page-slot.component';
import * as i0 from "@angular/core";
import * as i1 from "./page-slot.service";
export class PageSlotModule {
    // instantiate PageSlotService ASAP, so it can examine SSR pre-rendered DOM
    constructor(_pageSlot) { }
}
PageSlotModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageSlotModule, deps: [{ token: i1.PageSlotService }], target: i0.ɵɵFactoryTarget.NgModule });
PageSlotModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageSlotModule, declarations: [PageSlotComponent], imports: [CommonModule, OutletModule, PageComponentModule], exports: [PageSlotComponent] });
PageSlotModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageSlotModule, imports: [[CommonModule, OutletModule, PageComponentModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PageSlotModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, OutletModule, PageComponentModule],
                    declarations: [PageSlotComponent],
                    exports: [PageSlotComponent],
                }]
        }], ctorParameters: function () { return [{ type: i1.PageSlotService }]; } });
