// @spartacus/cart/esm2015/import-export/components/import-export/import-export-order-entries.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ConfigModule, I18nModule, UrlModule, } from '@spartacus/core';
import { PageComponentModule } from '@spartacus/storefront';
import { ExportOrderEntriesModule } from '../export-entries';
import { ImportOrderEntriesModule } from '../import-to-cart';
import { ImportExportOrderEntriesComponent } from './import-export-order-entries.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class ImportExportOrderEntriesModule {
}
ImportExportOrderEntriesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportOrderEntriesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ImportExportOrderEntriesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportOrderEntriesModule, declarations: [ImportExportOrderEntriesComponent], imports: [PageComponentModule, i1.ConfigModule, I18nModule,
        UrlModule,
        ImportOrderEntriesModule,
        ExportOrderEntriesModule,
        CommonModule], exports: [ImportExportOrderEntriesComponent] });
ImportExportOrderEntriesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportOrderEntriesModule, imports: [[
            PageComponentModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    ImportExportOrderEntriesComponent: {
                        component: ImportExportOrderEntriesComponent,
                    },
                },
            }),
            I18nModule,
            UrlModule,
            ImportOrderEntriesModule,
            ExportOrderEntriesModule,
            CommonModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ImportExportOrderEntriesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        PageComponentModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                ImportExportOrderEntriesComponent: {
                                    component: ImportExportOrderEntriesComponent,
                                },
                            },
                        }),
                        I18nModule,
                        UrlModule,
                        ImportOrderEntriesModule,
                        ExportOrderEntriesModule,
                        CommonModule,
                    ],
                    exports: [ImportExportOrderEntriesComponent],
                    declarations: [ImportExportOrderEntriesComponent],
                }]
        }] });
