// @spartacus/checkout/esm2015/core/store/reducers/payment-types.reducer.js
import { CheckoutActions } from '../actions/index';
export const initialState = {
    entities: {},
    selected: undefined,
};
export function reducer(state = initialState, action) {
    var _a;
    switch (action.type) {
        case CheckoutActions.LOAD_PAYMENT_TYPES_SUCCESS: {
            const paymentTypes = action.payload;
            const entities = paymentTypes.reduce((paymentTypesEntities, name) => {
                return Object.assign(Object.assign({}, paymentTypesEntities), { [name.code]: name });
            }, Object.assign({}, state.entities));
            return Object.assign(Object.assign({}, state), { entities });
        }
        case CheckoutActions.SET_PAYMENT_TYPE_SUCCESS: {
            return Object.assign(Object.assign({}, state), { selected: (_a = action.payload.paymentType) === null || _a === void 0 ? void 0 : _a.code });
        }
        case CheckoutActions.CLEAR_CHECKOUT_DATA: {
            return Object.assign(Object.assign({}, state), { selected: undefined });
        }
        case CheckoutActions.CHECKOUT_CLEAR_MISCS_DATA: {
            return initialState;
        }
    }
    return state;
}
export const getPaymentTypesEntites = (state) => state.entities;
export const getSelectedPaymentType = (state) => state.selected;
