// @spartacus/order/esm2015/root/pages/order-details-order-entries-context.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { OrderEntriesSource, } from '@spartacus/storefront';
import * as i0 from "@angular/core";
import * as i1 from "../facade/order.facade";
export class OrderDetailsOrderEntriesContext {
    constructor(userOrderService) {
        this.userOrderService = userOrderService;
        this.type = OrderEntriesSource.ORDER_DETAILS;
    }
    getEntries() {
        return this.userOrderService
            .getOrderDetails()
            .pipe(map((order) => { var _a; return (_a = order === null || order === void 0 ? void 0 : order.entries) !== null && _a !== void 0 ? _a : []; }));
    }
}
OrderDetailsOrderEntriesContext.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailsOrderEntriesContext, deps: [{ token: i1.OrderFacade }], target: i0.ɵɵFactoryTarget.Injectable });
OrderDetailsOrderEntriesContext.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailsOrderEntriesContext, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderDetailsOrderEntriesContext, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.OrderFacade }]; } });
