// @spartacus/product/schematics/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IMAGE_ZOOM_TRANSLATION_CHUNKS_CONFIG = exports.IMAGE_ZOOM_TRANSLATIONS = exports.SPARTACUS_IMAGE_ZOOM_ASSETS = exports.SPARTACUS_IMAGE_ZOOM_ROOT = exports.SPARTACUS_IMAGE_ZOOM = exports.IMAGE_ZOOM_ROOT_MODULE = exports.IMAGE_ZOOM_MODULE = exports.IMAGE_ZOOM_FEATURE_NAME_CONSTANT = exports.IMAGE_ZOOM_MODULE_NAME = exports.VARIANTS_TRANSLATION_CHUNKS_CONFIG = exports.VARIANTS_TRANSLATIONS = exports.SPARTACUS_VARIANTS_ASSETS = exports.SPARTACUS_VARIANTS_ROOT = exports.SPARTACUS_VARIANTS = exports.VARIANTS_ROOT_MODULE = exports.VARIANTS_MODULE = exports.VARIANTS_FEATURE_NAME_CONSTANT = exports.VARIANTS_MODULE_NAME = exports.BULK_PRICING_TRANSLATION_CHUNKS_CONFIG = exports.BULK_PRICING_TRANSLATIONS = exports.SPARTACUS_BULK_PRICING_ASSETS = exports.SPARTACUS_BULK_PRICING_ROOT = exports.SPARTACUS_BULK_PRICING = exports.BULK_PRICING_ROOT_MODULE = exports.BULK_PRICING_MODULE = exports.BULK_PRICING_FEATURE_NAME_CONSTANT = exports.BULK_PRICING_MODULE_NAME = exports.PRODUCT_SCSS_FILE_NAME = exports.PRODUCT_FOLDER_NAME = void 0;
const schematics_1 = require("@spartacus/schematics");
exports.PRODUCT_FOLDER_NAME = 'product';
exports.PRODUCT_SCSS_FILE_NAME = 'product.scss';
exports.BULK_PRICING_MODULE_NAME = 'ProductBulkPricing';
exports.BULK_PRICING_FEATURE_NAME_CONSTANT = 'PRODUCT_BULK_PRICING_FEATURE';
exports.BULK_PRICING_MODULE = 'BulkPricingModule';
exports.BULK_PRICING_ROOT_MODULE = 'BulkPricingRootModule';
exports.SPARTACUS_BULK_PRICING = `${schematics_1.SPARTACUS_PRODUCT}/bulk-pricing`;
exports.SPARTACUS_BULK_PRICING_ROOT = `${exports.SPARTACUS_BULK_PRICING}/root`;
exports.SPARTACUS_BULK_PRICING_ASSETS = `${exports.SPARTACUS_BULK_PRICING}/assets`;
exports.BULK_PRICING_TRANSLATIONS = 'bulkPricingTranslations';
exports.BULK_PRICING_TRANSLATION_CHUNKS_CONFIG = 'bulkPricingTranslationChunksConfig';
exports.VARIANTS_MODULE_NAME = 'ProductVariants';
exports.VARIANTS_FEATURE_NAME_CONSTANT = 'PRODUCT_VARIANTS_FEATURE';
exports.VARIANTS_MODULE = 'ProductVariantsModule';
exports.VARIANTS_ROOT_MODULE = 'ProductVariantsRootModule';
exports.SPARTACUS_VARIANTS = `${schematics_1.SPARTACUS_PRODUCT}/variants`;
exports.SPARTACUS_VARIANTS_ROOT = `${exports.SPARTACUS_VARIANTS}/root`;
exports.SPARTACUS_VARIANTS_ASSETS = `${exports.SPARTACUS_VARIANTS}/assets`;
exports.VARIANTS_TRANSLATIONS = 'productVariantsTranslations';
exports.VARIANTS_TRANSLATION_CHUNKS_CONFIG = 'productVariantsTranslationChunksConfig';
exports.IMAGE_ZOOM_MODULE_NAME = 'ProductImageZoom';
exports.IMAGE_ZOOM_FEATURE_NAME_CONSTANT = 'PRODUCT_IMAGE_ZOOM_FEATURE';
exports.IMAGE_ZOOM_MODULE = 'ProductImageZoomModule';
exports.IMAGE_ZOOM_ROOT_MODULE = 'ProductImageZoomRootModule';
exports.SPARTACUS_IMAGE_ZOOM = `${schematics_1.SPARTACUS_PRODUCT}/image-zoom`;
exports.SPARTACUS_IMAGE_ZOOM_ROOT = `${exports.SPARTACUS_IMAGE_ZOOM}/root`;
exports.SPARTACUS_IMAGE_ZOOM_ASSETS = `${exports.SPARTACUS_IMAGE_ZOOM}/assets`;
exports.IMAGE_ZOOM_TRANSLATIONS = 'productImageZoomTranslations';
exports.IMAGE_ZOOM_TRANSLATION_CHUNKS_CONFIG = 'productImageZoomTranslationChunksConfig';
