// @spartacus/storefront/esm2015/shared/components/item-counter/item-counter.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { I18nModule } from '@spartacus/core';
import { ItemCounterComponent } from './item-counter.component';
import * as i0 from "@angular/core";
export class ItemCounterModule {
}
ItemCounterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemCounterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ItemCounterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemCounterModule, declarations: [ItemCounterComponent], imports: [CommonModule, ReactiveFormsModule, I18nModule], exports: [ItemCounterComponent] });
ItemCounterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemCounterModule, imports: [[CommonModule, ReactiveFormsModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemCounterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, ReactiveFormsModule, I18nModule],
                    declarations: [ItemCounterComponent],
                    exports: [ItemCounterComponent],
                }]
        }] });
