// @spartacus/checkout/esm2015/components/components/payment-method/payment-form/payment-form.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { NgSelectModule } from '@ng-select/ng-select';
import { I18nModule } from '@spartacus/core';
import { CardModule, FormErrorsModule, IconModule, SpinnerModule, } from '@spartacus/storefront';
import { PaymentFormComponent } from './payment-form.component';
import * as i0 from "@angular/core";
export class PaymentFormModule {
}
PaymentFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PaymentFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PaymentFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PaymentFormModule, declarations: [PaymentFormComponent], imports: [CommonModule,
        ReactiveFormsModule,
        NgSelectModule,
        CardModule,
        I18nModule,
        IconModule,
        SpinnerModule,
        FormErrorsModule], exports: [PaymentFormComponent] });
PaymentFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PaymentFormModule, imports: [[
            CommonModule,
            ReactiveFormsModule,
            NgSelectModule,
            CardModule,
            I18nModule,
            IconModule,
            SpinnerModule,
            FormErrorsModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PaymentFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        NgSelectModule,
                        CardModule,
                        I18nModule,
                        IconModule,
                        SpinnerModule,
                        FormErrorsModule,
                    ],
                    declarations: [PaymentFormComponent],
                    exports: [PaymentFormComponent],
                }]
        }] });
